<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="natasha.santhanam">
<meta name="author" content="Sabrina">
<meta name="dcterms.date" content="2022-02-07">

<title>Rat Genomics Pipeline Rendered - 01.Heritability_Sparsity Sabrina</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-listing/list.min.js"></script>
<script src="../site_libs/quarto-listing/quarto-listing.js"></script>
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<script>

  window.document.addEventListener("DOMContentLoaded", function (_event) {
    const listingTargetEl = window.document.querySelector('#listing-listing .list');
    if (!listingTargetEl) {
      // No listing discovered, do not attach.
      return; 
    }

    const options = {
      valueNames: [{ data: ['index'] },{ data: ['categories'] },{ data: ['listing-date-sort'] },{ data: ['listing-file-modified-sort'] }],
      
      searchColumns: [],
    };

    window['quarto-listings'] = window['quarto-listings'] || {};
    window['quarto-listings']['listing-listing'] = new List('listing-listing', options);

    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  });

  window.addEventListener('hashchange',() => {
    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  })
  </script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Rat Genomics Pipeline Rendered</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html">
 <span class="menu-text">Home</span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#generate-heritability-and-sparsity-estimates-for-all-5-tissues" id="toc-generate-heritability-and-sparsity-estimates-for-all-5-tissues" class="nav-link active" data-scroll-target="#generate-heritability-and-sparsity-estimates-for-all-5-tissues">Generate Heritability and Sparsity Estimates for all 5 tissues</a>
  <ul class="collapse">
  <li><a href="#definitions" id="toc-definitions" class="nav-link" data-scroll-target="#definitions">Definitions</a></li>
  <li><a href="#calculate-cis-heritability-within-1mb" id="toc-calculate-cis-heritability-within-1mb" class="nav-link" data-scroll-target="#calculate-cis-heritability-within-1mb">Calculate Cis Heritability within 1MB</a></li>
  </ul></li>
  <li><a href="#calculate-sparsity-estimates-for-all-5-tissues" id="toc-calculate-sparsity-estimates-for-all-5-tissues" class="nav-link" data-scroll-target="#calculate-sparsity-estimates-for-all-5-tissues">Calculate Sparsity Estimates for all 5 tissues</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">01.Heritability_Sparsity Sabrina</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Authors</div>
    <div class="quarto-title-meta-contents">
             <p>natasha.santhanam </p>
             <p>Sabrina </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 7, 2022</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="generate-heritability-and-sparsity-estimates-for-all-5-tissues" class="level1">
<h1>Generate Heritability and Sparsity Estimates for all 5 tissues</h1>
<section id="definitions" class="level2">
<h2 class="anchored" data-anchor-id="definitions">Definitions</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/data/"</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="st">"%&amp;%"</span> <span class="ot">=</span> <span class="cf">function</span>(a,b) <span class="fu">paste</span>(a,b,<span class="at">sep=</span><span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="calculate-cis-heritability-within-1mb" class="level2">
<h2 class="anchored" data-anchor-id="calculate-cis-heritability-within-1mb">Calculate Cis Heritability within 1MB</h2>
<p>For each gene, we calculate heritability from its local GRM. Start with creating list of genes for each of the gene expression file:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"~/Box/imlab-data/Projects/PTRS-PGRS-Rosetta/Data-From-Abe-Palmer-Lab/Rdata/genoGex.RData"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>ensidlist <span class="ot">&lt;-</span> gexAc<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>ensidlist_Il <span class="ot">&lt;-</span> gexIl<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>ensidlist_Lh <span class="ot">&lt;-</span> gexLh<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>ensidlist_Pl <span class="ot">&lt;-</span> gexPl<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>ensidlist_Vo <span class="ot">&lt;-</span> gexVo<span class="sc">$</span>EnsemblGeneID</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in bim files for each tissue</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>bimfile <span class="ot">&lt;-</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Ac.bim"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>bimfile_Lh <span class="ot">&lt;-</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Lh.bim"</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>bimfile_Il <span class="ot">&lt;-</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Il.bim"</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>bimfile_Pl <span class="ot">&lt;-</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Pl.bim"</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>bimfile_Vo <span class="ot">&lt;-</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Vo.bim"</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>bim <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimfile)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>bim_Lh <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimfile_Lh)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>bim_Il <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimfile_Il)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>bim_Pl <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimfile_Pl)</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>bim_Vo <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimfile_Vo)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(bim) <span class="ot">&lt;-</span> bim<span class="sc">$</span>V2</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(bim_Lh) <span class="ot">&lt;-</span> bim_Lh<span class="sc">$</span>V2</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(bim_Il) <span class="ot">&lt;-</span> bim_Il<span class="sc">$</span>V2</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(bim_Pl) <span class="ot">&lt;-</span> bim_Pl<span class="sc">$</span>V2</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(bim_Vo) <span class="ot">&lt;-</span> bim_Vo<span class="sc">$</span>V2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We run gcta Ac plink files and gene annotation to generate local GRMs, then h2 calculations.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>gt.dir <span class="ot">&lt;-</span> <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/output/Ac/"</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/"</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>grm.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"GRMs/"</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>h2.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"h2_output/"</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Make local GRMs for each gene</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span>geneinfo[<span class="dv">1</span>]</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bim,bim[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">4</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">4</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">2</span>]    </span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Ac.SNPlist"</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">col.names=</span><span class="cn">FALSE</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>    runGCTAgrm <span class="ot">&lt;-</span> <span class="st">"gcta --bfile "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Ac --make-grm-bin --extract "</span> <span class="sc">%&amp;%</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Ac.SNPlist"</span> <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span>  gene</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGCTAgrm)</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculate h2</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"of"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>    ensid <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">10</span>])</span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">1</span>])</span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>  individuals <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexAc)[<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)]</span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>  expression <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gexAc[i,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)])</span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>    <span class="co">#output expression pheno for gcta</span></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>      <span class="at">famid =</span> individuals,</span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>      <span class="at">id =</span> individuals,</span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a>      <span class="at">expr =</span> expression</span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid, <span class="at">col.names=</span><span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>) <span class="co">#output pheno for gcta</span></span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Y ~ localGRM</span></span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a>    runLOC <span class="ot">&lt;-</span> <span class="st">"gcta --grm "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --reml --pheno "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> h2.dir <span class="sc">%&amp;%</span> <span class="st">"tmp."</span> <span class="sc">%&amp;%</span> ensid</span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runLOC)</span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Repeat for Il:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>gt.dir <span class="ot">&lt;-</span> <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/output/Il/"</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/"</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>grm.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"GRMs/"</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>h2.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"h2_output/"</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Make local GRMs for each gene</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Il)){</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist_Il),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist_Il[i]</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span>geneinfo[<span class="dv">1</span>]</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bim_Il,bim_Il[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">4</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">4</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">2</span>]    </span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Il.SNPlist"</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">col.names=</span><span class="cn">FALSE</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>    runGCTAgrm <span class="ot">&lt;-</span> <span class="st">"gcta --bfile "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Il --make-grm-bin --extract "</span> <span class="sc">%&amp;%</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Il.SNPlist"</span> <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span>  gene</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGCTAgrm)</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculate h2</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Il)){</span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"of"</span>,<span class="fu">length</span>(ensidlist_Il),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>    ensid <span class="ot">&lt;-</span> ensidlist_Il[i]</span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">10</span>])</span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">1</span>])</span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>  individuals <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexIl)[<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)]</span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a>  expression <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gexIl[i,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)])</span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a>    <span class="co">#output expression pheno for gcta</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>      <span class="at">famid =</span> individuals,</span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a>      <span class="at">id =</span> individuals,</span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a>      <span class="at">expr =</span> expression</span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid, <span class="at">col.names=</span><span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>) <span class="co">#output pheno for gcta</span></span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Y ~ localGRM</span></span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a>    runLOC <span class="ot">&lt;-</span> <span class="st">"gcta --grm "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --reml --pheno "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> h2.dir <span class="sc">%&amp;%</span> <span class="st">"tmp."</span> <span class="sc">%&amp;%</span> ensid</span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runLOC)</span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Lh:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>gt.dir <span class="ot">&lt;-</span> <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/output/Lh/"</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/"</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>grm.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"GRMs/"</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>h2.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"h2_output/"</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Make local GRMs for each gene</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Lh)){</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist_Lh),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist_Lh[i]</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span>geneinfo[<span class="dv">1</span>]</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bim_Lh,bim_Lh[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">4</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">4</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">2</span>]    </span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Lh.SNPlist"</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">col.names=</span><span class="cn">FALSE</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>    runGCTAgrm <span class="ot">&lt;-</span> <span class="st">"gcta --bfile "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Lh --make-grm-bin --extract "</span> <span class="sc">%&amp;%</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Lh.SNPlist"</span> <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span>  gene</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGCTAgrm)</span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculate h2</span></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Lh)){</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"of"</span>,<span class="fu">length</span>(ensidlist_Lh),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>    ensid <span class="ot">&lt;-</span> ensidlist_Lh[i]</span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">10</span>])</span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">1</span>])</span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>  individuals <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexLh)[<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)]</span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>  expression <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gexLh[i,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)])</span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a>    <span class="co">#output expression pheno for gcta</span></span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a>      <span class="at">famid =</span> individuals,</span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a>      <span class="at">id =</span> individuals,</span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a>      <span class="at">expr =</span> expression</span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid, <span class="at">col.names=</span><span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>) <span class="co">#output pheno for gcta</span></span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Y ~ localGRM</span></span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a>    runLOC <span class="ot">&lt;-</span> <span class="st">"gcta --grm "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --reml --pheno "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> h2.dir <span class="sc">%&amp;%</span> <span class="st">"tmp."</span> <span class="sc">%&amp;%</span> ensid</span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runLOC)</span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Pl:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>gt.dir <span class="ot">&lt;-</span> <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/output/Pl/"</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/"</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>grm.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"GRMs/"</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>h2.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"h2_output/"</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Make local GRMs for each gene</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Pl)){</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist_Pl),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist_Pl[i]</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span>geneinfo[<span class="dv">1</span>]</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bim_Pl,bim_Pl[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">4</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">4</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">2</span>]    </span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Pl.SNPlist"</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">col.names=</span><span class="cn">FALSE</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>    runGCTAgrm <span class="ot">&lt;-</span> <span class="st">"gcta --bfile "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Pl --make-grm-bin --extract "</span> <span class="sc">%&amp;%</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Pl.SNPlist"</span> <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span>  gene</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGCTAgrm)</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculate h2</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Pl)){</span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"of"</span>,<span class="fu">length</span>(ensidlist_Pl),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>    ensid <span class="ot">&lt;-</span> ensidlist_Pl[i]</span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">10</span>])</span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">1</span>])</span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>  individuals <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexPl)[<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)]</span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>  expression <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gexPl[i,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)])</span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>    <span class="co">#output expression pheno for gcta</span></span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a>      <span class="at">famid =</span> individuals,</span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a>      <span class="at">id =</span> individuals,</span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>      <span class="at">expr =</span> expression</span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid, <span class="at">col.names=</span><span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>) <span class="co">#output pheno for gcta</span></span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Y ~ localGRM</span></span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a>    runLOC <span class="ot">&lt;-</span> <span class="st">"gcta --grm "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --reml --pheno "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> h2.dir <span class="sc">%&amp;%</span> <span class="st">"tmp."</span> <span class="sc">%&amp;%</span> ensid</span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runLOC)</span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Vo</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>gt.dir <span class="ot">&lt;-</span> <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/output/Vo/"</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/"</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>grm.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"GRMs/"</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>h2.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"h2_output/"</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Make local GRMs for each gene</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Vo)){</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist_Vo),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist_Vo[i]</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span>geneinfo[<span class="dv">1</span>]</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bim_Vo,bim_Vo[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">4</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">4</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">2</span>]    </span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Vo.SNPlist"</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">col.names=</span><span class="cn">FALSE</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>    runGCTAgrm <span class="ot">&lt;-</span> <span class="st">"gcta --bfile "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Vo --make-grm-bin --extract "</span> <span class="sc">%&amp;%</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Vo.SNPlist"</span> <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span>  gene</span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGCTAgrm)</span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculate h2</span></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Vo)){</span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"of"</span>,<span class="fu">length</span>(ensidlist_Vo),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a>    ensid <span class="ot">&lt;-</span> ensidlist_Vo[i]</span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">10</span>])</span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">1</span>])</span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a>  individuals <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexVo)[<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)]</span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a>  expression <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gexVo[i,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)])</span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a>    <span class="co">#output expression pheno for gcta</span></span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a>      <span class="at">famid =</span> individuals,</span>
<span id="cb8-34"><a href="#cb8-34" aria-hidden="true" tabindex="-1"></a>      <span class="at">id =</span> individuals,</span>
<span id="cb8-35"><a href="#cb8-35" aria-hidden="true" tabindex="-1"></a>      <span class="at">expr =</span> expression</span>
<span id="cb8-36"><a href="#cb8-36" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb8-37"><a href="#cb8-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid, <span class="at">col.names=</span><span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>) <span class="co">#output pheno for gcta</span></span>
<span id="cb8-38"><a href="#cb8-38" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Y ~ localGRM</span></span>
<span id="cb8-39"><a href="#cb8-39" aria-hidden="true" tabindex="-1"></a>    runLOC <span class="ot">&lt;-</span> <span class="st">"gcta --grm "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --reml --pheno "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> h2.dir <span class="sc">%&amp;%</span> <span class="st">"tmp."</span> <span class="sc">%&amp;%</span> ensid</span>
<span id="cb8-40"><a href="#cb8-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runLOC)</span>
<span id="cb8-41"><a href="#cb8-41" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="calculate-sparsity-estimates-for-all-5-tissues" class="level1">
<h1>Calculate Sparsity Estimates for all 5 tissues</h1>
<p>The following code uses <code>bim_bam</code> files generated through Palmer lab data, steps <a href="../posts/convert_genotypes_old.html">here</a>. The code</p>
<p>Ac</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/phenotype_files/"</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/genotype_files/"</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/bim_bam/"</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Ac_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/"</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Ac.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexAc_transpose[,i])</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Ac.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Ac_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Il</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/phenotype_files/"</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/genotype_files/"</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/bim_bam/"</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Il_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/"</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr </span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]    </span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Il.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexIl_transpose[,i])</span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Il.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Il_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Lh</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/phenotype_files/"</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/genotype_files/"</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/bim_bam/"</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Lh_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/"</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]    </span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Lh.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexLh_transpose[,i])</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Lh.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Lh_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Pl</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/phenotype_files/"</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/genotype_files/"</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/bim_bam/"</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Pl_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/"</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]    </span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Pl.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexPl_transpose[,i])</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Pl.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Pl_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Vo</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/phenotype_files/"</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/genotype_files/"</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/bim_bam/"</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Vo_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/"</span>)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]    </span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Vo.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexVo_transpose[,i])</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Vo.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Vo_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>



<!-- -->

</section>

<div class="quarto-listing quarto-listing-container-default" id="listing-listing">
<div class="list quarto-listing-default">

</div>
<div class="listing-no-matching d-none">
No matching items
</div>
</div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb19" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "01.Heritability_Sparsity Sabrina"</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> ["natasha.santhanam", "Sabrina"]</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2022-02-07"</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="an">output:</span><span class="co"> workflowr::wflow_html</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="an">editor_options:</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="co">  chunk_output_type: console</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a><span class="fu"># Generate Heritability and Sparsity Estimates for all 5 tissues</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a><span class="fu">## Definitions</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a><span class="in">```{r definitions, eval=FALSE}</span></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/data/"</span></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a><span class="st">"%&amp;%"</span> <span class="ot">=</span> <span class="cf">function</span>(a,b) <span class="fu">paste</span>(a,b,<span class="at">sep=</span><span class="st">""</span>)</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a><span class="fu">## Calculate Cis Heritability within 1MB</span></span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a>For each gene, we calculate heritability from its local GRM. Start with creating list of genes for each of the gene expression file:</span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a><span class="in">```{r read in expression files, eval=FALSE}</span></span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"~/Box/imlab-data/Projects/PTRS-PGRS-Rosetta/Data-From-Abe-Palmer-Lab/Rdata/genoGex.RData"</span>)</span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true" tabindex="-1"></a>ensidlist <span class="ot">&lt;-</span> gexAc<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb19-27"><a href="#cb19-27" aria-hidden="true" tabindex="-1"></a>ensidlist_Il <span class="ot">&lt;-</span> gexIl<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb19-28"><a href="#cb19-28" aria-hidden="true" tabindex="-1"></a>ensidlist_Lh <span class="ot">&lt;-</span> gexLh<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb19-29"><a href="#cb19-29" aria-hidden="true" tabindex="-1"></a>ensidlist_Pl <span class="ot">&lt;-</span> gexPl<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb19-30"><a href="#cb19-30" aria-hidden="true" tabindex="-1"></a>ensidlist_Vo <span class="ot">&lt;-</span> gexVo<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb19-31"><a href="#cb19-31" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-32"><a href="#cb19-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-33"><a href="#cb19-33" aria-hidden="true" tabindex="-1"></a><span class="in">```{r read in bim files for each tissue, eval=FALSE}</span></span>
<span id="cb19-34"><a href="#cb19-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in bim files for each tissue</span></span>
<span id="cb19-35"><a href="#cb19-35" aria-hidden="true" tabindex="-1"></a>bimfile <span class="ot">&lt;-</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Ac.bim"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb19-36"><a href="#cb19-36" aria-hidden="true" tabindex="-1"></a>bimfile_Lh <span class="ot">&lt;-</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Lh.bim"</span></span>
<span id="cb19-37"><a href="#cb19-37" aria-hidden="true" tabindex="-1"></a>bimfile_Il <span class="ot">&lt;-</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Il.bim"</span></span>
<span id="cb19-38"><a href="#cb19-38" aria-hidden="true" tabindex="-1"></a>bimfile_Pl <span class="ot">&lt;-</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Pl.bim"</span></span>
<span id="cb19-39"><a href="#cb19-39" aria-hidden="true" tabindex="-1"></a>bimfile_Vo <span class="ot">&lt;-</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Vo.bim"</span></span>
<span id="cb19-40"><a href="#cb19-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-41"><a href="#cb19-41" aria-hidden="true" tabindex="-1"></a>bim <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimfile)</span>
<span id="cb19-42"><a href="#cb19-42" aria-hidden="true" tabindex="-1"></a>bim_Lh <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimfile_Lh)</span>
<span id="cb19-43"><a href="#cb19-43" aria-hidden="true" tabindex="-1"></a>bim_Il <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimfile_Il)</span>
<span id="cb19-44"><a href="#cb19-44" aria-hidden="true" tabindex="-1"></a>bim_Pl <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimfile_Pl)</span>
<span id="cb19-45"><a href="#cb19-45" aria-hidden="true" tabindex="-1"></a>bim_Vo <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimfile_Vo)</span>
<span id="cb19-46"><a href="#cb19-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-47"><a href="#cb19-47" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(bim) <span class="ot">&lt;-</span> bim<span class="sc">$</span>V2</span>
<span id="cb19-48"><a href="#cb19-48" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(bim_Lh) <span class="ot">&lt;-</span> bim_Lh<span class="sc">$</span>V2</span>
<span id="cb19-49"><a href="#cb19-49" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(bim_Il) <span class="ot">&lt;-</span> bim_Il<span class="sc">$</span>V2</span>
<span id="cb19-50"><a href="#cb19-50" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(bim_Pl) <span class="ot">&lt;-</span> bim_Pl<span class="sc">$</span>V2</span>
<span id="cb19-51"><a href="#cb19-51" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(bim_Vo) <span class="ot">&lt;-</span> bim_Vo<span class="sc">$</span>V2</span>
<span id="cb19-52"><a href="#cb19-52" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-53"><a href="#cb19-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-54"><a href="#cb19-54" aria-hidden="true" tabindex="-1"></a>We run gcta Ac plink files and gene annotation to generate local GRMs, then h2 calculations.</span>
<span id="cb19-55"><a href="#cb19-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-56"><a href="#cb19-56" aria-hidden="true" tabindex="-1"></a><span class="in">```{r generate cis herit for Ac, eval=FALSE}</span></span>
<span id="cb19-57"><a href="#cb19-57" aria-hidden="true" tabindex="-1"></a>gt.dir <span class="ot">&lt;-</span> <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/output/Ac/"</span></span>
<span id="cb19-58"><a href="#cb19-58" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/"</span></span>
<span id="cb19-59"><a href="#cb19-59" aria-hidden="true" tabindex="-1"></a>grm.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"GRMs/"</span></span>
<span id="cb19-60"><a href="#cb19-60" aria-hidden="true" tabindex="-1"></a>h2.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"h2_output/"</span></span>
<span id="cb19-61"><a href="#cb19-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-62"><a href="#cb19-62" aria-hidden="true" tabindex="-1"></a><span class="co">#Make local GRMs for each gene</span></span>
<span id="cb19-63"><a href="#cb19-63" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb19-64"><a href="#cb19-64" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb19-65"><a href="#cb19-65" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb19-66"><a href="#cb19-66" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb19-67"><a href="#cb19-67" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span>geneinfo[<span class="dv">1</span>]</span>
<span id="cb19-68"><a href="#cb19-68" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb19-69"><a href="#cb19-69" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb19-70"><a href="#cb19-70" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb19-71"><a href="#cb19-71" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bim,bim[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb19-72"><a href="#cb19-72" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">4</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">4</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb19-73"><a href="#cb19-73" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">2</span>]    </span>
<span id="cb19-74"><a href="#cb19-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Ac.SNPlist"</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">col.names=</span><span class="cn">FALSE</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb19-75"><a href="#cb19-75" aria-hidden="true" tabindex="-1"></a>    runGCTAgrm <span class="ot">&lt;-</span> <span class="st">"gcta --bfile "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Ac --make-grm-bin --extract "</span> <span class="sc">%&amp;%</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Ac.SNPlist"</span> <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span>  gene</span>
<span id="cb19-76"><a href="#cb19-76" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGCTAgrm)</span>
<span id="cb19-77"><a href="#cb19-77" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-78"><a href="#cb19-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-79"><a href="#cb19-79" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculate h2</span></span>
<span id="cb19-80"><a href="#cb19-80" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb19-81"><a href="#cb19-81" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"of"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb19-82"><a href="#cb19-82" aria-hidden="true" tabindex="-1"></a>    ensid <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb19-83"><a href="#cb19-83" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">10</span>])</span>
<span id="cb19-84"><a href="#cb19-84" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">1</span>])</span>
<span id="cb19-85"><a href="#cb19-85" aria-hidden="true" tabindex="-1"></a>  individuals <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexAc)[<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)]</span>
<span id="cb19-86"><a href="#cb19-86" aria-hidden="true" tabindex="-1"></a>  expression <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gexAc[i,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)])</span>
<span id="cb19-87"><a href="#cb19-87" aria-hidden="true" tabindex="-1"></a>    <span class="co">#output expression pheno for gcta</span></span>
<span id="cb19-88"><a href="#cb19-88" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb19-89"><a href="#cb19-89" aria-hidden="true" tabindex="-1"></a>      <span class="at">famid =</span> individuals,</span>
<span id="cb19-90"><a href="#cb19-90" aria-hidden="true" tabindex="-1"></a>      <span class="at">id =</span> individuals,</span>
<span id="cb19-91"><a href="#cb19-91" aria-hidden="true" tabindex="-1"></a>      <span class="at">expr =</span> expression</span>
<span id="cb19-92"><a href="#cb19-92" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-93"><a href="#cb19-93" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid, <span class="at">col.names=</span><span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>) <span class="co">#output pheno for gcta</span></span>
<span id="cb19-94"><a href="#cb19-94" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Y ~ localGRM</span></span>
<span id="cb19-95"><a href="#cb19-95" aria-hidden="true" tabindex="-1"></a>    runLOC <span class="ot">&lt;-</span> <span class="st">"gcta --grm "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --reml --pheno "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> h2.dir <span class="sc">%&amp;%</span> <span class="st">"tmp."</span> <span class="sc">%&amp;%</span> ensid</span>
<span id="cb19-96"><a href="#cb19-96" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runLOC)</span>
<span id="cb19-97"><a href="#cb19-97" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-98"><a href="#cb19-98" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-99"><a href="#cb19-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-100"><a href="#cb19-100" aria-hidden="true" tabindex="-1"></a>Repeat for Il:</span>
<span id="cb19-101"><a href="#cb19-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-102"><a href="#cb19-102" aria-hidden="true" tabindex="-1"></a><span class="in">```{r generate cis herit for Il, eval=FALSE}</span></span>
<span id="cb19-103"><a href="#cb19-103" aria-hidden="true" tabindex="-1"></a>gt.dir <span class="ot">&lt;-</span> <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/output/Il/"</span></span>
<span id="cb19-104"><a href="#cb19-104" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/"</span></span>
<span id="cb19-105"><a href="#cb19-105" aria-hidden="true" tabindex="-1"></a>grm.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"GRMs/"</span></span>
<span id="cb19-106"><a href="#cb19-106" aria-hidden="true" tabindex="-1"></a>h2.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"h2_output/"</span></span>
<span id="cb19-107"><a href="#cb19-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-108"><a href="#cb19-108" aria-hidden="true" tabindex="-1"></a><span class="co">#Make local GRMs for each gene</span></span>
<span id="cb19-109"><a href="#cb19-109" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Il)){</span>
<span id="cb19-110"><a href="#cb19-110" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist_Il),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb19-111"><a href="#cb19-111" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist_Il[i]</span>
<span id="cb19-112"><a href="#cb19-112" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb19-113"><a href="#cb19-113" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span>geneinfo[<span class="dv">1</span>]</span>
<span id="cb19-114"><a href="#cb19-114" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb19-115"><a href="#cb19-115" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb19-116"><a href="#cb19-116" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb19-117"><a href="#cb19-117" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bim_Il,bim_Il[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb19-118"><a href="#cb19-118" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">4</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">4</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb19-119"><a href="#cb19-119" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">2</span>]    </span>
<span id="cb19-120"><a href="#cb19-120" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Il.SNPlist"</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">col.names=</span><span class="cn">FALSE</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb19-121"><a href="#cb19-121" aria-hidden="true" tabindex="-1"></a>    runGCTAgrm <span class="ot">&lt;-</span> <span class="st">"gcta --bfile "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Il --make-grm-bin --extract "</span> <span class="sc">%&amp;%</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Il.SNPlist"</span> <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span>  gene</span>
<span id="cb19-122"><a href="#cb19-122" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGCTAgrm)</span>
<span id="cb19-123"><a href="#cb19-123" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-124"><a href="#cb19-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-125"><a href="#cb19-125" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculate h2</span></span>
<span id="cb19-126"><a href="#cb19-126" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Il)){</span>
<span id="cb19-127"><a href="#cb19-127" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"of"</span>,<span class="fu">length</span>(ensidlist_Il),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb19-128"><a href="#cb19-128" aria-hidden="true" tabindex="-1"></a>    ensid <span class="ot">&lt;-</span> ensidlist_Il[i]</span>
<span id="cb19-129"><a href="#cb19-129" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">10</span>])</span>
<span id="cb19-130"><a href="#cb19-130" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">1</span>])</span>
<span id="cb19-131"><a href="#cb19-131" aria-hidden="true" tabindex="-1"></a>  individuals <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexIl)[<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)]</span>
<span id="cb19-132"><a href="#cb19-132" aria-hidden="true" tabindex="-1"></a>  expression <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gexIl[i,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)])</span>
<span id="cb19-133"><a href="#cb19-133" aria-hidden="true" tabindex="-1"></a>    <span class="co">#output expression pheno for gcta</span></span>
<span id="cb19-134"><a href="#cb19-134" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb19-135"><a href="#cb19-135" aria-hidden="true" tabindex="-1"></a>      <span class="at">famid =</span> individuals,</span>
<span id="cb19-136"><a href="#cb19-136" aria-hidden="true" tabindex="-1"></a>      <span class="at">id =</span> individuals,</span>
<span id="cb19-137"><a href="#cb19-137" aria-hidden="true" tabindex="-1"></a>      <span class="at">expr =</span> expression</span>
<span id="cb19-138"><a href="#cb19-138" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-139"><a href="#cb19-139" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid, <span class="at">col.names=</span><span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>) <span class="co">#output pheno for gcta</span></span>
<span id="cb19-140"><a href="#cb19-140" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Y ~ localGRM</span></span>
<span id="cb19-141"><a href="#cb19-141" aria-hidden="true" tabindex="-1"></a>    runLOC <span class="ot">&lt;-</span> <span class="st">"gcta --grm "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --reml --pheno "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> h2.dir <span class="sc">%&amp;%</span> <span class="st">"tmp."</span> <span class="sc">%&amp;%</span> ensid</span>
<span id="cb19-142"><a href="#cb19-142" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runLOC)</span>
<span id="cb19-143"><a href="#cb19-143" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-144"><a href="#cb19-144" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-145"><a href="#cb19-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-146"><a href="#cb19-146" aria-hidden="true" tabindex="-1"></a>Lh:</span>
<span id="cb19-147"><a href="#cb19-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-148"><a href="#cb19-148" aria-hidden="true" tabindex="-1"></a><span class="in">```{r generate cis herit for Lh, eval=FALSE}</span></span>
<span id="cb19-149"><a href="#cb19-149" aria-hidden="true" tabindex="-1"></a>gt.dir <span class="ot">&lt;-</span> <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/output/Lh/"</span></span>
<span id="cb19-150"><a href="#cb19-150" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/"</span></span>
<span id="cb19-151"><a href="#cb19-151" aria-hidden="true" tabindex="-1"></a>grm.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"GRMs/"</span></span>
<span id="cb19-152"><a href="#cb19-152" aria-hidden="true" tabindex="-1"></a>h2.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"h2_output/"</span></span>
<span id="cb19-153"><a href="#cb19-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-154"><a href="#cb19-154" aria-hidden="true" tabindex="-1"></a><span class="co">#Make local GRMs for each gene</span></span>
<span id="cb19-155"><a href="#cb19-155" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Lh)){</span>
<span id="cb19-156"><a href="#cb19-156" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist_Lh),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb19-157"><a href="#cb19-157" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist_Lh[i]</span>
<span id="cb19-158"><a href="#cb19-158" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb19-159"><a href="#cb19-159" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span>geneinfo[<span class="dv">1</span>]</span>
<span id="cb19-160"><a href="#cb19-160" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb19-161"><a href="#cb19-161" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb19-162"><a href="#cb19-162" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb19-163"><a href="#cb19-163" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bim_Lh,bim_Lh[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb19-164"><a href="#cb19-164" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">4</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">4</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb19-165"><a href="#cb19-165" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">2</span>]    </span>
<span id="cb19-166"><a href="#cb19-166" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Lh.SNPlist"</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">col.names=</span><span class="cn">FALSE</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb19-167"><a href="#cb19-167" aria-hidden="true" tabindex="-1"></a>    runGCTAgrm <span class="ot">&lt;-</span> <span class="st">"gcta --bfile "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Lh --make-grm-bin --extract "</span> <span class="sc">%&amp;%</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Lh.SNPlist"</span> <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span>  gene</span>
<span id="cb19-168"><a href="#cb19-168" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGCTAgrm)</span>
<span id="cb19-169"><a href="#cb19-169" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-170"><a href="#cb19-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-171"><a href="#cb19-171" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculate h2</span></span>
<span id="cb19-172"><a href="#cb19-172" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Lh)){</span>
<span id="cb19-173"><a href="#cb19-173" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"of"</span>,<span class="fu">length</span>(ensidlist_Lh),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb19-174"><a href="#cb19-174" aria-hidden="true" tabindex="-1"></a>    ensid <span class="ot">&lt;-</span> ensidlist_Lh[i]</span>
<span id="cb19-175"><a href="#cb19-175" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">10</span>])</span>
<span id="cb19-176"><a href="#cb19-176" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">1</span>])</span>
<span id="cb19-177"><a href="#cb19-177" aria-hidden="true" tabindex="-1"></a>  individuals <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexLh)[<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)]</span>
<span id="cb19-178"><a href="#cb19-178" aria-hidden="true" tabindex="-1"></a>  expression <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gexLh[i,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)])</span>
<span id="cb19-179"><a href="#cb19-179" aria-hidden="true" tabindex="-1"></a>    <span class="co">#output expression pheno for gcta</span></span>
<span id="cb19-180"><a href="#cb19-180" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb19-181"><a href="#cb19-181" aria-hidden="true" tabindex="-1"></a>      <span class="at">famid =</span> individuals,</span>
<span id="cb19-182"><a href="#cb19-182" aria-hidden="true" tabindex="-1"></a>      <span class="at">id =</span> individuals,</span>
<span id="cb19-183"><a href="#cb19-183" aria-hidden="true" tabindex="-1"></a>      <span class="at">expr =</span> expression</span>
<span id="cb19-184"><a href="#cb19-184" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-185"><a href="#cb19-185" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid, <span class="at">col.names=</span><span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>) <span class="co">#output pheno for gcta</span></span>
<span id="cb19-186"><a href="#cb19-186" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Y ~ localGRM</span></span>
<span id="cb19-187"><a href="#cb19-187" aria-hidden="true" tabindex="-1"></a>    runLOC <span class="ot">&lt;-</span> <span class="st">"gcta --grm "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --reml --pheno "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> h2.dir <span class="sc">%&amp;%</span> <span class="st">"tmp."</span> <span class="sc">%&amp;%</span> ensid</span>
<span id="cb19-188"><a href="#cb19-188" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runLOC)</span>
<span id="cb19-189"><a href="#cb19-189" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-190"><a href="#cb19-190" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-191"><a href="#cb19-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-192"><a href="#cb19-192" aria-hidden="true" tabindex="-1"></a>Pl:</span>
<span id="cb19-193"><a href="#cb19-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-194"><a href="#cb19-194" aria-hidden="true" tabindex="-1"></a><span class="in">```{r generate cis herit for Pl, eval=FALSE}</span></span>
<span id="cb19-195"><a href="#cb19-195" aria-hidden="true" tabindex="-1"></a>gt.dir <span class="ot">&lt;-</span> <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/output/Pl/"</span></span>
<span id="cb19-196"><a href="#cb19-196" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/"</span></span>
<span id="cb19-197"><a href="#cb19-197" aria-hidden="true" tabindex="-1"></a>grm.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"GRMs/"</span></span>
<span id="cb19-198"><a href="#cb19-198" aria-hidden="true" tabindex="-1"></a>h2.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"h2_output/"</span></span>
<span id="cb19-199"><a href="#cb19-199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-200"><a href="#cb19-200" aria-hidden="true" tabindex="-1"></a><span class="co">#Make local GRMs for each gene</span></span>
<span id="cb19-201"><a href="#cb19-201" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Pl)){</span>
<span id="cb19-202"><a href="#cb19-202" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist_Pl),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb19-203"><a href="#cb19-203" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist_Pl[i]</span>
<span id="cb19-204"><a href="#cb19-204" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb19-205"><a href="#cb19-205" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span>geneinfo[<span class="dv">1</span>]</span>
<span id="cb19-206"><a href="#cb19-206" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb19-207"><a href="#cb19-207" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb19-208"><a href="#cb19-208" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb19-209"><a href="#cb19-209" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bim_Pl,bim_Pl[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb19-210"><a href="#cb19-210" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">4</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">4</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb19-211"><a href="#cb19-211" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">2</span>]    </span>
<span id="cb19-212"><a href="#cb19-212" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Pl.SNPlist"</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">col.names=</span><span class="cn">FALSE</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb19-213"><a href="#cb19-213" aria-hidden="true" tabindex="-1"></a>    runGCTAgrm <span class="ot">&lt;-</span> <span class="st">"gcta --bfile "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Pl --make-grm-bin --extract "</span> <span class="sc">%&amp;%</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Pl.SNPlist"</span> <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span>  gene</span>
<span id="cb19-214"><a href="#cb19-214" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGCTAgrm)</span>
<span id="cb19-215"><a href="#cb19-215" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-216"><a href="#cb19-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-217"><a href="#cb19-217" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculate h2</span></span>
<span id="cb19-218"><a href="#cb19-218" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Pl)){</span>
<span id="cb19-219"><a href="#cb19-219" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"of"</span>,<span class="fu">length</span>(ensidlist_Pl),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb19-220"><a href="#cb19-220" aria-hidden="true" tabindex="-1"></a>    ensid <span class="ot">&lt;-</span> ensidlist_Pl[i]</span>
<span id="cb19-221"><a href="#cb19-221" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">10</span>])</span>
<span id="cb19-222"><a href="#cb19-222" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">1</span>])</span>
<span id="cb19-223"><a href="#cb19-223" aria-hidden="true" tabindex="-1"></a>  individuals <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexPl)[<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)]</span>
<span id="cb19-224"><a href="#cb19-224" aria-hidden="true" tabindex="-1"></a>  expression <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gexPl[i,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)])</span>
<span id="cb19-225"><a href="#cb19-225" aria-hidden="true" tabindex="-1"></a>    <span class="co">#output expression pheno for gcta</span></span>
<span id="cb19-226"><a href="#cb19-226" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb19-227"><a href="#cb19-227" aria-hidden="true" tabindex="-1"></a>      <span class="at">famid =</span> individuals,</span>
<span id="cb19-228"><a href="#cb19-228" aria-hidden="true" tabindex="-1"></a>      <span class="at">id =</span> individuals,</span>
<span id="cb19-229"><a href="#cb19-229" aria-hidden="true" tabindex="-1"></a>      <span class="at">expr =</span> expression</span>
<span id="cb19-230"><a href="#cb19-230" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-231"><a href="#cb19-231" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid, <span class="at">col.names=</span><span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>) <span class="co">#output pheno for gcta</span></span>
<span id="cb19-232"><a href="#cb19-232" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Y ~ localGRM</span></span>
<span id="cb19-233"><a href="#cb19-233" aria-hidden="true" tabindex="-1"></a>    runLOC <span class="ot">&lt;-</span> <span class="st">"gcta --grm "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --reml --pheno "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> h2.dir <span class="sc">%&amp;%</span> <span class="st">"tmp."</span> <span class="sc">%&amp;%</span> ensid</span>
<span id="cb19-234"><a href="#cb19-234" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runLOC)</span>
<span id="cb19-235"><a href="#cb19-235" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-236"><a href="#cb19-236" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-237"><a href="#cb19-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-238"><a href="#cb19-238" aria-hidden="true" tabindex="-1"></a>Vo</span>
<span id="cb19-239"><a href="#cb19-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-240"><a href="#cb19-240" aria-hidden="true" tabindex="-1"></a><span class="in">```{r generate cis herit for Vo, eval=FALSE}</span></span>
<span id="cb19-241"><a href="#cb19-241" aria-hidden="true" tabindex="-1"></a>gt.dir <span class="ot">&lt;-</span> <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/output/Vo/"</span></span>
<span id="cb19-242"><a href="#cb19-242" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/"</span></span>
<span id="cb19-243"><a href="#cb19-243" aria-hidden="true" tabindex="-1"></a>grm.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"GRMs/"</span></span>
<span id="cb19-244"><a href="#cb19-244" aria-hidden="true" tabindex="-1"></a>h2.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"h2_output/"</span></span>
<span id="cb19-245"><a href="#cb19-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-246"><a href="#cb19-246" aria-hidden="true" tabindex="-1"></a><span class="co">#Make local GRMs for each gene</span></span>
<span id="cb19-247"><a href="#cb19-247" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Vo)){</span>
<span id="cb19-248"><a href="#cb19-248" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist_Vo),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb19-249"><a href="#cb19-249" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist_Vo[i]</span>
<span id="cb19-250"><a href="#cb19-250" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb19-251"><a href="#cb19-251" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span>geneinfo[<span class="dv">1</span>]</span>
<span id="cb19-252"><a href="#cb19-252" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb19-253"><a href="#cb19-253" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb19-254"><a href="#cb19-254" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb19-255"><a href="#cb19-255" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bim_Vo,bim_Vo[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb19-256"><a href="#cb19-256" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">4</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">4</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb19-257"><a href="#cb19-257" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">2</span>]    </span>
<span id="cb19-258"><a href="#cb19-258" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Vo.SNPlist"</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">col.names=</span><span class="cn">FALSE</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb19-259"><a href="#cb19-259" aria-hidden="true" tabindex="-1"></a>    runGCTAgrm <span class="ot">&lt;-</span> <span class="st">"gcta --bfile "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Vo --make-grm-bin --extract "</span> <span class="sc">%&amp;%</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Vo.SNPlist"</span> <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span>  gene</span>
<span id="cb19-260"><a href="#cb19-260" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGCTAgrm)</span>
<span id="cb19-261"><a href="#cb19-261" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-262"><a href="#cb19-262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-263"><a href="#cb19-263" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculate h2</span></span>
<span id="cb19-264"><a href="#cb19-264" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Vo)){</span>
<span id="cb19-265"><a href="#cb19-265" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"of"</span>,<span class="fu">length</span>(ensidlist_Vo),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb19-266"><a href="#cb19-266" aria-hidden="true" tabindex="-1"></a>    ensid <span class="ot">&lt;-</span> ensidlist_Vo[i]</span>
<span id="cb19-267"><a href="#cb19-267" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">10</span>])</span>
<span id="cb19-268"><a href="#cb19-268" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">1</span>])</span>
<span id="cb19-269"><a href="#cb19-269" aria-hidden="true" tabindex="-1"></a>  individuals <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexVo)[<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)]</span>
<span id="cb19-270"><a href="#cb19-270" aria-hidden="true" tabindex="-1"></a>  expression <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gexVo[i,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)])</span>
<span id="cb19-271"><a href="#cb19-271" aria-hidden="true" tabindex="-1"></a>    <span class="co">#output expression pheno for gcta</span></span>
<span id="cb19-272"><a href="#cb19-272" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb19-273"><a href="#cb19-273" aria-hidden="true" tabindex="-1"></a>      <span class="at">famid =</span> individuals,</span>
<span id="cb19-274"><a href="#cb19-274" aria-hidden="true" tabindex="-1"></a>      <span class="at">id =</span> individuals,</span>
<span id="cb19-275"><a href="#cb19-275" aria-hidden="true" tabindex="-1"></a>      <span class="at">expr =</span> expression</span>
<span id="cb19-276"><a href="#cb19-276" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-277"><a href="#cb19-277" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid, <span class="at">col.names=</span><span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>) <span class="co">#output pheno for gcta</span></span>
<span id="cb19-278"><a href="#cb19-278" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Y ~ localGRM</span></span>
<span id="cb19-279"><a href="#cb19-279" aria-hidden="true" tabindex="-1"></a>    runLOC <span class="ot">&lt;-</span> <span class="st">"gcta --grm "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --reml --pheno "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> h2.dir <span class="sc">%&amp;%</span> <span class="st">"tmp."</span> <span class="sc">%&amp;%</span> ensid</span>
<span id="cb19-280"><a href="#cb19-280" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runLOC)</span>
<span id="cb19-281"><a href="#cb19-281" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-282"><a href="#cb19-282" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-283"><a href="#cb19-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-284"><a href="#cb19-284" aria-hidden="true" tabindex="-1"></a><span class="fu"># Calculate Sparsity Estimates for all 5 tissues</span></span>
<span id="cb19-285"><a href="#cb19-285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-286"><a href="#cb19-286" aria-hidden="true" tabindex="-1"></a>The following code uses <span class="in">`bim_bam`</span> files generated through Palmer lab data, steps <span class="co">[</span><span class="ot">here</span><span class="co">](convert_genotypes_old.Rmd)</span>. The code</span>
<span id="cb19-287"><a href="#cb19-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-288"><a href="#cb19-288" aria-hidden="true" tabindex="-1"></a>Ac</span>
<span id="cb19-289"><a href="#cb19-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-290"><a href="#cb19-290" aria-hidden="true" tabindex="-1"></a><span class="in">```{r dir for Ac, eval=FALSE}</span></span>
<span id="cb19-291"><a href="#cb19-291" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/phenotype_files/"</span></span>
<span id="cb19-292"><a href="#cb19-292" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/genotype_files/"</span></span>
<span id="cb19-293"><a href="#cb19-293" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/bim_bam/"</span></span>
<span id="cb19-294"><a href="#cb19-294" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb19-295"><a href="#cb19-295" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Ac_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb19-296"><a href="#cb19-296" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span>
<span id="cb19-297"><a href="#cb19-297" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-298"><a href="#cb19-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-299"><a href="#cb19-299" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Make local GRMs for each gene for Ac, eval=FALSE}</span></span>
<span id="cb19-300"><a href="#cb19-300" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/"</span>)</span>
<span id="cb19-301"><a href="#cb19-301" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb19-302"><a href="#cb19-302" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb19-303"><a href="#cb19-303" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb19-304"><a href="#cb19-304" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb19-305"><a href="#cb19-305" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb19-306"><a href="#cb19-306" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb19-307"><a href="#cb19-307" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb19-308"><a href="#cb19-308" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb19-309"><a href="#cb19-309" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb19-310"><a href="#cb19-310" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb19-311"><a href="#cb19-311" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]</span>
<span id="cb19-312"><a href="#cb19-312" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Ac.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb19-313"><a href="#cb19-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-314"><a href="#cb19-314" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexAc_transpose[,i])</span>
<span id="cb19-315"><a href="#cb19-315" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb19-316"><a href="#cb19-316" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Ac.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Ac_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb19-317"><a href="#cb19-317" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb19-318"><a href="#cb19-318" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-319"><a href="#cb19-319" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-320"><a href="#cb19-320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-321"><a href="#cb19-321" aria-hidden="true" tabindex="-1"></a>Il</span>
<span id="cb19-322"><a href="#cb19-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-323"><a href="#cb19-323" aria-hidden="true" tabindex="-1"></a><span class="in">```{r dir for Il,eval=FALSE}</span></span>
<span id="cb19-324"><a href="#cb19-324" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/phenotype_files/"</span></span>
<span id="cb19-325"><a href="#cb19-325" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/genotype_files/"</span></span>
<span id="cb19-326"><a href="#cb19-326" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/bim_bam/"</span></span>
<span id="cb19-327"><a href="#cb19-327" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb19-328"><a href="#cb19-328" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Il_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb19-329"><a href="#cb19-329" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span>
<span id="cb19-330"><a href="#cb19-330" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-331"><a href="#cb19-331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-332"><a href="#cb19-332" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Make local GRMs for each gene for Il, eval=FALSE}</span></span>
<span id="cb19-333"><a href="#cb19-333" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/"</span>)</span>
<span id="cb19-334"><a href="#cb19-334" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb19-335"><a href="#cb19-335" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb19-336"><a href="#cb19-336" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb19-337"><a href="#cb19-337" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb19-338"><a href="#cb19-338" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb19-339"><a href="#cb19-339" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr </span>
<span id="cb19-340"><a href="#cb19-340" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb19-341"><a href="#cb19-341" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb19-342"><a href="#cb19-342" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb19-343"><a href="#cb19-343" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb19-344"><a href="#cb19-344" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]    </span>
<span id="cb19-345"><a href="#cb19-345" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Il.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb19-346"><a href="#cb19-346" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb19-347"><a href="#cb19-347" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexIl_transpose[,i])</span>
<span id="cb19-348"><a href="#cb19-348" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb19-349"><a href="#cb19-349" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Il.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Il_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb19-350"><a href="#cb19-350" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb19-351"><a href="#cb19-351" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-352"><a href="#cb19-352" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-353"><a href="#cb19-353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-354"><a href="#cb19-354" aria-hidden="true" tabindex="-1"></a>Lh</span>
<span id="cb19-355"><a href="#cb19-355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-356"><a href="#cb19-356" aria-hidden="true" tabindex="-1"></a><span class="in">```{r dir for Lh, eval=FALSE}</span></span>
<span id="cb19-357"><a href="#cb19-357" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/phenotype_files/"</span></span>
<span id="cb19-358"><a href="#cb19-358" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/genotype_files/"</span></span>
<span id="cb19-359"><a href="#cb19-359" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/bim_bam/"</span></span>
<span id="cb19-360"><a href="#cb19-360" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb19-361"><a href="#cb19-361" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Lh_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb19-362"><a href="#cb19-362" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span>
<span id="cb19-363"><a href="#cb19-363" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-364"><a href="#cb19-364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-365"><a href="#cb19-365" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Make local GRMs for each gene for Lh, eval=FALSE}</span></span>
<span id="cb19-366"><a href="#cb19-366" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/"</span>)</span>
<span id="cb19-367"><a href="#cb19-367" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb19-368"><a href="#cb19-368" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb19-369"><a href="#cb19-369" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb19-370"><a href="#cb19-370" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb19-371"><a href="#cb19-371" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb19-372"><a href="#cb19-372" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb19-373"><a href="#cb19-373" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb19-374"><a href="#cb19-374" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb19-375"><a href="#cb19-375" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb19-376"><a href="#cb19-376" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb19-377"><a href="#cb19-377" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]    </span>
<span id="cb19-378"><a href="#cb19-378" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Lh.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb19-379"><a href="#cb19-379" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb19-380"><a href="#cb19-380" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexLh_transpose[,i])</span>
<span id="cb19-381"><a href="#cb19-381" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb19-382"><a href="#cb19-382" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Lh.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Lh_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb19-383"><a href="#cb19-383" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb19-384"><a href="#cb19-384" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-385"><a href="#cb19-385" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-386"><a href="#cb19-386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-387"><a href="#cb19-387" aria-hidden="true" tabindex="-1"></a>Pl</span>
<span id="cb19-388"><a href="#cb19-388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-389"><a href="#cb19-389" aria-hidden="true" tabindex="-1"></a><span class="in">```{r dir for Pl, eval=FALSE}</span></span>
<span id="cb19-390"><a href="#cb19-390" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/phenotype_files/"</span></span>
<span id="cb19-391"><a href="#cb19-391" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/genotype_files/"</span></span>
<span id="cb19-392"><a href="#cb19-392" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/bim_bam/"</span></span>
<span id="cb19-393"><a href="#cb19-393" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb19-394"><a href="#cb19-394" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Pl_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb19-395"><a href="#cb19-395" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span>
<span id="cb19-396"><a href="#cb19-396" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-397"><a href="#cb19-397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-398"><a href="#cb19-398" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Make local GRMs for each gene for Pl, eval=FALSE}</span></span>
<span id="cb19-399"><a href="#cb19-399" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/"</span>)</span>
<span id="cb19-400"><a href="#cb19-400" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb19-401"><a href="#cb19-401" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb19-402"><a href="#cb19-402" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb19-403"><a href="#cb19-403" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb19-404"><a href="#cb19-404" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb19-405"><a href="#cb19-405" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb19-406"><a href="#cb19-406" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb19-407"><a href="#cb19-407" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb19-408"><a href="#cb19-408" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb19-409"><a href="#cb19-409" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb19-410"><a href="#cb19-410" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]    </span>
<span id="cb19-411"><a href="#cb19-411" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Pl.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb19-412"><a href="#cb19-412" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb19-413"><a href="#cb19-413" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexPl_transpose[,i])</span>
<span id="cb19-414"><a href="#cb19-414" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb19-415"><a href="#cb19-415" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Pl.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Pl_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb19-416"><a href="#cb19-416" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb19-417"><a href="#cb19-417" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-418"><a href="#cb19-418" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-419"><a href="#cb19-419" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-420"><a href="#cb19-420" aria-hidden="true" tabindex="-1"></a>Vo</span>
<span id="cb19-421"><a href="#cb19-421" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-422"><a href="#cb19-422" aria-hidden="true" tabindex="-1"></a><span class="in">```{r dir for Vo, eval=FALSE}</span></span>
<span id="cb19-423"><a href="#cb19-423" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/phenotype_files/"</span></span>
<span id="cb19-424"><a href="#cb19-424" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/genotype_files/"</span></span>
<span id="cb19-425"><a href="#cb19-425" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/bim_bam/"</span></span>
<span id="cb19-426"><a href="#cb19-426" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb19-427"><a href="#cb19-427" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Vo_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb19-428"><a href="#cb19-428" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span>
<span id="cb19-429"><a href="#cb19-429" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-430"><a href="#cb19-430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-431"><a href="#cb19-431" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Make local GRMs for each gene for Vo, eval=FALSE}</span></span>
<span id="cb19-432"><a href="#cb19-432" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/"</span>)</span>
<span id="cb19-433"><a href="#cb19-433" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb19-434"><a href="#cb19-434" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb19-435"><a href="#cb19-435" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb19-436"><a href="#cb19-436" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb19-437"><a href="#cb19-437" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb19-438"><a href="#cb19-438" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb19-439"><a href="#cb19-439" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb19-440"><a href="#cb19-440" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb19-441"><a href="#cb19-441" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb19-442"><a href="#cb19-442" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb19-443"><a href="#cb19-443" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]    </span>
<span id="cb19-444"><a href="#cb19-444" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Vo.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb19-445"><a href="#cb19-445" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb19-446"><a href="#cb19-446" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexVo_transpose[,i])</span>
<span id="cb19-447"><a href="#cb19-447" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb19-448"><a href="#cb19-448" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Vo.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Vo_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb19-449"><a href="#cb19-449" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb19-450"><a href="#cb19-450" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-451"><a href="#cb19-451" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>