<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="natasha.santhanam">
<meta name="dcterms.date" content="2022-02-07">

<title>Rat Genomics Pipeline Rendered - 01.Heritability_Sparsity</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-listing/list.min.js"></script>
<script src="../site_libs/quarto-listing/quarto-listing.js"></script>
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script>

  window.document.addEventListener("DOMContentLoaded", function (_event) {
    const listingTargetEl = window.document.querySelector('#listing-listing .list');
    if (!listingTargetEl) {
      // No listing discovered, do not attach.
      return; 
    }

    const options = {
      valueNames: [{ data: ['index'] },{ data: ['categories'] },{ data: ['listing-date-sort'] },{ data: ['listing-file-modified-sort'] }],
      
      searchColumns: [],
    };

    window['quarto-listings'] = window['quarto-listings'] || {};
    window['quarto-listings']['listing-listing'] = new List('listing-listing', options);

    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  });

  window.addEventListener('hashchange',() => {
    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  })
  </script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Rat Genomics Pipeline Rendered</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html" rel="" target="">
 <span class="menu-text">paper_code</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../other_code.html" rel="" target="">
 <span class="menu-text">more_code</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#generate-heritability-and-sparsity-estimates-for-all-5-tissues" id="toc-generate-heritability-and-sparsity-estimates-for-all-5-tissues" class="nav-link active" data-scroll-target="#generate-heritability-and-sparsity-estimates-for-all-5-tissues">Generate Heritability and Sparsity Estimates for all 5 tissues</a></li>
  <li><a href="#calculate-cis-heritability-within-1mb" id="toc-calculate-cis-heritability-within-1mb" class="nav-link" data-scroll-target="#calculate-cis-heritability-within-1mb">Calculate Cis Heritability within 1MB</a></li>
  <li><a href="#ac" id="toc-ac" class="nav-link" data-scroll-target="#ac">Ac</a></li>
  <li><a href="#il" id="toc-il" class="nav-link" data-scroll-target="#il">Il</a></li>
  <li><a href="#lh" id="toc-lh" class="nav-link" data-scroll-target="#lh">Lh</a></li>
  <li><a href="#pl" id="toc-pl" class="nav-link" data-scroll-target="#pl">Pl</a>
  <ul class="collapse">
  <li><a href="#sparsity-analysis" id="toc-sparsity-analysis" class="nav-link" data-scroll-target="#sparsity-analysis">Sparsity Analysis</a></li>
  </ul></li>
  <li><a href="#ac-1" id="toc-ac-1" class="nav-link" data-scroll-target="#ac-1">Ac</a></li>
  <li><a href="#il-1" id="toc-il-1" class="nav-link" data-scroll-target="#il-1">Il</a></li>
  <li><a href="#lh-1" id="toc-lh-1" class="nav-link" data-scroll-target="#lh-1">Lh</a></li>
  <li><a href="#pl-1" id="toc-pl-1" class="nav-link" data-scroll-target="#pl-1">Pl</a></li>
  <li><a href="#vo" id="toc-vo" class="nav-link" data-scroll-target="#vo">Vo</a></li>
  <li><a href="#generate-fitted-heritability-curves-with-r2" id="toc-generate-fitted-heritability-curves-with-r2" class="nav-link" data-scroll-target="#generate-fitted-heritability-curves-with-r2">Generate Fitted Heritability Curves with R2</a></li>
  <li><a href="#generate-analagous-plot-in-humans" id="toc-generate-analagous-plot-in-humans" class="nav-link" data-scroll-target="#generate-analagous-plot-in-humans">Generate Analagous Plot in Humans</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">01.Heritability_Sparsity</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>natasha.santhanam </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 7, 2022</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="generate-heritability-and-sparsity-estimates-for-all-5-tissues" class="level2">
<h2 class="anchored" data-anchor-id="generate-heritability-and-sparsity-estimates-for-all-5-tissues">Generate Heritability and Sparsity Estimates for all 5 tissues</h2>
</section>
<section id="calculate-cis-heritability-within-1mb" class="level2">
<h2 class="anchored" data-anchor-id="calculate-cis-heritability-within-1mb">Calculate Cis Heritability within 1MB</h2>
<p>First we create bimbam formats for genotypes from the original genotype file. The bimbam format is the input for gemma, which we will use for both heritability and sparisty estiamtes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"~/Box/imlab-data/Projects/PTRS-PGRS-Rosetta/Data-From-Abe-Palmer-Lab/Rdata/genoGex.RData"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="st">"%&amp;%"</span> <span class="ot">=</span> <span class="cf">function</span>(a,b) <span class="fu">paste</span>(a,b,<span class="at">sep=</span><span class="st">""</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>wd <span class="ot">&lt;-</span> <span class="st">"/Users/natashasanthanam/Github/rat-genomic-analysis/data/"</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>geno_Ac <span class="ot">=</span> geno[,<span class="fu">match</span>(<span class="fu">rownames</span>(gexAc_transpose), <span class="fu">colnames</span>(geno))]</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>geno_Il <span class="ot">=</span> geno[,<span class="fu">match</span>(<span class="fu">rownames</span>(gexIl_transpose), <span class="fu">colnames</span>(geno))]</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>geno_Lh <span class="ot">=</span> geno[,<span class="fu">match</span>(<span class="fu">rownames</span>(gexLh_transpose), <span class="fu">colnames</span>(geno))]</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>geno_Pl <span class="ot">=</span> geno[,<span class="fu">match</span>(<span class="fu">rownames</span>(gexPl_transpose), <span class="fu">colnames</span>(geno))]</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>geno_Vo <span class="ot">=</span> geno[,<span class="fu">match</span>(<span class="fu">rownames</span>(gexVo_transpose), <span class="fu">colnames</span>(geno))]</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>Ac_bimbam <span class="ot">&lt;-</span> <span class="fu">cbind</span>(phyMap<span class="sc">$</span>chr, phyMap<span class="sc">$</span>pos, <span class="fu">rownames</span>(geno_Ac), phyMap<span class="sc">$</span>refAllele, phyMap<span class="sc">$</span>effectAllele, geno_Ac)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>Il_bimbam <span class="ot">&lt;-</span> <span class="fu">cbind</span>(phyMap<span class="sc">$</span>chr, phyMap<span class="sc">$</span>pos, <span class="fu">rownames</span>(geno_Il),phyMap<span class="sc">$</span>refAllele, phyMap<span class="sc">$</span>effectAllele,  geno_Il)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>Lh_bimbam <span class="ot">&lt;-</span> <span class="fu">cbind</span>(phyMap<span class="sc">$</span>chr, phyMap<span class="sc">$</span>pos, <span class="fu">rownames</span>(geno_Lh),phyMap<span class="sc">$</span>refAllele, phyMap<span class="sc">$</span>effectAllele,  geno_Lh)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>Pl_bimbam <span class="ot">&lt;-</span> <span class="fu">cbind</span>(phyMap<span class="sc">$</span>chr, phyMap<span class="sc">$</span>pos, <span class="fu">rownames</span>(geno_Pl),phyMap<span class="sc">$</span>refAllele, phyMap<span class="sc">$</span>effectAllele,  geno_Pl)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>Vo_bimbam <span class="ot">&lt;-</span> <span class="fu">cbind</span>(phyMap<span class="sc">$</span>chr, phyMap<span class="sc">$</span>pos, <span class="fu">rownames</span>(geno_Vo),phyMap<span class="sc">$</span>refAllele, phyMap<span class="sc">$</span>effectAllele,  geno_Vo)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(Ac_bimbam, <span class="at">file =</span> wd <span class="sc">%&amp;%</span> <span class="st">"Ac_bimbam"</span>,<span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(Il_bimbam, <span class="at">file =</span> wd <span class="sc">%&amp;%</span> <span class="st">"Il_bimbam"</span>,<span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(Lh_bimbam, <span class="at">file =</span> wd <span class="sc">%&amp;%</span> <span class="st">"Lh_bimbam"</span>,<span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(Pl_bimbam, <span class="at">file =</span> wd <span class="sc">%&amp;%</span><span class="st">"Pl_bimbam"</span>,<span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(Vo_bimbam, <span class="at">file =</span> wd <span class="sc">%&amp;%</span><span class="st">"Vo_bimbam"</span>,<span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Collect list of individuals from the expression files</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>gtf <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/gtf.txt"</span>, <span class="at">col_names=</span><span class="cn">TRUE</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>gexAc_transpose <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/gexAc_transpose.txt"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>gexIl_transpose <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/gexIl_transpose.txt"</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>gexLh_transpose <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/gexLh_transpose.txt"</span>)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>gexPl_transpose <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/gexPl_transpose.txt"</span>)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>gexVo_transpose <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/gexVo_transpose.txt"</span>)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>ensidlist <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexAc_transpose)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>ensidlist_Il <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexIl_transpose)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>ensidlist_Lh <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexLh_transpose)</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>ensidlist_Pl <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexPl_transpose)</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>ensidlist_Vo <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexVo_transpose)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ac" class="level1">
<h1>Ac</h1>
<p>set directory</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/phenotype_files/"</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/genotype_files/"</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/bim_bam/"</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Ac_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Make local GRMs for each gene</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Ac.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexAc_transpose[,i])</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Ac.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Ac_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now we do the above process for the rest of tissues</p>
</section>
<section id="il" class="level1">
<h1>Il</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/phenotype_files/"</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/genotype_files/"</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/bim_bam/"</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Il_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Make local GRMs for each gene</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/"</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]    </span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Il.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexIl_transpose[,i])</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Il.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Il_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="lh" class="level1">
<h1>Lh</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/phenotype_files/"</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/genotype_files/"</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/bim_bam/"</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Lh_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Make local GRMs for each gene</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/"</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]    </span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Lh.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexLh_transpose[,i])</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Lh.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Lh_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="pl" class="level1">
<h1>Pl</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/phenotype_files/"</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/genotype_files/"</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/bim_bam/"</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Pl_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Make local GRMs for each gene</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/"</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]    </span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Pl.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexPl_transpose[,i])</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Pl.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Pl_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>#Vo</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/phenotype_files/"</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/genotype_files/"</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/bim_bam/"</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Vo_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Make local GRMs for each gene</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/"</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]    </span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Vo.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexVo_transpose[,i])</span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Vo.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Vo_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="sparsity-analysis" class="level2">
<h2 class="anchored" data-anchor-id="sparsity-analysis">Sparsity Analysis</h2>
<p>All the code above generates the local GRM for each phenotype (gene). With the GRM we then run gemma again to calculate both PVE (heritability) and PGE (sparsity). I used a badger template to calculate h2 and sparsity for each tissue. This steps takes a lot of computing power, so we use Badger. It takes approximatley 2-3 days to run.</p>
<p>The code to run badger is here</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"./Rat_Genomics_Paper_Pipeline/analysis/Sparsity_Badger_Template.Rmd"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>GEMMA then generates a .hyp file for each phenotype or in our case gene of interest. The hyp file contains the posterior samples for the hyper-parameters (h, PVE, rho, PGE, pi and gamma) for every 10th iteration. For our purposes, we are interested in the PGE and PVE parameters.</p>
<p>To then calculate the point estimate and credible set for Proportion of Variance Explained (PVE) and Proportion of genetic variance explained by the sparse effects terms (PGE), we calculate the posterior probability for each gene for each tissue.</p>
<p>I generated a function that calculates the beta of the posterior distribution and can be found <a href="https://gist.github.com/natashasanthanam/6eb539dcadecafe3a6dec448cedc7046">here</a></p>
</section>
</section>
<section id="ac-1" class="level1">
<h1>Ac</h1>
<p>Find point estimate and 95% credible interval in CRI for PVE</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/output"</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> ge.dir, <span class="at">pattern =</span> <span class="st">".hyp.txt"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>PVE_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="dv">0</span>, <span class="dv">4</span>)) </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(files[i],<span class="st">"/"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">11</span>), <span class="dv">8</span>, <span class="dv">25</span>)</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(files[i])</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pve, <span class="fl">0.1</span>)</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>  q2 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pve, <span class="fl">0.9</span>)</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>  quantile1<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">1</span> ,<span class="at">x=</span>q1)</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>  quantile2<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">9</span>, <span class="at">x=</span>q2)</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quantile1<span class="sc">$</span>x <span class="sc">!=</span> quantile2<span class="sc">$</span>x) {</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>  prior <span class="ot">&lt;-</span> <span class="fu">beta.select</span>(quantile1, quantile2)</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>  credible_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">qbeta</span>(<span class="fl">0.025</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]), <span class="fu">qbeta</span>(<span class="fl">0.975</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]))</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> gene</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.5</span>, prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>])</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">1</span>]</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">2</span>]</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> </span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PVE_df, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/PVE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> )</span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>PGE_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="dv">0</span>, <span class="dv">4</span>))</span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(files[i],<span class="st">"/"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">11</span>), <span class="dv">8</span>, <span class="dv">25</span>)</span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(files[i])</span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pge, <span class="fl">0.5</span>)</span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a>  q2 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pge, <span class="fl">0.9</span>)</span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a>  quantile1<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">5</span> ,<span class="at">x=</span>q1)</span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a>  quantile2<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">9</span>, <span class="at">x=</span>q2)</span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quantile1<span class="sc">$</span>x <span class="sc">!=</span> quantile2<span class="sc">$</span>x) {</span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a>  prior <span class="ot">&lt;-</span> <span class="fu">beta.select</span>(quantile1, quantile2)</span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a>  credible_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">qbeta</span>(<span class="fl">0.025</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]), <span class="fu">qbeta</span>(<span class="fl">0.975</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]))</span>
<span id="cb15-38"><a href="#cb15-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-39"><a href="#cb15-39" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> gene</span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.5</span>, prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>])</span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">1</span>]</span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">2</span>]</span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb15-44"><a href="#cb15-44" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> </span>
<span id="cb15-45"><a href="#cb15-45" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb15-46"><a href="#cb15-46" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-47"><a href="#cb15-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-48"><a href="#cb15-48" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PGE_df, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/PGE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="il-1" class="level1">
<h1>Il</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/output"</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> ge.dir, <span class="at">pattern =</span> <span class="st">".hyp.txt"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>PVE_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="dv">0</span>, <span class="dv">4</span>)) </span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(files[i],<span class="st">"/"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">11</span>), <span class="dv">8</span>, <span class="dv">25</span>)</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(files[i])</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pve, <span class="fl">0.1</span>)</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  q2 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pve, <span class="fl">0.9</span>)</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  quantile1<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">1</span> ,<span class="at">x=</span>q1)</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  quantile2<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">9</span>, <span class="at">x=</span>q2)</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quantile1<span class="sc">$</span>x <span class="sc">!=</span> quantile2<span class="sc">$</span>x) {</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>  prior <span class="ot">&lt;-</span> <span class="fu">beta.select</span>(quantile1, quantile2)</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>  credible_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">qbeta</span>(<span class="fl">0.025</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]), <span class="fu">qbeta</span>(<span class="fl">0.975</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]))</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> gene</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.5</span>, prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>])</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">1</span>]</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">2</span>]</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> </span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PVE_df, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/PVE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> )</span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>PGE_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="dv">0</span>, <span class="dv">4</span>))</span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(files[i],<span class="st">"/"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">11</span>), <span class="dv">8</span>, <span class="dv">25</span>)</span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(files[i])</span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pge, <span class="fl">0.5</span>)</span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a>  q2 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pge, <span class="fl">0.9</span>)</span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a>  quantile1<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">5</span> ,<span class="at">x=</span>q1)</span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a>  quantile2<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">9</span>, <span class="at">x=</span>q2)</span>
<span id="cb17-35"><a href="#cb17-35" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quantile1<span class="sc">$</span>x <span class="sc">!=</span> quantile2<span class="sc">$</span>x) {</span>
<span id="cb17-36"><a href="#cb17-36" aria-hidden="true" tabindex="-1"></a>  prior <span class="ot">&lt;-</span> <span class="fu">beta.select</span>(quantile1, quantile2)</span>
<span id="cb17-37"><a href="#cb17-37" aria-hidden="true" tabindex="-1"></a>  credible_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">qbeta</span>(<span class="fl">0.025</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]), <span class="fu">qbeta</span>(<span class="fl">0.975</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]))</span>
<span id="cb17-38"><a href="#cb17-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-39"><a href="#cb17-39" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> gene</span>
<span id="cb17-40"><a href="#cb17-40" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.5</span>, prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>])</span>
<span id="cb17-41"><a href="#cb17-41" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">1</span>]</span>
<span id="cb17-42"><a href="#cb17-42" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">2</span>]</span>
<span id="cb17-43"><a href="#cb17-43" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb17-44"><a href="#cb17-44" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> </span>
<span id="cb17-45"><a href="#cb17-45" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb17-46"><a href="#cb17-46" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-47"><a href="#cb17-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-48"><a href="#cb17-48" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PGE_df, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/PGE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="lh-1" class="level1">
<h1>Lh</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/output"</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> ge.dir, <span class="at">pattern =</span> <span class="st">".hyp.txt"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>PVE_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="dv">0</span>, <span class="dv">4</span>)) </span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(files[i],<span class="st">"/"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">11</span>), <span class="dv">8</span>, <span class="dv">25</span>)</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(files[i])</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pve, <span class="fl">0.1</span>)</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>  q2 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pve, <span class="fl">0.9</span>)</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  quantile1<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">1</span> ,<span class="at">x=</span>q1)</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  quantile2<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">9</span>, <span class="at">x=</span>q2)</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quantile1<span class="sc">$</span>x <span class="sc">!=</span> quantile2<span class="sc">$</span>x) {</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>  prior <span class="ot">&lt;-</span> <span class="fu">beta.select</span>(quantile1, quantile2)</span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>  credible_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">qbeta</span>(<span class="fl">0.025</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]), <span class="fu">qbeta</span>(<span class="fl">0.975</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]))</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> gene</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.5</span>, prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>])</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">1</span>]</span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">2</span>]</span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> </span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PVE_df, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/PVE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> )</span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true" tabindex="-1"></a>PGE_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="dv">0</span>, <span class="dv">4</span>))</span>
<span id="cb19-27"><a href="#cb19-27" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb19-28"><a href="#cb19-28" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(files[i],<span class="st">"/"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">11</span>), <span class="dv">8</span>, <span class="dv">25</span>)</span>
<span id="cb19-29"><a href="#cb19-29" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(files[i])</span>
<span id="cb19-30"><a href="#cb19-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-31"><a href="#cb19-31" aria-hidden="true" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pge, <span class="fl">0.5</span>)</span>
<span id="cb19-32"><a href="#cb19-32" aria-hidden="true" tabindex="-1"></a>  q2 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pge, <span class="fl">0.9</span>)</span>
<span id="cb19-33"><a href="#cb19-33" aria-hidden="true" tabindex="-1"></a>  quantile1<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">5</span> ,<span class="at">x=</span>q1)</span>
<span id="cb19-34"><a href="#cb19-34" aria-hidden="true" tabindex="-1"></a>  quantile2<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">9</span>, <span class="at">x=</span>q2)</span>
<span id="cb19-35"><a href="#cb19-35" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quantile1<span class="sc">$</span>x <span class="sc">!=</span> quantile2<span class="sc">$</span>x) {</span>
<span id="cb19-36"><a href="#cb19-36" aria-hidden="true" tabindex="-1"></a>  prior <span class="ot">&lt;-</span> <span class="fu">beta.select</span>(quantile1, quantile2)</span>
<span id="cb19-37"><a href="#cb19-37" aria-hidden="true" tabindex="-1"></a>  credible_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">qbeta</span>(<span class="fl">0.025</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]), <span class="fu">qbeta</span>(<span class="fl">0.975</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]))</span>
<span id="cb19-38"><a href="#cb19-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-39"><a href="#cb19-39" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> gene</span>
<span id="cb19-40"><a href="#cb19-40" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.5</span>, prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>])</span>
<span id="cb19-41"><a href="#cb19-41" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">1</span>]</span>
<span id="cb19-42"><a href="#cb19-42" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">2</span>]</span>
<span id="cb19-43"><a href="#cb19-43" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb19-44"><a href="#cb19-44" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> </span>
<span id="cb19-45"><a href="#cb19-45" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb19-46"><a href="#cb19-46" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-47"><a href="#cb19-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-48"><a href="#cb19-48" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PGE_df, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/PGE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="pl-1" class="level1">
<h1>Pl</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/output"</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> ge.dir, <span class="at">pattern =</span> <span class="st">".hyp.txt"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>PVE_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="dv">0</span>, <span class="dv">4</span>)) </span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(files[i],<span class="st">"/"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">11</span>), <span class="dv">8</span>, <span class="dv">25</span>)</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(files[i])</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pve, <span class="fl">0.1</span>)</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>  q2 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pve, <span class="fl">0.9</span>)</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>  quantile1<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">1</span> ,<span class="at">x=</span>q1)</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>  quantile2<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">9</span>, <span class="at">x=</span>q2)</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quantile1<span class="sc">$</span>x <span class="sc">!=</span> quantile2<span class="sc">$</span>x) {</span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>  prior <span class="ot">&lt;-</span> <span class="fu">beta.select</span>(quantile1, quantile2)</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>  credible_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">qbeta</span>(<span class="fl">0.025</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]), <span class="fu">qbeta</span>(<span class="fl">0.975</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]))</span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> gene</span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.5</span>, prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>])</span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">1</span>]</span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">2</span>]</span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> </span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PVE_df, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/PVE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> )</span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a>PGE_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="dv">0</span>, <span class="dv">4</span>))</span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(files[i],<span class="st">"/"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">11</span>), <span class="dv">8</span>, <span class="dv">25</span>)</span>
<span id="cb21-29"><a href="#cb21-29" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(files[i])</span>
<span id="cb21-30"><a href="#cb21-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb21-31"><a href="#cb21-31" aria-hidden="true" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pge, <span class="fl">0.5</span>)</span>
<span id="cb21-32"><a href="#cb21-32" aria-hidden="true" tabindex="-1"></a>  q2 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pge, <span class="fl">0.9</span>)</span>
<span id="cb21-33"><a href="#cb21-33" aria-hidden="true" tabindex="-1"></a>  quantile1<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">5</span> ,<span class="at">x=</span>q1)</span>
<span id="cb21-34"><a href="#cb21-34" aria-hidden="true" tabindex="-1"></a>  quantile2<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">9</span>, <span class="at">x=</span>q2)</span>
<span id="cb21-35"><a href="#cb21-35" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quantile1<span class="sc">$</span>x <span class="sc">!=</span> quantile2<span class="sc">$</span>x) {</span>
<span id="cb21-36"><a href="#cb21-36" aria-hidden="true" tabindex="-1"></a>  prior <span class="ot">&lt;-</span> <span class="fu">beta.select</span>(quantile1, quantile2)</span>
<span id="cb21-37"><a href="#cb21-37" aria-hidden="true" tabindex="-1"></a>  credible_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">qbeta</span>(<span class="fl">0.025</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]), <span class="fu">qbeta</span>(<span class="fl">0.975</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]))</span>
<span id="cb21-38"><a href="#cb21-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb21-39"><a href="#cb21-39" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> gene</span>
<span id="cb21-40"><a href="#cb21-40" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.5</span>, prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>])</span>
<span id="cb21-41"><a href="#cb21-41" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">1</span>]</span>
<span id="cb21-42"><a href="#cb21-42" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">2</span>]</span>
<span id="cb21-43"><a href="#cb21-43" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb21-44"><a href="#cb21-44" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> </span>
<span id="cb21-45"><a href="#cb21-45" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb21-46"><a href="#cb21-46" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb21-47"><a href="#cb21-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-48"><a href="#cb21-48" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PGE_df, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/PGE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="vo" class="level1">
<h1>Vo</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/output"</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> ge.dir, <span class="at">pattern =</span> <span class="st">".hyp.txt"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>PVE_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="dv">0</span>, <span class="dv">4</span>)) </span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(files[i],<span class="st">"/"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">11</span>), <span class="dv">8</span>, <span class="dv">25</span>)</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(files[i])</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pve, <span class="fl">0.1</span>)</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>  q2 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pve, <span class="fl">0.9</span>)</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>  quantile1<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">1</span> ,<span class="at">x=</span>q1)</span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>  quantile2<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">9</span>, <span class="at">x=</span>q2)</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quantile1<span class="sc">$</span>x <span class="sc">!=</span> quantile2<span class="sc">$</span>x) {</span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>  prior <span class="ot">&lt;-</span> <span class="fu">beta.select</span>(quantile1, quantile2)</span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>  credible_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">qbeta</span>(<span class="fl">0.025</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]), <span class="fu">qbeta</span>(<span class="fl">0.975</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]))</span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> gene</span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.5</span>, prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>])</span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">1</span>]</span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">2</span>]</span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> </span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb23-22"><a href="#cb23-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb23-23"><a href="#cb23-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-24"><a href="#cb23-24" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PVE_df, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/PVE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> )</span>
<span id="cb23-25"><a href="#cb23-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-26"><a href="#cb23-26" aria-hidden="true" tabindex="-1"></a>PGE_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="dv">0</span>, <span class="dv">4</span>))</span>
<span id="cb23-27"><a href="#cb23-27" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb23-28"><a href="#cb23-28" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(files[i],<span class="st">"/"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">11</span>), <span class="dv">8</span>, <span class="dv">25</span>)</span>
<span id="cb23-29"><a href="#cb23-29" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(files[i])</span>
<span id="cb23-30"><a href="#cb23-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-31"><a href="#cb23-31" aria-hidden="true" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pge, <span class="fl">0.5</span>)</span>
<span id="cb23-32"><a href="#cb23-32" aria-hidden="true" tabindex="-1"></a>  q2 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pge, <span class="fl">0.9</span>)</span>
<span id="cb23-33"><a href="#cb23-33" aria-hidden="true" tabindex="-1"></a>  quantile1<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">5</span> ,<span class="at">x=</span>q1)</span>
<span id="cb23-34"><a href="#cb23-34" aria-hidden="true" tabindex="-1"></a>  quantile2<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">9</span>, <span class="at">x=</span>q2)</span>
<span id="cb23-35"><a href="#cb23-35" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quantile1<span class="sc">$</span>x <span class="sc">!=</span> quantile2<span class="sc">$</span>x) {</span>
<span id="cb23-36"><a href="#cb23-36" aria-hidden="true" tabindex="-1"></a>  prior <span class="ot">&lt;-</span> <span class="fu">beta.select</span>(quantile1, quantile2)</span>
<span id="cb23-37"><a href="#cb23-37" aria-hidden="true" tabindex="-1"></a>  credible_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">qbeta</span>(<span class="fl">0.025</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]), <span class="fu">qbeta</span>(<span class="fl">0.975</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]))</span>
<span id="cb23-38"><a href="#cb23-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-39"><a href="#cb23-39" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> gene</span>
<span id="cb23-40"><a href="#cb23-40" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.5</span>, prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>])</span>
<span id="cb23-41"><a href="#cb23-41" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">1</span>]</span>
<span id="cb23-42"><a href="#cb23-42" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">2</span>]</span>
<span id="cb23-43"><a href="#cb23-43" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb23-44"><a href="#cb23-44" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> </span>
<span id="cb23-45"><a href="#cb23-45" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb23-46"><a href="#cb23-46" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb23-47"><a href="#cb23-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-48"><a href="#cb23-48" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PGE_df, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/PGE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now we have PVE and PGE estimates for each tissue and can generate the fitted curve figure for Heritability.</p>
</section>
<section id="generate-fitted-heritability-curves-with-r2" class="level1">
<h1>Generate Fitted Heritability Curves with R2</h1>
<p>I’m showing the curve for Ac tissue but I generated the one for all other tissues as well.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>))</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>Data <span class="ot">&lt;-</span> <span class="st">"/Users/natashasanthanam/Box/imlab-data/Projects/PTRS-PGRS-Rosetta/Rat-Genomics/Tyson-PalmerLab_PredictDB_Results/sql/"</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>poly_dir <span class="ot">&lt;-</span> <span class="st">"/Users/natashasanthanam/Github/rat-genomic-analysis/data/"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>load_pve <span class="ot">&lt;-</span> <span class="cf">function</span>(df){</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> df[<span class="fu">order</span>(df<span class="sc">$</span>point_estimate),]</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  df<span class="sc">$</span>index <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(df)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>First we get the R2 for the tissue from the model we generated</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>filename <span class="ot">&lt;-</span> Data <span class="sc">%&amp;%</span> <span class="st">"Ac_output_db.db"</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>sqlite.driver <span class="ot">&lt;-</span> <span class="fu">dbDriver</span>(<span class="st">"SQLite"</span>)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>conn <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(), filename)</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>extra <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>(conn, <span class="st">'select * from extra'</span>)</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a><span class="fu">dbDisconnect</span>(conn)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Next we create the ordered plot for heritability for Ac. We use PVE as h2, the proportion of variance in phenotypes explained by typed genotypes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>human_h2 <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"/Users/natashasanthanam/Github/rat-genomic-analysis/data/human_PVE.csv"</span>) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">point_estimate =</span> pve) </span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>PVE_Ac <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(poly_dir <span class="sc">%&amp;%</span> <span class="st">"Ac_PVE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>human_h2<span class="sc">$</span>ensid <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(human_h2<span class="sc">$</span>ensid, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(PVE_Ac) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"gene"</span>, <span class="st">"point_estimate"</span>, <span class="st">"credible_set_1"</span>, <span class="st">"credible_set_2"</span>)</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>overlap <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">rat_genes =</span> PVE_Ac<span class="sc">$</span>gene, <span class="at">ensid =</span> orth.rats[<span class="fu">match</span>(PVE_Ac<span class="sc">$</span>gene, orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene), <span class="dv">1</span>]<span class="sc">$</span>ensembl_gene_id) </span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>overlap <span class="ot">&lt;-</span> overlap[<span class="fu">na.omit</span>(<span class="fu">match</span>(human_h2<span class="sc">$</span>ensid, overlap<span class="sc">$</span>ensid)), ]</span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>PVE_Ac <span class="ot">&lt;-</span> PVE_Ac <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene <span class="sc">%in%</span> overlap<span class="sc">$</span>rat_genes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>PVE_Ac <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(PVE_Ac, extra, <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>A_df_Ac <span class="ot">&lt;-</span> <span class="fu">load_pve</span>(PVE_Ac)</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>plt_1 <span class="ot">&lt;-</span> (<span class="fu">ggplot</span>(<span class="at">data =</span> A_df_Ac, <span class="fu">aes</span>(<span class="at">x =</span> index))</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>index, <span class="at">y=</span>R2), <span class="at">colour =</span> <span class="st">"dodgerblue2"</span>, <span class="at">size =</span> <span class="fl">0.2</span>)</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> point_estimate), <span class="at">lwd=</span><span class="fl">1.5</span>)</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype=</span><span class="dv">2</span>)</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">geom_ribbon</span>(<span class="fu">aes</span>(<span class="at">ymax =</span> credible_set_2, <span class="at">ymin =</span> credible_set_1),</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>                         <span class="at">alpha =</span> <span class="fl">0.25</span>)</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">'Genes Sorted by Proportion of Variance Explained (PVE)'</span>,</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y =</span> <span class="st">'PVE'</span>)</span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>           <span class="sc">+</span> <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>))</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>           <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">1250</span>, <span class="at">y =</span> <span class="fl">0.9</span>, <span class="at">label =</span> <span class="st">"Mean h2 =  0.098"</span>, <span class="at">size =</span> <span class="dv">6</span>)</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>           <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">1250</span>, <span class="at">y =</span> <span class="fl">0.8</span>, <span class="at">label =</span> <span class="st">"Mean R2 =  0.085"</span>, <span class="at">size =</span> <span class="dv">6</span>)) </span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>plt_1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="generate-analagous-plot-in-humans" class="level1">
<h1>Generate Analagous Plot in Humans</h1>
<p>Get R2 and H2 from Heather’s Models for Brain</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode sql code-with-copy"><code class="sourceCode sql"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>sqlite3 genarch.db</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>.headers <span class="kw">on</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>.<span class="kw">mode</span> csv</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>.output human_PVE.csv</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="kw">select</span> gene, ensid, en_r2 ,pve,pve_CI <span class="kw">from</span> results <span class="kw">where</span> tissue <span class="op">=</span> <span class="ot">"DGN-WB"</span>;</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>.quit</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>human_h2<span class="sc">$</span>credible_set_1 <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(<span class="fu">substr</span>(human_h2<span class="sc">$</span>pve_CI ,<span class="dv">1</span>,<span class="dv">1</span>) <span class="sc">==</span> <span class="st">"-"</span>, <span class="fu">paste</span>(<span class="st">"-"</span>, <span class="fu">sapply</span>(<span class="fu">strsplit</span>(human_h2<span class="sc">$</span>pve_CI, <span class="st">"-"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">2</span>), <span class="at">sep =</span> <span class="st">""</span>), <span class="fu">sapply</span>(<span class="fu">strsplit</span>(human_h2<span class="sc">$</span>pve_CI, <span class="st">"-"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)))</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>human_h2<span class="sc">$</span>credible_set_2 <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(<span class="fu">substr</span>(human_h2<span class="sc">$</span>pve_CI ,<span class="dv">1</span>,<span class="dv">1</span>) <span class="sc">==</span> <span class="st">"-"</span>, <span class="fu">sapply</span>(<span class="fu">strsplit</span>(human_h2<span class="sc">$</span>pve_CI, <span class="st">"-"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">3</span>), <span class="fu">sapply</span>(<span class="fu">strsplit</span>(human_h2<span class="sc">$</span>pve_CI, <span class="st">"-"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">2</span>)))</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>human_h2 <span class="ot">&lt;-</span> human_h2 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ensid <span class="sc">%in%</span> overlap<span class="sc">$</span>ensid)</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>hum_df <span class="ot">&lt;-</span> <span class="fu">load_pve</span>(human_h2)</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>plt_2  <span class="ot">&lt;-</span> (<span class="fu">ggplot</span>(<span class="at">data =</span> hum_df, <span class="fu">aes</span>(<span class="at">x =</span> index))</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>index, <span class="at">y=</span>en_r2), <span class="at">colour =</span> <span class="st">"maroon1"</span>, <span class="at">size =</span> <span class="fl">0.2</span>)</span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> point_estimate), <span class="at">lwd =</span> <span class="fl">1.5</span>)</span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype=</span><span class="dv">2</span>)</span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">geom_ribbon</span>(<span class="fu">aes</span>(<span class="at">ymax =</span> credible_set_2, <span class="at">ymin =</span> credible_set_1),</span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>                         <span class="at">alpha =</span> <span class="fl">0.25</span>)</span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">'Genes Sorted by Proportion of Variance Explained (PVE)'</span>,</span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y =</span> <span class="st">'PVE'</span>)</span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>))</span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>           <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">1690</span>, <span class="at">y =</span> <span class="fl">0.9</span>, <span class="at">label =</span> <span class="st">"Mean h2 =  0.124"</span>, <span class="at">size =</span> <span class="dv">6</span>)</span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>           <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">1710</span>, <span class="at">y =</span> <span class="fl">0.8</span>, <span class="at">label =</span> <span class="st">"Mean R2 =  0.114"</span>, <span class="at">size =</span> <span class="dv">6</span>))</span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>plt_2 </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>all_h2 <span class="ot">&lt;-</span> <span class="fu">tibble</span>(gene)</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(l <span class="cf">in</span> all) {</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  l <span class="ot">&lt;-</span> l <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, point_estimate))</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>  all_h2 <span class="ot">&lt;-</span> <span class="fu">full_join</span>(all_h2, l, <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>



<!-- -->

</section>

<div class="quarto-listing quarto-listing-container-default" id="listing-listing">
<div class="list quarto-listing-default">

</div>
<div class="listing-no-matching d-none">
No matching items
</div>
</div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb32" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "01.Heritability_Sparsity"</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "natasha.santhanam"</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2022-02-07"</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a><span class="an">output:</span><span class="co"> workflowr::wflow_html</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a><span class="an">editor_options:</span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a><span class="an">chunk_output_type:</span><span class="co"> console</span></span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a><span class="fu">## Generate Heritability and Sparsity Estimates for all 5 tissues</span></span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a><span class="fu">## Calculate Cis Heritability within 1MB</span></span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a>First we create bimbam formats for genotypes from the original genotype file. The bimbam format is the input for gemma, which we will use for both heritability and sparisty estiamtes.</span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a><span class="in">```{r make bimbam, eval=FALSE}</span></span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"~/Box/imlab-data/Projects/PTRS-PGRS-Rosetta/Data-From-Abe-Palmer-Lab/Rdata/genoGex.RData"</span>)</span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true" tabindex="-1"></a><span class="st">"%&amp;%"</span> <span class="ot">=</span> <span class="cf">function</span>(a,b) <span class="fu">paste</span>(a,b,<span class="at">sep=</span><span class="st">""</span>)</span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true" tabindex="-1"></a>wd <span class="ot">&lt;-</span> <span class="st">"/Users/natashasanthanam/Github/rat-genomic-analysis/data/"</span></span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true" tabindex="-1"></a>geno_Ac <span class="ot">=</span> geno[,<span class="fu">match</span>(<span class="fu">rownames</span>(gexAc_transpose), <span class="fu">colnames</span>(geno))]</span>
<span id="cb32-23"><a href="#cb32-23" aria-hidden="true" tabindex="-1"></a>geno_Il <span class="ot">=</span> geno[,<span class="fu">match</span>(<span class="fu">rownames</span>(gexIl_transpose), <span class="fu">colnames</span>(geno))]</span>
<span id="cb32-24"><a href="#cb32-24" aria-hidden="true" tabindex="-1"></a>geno_Lh <span class="ot">=</span> geno[,<span class="fu">match</span>(<span class="fu">rownames</span>(gexLh_transpose), <span class="fu">colnames</span>(geno))]</span>
<span id="cb32-25"><a href="#cb32-25" aria-hidden="true" tabindex="-1"></a>geno_Pl <span class="ot">=</span> geno[,<span class="fu">match</span>(<span class="fu">rownames</span>(gexPl_transpose), <span class="fu">colnames</span>(geno))]</span>
<span id="cb32-26"><a href="#cb32-26" aria-hidden="true" tabindex="-1"></a>geno_Vo <span class="ot">=</span> geno[,<span class="fu">match</span>(<span class="fu">rownames</span>(gexVo_transpose), <span class="fu">colnames</span>(geno))]</span>
<span id="cb32-27"><a href="#cb32-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-28"><a href="#cb32-28" aria-hidden="true" tabindex="-1"></a>Ac_bimbam <span class="ot">&lt;-</span> <span class="fu">cbind</span>(phyMap<span class="sc">$</span>chr, phyMap<span class="sc">$</span>pos, <span class="fu">rownames</span>(geno_Ac), phyMap<span class="sc">$</span>refAllele, phyMap<span class="sc">$</span>effectAllele, geno_Ac)</span>
<span id="cb32-29"><a href="#cb32-29" aria-hidden="true" tabindex="-1"></a>Il_bimbam <span class="ot">&lt;-</span> <span class="fu">cbind</span>(phyMap<span class="sc">$</span>chr, phyMap<span class="sc">$</span>pos, <span class="fu">rownames</span>(geno_Il),phyMap<span class="sc">$</span>refAllele, phyMap<span class="sc">$</span>effectAllele,  geno_Il)</span>
<span id="cb32-30"><a href="#cb32-30" aria-hidden="true" tabindex="-1"></a>Lh_bimbam <span class="ot">&lt;-</span> <span class="fu">cbind</span>(phyMap<span class="sc">$</span>chr, phyMap<span class="sc">$</span>pos, <span class="fu">rownames</span>(geno_Lh),phyMap<span class="sc">$</span>refAllele, phyMap<span class="sc">$</span>effectAllele,  geno_Lh)</span>
<span id="cb32-31"><a href="#cb32-31" aria-hidden="true" tabindex="-1"></a>Pl_bimbam <span class="ot">&lt;-</span> <span class="fu">cbind</span>(phyMap<span class="sc">$</span>chr, phyMap<span class="sc">$</span>pos, <span class="fu">rownames</span>(geno_Pl),phyMap<span class="sc">$</span>refAllele, phyMap<span class="sc">$</span>effectAllele,  geno_Pl)</span>
<span id="cb32-32"><a href="#cb32-32" aria-hidden="true" tabindex="-1"></a>Vo_bimbam <span class="ot">&lt;-</span> <span class="fu">cbind</span>(phyMap<span class="sc">$</span>chr, phyMap<span class="sc">$</span>pos, <span class="fu">rownames</span>(geno_Vo),phyMap<span class="sc">$</span>refAllele, phyMap<span class="sc">$</span>effectAllele,  geno_Vo)</span>
<span id="cb32-33"><a href="#cb32-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-34"><a href="#cb32-34" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(Ac_bimbam, <span class="at">file =</span> wd <span class="sc">%&amp;%</span> <span class="st">"Ac_bimbam"</span>,<span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb32-35"><a href="#cb32-35" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(Il_bimbam, <span class="at">file =</span> wd <span class="sc">%&amp;%</span> <span class="st">"Il_bimbam"</span>,<span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb32-36"><a href="#cb32-36" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(Lh_bimbam, <span class="at">file =</span> wd <span class="sc">%&amp;%</span> <span class="st">"Lh_bimbam"</span>,<span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb32-37"><a href="#cb32-37" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(Pl_bimbam, <span class="at">file =</span> wd <span class="sc">%&amp;%</span><span class="st">"Pl_bimbam"</span>,<span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb32-38"><a href="#cb32-38" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(Vo_bimbam, <span class="at">file =</span> wd <span class="sc">%&amp;%</span><span class="st">"Vo_bimbam"</span>,<span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb32-39"><a href="#cb32-39" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-40"><a href="#cb32-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-41"><a href="#cb32-41" aria-hidden="true" tabindex="-1"></a>Collect list of individuals from the expression files</span>
<span id="cb32-42"><a href="#cb32-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-43"><a href="#cb32-43" aria-hidden="true" tabindex="-1"></a><span class="in">```{r get ids of rats, eval=FALSE}</span></span>
<span id="cb32-44"><a href="#cb32-44" aria-hidden="true" tabindex="-1"></a>gtf <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/gtf.txt"</span>, <span class="at">col_names=</span><span class="cn">TRUE</span>)</span>
<span id="cb32-45"><a href="#cb32-45" aria-hidden="true" tabindex="-1"></a>gexAc_transpose <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/gexAc_transpose.txt"</span>)</span>
<span id="cb32-46"><a href="#cb32-46" aria-hidden="true" tabindex="-1"></a>gexIl_transpose <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/gexIl_transpose.txt"</span>)</span>
<span id="cb32-47"><a href="#cb32-47" aria-hidden="true" tabindex="-1"></a>gexLh_transpose <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/gexLh_transpose.txt"</span>)</span>
<span id="cb32-48"><a href="#cb32-48" aria-hidden="true" tabindex="-1"></a>gexPl_transpose <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/gexPl_transpose.txt"</span>)</span>
<span id="cb32-49"><a href="#cb32-49" aria-hidden="true" tabindex="-1"></a>gexVo_transpose <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/gexVo_transpose.txt"</span>)</span>
<span id="cb32-50"><a href="#cb32-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-51"><a href="#cb32-51" aria-hidden="true" tabindex="-1"></a>ensidlist <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexAc_transpose)</span>
<span id="cb32-52"><a href="#cb32-52" aria-hidden="true" tabindex="-1"></a>ensidlist_Il <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexIl_transpose)</span>
<span id="cb32-53"><a href="#cb32-53" aria-hidden="true" tabindex="-1"></a>ensidlist_Lh <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexLh_transpose)</span>
<span id="cb32-54"><a href="#cb32-54" aria-hidden="true" tabindex="-1"></a>ensidlist_Pl <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexPl_transpose)</span>
<span id="cb32-55"><a href="#cb32-55" aria-hidden="true" tabindex="-1"></a>ensidlist_Vo <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexVo_transpose)</span>
<span id="cb32-56"><a href="#cb32-56" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-57"><a href="#cb32-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-58"><a href="#cb32-58" aria-hidden="true" tabindex="-1"></a><span class="fu"># Ac</span></span>
<span id="cb32-59"><a href="#cb32-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-60"><a href="#cb32-60" aria-hidden="true" tabindex="-1"></a>set directory</span>
<span id="cb32-61"><a href="#cb32-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-62"><a href="#cb32-62" aria-hidden="true" tabindex="-1"></a><span class="in">```{r set dir, eval=FALSE}</span></span>
<span id="cb32-63"><a href="#cb32-63" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/phenotype_files/"</span></span>
<span id="cb32-64"><a href="#cb32-64" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/genotype_files/"</span></span>
<span id="cb32-65"><a href="#cb32-65" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/bim_bam/"</span></span>
<span id="cb32-66"><a href="#cb32-66" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb32-67"><a href="#cb32-67" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Ac_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb32-68"><a href="#cb32-68" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span>
<span id="cb32-69"><a href="#cb32-69" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-70"><a href="#cb32-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-71"><a href="#cb32-71" aria-hidden="true" tabindex="-1"></a>Make local GRMs for each gene</span>
<span id="cb32-72"><a href="#cb32-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-73"><a href="#cb32-73" aria-hidden="true" tabindex="-1"></a><span class="in">```{r generate grm, eval=FALSE }</span></span>
<span id="cb32-74"><a href="#cb32-74" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/"</span>)</span>
<span id="cb32-75"><a href="#cb32-75" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb32-76"><a href="#cb32-76" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb32-77"><a href="#cb32-77" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb32-78"><a href="#cb32-78" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb32-79"><a href="#cb32-79" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb32-80"><a href="#cb32-80" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb32-81"><a href="#cb32-81" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb32-82"><a href="#cb32-82" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb32-83"><a href="#cb32-83" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb32-84"><a href="#cb32-84" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb32-85"><a href="#cb32-85" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]</span>
<span id="cb32-86"><a href="#cb32-86" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Ac.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb32-87"><a href="#cb32-87" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb32-88"><a href="#cb32-88" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexAc_transpose[,i])</span>
<span id="cb32-89"><a href="#cb32-89" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb32-90"><a href="#cb32-90" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Ac.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Ac_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb32-91"><a href="#cb32-91" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb32-92"><a href="#cb32-92" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-93"><a href="#cb32-93" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-94"><a href="#cb32-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-95"><a href="#cb32-95" aria-hidden="true" tabindex="-1"></a>Now we do the above process for the rest of tissues</span>
<span id="cb32-96"><a href="#cb32-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-97"><a href="#cb32-97" aria-hidden="true" tabindex="-1"></a><span class="fu"># Il</span></span>
<span id="cb32-98"><a href="#cb32-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-99"><a href="#cb32-99" aria-hidden="true" tabindex="-1"></a><span class="in">```{r set Il dir , eval=FALSE}</span></span>
<span id="cb32-100"><a href="#cb32-100" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/phenotype_files/"</span></span>
<span id="cb32-101"><a href="#cb32-101" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/genotype_files/"</span></span>
<span id="cb32-102"><a href="#cb32-102" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/bim_bam/"</span></span>
<span id="cb32-103"><a href="#cb32-103" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb32-104"><a href="#cb32-104" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Il_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb32-105"><a href="#cb32-105" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span>
<span id="cb32-106"><a href="#cb32-106" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-107"><a href="#cb32-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-108"><a href="#cb32-108" aria-hidden="true" tabindex="-1"></a>Make local GRMs for each gene</span>
<span id="cb32-109"><a href="#cb32-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-110"><a href="#cb32-110" aria-hidden="true" tabindex="-1"></a><span class="in">```{r create local grm, eval=FALSE}</span></span>
<span id="cb32-111"><a href="#cb32-111" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/"</span>)</span>
<span id="cb32-112"><a href="#cb32-112" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb32-113"><a href="#cb32-113" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb32-114"><a href="#cb32-114" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb32-115"><a href="#cb32-115" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb32-116"><a href="#cb32-116" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb32-117"><a href="#cb32-117" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb32-118"><a href="#cb32-118" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb32-119"><a href="#cb32-119" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb32-120"><a href="#cb32-120" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb32-121"><a href="#cb32-121" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb32-122"><a href="#cb32-122" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]    </span>
<span id="cb32-123"><a href="#cb32-123" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Il.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb32-124"><a href="#cb32-124" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb32-125"><a href="#cb32-125" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexIl_transpose[,i])</span>
<span id="cb32-126"><a href="#cb32-126" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb32-127"><a href="#cb32-127" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Il.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Il_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb32-128"><a href="#cb32-128" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb32-129"><a href="#cb32-129" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-130"><a href="#cb32-130" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-131"><a href="#cb32-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-132"><a href="#cb32-132" aria-hidden="true" tabindex="-1"></a><span class="fu"># Lh</span></span>
<span id="cb32-133"><a href="#cb32-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-134"><a href="#cb32-134" aria-hidden="true" tabindex="-1"></a><span class="in">```{r set Lh dir, eval=FALSE}</span></span>
<span id="cb32-135"><a href="#cb32-135" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/phenotype_files/"</span></span>
<span id="cb32-136"><a href="#cb32-136" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/genotype_files/"</span></span>
<span id="cb32-137"><a href="#cb32-137" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/bim_bam/"</span></span>
<span id="cb32-138"><a href="#cb32-138" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb32-139"><a href="#cb32-139" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Lh_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb32-140"><a href="#cb32-140" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span>
<span id="cb32-141"><a href="#cb32-141" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-142"><a href="#cb32-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-143"><a href="#cb32-143" aria-hidden="true" tabindex="-1"></a>Make local GRMs for each gene</span>
<span id="cb32-144"><a href="#cb32-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-145"><a href="#cb32-145" aria-hidden="true" tabindex="-1"></a><span class="in">```{r local dir for Lh, eval=FALSE}</span></span>
<span id="cb32-146"><a href="#cb32-146" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/"</span>)</span>
<span id="cb32-147"><a href="#cb32-147" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb32-148"><a href="#cb32-148" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb32-149"><a href="#cb32-149" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb32-150"><a href="#cb32-150" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb32-151"><a href="#cb32-151" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb32-152"><a href="#cb32-152" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb32-153"><a href="#cb32-153" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb32-154"><a href="#cb32-154" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb32-155"><a href="#cb32-155" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb32-156"><a href="#cb32-156" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb32-157"><a href="#cb32-157" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]    </span>
<span id="cb32-158"><a href="#cb32-158" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Lh.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb32-159"><a href="#cb32-159" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb32-160"><a href="#cb32-160" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexLh_transpose[,i])</span>
<span id="cb32-161"><a href="#cb32-161" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb32-162"><a href="#cb32-162" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Lh.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Lh_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb32-163"><a href="#cb32-163" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb32-164"><a href="#cb32-164" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-165"><a href="#cb32-165" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-166"><a href="#cb32-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-167"><a href="#cb32-167" aria-hidden="true" tabindex="-1"></a><span class="fu"># Pl</span></span>
<span id="cb32-168"><a href="#cb32-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-169"><a href="#cb32-169" aria-hidden="true" tabindex="-1"></a><span class="in">```{r pl dir, eval=FALSE}</span></span>
<span id="cb32-170"><a href="#cb32-170" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/phenotype_files/"</span></span>
<span id="cb32-171"><a href="#cb32-171" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/genotype_files/"</span></span>
<span id="cb32-172"><a href="#cb32-172" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/bim_bam/"</span></span>
<span id="cb32-173"><a href="#cb32-173" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb32-174"><a href="#cb32-174" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Pl_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb32-175"><a href="#cb32-175" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span>
<span id="cb32-176"><a href="#cb32-176" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-177"><a href="#cb32-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-178"><a href="#cb32-178" aria-hidden="true" tabindex="-1"></a>Make local GRMs for each gene</span>
<span id="cb32-179"><a href="#cb32-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-180"><a href="#cb32-180" aria-hidden="true" tabindex="-1"></a><span class="in">```{r pl grm, eval=FALSE}</span></span>
<span id="cb32-181"><a href="#cb32-181" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/"</span>)</span>
<span id="cb32-182"><a href="#cb32-182" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb32-183"><a href="#cb32-183" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb32-184"><a href="#cb32-184" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb32-185"><a href="#cb32-185" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb32-186"><a href="#cb32-186" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb32-187"><a href="#cb32-187" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb32-188"><a href="#cb32-188" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb32-189"><a href="#cb32-189" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb32-190"><a href="#cb32-190" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb32-191"><a href="#cb32-191" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb32-192"><a href="#cb32-192" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]    </span>
<span id="cb32-193"><a href="#cb32-193" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Pl.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb32-194"><a href="#cb32-194" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb32-195"><a href="#cb32-195" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexPl_transpose[,i])</span>
<span id="cb32-196"><a href="#cb32-196" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb32-197"><a href="#cb32-197" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Pl.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Pl_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb32-198"><a href="#cb32-198" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb32-199"><a href="#cb32-199" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-200"><a href="#cb32-200" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-201"><a href="#cb32-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-202"><a href="#cb32-202" aria-hidden="true" tabindex="-1"></a>#Vo</span>
<span id="cb32-203"><a href="#cb32-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-204"><a href="#cb32-204" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Vo dir, eval=FALSE}</span></span>
<span id="cb32-205"><a href="#cb32-205" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/phenotype_files/"</span></span>
<span id="cb32-206"><a href="#cb32-206" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/genotype_files/"</span></span>
<span id="cb32-207"><a href="#cb32-207" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/bim_bam/"</span></span>
<span id="cb32-208"><a href="#cb32-208" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb32-209"><a href="#cb32-209" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Vo_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb32-210"><a href="#cb32-210" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span>
<span id="cb32-211"><a href="#cb32-211" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-212"><a href="#cb32-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-213"><a href="#cb32-213" aria-hidden="true" tabindex="-1"></a>Make local GRMs for each gene</span>
<span id="cb32-214"><a href="#cb32-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-215"><a href="#cb32-215" aria-hidden="true" tabindex="-1"></a><span class="in">```{r set GRM for Vo, eval=FALSE}</span></span>
<span id="cb32-216"><a href="#cb32-216" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/"</span>)</span>
<span id="cb32-217"><a href="#cb32-217" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb32-218"><a href="#cb32-218" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb32-219"><a href="#cb32-219" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb32-220"><a href="#cb32-220" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb32-221"><a href="#cb32-221" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb32-222"><a href="#cb32-222" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb32-223"><a href="#cb32-223" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb32-224"><a href="#cb32-224" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb32-225"><a href="#cb32-225" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb32-226"><a href="#cb32-226" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb32-227"><a href="#cb32-227" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]    </span>
<span id="cb32-228"><a href="#cb32-228" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Vo.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb32-229"><a href="#cb32-229" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb32-230"><a href="#cb32-230" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexVo_transpose[,i])</span>
<span id="cb32-231"><a href="#cb32-231" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb32-232"><a href="#cb32-232" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Vo.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Vo_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb32-233"><a href="#cb32-233" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb32-234"><a href="#cb32-234" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-235"><a href="#cb32-235" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-236"><a href="#cb32-236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-237"><a href="#cb32-237" aria-hidden="true" tabindex="-1"></a><span class="fu">## Sparsity Analysis</span></span>
<span id="cb32-238"><a href="#cb32-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-239"><a href="#cb32-239" aria-hidden="true" tabindex="-1"></a>All the code above generates the local GRM for each phenotype (gene). With the GRM we then run gemma again to calculate both PVE (heritability) and PGE (sparsity). I used a badger template to calculate h2 and sparsity for each tissue. This steps takes a lot of computing power, so we use Badger. It takes approximatley 2-3 days to run.</span>
<span id="cb32-240"><a href="#cb32-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-241"><a href="#cb32-241" aria-hidden="true" tabindex="-1"></a>The code to run badger is here</span>
<span id="cb32-242"><a href="#cb32-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-245"><a href="#cb32-245" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb32-246"><a href="#cb32-246" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"./Rat_Genomics_Paper_Pipeline/analysis/Sparsity_Badger_Template.Rmd"</span>)</span>
<span id="cb32-247"><a href="#cb32-247" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-248"><a href="#cb32-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-249"><a href="#cb32-249" aria-hidden="true" tabindex="-1"></a>GEMMA then generates a .hyp file for each phenotype or in our case gene of interest. The hyp file contains the posterior samples for the hyper-parameters (h, PVE, rho, PGE, pi and gamma) for every 10th iteration. For our purposes, we are interested in the PGE and PVE parameters.</span>
<span id="cb32-250"><a href="#cb32-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-251"><a href="#cb32-251" aria-hidden="true" tabindex="-1"></a>To then calculate the point estimate and credible set for Proportion of Variance Explained (PVE) and Proportion of genetic variance explained by the sparse effects terms (PGE), we calculate the posterior probability for each gene for each tissue.</span>
<span id="cb32-252"><a href="#cb32-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-253"><a href="#cb32-253" aria-hidden="true" tabindex="-1"></a>I generated a function that calculates the beta of the posterior distribution and can be found <span class="co">[</span><span class="ot">here</span><span class="co">](https://gist.github.com/natashasanthanam/6eb539dcadecafe3a6dec448cedc7046)</span></span>
<span id="cb32-254"><a href="#cb32-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-255"><a href="#cb32-255" aria-hidden="true" tabindex="-1"></a><span class="fu"># Ac</span></span>
<span id="cb32-256"><a href="#cb32-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-257"><a href="#cb32-257" aria-hidden="true" tabindex="-1"></a>Find point estimate and 95% credible interval in CRI for PVE</span>
<span id="cb32-258"><a href="#cb32-258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-259"><a href="#cb32-259" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb32-260"><a href="#cb32-260" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/output"</span></span>
<span id="cb32-261"><a href="#cb32-261" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> ge.dir, <span class="at">pattern =</span> <span class="st">".hyp.txt"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb32-262"><a href="#cb32-262" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-263"><a href="#cb32-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-264"><a href="#cb32-264" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb32-265"><a href="#cb32-265" aria-hidden="true" tabindex="-1"></a>PVE_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="dv">0</span>, <span class="dv">4</span>)) </span>
<span id="cb32-266"><a href="#cb32-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-267"><a href="#cb32-267" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb32-268"><a href="#cb32-268" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(files[i],<span class="st">"/"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">11</span>), <span class="dv">8</span>, <span class="dv">25</span>)</span>
<span id="cb32-269"><a href="#cb32-269" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(files[i])</span>
<span id="cb32-270"><a href="#cb32-270" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-271"><a href="#cb32-271" aria-hidden="true" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pve, <span class="fl">0.1</span>)</span>
<span id="cb32-272"><a href="#cb32-272" aria-hidden="true" tabindex="-1"></a>  q2 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pve, <span class="fl">0.9</span>)</span>
<span id="cb32-273"><a href="#cb32-273" aria-hidden="true" tabindex="-1"></a>  quantile1<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">1</span> ,<span class="at">x=</span>q1)</span>
<span id="cb32-274"><a href="#cb32-274" aria-hidden="true" tabindex="-1"></a>  quantile2<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">9</span>, <span class="at">x=</span>q2)</span>
<span id="cb32-275"><a href="#cb32-275" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quantile1<span class="sc">$</span>x <span class="sc">!=</span> quantile2<span class="sc">$</span>x) {</span>
<span id="cb32-276"><a href="#cb32-276" aria-hidden="true" tabindex="-1"></a>  prior <span class="ot">&lt;-</span> <span class="fu">beta.select</span>(quantile1, quantile2)</span>
<span id="cb32-277"><a href="#cb32-277" aria-hidden="true" tabindex="-1"></a>  credible_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">qbeta</span>(<span class="fl">0.025</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]), <span class="fu">qbeta</span>(<span class="fl">0.975</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]))</span>
<span id="cb32-278"><a href="#cb32-278" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-279"><a href="#cb32-279" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> gene</span>
<span id="cb32-280"><a href="#cb32-280" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.5</span>, prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>])</span>
<span id="cb32-281"><a href="#cb32-281" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">1</span>]</span>
<span id="cb32-282"><a href="#cb32-282" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">2</span>]</span>
<span id="cb32-283"><a href="#cb32-283" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb32-284"><a href="#cb32-284" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> </span>
<span id="cb32-285"><a href="#cb32-285" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb32-286"><a href="#cb32-286" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-287"><a href="#cb32-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-288"><a href="#cb32-288" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PVE_df, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/PVE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> )</span>
<span id="cb32-289"><a href="#cb32-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-290"><a href="#cb32-290" aria-hidden="true" tabindex="-1"></a>PGE_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="dv">0</span>, <span class="dv">4</span>))</span>
<span id="cb32-291"><a href="#cb32-291" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb32-292"><a href="#cb32-292" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(files[i],<span class="st">"/"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">11</span>), <span class="dv">8</span>, <span class="dv">25</span>)</span>
<span id="cb32-293"><a href="#cb32-293" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(files[i])</span>
<span id="cb32-294"><a href="#cb32-294" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-295"><a href="#cb32-295" aria-hidden="true" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pge, <span class="fl">0.5</span>)</span>
<span id="cb32-296"><a href="#cb32-296" aria-hidden="true" tabindex="-1"></a>  q2 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pge, <span class="fl">0.9</span>)</span>
<span id="cb32-297"><a href="#cb32-297" aria-hidden="true" tabindex="-1"></a>  quantile1<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">5</span> ,<span class="at">x=</span>q1)</span>
<span id="cb32-298"><a href="#cb32-298" aria-hidden="true" tabindex="-1"></a>  quantile2<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">9</span>, <span class="at">x=</span>q2)</span>
<span id="cb32-299"><a href="#cb32-299" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quantile1<span class="sc">$</span>x <span class="sc">!=</span> quantile2<span class="sc">$</span>x) {</span>
<span id="cb32-300"><a href="#cb32-300" aria-hidden="true" tabindex="-1"></a>  prior <span class="ot">&lt;-</span> <span class="fu">beta.select</span>(quantile1, quantile2)</span>
<span id="cb32-301"><a href="#cb32-301" aria-hidden="true" tabindex="-1"></a>  credible_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">qbeta</span>(<span class="fl">0.025</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]), <span class="fu">qbeta</span>(<span class="fl">0.975</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]))</span>
<span id="cb32-302"><a href="#cb32-302" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-303"><a href="#cb32-303" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> gene</span>
<span id="cb32-304"><a href="#cb32-304" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.5</span>, prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>])</span>
<span id="cb32-305"><a href="#cb32-305" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">1</span>]</span>
<span id="cb32-306"><a href="#cb32-306" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">2</span>]</span>
<span id="cb32-307"><a href="#cb32-307" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb32-308"><a href="#cb32-308" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> </span>
<span id="cb32-309"><a href="#cb32-309" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb32-310"><a href="#cb32-310" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-311"><a href="#cb32-311" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-312"><a href="#cb32-312" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PGE_df, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/PGE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> )</span>
<span id="cb32-313"><a href="#cb32-313" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-314"><a href="#cb32-314" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-315"><a href="#cb32-315" aria-hidden="true" tabindex="-1"></a><span class="fu"># Il</span></span>
<span id="cb32-316"><a href="#cb32-316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-317"><a href="#cb32-317" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb32-318"><a href="#cb32-318" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/output"</span></span>
<span id="cb32-319"><a href="#cb32-319" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> ge.dir, <span class="at">pattern =</span> <span class="st">".hyp.txt"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb32-320"><a href="#cb32-320" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-321"><a href="#cb32-321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-322"><a href="#cb32-322" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb32-323"><a href="#cb32-323" aria-hidden="true" tabindex="-1"></a>PVE_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="dv">0</span>, <span class="dv">4</span>)) </span>
<span id="cb32-324"><a href="#cb32-324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-325"><a href="#cb32-325" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb32-326"><a href="#cb32-326" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(files[i],<span class="st">"/"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">11</span>), <span class="dv">8</span>, <span class="dv">25</span>)</span>
<span id="cb32-327"><a href="#cb32-327" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(files[i])</span>
<span id="cb32-328"><a href="#cb32-328" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-329"><a href="#cb32-329" aria-hidden="true" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pve, <span class="fl">0.1</span>)</span>
<span id="cb32-330"><a href="#cb32-330" aria-hidden="true" tabindex="-1"></a>  q2 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pve, <span class="fl">0.9</span>)</span>
<span id="cb32-331"><a href="#cb32-331" aria-hidden="true" tabindex="-1"></a>  quantile1<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">1</span> ,<span class="at">x=</span>q1)</span>
<span id="cb32-332"><a href="#cb32-332" aria-hidden="true" tabindex="-1"></a>  quantile2<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">9</span>, <span class="at">x=</span>q2)</span>
<span id="cb32-333"><a href="#cb32-333" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quantile1<span class="sc">$</span>x <span class="sc">!=</span> quantile2<span class="sc">$</span>x) {</span>
<span id="cb32-334"><a href="#cb32-334" aria-hidden="true" tabindex="-1"></a>  prior <span class="ot">&lt;-</span> <span class="fu">beta.select</span>(quantile1, quantile2)</span>
<span id="cb32-335"><a href="#cb32-335" aria-hidden="true" tabindex="-1"></a>  credible_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">qbeta</span>(<span class="fl">0.025</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]), <span class="fu">qbeta</span>(<span class="fl">0.975</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]))</span>
<span id="cb32-336"><a href="#cb32-336" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-337"><a href="#cb32-337" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> gene</span>
<span id="cb32-338"><a href="#cb32-338" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.5</span>, prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>])</span>
<span id="cb32-339"><a href="#cb32-339" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">1</span>]</span>
<span id="cb32-340"><a href="#cb32-340" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">2</span>]</span>
<span id="cb32-341"><a href="#cb32-341" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb32-342"><a href="#cb32-342" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> </span>
<span id="cb32-343"><a href="#cb32-343" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb32-344"><a href="#cb32-344" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-345"><a href="#cb32-345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-346"><a href="#cb32-346" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PVE_df, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/PVE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> )</span>
<span id="cb32-347"><a href="#cb32-347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-348"><a href="#cb32-348" aria-hidden="true" tabindex="-1"></a>PGE_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="dv">0</span>, <span class="dv">4</span>))</span>
<span id="cb32-349"><a href="#cb32-349" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb32-350"><a href="#cb32-350" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(files[i],<span class="st">"/"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">11</span>), <span class="dv">8</span>, <span class="dv">25</span>)</span>
<span id="cb32-351"><a href="#cb32-351" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(files[i])</span>
<span id="cb32-352"><a href="#cb32-352" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-353"><a href="#cb32-353" aria-hidden="true" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pge, <span class="fl">0.5</span>)</span>
<span id="cb32-354"><a href="#cb32-354" aria-hidden="true" tabindex="-1"></a>  q2 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pge, <span class="fl">0.9</span>)</span>
<span id="cb32-355"><a href="#cb32-355" aria-hidden="true" tabindex="-1"></a>  quantile1<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">5</span> ,<span class="at">x=</span>q1)</span>
<span id="cb32-356"><a href="#cb32-356" aria-hidden="true" tabindex="-1"></a>  quantile2<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">9</span>, <span class="at">x=</span>q2)</span>
<span id="cb32-357"><a href="#cb32-357" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quantile1<span class="sc">$</span>x <span class="sc">!=</span> quantile2<span class="sc">$</span>x) {</span>
<span id="cb32-358"><a href="#cb32-358" aria-hidden="true" tabindex="-1"></a>  prior <span class="ot">&lt;-</span> <span class="fu">beta.select</span>(quantile1, quantile2)</span>
<span id="cb32-359"><a href="#cb32-359" aria-hidden="true" tabindex="-1"></a>  credible_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">qbeta</span>(<span class="fl">0.025</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]), <span class="fu">qbeta</span>(<span class="fl">0.975</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]))</span>
<span id="cb32-360"><a href="#cb32-360" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-361"><a href="#cb32-361" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> gene</span>
<span id="cb32-362"><a href="#cb32-362" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.5</span>, prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>])</span>
<span id="cb32-363"><a href="#cb32-363" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">1</span>]</span>
<span id="cb32-364"><a href="#cb32-364" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">2</span>]</span>
<span id="cb32-365"><a href="#cb32-365" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb32-366"><a href="#cb32-366" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> </span>
<span id="cb32-367"><a href="#cb32-367" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb32-368"><a href="#cb32-368" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-369"><a href="#cb32-369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-370"><a href="#cb32-370" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PGE_df, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/PGE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> )</span>
<span id="cb32-371"><a href="#cb32-371" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-372"><a href="#cb32-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-373"><a href="#cb32-373" aria-hidden="true" tabindex="-1"></a><span class="fu"># Lh</span></span>
<span id="cb32-374"><a href="#cb32-374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-375"><a href="#cb32-375" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb32-376"><a href="#cb32-376" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/output"</span></span>
<span id="cb32-377"><a href="#cb32-377" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> ge.dir, <span class="at">pattern =</span> <span class="st">".hyp.txt"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb32-378"><a href="#cb32-378" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-379"><a href="#cb32-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-380"><a href="#cb32-380" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb32-381"><a href="#cb32-381" aria-hidden="true" tabindex="-1"></a>PVE_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="dv">0</span>, <span class="dv">4</span>)) </span>
<span id="cb32-382"><a href="#cb32-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-383"><a href="#cb32-383" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb32-384"><a href="#cb32-384" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(files[i],<span class="st">"/"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">11</span>), <span class="dv">8</span>, <span class="dv">25</span>)</span>
<span id="cb32-385"><a href="#cb32-385" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(files[i])</span>
<span id="cb32-386"><a href="#cb32-386" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-387"><a href="#cb32-387" aria-hidden="true" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pve, <span class="fl">0.1</span>)</span>
<span id="cb32-388"><a href="#cb32-388" aria-hidden="true" tabindex="-1"></a>  q2 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pve, <span class="fl">0.9</span>)</span>
<span id="cb32-389"><a href="#cb32-389" aria-hidden="true" tabindex="-1"></a>  quantile1<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">1</span> ,<span class="at">x=</span>q1)</span>
<span id="cb32-390"><a href="#cb32-390" aria-hidden="true" tabindex="-1"></a>  quantile2<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">9</span>, <span class="at">x=</span>q2)</span>
<span id="cb32-391"><a href="#cb32-391" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quantile1<span class="sc">$</span>x <span class="sc">!=</span> quantile2<span class="sc">$</span>x) {</span>
<span id="cb32-392"><a href="#cb32-392" aria-hidden="true" tabindex="-1"></a>  prior <span class="ot">&lt;-</span> <span class="fu">beta.select</span>(quantile1, quantile2)</span>
<span id="cb32-393"><a href="#cb32-393" aria-hidden="true" tabindex="-1"></a>  credible_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">qbeta</span>(<span class="fl">0.025</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]), <span class="fu">qbeta</span>(<span class="fl">0.975</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]))</span>
<span id="cb32-394"><a href="#cb32-394" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-395"><a href="#cb32-395" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> gene</span>
<span id="cb32-396"><a href="#cb32-396" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.5</span>, prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>])</span>
<span id="cb32-397"><a href="#cb32-397" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">1</span>]</span>
<span id="cb32-398"><a href="#cb32-398" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">2</span>]</span>
<span id="cb32-399"><a href="#cb32-399" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb32-400"><a href="#cb32-400" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> </span>
<span id="cb32-401"><a href="#cb32-401" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb32-402"><a href="#cb32-402" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-403"><a href="#cb32-403" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-404"><a href="#cb32-404" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PVE_df, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/PVE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> )</span>
<span id="cb32-405"><a href="#cb32-405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-406"><a href="#cb32-406" aria-hidden="true" tabindex="-1"></a>PGE_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="dv">0</span>, <span class="dv">4</span>))</span>
<span id="cb32-407"><a href="#cb32-407" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb32-408"><a href="#cb32-408" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(files[i],<span class="st">"/"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">11</span>), <span class="dv">8</span>, <span class="dv">25</span>)</span>
<span id="cb32-409"><a href="#cb32-409" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(files[i])</span>
<span id="cb32-410"><a href="#cb32-410" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-411"><a href="#cb32-411" aria-hidden="true" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pge, <span class="fl">0.5</span>)</span>
<span id="cb32-412"><a href="#cb32-412" aria-hidden="true" tabindex="-1"></a>  q2 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pge, <span class="fl">0.9</span>)</span>
<span id="cb32-413"><a href="#cb32-413" aria-hidden="true" tabindex="-1"></a>  quantile1<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">5</span> ,<span class="at">x=</span>q1)</span>
<span id="cb32-414"><a href="#cb32-414" aria-hidden="true" tabindex="-1"></a>  quantile2<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">9</span>, <span class="at">x=</span>q2)</span>
<span id="cb32-415"><a href="#cb32-415" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quantile1<span class="sc">$</span>x <span class="sc">!=</span> quantile2<span class="sc">$</span>x) {</span>
<span id="cb32-416"><a href="#cb32-416" aria-hidden="true" tabindex="-1"></a>  prior <span class="ot">&lt;-</span> <span class="fu">beta.select</span>(quantile1, quantile2)</span>
<span id="cb32-417"><a href="#cb32-417" aria-hidden="true" tabindex="-1"></a>  credible_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">qbeta</span>(<span class="fl">0.025</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]), <span class="fu">qbeta</span>(<span class="fl">0.975</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]))</span>
<span id="cb32-418"><a href="#cb32-418" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-419"><a href="#cb32-419" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> gene</span>
<span id="cb32-420"><a href="#cb32-420" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.5</span>, prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>])</span>
<span id="cb32-421"><a href="#cb32-421" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">1</span>]</span>
<span id="cb32-422"><a href="#cb32-422" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">2</span>]</span>
<span id="cb32-423"><a href="#cb32-423" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb32-424"><a href="#cb32-424" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> </span>
<span id="cb32-425"><a href="#cb32-425" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb32-426"><a href="#cb32-426" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-427"><a href="#cb32-427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-428"><a href="#cb32-428" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PGE_df, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/PGE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> )</span>
<span id="cb32-429"><a href="#cb32-429" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-430"><a href="#cb32-430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-431"><a href="#cb32-431" aria-hidden="true" tabindex="-1"></a><span class="fu"># Pl</span></span>
<span id="cb32-432"><a href="#cb32-432" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-433"><a href="#cb32-433" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb32-434"><a href="#cb32-434" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/output"</span></span>
<span id="cb32-435"><a href="#cb32-435" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> ge.dir, <span class="at">pattern =</span> <span class="st">".hyp.txt"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb32-436"><a href="#cb32-436" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-437"><a href="#cb32-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-438"><a href="#cb32-438" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb32-439"><a href="#cb32-439" aria-hidden="true" tabindex="-1"></a>PVE_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="dv">0</span>, <span class="dv">4</span>)) </span>
<span id="cb32-440"><a href="#cb32-440" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-441"><a href="#cb32-441" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb32-442"><a href="#cb32-442" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(files[i],<span class="st">"/"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">11</span>), <span class="dv">8</span>, <span class="dv">25</span>)</span>
<span id="cb32-443"><a href="#cb32-443" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(files[i])</span>
<span id="cb32-444"><a href="#cb32-444" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-445"><a href="#cb32-445" aria-hidden="true" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pve, <span class="fl">0.1</span>)</span>
<span id="cb32-446"><a href="#cb32-446" aria-hidden="true" tabindex="-1"></a>  q2 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pve, <span class="fl">0.9</span>)</span>
<span id="cb32-447"><a href="#cb32-447" aria-hidden="true" tabindex="-1"></a>  quantile1<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">1</span> ,<span class="at">x=</span>q1)</span>
<span id="cb32-448"><a href="#cb32-448" aria-hidden="true" tabindex="-1"></a>  quantile2<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">9</span>, <span class="at">x=</span>q2)</span>
<span id="cb32-449"><a href="#cb32-449" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quantile1<span class="sc">$</span>x <span class="sc">!=</span> quantile2<span class="sc">$</span>x) {</span>
<span id="cb32-450"><a href="#cb32-450" aria-hidden="true" tabindex="-1"></a>  prior <span class="ot">&lt;-</span> <span class="fu">beta.select</span>(quantile1, quantile2)</span>
<span id="cb32-451"><a href="#cb32-451" aria-hidden="true" tabindex="-1"></a>  credible_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">qbeta</span>(<span class="fl">0.025</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]), <span class="fu">qbeta</span>(<span class="fl">0.975</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]))</span>
<span id="cb32-452"><a href="#cb32-452" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-453"><a href="#cb32-453" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> gene</span>
<span id="cb32-454"><a href="#cb32-454" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.5</span>, prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>])</span>
<span id="cb32-455"><a href="#cb32-455" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">1</span>]</span>
<span id="cb32-456"><a href="#cb32-456" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">2</span>]</span>
<span id="cb32-457"><a href="#cb32-457" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb32-458"><a href="#cb32-458" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> </span>
<span id="cb32-459"><a href="#cb32-459" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb32-460"><a href="#cb32-460" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-461"><a href="#cb32-461" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-462"><a href="#cb32-462" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PVE_df, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/PVE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> )</span>
<span id="cb32-463"><a href="#cb32-463" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-464"><a href="#cb32-464" aria-hidden="true" tabindex="-1"></a>PGE_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="dv">0</span>, <span class="dv">4</span>))</span>
<span id="cb32-465"><a href="#cb32-465" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb32-466"><a href="#cb32-466" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(files[i],<span class="st">"/"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">11</span>), <span class="dv">8</span>, <span class="dv">25</span>)</span>
<span id="cb32-467"><a href="#cb32-467" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(files[i])</span>
<span id="cb32-468"><a href="#cb32-468" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-469"><a href="#cb32-469" aria-hidden="true" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pge, <span class="fl">0.5</span>)</span>
<span id="cb32-470"><a href="#cb32-470" aria-hidden="true" tabindex="-1"></a>  q2 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pge, <span class="fl">0.9</span>)</span>
<span id="cb32-471"><a href="#cb32-471" aria-hidden="true" tabindex="-1"></a>  quantile1<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">5</span> ,<span class="at">x=</span>q1)</span>
<span id="cb32-472"><a href="#cb32-472" aria-hidden="true" tabindex="-1"></a>  quantile2<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">9</span>, <span class="at">x=</span>q2)</span>
<span id="cb32-473"><a href="#cb32-473" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quantile1<span class="sc">$</span>x <span class="sc">!=</span> quantile2<span class="sc">$</span>x) {</span>
<span id="cb32-474"><a href="#cb32-474" aria-hidden="true" tabindex="-1"></a>  prior <span class="ot">&lt;-</span> <span class="fu">beta.select</span>(quantile1, quantile2)</span>
<span id="cb32-475"><a href="#cb32-475" aria-hidden="true" tabindex="-1"></a>  credible_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">qbeta</span>(<span class="fl">0.025</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]), <span class="fu">qbeta</span>(<span class="fl">0.975</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]))</span>
<span id="cb32-476"><a href="#cb32-476" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-477"><a href="#cb32-477" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> gene</span>
<span id="cb32-478"><a href="#cb32-478" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.5</span>, prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>])</span>
<span id="cb32-479"><a href="#cb32-479" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">1</span>]</span>
<span id="cb32-480"><a href="#cb32-480" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">2</span>]</span>
<span id="cb32-481"><a href="#cb32-481" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb32-482"><a href="#cb32-482" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> </span>
<span id="cb32-483"><a href="#cb32-483" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb32-484"><a href="#cb32-484" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-485"><a href="#cb32-485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-486"><a href="#cb32-486" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PGE_df, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/PGE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> )</span>
<span id="cb32-487"><a href="#cb32-487" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-488"><a href="#cb32-488" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-489"><a href="#cb32-489" aria-hidden="true" tabindex="-1"></a><span class="fu"># Vo</span></span>
<span id="cb32-490"><a href="#cb32-490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-491"><a href="#cb32-491" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb32-492"><a href="#cb32-492" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/output"</span></span>
<span id="cb32-493"><a href="#cb32-493" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> ge.dir, <span class="at">pattern =</span> <span class="st">".hyp.txt"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb32-494"><a href="#cb32-494" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-495"><a href="#cb32-495" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-496"><a href="#cb32-496" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb32-497"><a href="#cb32-497" aria-hidden="true" tabindex="-1"></a>PVE_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="dv">0</span>, <span class="dv">4</span>)) </span>
<span id="cb32-498"><a href="#cb32-498" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-499"><a href="#cb32-499" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb32-500"><a href="#cb32-500" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(files[i],<span class="st">"/"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">11</span>), <span class="dv">8</span>, <span class="dv">25</span>)</span>
<span id="cb32-501"><a href="#cb32-501" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(files[i])</span>
<span id="cb32-502"><a href="#cb32-502" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-503"><a href="#cb32-503" aria-hidden="true" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pve, <span class="fl">0.1</span>)</span>
<span id="cb32-504"><a href="#cb32-504" aria-hidden="true" tabindex="-1"></a>  q2 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pve, <span class="fl">0.9</span>)</span>
<span id="cb32-505"><a href="#cb32-505" aria-hidden="true" tabindex="-1"></a>  quantile1<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">1</span> ,<span class="at">x=</span>q1)</span>
<span id="cb32-506"><a href="#cb32-506" aria-hidden="true" tabindex="-1"></a>  quantile2<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">9</span>, <span class="at">x=</span>q2)</span>
<span id="cb32-507"><a href="#cb32-507" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quantile1<span class="sc">$</span>x <span class="sc">!=</span> quantile2<span class="sc">$</span>x) {</span>
<span id="cb32-508"><a href="#cb32-508" aria-hidden="true" tabindex="-1"></a>  prior <span class="ot">&lt;-</span> <span class="fu">beta.select</span>(quantile1, quantile2)</span>
<span id="cb32-509"><a href="#cb32-509" aria-hidden="true" tabindex="-1"></a>  credible_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">qbeta</span>(<span class="fl">0.025</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]), <span class="fu">qbeta</span>(<span class="fl">0.975</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]))</span>
<span id="cb32-510"><a href="#cb32-510" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-511"><a href="#cb32-511" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> gene</span>
<span id="cb32-512"><a href="#cb32-512" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.5</span>, prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>])</span>
<span id="cb32-513"><a href="#cb32-513" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">1</span>]</span>
<span id="cb32-514"><a href="#cb32-514" aria-hidden="true" tabindex="-1"></a>  PVE_df[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">2</span>]</span>
<span id="cb32-515"><a href="#cb32-515" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb32-516"><a href="#cb32-516" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> </span>
<span id="cb32-517"><a href="#cb32-517" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb32-518"><a href="#cb32-518" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-519"><a href="#cb32-519" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-520"><a href="#cb32-520" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PVE_df, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/PVE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> )</span>
<span id="cb32-521"><a href="#cb32-521" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-522"><a href="#cb32-522" aria-hidden="true" tabindex="-1"></a>PGE_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="dv">0</span>, <span class="dv">4</span>))</span>
<span id="cb32-523"><a href="#cb32-523" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb32-524"><a href="#cb32-524" aria-hidden="true" tabindex="-1"></a>  gene <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(files[i],<span class="st">"/"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">11</span>), <span class="dv">8</span>, <span class="dv">25</span>)</span>
<span id="cb32-525"><a href="#cb32-525" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(files[i])</span>
<span id="cb32-526"><a href="#cb32-526" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-527"><a href="#cb32-527" aria-hidden="true" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pge, <span class="fl">0.5</span>)</span>
<span id="cb32-528"><a href="#cb32-528" aria-hidden="true" tabindex="-1"></a>  q2 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(df<span class="sc">$</span>pge, <span class="fl">0.9</span>)</span>
<span id="cb32-529"><a href="#cb32-529" aria-hidden="true" tabindex="-1"></a>  quantile1<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">5</span> ,<span class="at">x=</span>q1)</span>
<span id="cb32-530"><a href="#cb32-530" aria-hidden="true" tabindex="-1"></a>  quantile2<span class="ot">=</span><span class="fu">list</span>(<span class="at">p=</span>.<span class="dv">9</span>, <span class="at">x=</span>q2)</span>
<span id="cb32-531"><a href="#cb32-531" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(quantile1<span class="sc">$</span>x <span class="sc">!=</span> quantile2<span class="sc">$</span>x) {</span>
<span id="cb32-532"><a href="#cb32-532" aria-hidden="true" tabindex="-1"></a>  prior <span class="ot">&lt;-</span> <span class="fu">beta.select</span>(quantile1, quantile2)</span>
<span id="cb32-533"><a href="#cb32-533" aria-hidden="true" tabindex="-1"></a>  credible_set <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">qbeta</span>(<span class="fl">0.025</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]), <span class="fu">qbeta</span>(<span class="fl">0.975</span>,prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>]))</span>
<span id="cb32-534"><a href="#cb32-534" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-535"><a href="#cb32-535" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> gene</span>
<span id="cb32-536"><a href="#cb32-536" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">qbeta</span>(<span class="fl">0.5</span>, prior[<span class="dv">1</span>],  prior[<span class="dv">2</span>])</span>
<span id="cb32-537"><a href="#cb32-537" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">1</span>]</span>
<span id="cb32-538"><a href="#cb32-538" aria-hidden="true" tabindex="-1"></a>  PGE_df[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> credible_set[<span class="dv">2</span>]</span>
<span id="cb32-539"><a href="#cb32-539" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb32-540"><a href="#cb32-540" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> </span>
<span id="cb32-541"><a href="#cb32-541" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb32-542"><a href="#cb32-542" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-543"><a href="#cb32-543" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-544"><a href="#cb32-544" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PGE_df, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/PGE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> )</span>
<span id="cb32-545"><a href="#cb32-545" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-546"><a href="#cb32-546" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-547"><a href="#cb32-547" aria-hidden="true" tabindex="-1"></a>Now we have PVE and PGE estimates for each tissue and can generate the fitted curve figure for Heritability.</span>
<span id="cb32-548"><a href="#cb32-548" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-549"><a href="#cb32-549" aria-hidden="true" tabindex="-1"></a><span class="fu"># Generate Fitted Heritability Curves with R2</span></span>
<span id="cb32-550"><a href="#cb32-550" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-551"><a href="#cb32-551" aria-hidden="true" tabindex="-1"></a>I'm showing the curve for Ac tissue but I generated the one for all other tissues as well.</span>
<span id="cb32-552"><a href="#cb32-552" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-553"><a href="#cb32-553" aria-hidden="true" tabindex="-1"></a><span class="in">```{r set parameters, eval=FALSE}</span></span>
<span id="cb32-554"><a href="#cb32-554" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>))</span>
<span id="cb32-555"><a href="#cb32-555" aria-hidden="true" tabindex="-1"></a>Data <span class="ot">&lt;-</span> <span class="st">"/Users/natashasanthanam/Box/imlab-data/Projects/PTRS-PGRS-Rosetta/Rat-Genomics/Tyson-PalmerLab_PredictDB_Results/sql/"</span></span>
<span id="cb32-556"><a href="#cb32-556" aria-hidden="true" tabindex="-1"></a>poly_dir <span class="ot">&lt;-</span> <span class="st">"/Users/natashasanthanam/Github/rat-genomic-analysis/data/"</span></span>
<span id="cb32-557"><a href="#cb32-557" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-558"><a href="#cb32-558" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-559"><a href="#cb32-559" aria-hidden="true" tabindex="-1"></a><span class="in">```{r generate order function, eval=FALSE}</span></span>
<span id="cb32-560"><a href="#cb32-560" aria-hidden="true" tabindex="-1"></a>load_pve <span class="ot">&lt;-</span> <span class="cf">function</span>(df){</span>
<span id="cb32-561"><a href="#cb32-561" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> df[<span class="fu">order</span>(df<span class="sc">$</span>point_estimate),]</span>
<span id="cb32-562"><a href="#cb32-562" aria-hidden="true" tabindex="-1"></a>  df<span class="sc">$</span>index <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(df)</span>
<span id="cb32-563"><a href="#cb32-563" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb32-564"><a href="#cb32-564" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-565"><a href="#cb32-565" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-566"><a href="#cb32-566" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-567"><a href="#cb32-567" aria-hidden="true" tabindex="-1"></a>First we get the R2 for the tissue from the model we generated</span>
<span id="cb32-568"><a href="#cb32-568" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-569"><a href="#cb32-569" aria-hidden="true" tabindex="-1"></a><span class="in">```{r get R2 for tissue, eval=FALSE}</span></span>
<span id="cb32-570"><a href="#cb32-570" aria-hidden="true" tabindex="-1"></a>filename <span class="ot">&lt;-</span> Data <span class="sc">%&amp;%</span> <span class="st">"Ac_output_db.db"</span></span>
<span id="cb32-571"><a href="#cb32-571" aria-hidden="true" tabindex="-1"></a>sqlite.driver <span class="ot">&lt;-</span> <span class="fu">dbDriver</span>(<span class="st">"SQLite"</span>)</span>
<span id="cb32-572"><a href="#cb32-572" aria-hidden="true" tabindex="-1"></a>conn <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(), filename)</span>
<span id="cb32-573"><a href="#cb32-573" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-574"><a href="#cb32-574" aria-hidden="true" tabindex="-1"></a>extra <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>(conn, <span class="st">'select * from extra'</span>)</span>
<span id="cb32-575"><a href="#cb32-575" aria-hidden="true" tabindex="-1"></a><span class="fu">dbDisconnect</span>(conn)</span>
<span id="cb32-576"><a href="#cb32-576" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-577"><a href="#cb32-577" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-578"><a href="#cb32-578" aria-hidden="true" tabindex="-1"></a>Next we create the ordered plot for heritability for Ac. We use PVE as h2, the proportion of variance in phenotypes explained by typed genotypes.</span>
<span id="cb32-579"><a href="#cb32-579" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-582"><a href="#cb32-582" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb32-583"><a href="#cb32-583" aria-hidden="true" tabindex="-1"></a>human_h2 <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"/Users/natashasanthanam/Github/rat-genomic-analysis/data/human_PVE.csv"</span>) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">point_estimate =</span> pve) </span>
<span id="cb32-584"><a href="#cb32-584" aria-hidden="true" tabindex="-1"></a>PVE_Ac <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(poly_dir <span class="sc">%&amp;%</span> <span class="st">"Ac_PVE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb32-585"><a href="#cb32-585" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-586"><a href="#cb32-586" aria-hidden="true" tabindex="-1"></a>human_h2<span class="sc">$</span>ensid <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(human_h2<span class="sc">$</span>ensid, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb32-587"><a href="#cb32-587" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(PVE_Ac) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"gene"</span>, <span class="st">"point_estimate"</span>, <span class="st">"credible_set_1"</span>, <span class="st">"credible_set_2"</span>)</span>
<span id="cb32-588"><a href="#cb32-588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-589"><a href="#cb32-589" aria-hidden="true" tabindex="-1"></a>overlap <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">rat_genes =</span> PVE_Ac<span class="sc">$</span>gene, <span class="at">ensid =</span> orth.rats[<span class="fu">match</span>(PVE_Ac<span class="sc">$</span>gene, orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene), <span class="dv">1</span>]<span class="sc">$</span>ensembl_gene_id) </span>
<span id="cb32-590"><a href="#cb32-590" aria-hidden="true" tabindex="-1"></a>overlap <span class="ot">&lt;-</span> overlap[<span class="fu">na.omit</span>(<span class="fu">match</span>(human_h2<span class="sc">$</span>ensid, overlap<span class="sc">$</span>ensid)), ]</span>
<span id="cb32-591"><a href="#cb32-591" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-592"><a href="#cb32-592" aria-hidden="true" tabindex="-1"></a>PVE_Ac <span class="ot">&lt;-</span> PVE_Ac <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene <span class="sc">%in%</span> overlap<span class="sc">$</span>rat_genes)</span>
<span id="cb32-593"><a href="#cb32-593" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-594"><a href="#cb32-594" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-597"><a href="#cb32-597" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb32-598"><a href="#cb32-598" aria-hidden="true" tabindex="-1"></a>PVE_Ac <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(PVE_Ac, extra, <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb32-599"><a href="#cb32-599" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-600"><a href="#cb32-600" aria-hidden="true" tabindex="-1"></a>A_df_Ac <span class="ot">&lt;-</span> <span class="fu">load_pve</span>(PVE_Ac)</span>
<span id="cb32-601"><a href="#cb32-601" aria-hidden="true" tabindex="-1"></a>plt_1 <span class="ot">&lt;-</span> (<span class="fu">ggplot</span>(<span class="at">data =</span> A_df_Ac, <span class="fu">aes</span>(<span class="at">x =</span> index))</span>
<span id="cb32-602"><a href="#cb32-602" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>index, <span class="at">y=</span>R2), <span class="at">colour =</span> <span class="st">"dodgerblue2"</span>, <span class="at">size =</span> <span class="fl">0.2</span>)</span>
<span id="cb32-603"><a href="#cb32-603" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> point_estimate), <span class="at">lwd=</span><span class="fl">1.5</span>)</span>
<span id="cb32-604"><a href="#cb32-604" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype=</span><span class="dv">2</span>)</span>
<span id="cb32-605"><a href="#cb32-605" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">geom_ribbon</span>(<span class="fu">aes</span>(<span class="at">ymax =</span> credible_set_2, <span class="at">ymin =</span> credible_set_1),</span>
<span id="cb32-606"><a href="#cb32-606" aria-hidden="true" tabindex="-1"></a>                         <span class="at">alpha =</span> <span class="fl">0.25</span>)</span>
<span id="cb32-607"><a href="#cb32-607" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">'Genes Sorted by Proportion of Variance Explained (PVE)'</span>,</span>
<span id="cb32-608"><a href="#cb32-608" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y =</span> <span class="st">'PVE'</span>)</span>
<span id="cb32-609"><a href="#cb32-609" aria-hidden="true" tabindex="-1"></a>           <span class="sc">+</span> <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>))</span>
<span id="cb32-610"><a href="#cb32-610" aria-hidden="true" tabindex="-1"></a>           <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">1250</span>, <span class="at">y =</span> <span class="fl">0.9</span>, <span class="at">label =</span> <span class="st">"Mean h2 =  0.098"</span>, <span class="at">size =</span> <span class="dv">6</span>)</span>
<span id="cb32-611"><a href="#cb32-611" aria-hidden="true" tabindex="-1"></a>           <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">1250</span>, <span class="at">y =</span> <span class="fl">0.8</span>, <span class="at">label =</span> <span class="st">"Mean R2 =  0.085"</span>, <span class="at">size =</span> <span class="dv">6</span>)) </span>
<span id="cb32-612"><a href="#cb32-612" aria-hidden="true" tabindex="-1"></a>plt_1</span>
<span id="cb32-613"><a href="#cb32-613" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-614"><a href="#cb32-614" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-615"><a href="#cb32-615" aria-hidden="true" tabindex="-1"></a><span class="fu"># Generate Analagous Plot in Humans</span></span>
<span id="cb32-616"><a href="#cb32-616" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-617"><a href="#cb32-617" aria-hidden="true" tabindex="-1"></a>Get R2 and H2 from Heather's Models for Brain</span>
<span id="cb32-618"><a href="#cb32-618" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-619"><a href="#cb32-619" aria-hidden="true" tabindex="-1"></a><span class="in">```{sql connection, eval=FALSE}</span></span>
<span id="cb32-620"><a href="#cb32-620" aria-hidden="true" tabindex="-1"></a>sqlite3 genarch<span class="ch">.</span>db</span>
<span id="cb32-621"><a href="#cb32-621" aria-hidden="true" tabindex="-1"></a><span class="ch">.</span>headers <span class="kw">on</span></span>
<span id="cb32-622"><a href="#cb32-622" aria-hidden="true" tabindex="-1"></a><span class="ch">.</span>mode csv</span>
<span id="cb32-623"><a href="#cb32-623" aria-hidden="true" tabindex="-1"></a><span class="ch">.</span>output human_PVE<span class="ch">.</span>csv</span>
<span id="cb32-624"><a href="#cb32-624" aria-hidden="true" tabindex="-1"></a><span class="kw">select</span> gene, ensid, en_r2 ,pve,pve_CI <span class="kw">from</span> results <span class="kw">where</span> tissue <span class="op">=</span> <span class="st">"DGN-WB"</span>;</span>
<span id="cb32-625"><a href="#cb32-625" aria-hidden="true" tabindex="-1"></a><span class="ch">.</span>quit</span>
<span id="cb32-626"><a href="#cb32-626" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-627"><a href="#cb32-627" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-628"><a href="#cb32-628" aria-hidden="true" tabindex="-1"></a><span class="in">```{r plot human h2}</span></span>
<span id="cb32-629"><a href="#cb32-629" aria-hidden="true" tabindex="-1"></a>human_h2<span class="sc">$</span>credible_set_1 <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(<span class="fu">substr</span>(human_h2<span class="sc">$</span>pve_CI ,<span class="dv">1</span>,<span class="dv">1</span>) <span class="sc">==</span> <span class="st">"-"</span>, <span class="fu">paste</span>(<span class="st">"-"</span>, <span class="fu">sapply</span>(<span class="fu">strsplit</span>(human_h2<span class="sc">$</span>pve_CI, <span class="st">"-"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">2</span>), <span class="at">sep =</span> <span class="st">""</span>), <span class="fu">sapply</span>(<span class="fu">strsplit</span>(human_h2<span class="sc">$</span>pve_CI, <span class="st">"-"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)))</span>
<span id="cb32-630"><a href="#cb32-630" aria-hidden="true" tabindex="-1"></a>human_h2<span class="sc">$</span>credible_set_2 <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(<span class="fu">substr</span>(human_h2<span class="sc">$</span>pve_CI ,<span class="dv">1</span>,<span class="dv">1</span>) <span class="sc">==</span> <span class="st">"-"</span>, <span class="fu">sapply</span>(<span class="fu">strsplit</span>(human_h2<span class="sc">$</span>pve_CI, <span class="st">"-"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">3</span>), <span class="fu">sapply</span>(<span class="fu">strsplit</span>(human_h2<span class="sc">$</span>pve_CI, <span class="st">"-"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">2</span>)))</span>
<span id="cb32-631"><a href="#cb32-631" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-632"><a href="#cb32-632" aria-hidden="true" tabindex="-1"></a>human_h2 <span class="ot">&lt;-</span> human_h2 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ensid <span class="sc">%in%</span> overlap<span class="sc">$</span>ensid)</span>
<span id="cb32-633"><a href="#cb32-633" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-634"><a href="#cb32-634" aria-hidden="true" tabindex="-1"></a>hum_df <span class="ot">&lt;-</span> <span class="fu">load_pve</span>(human_h2)</span>
<span id="cb32-635"><a href="#cb32-635" aria-hidden="true" tabindex="-1"></a>plt_2  <span class="ot">&lt;-</span> (<span class="fu">ggplot</span>(<span class="at">data =</span> hum_df, <span class="fu">aes</span>(<span class="at">x =</span> index))</span>
<span id="cb32-636"><a href="#cb32-636" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>index, <span class="at">y=</span>en_r2), <span class="at">colour =</span> <span class="st">"maroon1"</span>, <span class="at">size =</span> <span class="fl">0.2</span>)</span>
<span id="cb32-637"><a href="#cb32-637" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> point_estimate), <span class="at">lwd =</span> <span class="fl">1.5</span>)</span>
<span id="cb32-638"><a href="#cb32-638" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype=</span><span class="dv">2</span>)</span>
<span id="cb32-639"><a href="#cb32-639" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">geom_ribbon</span>(<span class="fu">aes</span>(<span class="at">ymax =</span> credible_set_2, <span class="at">ymin =</span> credible_set_1),</span>
<span id="cb32-640"><a href="#cb32-640" aria-hidden="true" tabindex="-1"></a>                         <span class="at">alpha =</span> <span class="fl">0.25</span>)</span>
<span id="cb32-641"><a href="#cb32-641" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">'Genes Sorted by Proportion of Variance Explained (PVE)'</span>,</span>
<span id="cb32-642"><a href="#cb32-642" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y =</span> <span class="st">'PVE'</span>)</span>
<span id="cb32-643"><a href="#cb32-643" aria-hidden="true" tabindex="-1"></a>          <span class="sc">+</span> <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>))</span>
<span id="cb32-644"><a href="#cb32-644" aria-hidden="true" tabindex="-1"></a>           <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">1690</span>, <span class="at">y =</span> <span class="fl">0.9</span>, <span class="at">label =</span> <span class="st">"Mean h2 =  0.124"</span>, <span class="at">size =</span> <span class="dv">6</span>)</span>
<span id="cb32-645"><a href="#cb32-645" aria-hidden="true" tabindex="-1"></a>           <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">1710</span>, <span class="at">y =</span> <span class="fl">0.8</span>, <span class="at">label =</span> <span class="st">"Mean R2 =  0.114"</span>, <span class="at">size =</span> <span class="dv">6</span>))</span>
<span id="cb32-646"><a href="#cb32-646" aria-hidden="true" tabindex="-1"></a>plt_2 </span>
<span id="cb32-647"><a href="#cb32-647" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb32-648"><a href="#cb32-648" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-651"><a href="#cb32-651" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb32-652"><a href="#cb32-652" aria-hidden="true" tabindex="-1"></a>all_h2 <span class="ot">&lt;-</span> <span class="fu">tibble</span>(gene)</span>
<span id="cb32-653"><a href="#cb32-653" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(l <span class="cf">in</span> all) {</span>
<span id="cb32-654"><a href="#cb32-654" aria-hidden="true" tabindex="-1"></a>  l <span class="ot">&lt;-</span> l <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, point_estimate))</span>
<span id="cb32-655"><a href="#cb32-655" aria-hidden="true" tabindex="-1"></a>  all_h2 <span class="ot">&lt;-</span> <span class="fu">full_join</span>(all_h2, l, <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb32-656"><a href="#cb32-656" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-657"><a href="#cb32-657" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>