<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Natasha Santhanam">
<meta name="dcterms.date" content="2022-02-07">

<title>Rat Genomics Pipeline Rendered - 03.PrediXcan_MultiXcan</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-listing/list.min.js"></script>
<script src="../site_libs/quarto-listing/quarto-listing.js"></script>
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script>

  window.document.addEventListener("DOMContentLoaded", function (_event) {
    const listingTargetEl = window.document.querySelector('#listing-listing .list');
    if (!listingTargetEl) {
      // No listing discovered, do not attach.
      return; 
    }

    const options = {
      valueNames: [{ data: ['index'] },{ data: ['categories'] },{ data: ['listing-date-sort'] },{ data: ['listing-file-modified-sort'] }],
      
      searchColumns: [],
    };

    window['quarto-listings'] = window['quarto-listings'] || {};
    window['quarto-listings']['listing-listing'] = new List('listing-listing', options);

    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  });

  window.addEventListener('hashchange',() => {
    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  })
  </script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Rat Genomics Pipeline Rendered</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html" rel="" target="">
 <span class="menu-text">paper_code</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../other_code.html" rel="" target="">
 <span class="menu-text">more_code</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#run-predixcan-with-metabolic-phenotype-data" id="toc-run-predixcan-with-metabolic-phenotype-data" class="nav-link active" data-scroll-target="#run-predixcan-with-metabolic-phenotype-data">Run PrediXcan with Metabolic Phenotype Data</a></li>
  <li><a href="#create-model-with-right-column-names-for-predixcan-do-for-all-5-tissues" id="toc-create-model-with-right-column-names-for-predixcan-do-for-all-5-tissues" class="nav-link" data-scroll-target="#create-model-with-right-column-names-for-predixcan-do-for-all-5-tissues">Create model with right column names for PrediXcan (do for all 5 tissues)</a></li>
  <li><a href="#do-for-all-5-tissues" id="toc-do-for-all-5-tissues" class="nav-link" data-scroll-target="#do-for-all-5-tissues">Do for all 5 tissues</a>
  <ul class="collapse">
  <li><a href="#run-multixcan" id="toc-run-multixcan" class="nav-link" data-scroll-target="#run-multixcan">Run MultiXcan</a></li>
  </ul></li>
  <li><a href="#generate-folder-with-predicted-expression-data-for-each-tissue" id="toc-generate-folder-with-predicted-expression-data-for-each-tissue" class="nav-link" data-scroll-target="#generate-folder-with-predicted-expression-data-for-each-tissue">Generate Folder with Predicted Expression Data for each Tissue</a></li>
  <li><a href="#add-zscore-to-mutlixcan" id="toc-add-zscore-to-mutlixcan" class="nav-link" data-scroll-target="#add-zscore-to-mutlixcan">Add Zscore to MutliXcan</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">03.PrediXcan_MultiXcan</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Natasha Santhanam </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 7, 2022</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(broom)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RSQLite)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(qqman)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">source_gist</span>(<span class="st">"0ddc9c0ea03245bb30efbe3e899897be"</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="st">"%&amp;%"</span> <span class="ot">=</span> <span class="cf">function</span>(a,b) <span class="fu">paste</span>(a,b,<span class="at">sep=</span><span class="st">""</span>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>args <span class="ot">=</span> <span class="fu">commandArgs</span>(<span class="at">trailingOnly=</span><span class="cn">TRUE</span>)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>geno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/rat_genotypes_LD_pruned_0.95/"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="run-predixcan-with-metabolic-phenotype-data" class="level2">
<h2 class="anchored" data-anchor-id="run-predixcan-with-metabolic-phenotype-data">Run PrediXcan with Metabolic Phenotype Data</h2>
</section>
<section id="create-model-with-right-column-names-for-predixcan-do-for-all-5-tissues" class="level1">
<h1>Create model with right column names for PrediXcan (do for all 5 tissues)</h1>
<p>Sabrina’s note: I have no idea what is going on here, I don’t know where bimbam file came from. I tried running this script from genotype files generated from previous steps in the pipeline, but it seems like a different file was inputed below. We may be able to ignore this step for now, since we already have PrediXcan formatted genotype from an earlier step in the pipeline.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(geno.dir, <span class="at">pattern =</span> <span class="st">".bimbam"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co">#ids for rats are in the phenotype file under rat_rfid</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(file <span class="cf">in</span> filelist) {</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">fread</span>(geno.dir <span class="sc">%&amp;%</span> file)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> tempo <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">chr =</span>  <span class="fu">numextract</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(tempo<span class="sc">$</span>V1, <span class="st">":"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)), <span class="at">.before =</span> V1) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">pos =</span> <span class="fu">numextract</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(tempo<span class="sc">$</span>V1, <span class="st">":"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">2</span>)), <span class="at">.before =</span> V2) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">maf =</span> <span class="dv">0</span>, <span class="at">.before =</span> V4)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_tsv</span>(tempo, geno.dir <span class="sc">%&amp;%</span> <span class="fu">substr</span>(file, <span class="dv">1</span>, <span class="fu">nchar</span>(file) <span class="sc">-</span> <span class="dv">7</span>) <span class="sc">%&amp;%</span> <span class="st">".txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>tis <span class="ot">=</span> <span class="st">"Ac"</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>data.dir <span class="ot">&lt;-</span> <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/Results/"</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>filename <span class="ot">&lt;-</span>  data.dir <span class="sc">%&amp;%</span> <span class="st">"sql/"</span> <span class="sc">%&amp;%</span> tis <span class="sc">%&amp;%</span> <span class="st">"_output_db.db"</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>sqlite.driver <span class="ot">&lt;-</span> <span class="fu">dbDriver</span>(<span class="st">"SQLite"</span>)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>conn <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(), filename)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>extra <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>(conn, <span class="st">'select * from extra'</span>)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>weights <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>(conn, <span class="st">'select * from weights'</span>)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dbDisconnect</span>(conn)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>extra <span class="ot">&lt;-</span> extra <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, genename, n.snps, R2, pval)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">pred.perf.qval =</span> <span class="cn">NA</span>) </span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(extra) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"gene"</span>, <span class="st">"genename"</span>, <span class="st">"n.snps.in.model"</span>, <span class="st">"pred.perf.R2"</span>, <span class="st">"pred.perf.pval"</span>, <span class="st">"pred.perf.qval"</span>)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>cor_df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(data.dir <span class="sc">%&amp;%</span> <span class="st">"all_results_"</span> <span class="sc">%&amp;%</span> tis, <span class="at">col_names =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, cvm))</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>extra <span class="ot">&lt;-</span> extra <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pred.perf.R2 <span class="sc">&gt;</span> <span class="fl">0.01</span>)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>extra <span class="ot">&lt;-</span> <span class="fu">full_join</span>(extra, cor_df, <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>extra <span class="ot">&lt;-</span> extra <span class="sc">%&gt;%</span> <span class="fu">filter</span>(cvm <span class="sc">&gt;=</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">is.na</span>(cvm)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(cvm)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(pred.perf.R2))</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>weights <span class="ot">&lt;-</span> weights <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene <span class="sc">%in%</span> extra<span class="sc">$</span>gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>model_db <span class="ot">=</span> data.dir <span class="sc">%&amp;%</span> <span class="st">"sql/"</span> <span class="sc">%&amp;%</span> tis <span class="sc">%&amp;%</span> <span class="st">"_best_prediXcan_db.db"</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>conn <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(), model_db)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dbWriteTable</span>(conn, <span class="st">"weights"</span>, weights)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dbWriteTable</span>(conn, <span class="st">"extra"</span>, extra)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co">#check to see model is set up </span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="fu">dbListTables</span>(conn)</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dbGetQuery</span>(conn, <span class="st">'SELECT * FROM weights'</span>) <span class="sc">%&gt;%</span> head</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="fu">dbGetQuery</span>(conn, <span class="st">'SELECT * FROM extra'</span>) <span class="sc">%&gt;%</span> head</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="fu">dbDisconnect</span>(conn)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Note: I don’t know how these files were generated.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(GENO <span class="sc">%&amp;%</span> <span class="st">"samples_Rat_metab_phenos_file"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>all_rats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(PHENO <span class="sc">%&amp;%</span> <span class="st">"all_names.txt"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> samples <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>(X1  <span class="sc">%in%</span> all_rats<span class="sc">$</span>ID))</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(samples, GENO <span class="sc">%&amp;%</span> <span class="st">"samples_Rat_metab_abrv_phenos_file"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="do-for-all-5-tissues" class="level1">
<h1>Do for all 5 tissues</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co">#run prediXcan</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate imlabtools</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="va">METAXCAN</span><span class="op">=</span>/Users/sabrinami/Github/MetaXcan/software</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="va">GENO</span><span class="op">=</span>/Users/sabrinami/Box/imlab-data/data-Github/Rat_Genomics_Paper_Pipeline/data/rat_genotypes_LD_pruned_0.95</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="va">MODEL</span><span class="op">=</span>/Users/sabrinami/Box/imlab-data/data-Github/Rat_Genomics_Paper_Pipeline/Results/sql</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="va">OUTPUT</span><span class="op">=</span>/Users/sabrinami/Box/imlab-data/data-Github/Rat_Genomics_Paper_Pipeline/Results/PrediXcan/metabolic_traits/</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="ex">python</span> <span class="va">$METAXCAN</span>/Predict.py <span class="dt">\</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>--model_db_path <span class="va">$MODEL</span>/Ac_best_prediXcan_db.db <span class="dt">\</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>--text_genotypes <span class="dt">\</span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a> <span class="va">$GENO</span>/chr<span class="pp">*</span>.round2_impute2_3473.txt  <span class="dt">\</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>--on_the_fly_mapping METADATA <span class="st">"{}_{}_{}_{}"</span> <span class="dt">\</span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>--text_sample_ids <span class="va">$GENO</span>/samples_Rat_metab_phenos_file <span class="dt">\</span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>--prediction_output <span class="va">$OUTPUT</span>/rat_metabolic_Ac_best__predict.txt  <span class="dt">\</span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>--prediction_summary_output <span class="va">$OUTPUT</span>/rat_metabolic_Ac_best__summary.txt <span class="dt">\</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>--throw</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Note: Stop here, we have all the results needed to continue to PTRS analysis.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>pred_expr <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/metabolic_traits/rat_metabolic_"</span> <span class="sc">%&amp;%</span> tis <span class="sc">%&amp;%</span> <span class="st">"_best__predict.txt"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>all_rats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(PHENO <span class="sc">%&amp;%</span> <span class="st">"all_names.txt"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>pred_expr <span class="ot">&lt;-</span> pred_expr <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>(FID  <span class="sc">%in%</span> all_rats<span class="sc">$</span>ID))</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co">#write_tsv(pred_expr, "/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/metabolic_traits/rat_metabolic_" %&amp;% tis %&amp;% "_best__predict.txt", col_names = TRUE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">#run asssociation in prediXcan</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>PHENO <span class="ot">=</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/"</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>RESULTS <span class="ot">=</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/metabolic_traits/associations/"</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(PHENO <span class="sc">%&amp;%</span> <span class="st">"processed_obesity_rat_Palmer_phenotypes.csv"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>(rat_rfid  <span class="sc">%in%</span> all_rats<span class="sc">$</span>ID))</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(pheno, PHENO <span class="sc">%&amp;%</span> <span class="st">"processed_obesity_rat_Palmer_phenotypes_target_set.tsv"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(pheno))){</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>trait <span class="ot">&lt;-</span> <span class="fu">colnames</span>(pheno)[i]</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>runLOC <span class="ot">&lt;-</span> <span class="st">"python3 "</span> <span class="sc">%&amp;%</span> METAXCAN <span class="sc">%&amp;%</span> <span class="st">"/PrediXcanAssociation.py "</span> <span class="sc">%&amp;%</span> <span class="st">"--expression_file "</span> <span class="sc">%&amp;%</span>  <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/metabolic_traits/rat_metabolic_"</span> <span class="sc">%&amp;%</span> <span class="st">"Ac"</span> <span class="sc">%&amp;%</span> <span class="st">"__predict.txt --input_phenos_file "</span> <span class="sc">%&amp;%</span> PHENO <span class="sc">%&amp;%</span> <span class="st">"processed_obesity_rat_Palmer_phenotypes_target_set.tsv "</span> <span class="sc">%&amp;%</span> <span class="st">"--input_phenos_column "</span> <span class="sc">%&amp;%</span> trait <span class="sc">%&amp;%</span>  <span class="st">" --output "</span> <span class="sc">%&amp;%</span> RESULTS <span class="sc">%&amp;%</span> <span class="st">"associations/"</span> <span class="sc">%&amp;%</span> <span class="st">"Ac"</span> <span class="sc">%&amp;%</span> <span class="st">"__association_"</span> <span class="sc">%&amp;%</span> trait <span class="sc">%&amp;%</span> <span class="st">".txt --verbosity 9 --throw"</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(runLOC)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">Rscript</span> <span class="at">--vanilla</span> /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/metabolic_trait_assoc_all_tis.R <span class="va">$tissue</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="ex">qsub</span> <span class="at">-v</span> tissue=<span class="va">$tis</span> metabolic_assoc_all_tissues.pbs</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Format and Plot PrediXcan Results</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>results.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/metabolic_traits/associations"</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(results.dir, <span class="at">pattern =</span> <span class="st">"Ac__association_"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  assoc_fila <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(fila, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  pheno_id <span class="ot">&lt;-</span> <span class="fu">substr</span>(fila, <span class="dv">104</span>, (<span class="fu">str_length</span>(fila) <span class="sc">-</span> <span class="dv">4</span>))</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">cbind</span>(assoc_fila, <span class="at">metabolic_trait=</span>pheno_id) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(status))</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>  full_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(full_df, tempo)</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>} </span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a><span class="co">#full_df &lt;- read_tsv("/Users/natashasanthanam/Github/rat-genomic-analysis/data/rat_metabolic_traits_best_Ac_full_assocs.txt", col_names = TRUE)</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>tempo_df <span class="ot">&lt;-</span> full_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pvalue <span class="sc">&lt;</span>  <span class="fl">9.279881e-06</span>)</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="co">#428 sig genes</span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>tempo_df <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>())</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a><span class="co">#all 11 traits</span></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>tempo_df <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(metabolic_trait) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Filter prediXcan results for Supplementary Table</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> full_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(metabolic_trait <span class="sc">==</span> <span class="st">"bmi_bodylength_w_tail"</span><span class="sc">|</span>metabolic_trait <span class="sc">==</span>  <span class="st">"bodylength_w_tail"</span><span class="sc">|</span> metabolic_trait <span class="sc">==</span> <span class="st">"bodyweight"</span> <span class="sc">|</span> metabolic_trait <span class="sc">==</span> <span class="st">"fasting_glucose"</span> <span class="sc">|</span> metabolic_trait <span class="sc">==</span> <span class="st">"epifat"</span> <span class="sc">|</span>  metabolic_trait <span class="sc">==</span> <span class="st">"retrofat"</span> <span class="sc">|</span> metabolic_trait <span class="sc">==</span> <span class="st">"parafat"</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> full_df <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">gene_name =</span> orth.rats[<span class="fu">match</span>(full_df<span class="sc">$</span>gene, orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene),<span class="dv">4</span>]<span class="sc">$</span>rnorvegicus_homolog_associated_gene_name, <span class="at">.before =</span> effect)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>metabolic_trait[full_df<span class="sc">$</span>metabolic_trait <span class="sc">==</span> <span class="st">"bmi_bodylength_w_tail"</span> ] <span class="ot">&lt;-</span> <span class="st">"Body Mass Index (BMI) with tail"</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>metabolic_trait[full_df<span class="sc">$</span>metabolic_trait <span class="sc">==</span> <span class="st">"bodylength_w_tail"</span> ] <span class="ot">&lt;-</span> <span class="st">"Body length including tail"</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>metabolic_trait[full_df<span class="sc">$</span>metabolic_trait <span class="sc">==</span> <span class="st">"bodyweight"</span> ] <span class="ot">&lt;-</span> <span class="st">"Body weight"</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>metabolic_trait[full_df<span class="sc">$</span>metabolic_trait <span class="sc">==</span> <span class="st">"fasting_glucose"</span> ] <span class="ot">&lt;-</span> <span class="st">"Fasting Glucose"</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>metabolic_trait[full_df<span class="sc">$</span>metabolic_trait <span class="sc">==</span> <span class="st">"epifat"</span> ] <span class="ot">&lt;-</span> <span class="st">"Epididymal fat"</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>metabolic_trait[full_df<span class="sc">$</span>metabolic_trait <span class="sc">==</span> <span class="st">"retrofat"</span> ] <span class="ot">&lt;-</span> <span class="st">"Retroperitoneal fat"</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>metabolic_trait[full_df<span class="sc">$</span>metabolic_trait <span class="sc">==</span> <span class="st">"parafat"</span> ] <span class="ot">&lt;-</span> <span class="st">"Parametrial fat"</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> full_df <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">bf_sig =</span> <span class="fu">ifelse</span>(full_df<span class="sc">$</span>pvalue <span class="sc">&lt;=</span> <span class="fl">9.279881e-06</span>, <span class="st">"Yes"</span>, <span class="st">"No"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>human_height_genes <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Downloads/Human_phenomeXcan_all_traits.txt"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>human_height_genes <span class="ot">&lt;-</span> human_height_genes <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">rat_gene =</span> orth.rats[<span class="fu">match</span>(human_height_genes<span class="sc">$</span>gene_name, orth.rats<span class="sc">$</span>external_gene_name), <span class="dv">4</span>]<span class="sc">$</span>rnorvegicus_homolog_associated_gene_name) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pvalue_Height <span class="sc">&lt;=</span> <span class="fl">0.01</span>)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>human_bmi_genes <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Downloads/Human_phenomeXcan_all_traits.txt"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>) </span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(human_bmi_genes)[<span class="dv">2</span>] <span class="ot">=</span> <span class="st">"pvalue_BMI"</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>human_bmi_genes <span class="ot">&lt;-</span> human_bmi_genes <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">rat_gene =</span> orth.rats[<span class="fu">match</span>(human_bmi_genes<span class="sc">$</span>gene_name, orth.rats<span class="sc">$</span>external_gene_name), <span class="dv">4</span>]<span class="sc">$</span>rnorvegicus_homolog_associated_gene_name) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pvalue_BMI <span class="sc">&lt;=</span> <span class="fl">0.01</span> )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>gene_annot <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"/Users/natashasanthanam/Github/rat-genomic-analysis/data/gene_annotation.RDS"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(<span class="st">"chr"</span>, <span class="st">"gene_id"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>)) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">gene =</span> gene_id)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>tempo_manhatt <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(gene_annot, full_df, <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>tempo_manhatt<span class="sc">$</span>chr <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(tempo_manhatt<span class="sc">$</span>chr)</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>bmi_manhat <span class="ot">&lt;-</span> tempo_manhatt <span class="sc">%&gt;%</span> <span class="fu">filter</span>(metabolic_trait <span class="sc">==</span> <span class="st">"Body Mass Index (BMI) with tail"</span>) </span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>bmi_manhat <span class="ot">&lt;-</span> bmi_manhat <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">gene_name =</span> orth.rats[<span class="fu">match</span>(bmi_manhat<span class="sc">$</span>gene, orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene), <span class="dv">4</span>]<span class="sc">$</span>rnorvegicus_homolog_associated_gene_name)</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>height_manhat <span class="ot">&lt;-</span> tempo_manhatt <span class="sc">%&gt;%</span> <span class="fu">filter</span>(metabolic_trait <span class="sc">==</span> <span class="st">"Body length including tail"</span>) </span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>height_manhat <span class="ot">&lt;-</span> height_manhat <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">gene_name =</span> orth.rats[<span class="fu">match</span>(height_manhat<span class="sc">$</span>gene, orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene), <span class="dv">4</span>]<span class="sc">$</span>rnorvegicus_homolog_associated_gene_name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>data_cum <span class="ot">&lt;-</span> bmi_manhat <span class="sc">%&gt;%</span> </span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(chr) <span class="sc">%&gt;%</span> </span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">max_bp =</span> <span class="fu">as.numeric</span>(<span class="fu">max</span>(start))) <span class="sc">%&gt;%</span> </span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">bp_add =</span> <span class="fu">lag</span>(<span class="fu">cumsum</span>(max_bp), <span class="at">default =</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(chr, bp_add)</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>gwas_data <span class="ot">&lt;-</span> bmi_manhat <span class="sc">%&gt;%</span> </span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(data_cum, <span class="at">by =</span> <span class="st">"chr"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">bp_cum =</span> start <span class="sc">+</span> bp_add)</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>axis_set <span class="ot">&lt;-</span> gwas_data <span class="sc">%&gt;%</span> </span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(chr) <span class="sc">%&gt;%</span> </span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">center =</span> <span class="fu">mean</span>(bp_cum))</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>ylim <span class="ot">&lt;-</span> gwas_data <span class="sc">%&gt;%</span> </span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pvalue <span class="sc">==</span> <span class="fu">min</span>(pvalue)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ylim =</span> <span class="fu">abs</span>(<span class="fu">floor</span>(<span class="fu">log10</span>(pvalue))) <span class="sc">+</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(ylim)</span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>sig <span class="ot">&lt;-</span>  <span class="fl">0.05</span><span class="sc">/</span>(<span class="dv">5388</span>)</span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>bmi_manhplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(gwas_data, <span class="fu">aes</span>(<span class="at">x =</span> bp_cum, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pvalue), </span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">color =</span> <span class="fu">as_factor</span>(chr), <span class="at">size =</span> <span class="sc">-</span><span class="fu">log10</span>(pvalue))) <span class="sc">+</span></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(sig), <span class="at">color =</span> <span class="st">"grey40"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span> </span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.0001</span>), <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span> </span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.75</span>, <span class="at">shape =</span> <span class="fu">ifelse</span>((gwas_data<span class="sc">$</span>zscore <span class="sc">&gt;=</span> <span class="fl">4.863456</span>), <span class="dv">17</span>, <span class="fu">ifelse</span>(gwas_data<span class="sc">$</span>zscore <span class="sc">&lt;=</span> <span class="sc">-</span><span class="fl">4.863456</span>, <span class="dv">25</span>, <span class="dv">19</span>)), <span class="at">fill =</span> <span class="st">"dodgerblue4"</span>) <span class="sc">+</span></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label_repel</span>(<span class="fu">aes</span>(<span class="at">label=</span><span class="fu">ifelse</span>((pvalue <span class="sc">&lt;=</span>  sig <span class="sc">&amp;</span> gene_name <span class="sc">%in%</span> human_bmi_genes<span class="sc">$</span>rat_gene), gene_name, <span class="st">""</span>)), <span class="at">size =</span> <span class="dv">6</span>) <span class="sc">+</span> </span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">8</span>)) <span class="sc">+</span> </span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">label =</span> axis_set<span class="sc">$</span>chr, <span class="at">breaks =</span> axis_set<span class="sc">$</span>center) <span class="sc">+</span></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">"dodgerblue4"</span>, <span class="st">"midnightblue"</span>), <span class="fu">unique</span>(<span class="fu">length</span>(axis_set<span class="sc">$</span>chr)))) <span class="sc">+</span></span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size_continuous</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="fl">0.5</span>,<span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, </span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="sc">-</span>log[<span class="dv">10</span>](<span class="fu">italic</span>(p)))) <span class="sc">+</span> </span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>( </span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">size =</span> <span class="dv">12</span>), </span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">12</span>,  <span class="at">vjust =</span> <span class="dv">0</span>), </span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>data_cum <span class="ot">&lt;-</span> height_manhat <span class="sc">%&gt;%</span> </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(chr) <span class="sc">%&gt;%</span> </span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">max_bp =</span> <span class="fu">as.numeric</span>(<span class="fu">max</span>(start))) <span class="sc">%&gt;%</span> </span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">bp_add =</span> <span class="fu">lag</span>(<span class="fu">cumsum</span>(max_bp), <span class="at">default =</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(chr, bp_add)</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>gwas_data <span class="ot">&lt;-</span> height_manhat <span class="sc">%&gt;%</span> </span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(data_cum, <span class="at">by =</span> <span class="st">"chr"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">bp_cum =</span> start <span class="sc">+</span> bp_add)</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>axis_set <span class="ot">&lt;-</span> gwas_data <span class="sc">%&gt;%</span> </span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(chr) <span class="sc">%&gt;%</span> </span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">center =</span> <span class="fu">mean</span>(bp_cum))</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>ylim <span class="ot">&lt;-</span> gwas_data <span class="sc">%&gt;%</span> </span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pvalue <span class="sc">==</span> <span class="fu">min</span>(pvalue)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ylim =</span> <span class="fu">abs</span>(<span class="fu">floor</span>(<span class="fu">log10</span>(pvalue))) <span class="sc">+</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(ylim)</span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>sig <span class="ot">&lt;-</span>   <span class="fl">0.05</span><span class="sc">/</span>(<span class="dv">5388</span>)</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a>height_manhplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(gwas_data, <span class="fu">aes</span>(<span class="at">x =</span> bp_cum, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pvalue), </span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">color =</span> <span class="fu">as_factor</span>(chr), <span class="at">size =</span> <span class="sc">-</span><span class="fu">log10</span>(pvalue))) <span class="sc">+</span></span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(sig), <span class="at">color =</span> <span class="st">"grey40"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span> </span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.0001</span>), <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span> </span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.75</span>, <span class="at">shape =</span> <span class="fu">ifelse</span>((gwas_data<span class="sc">$</span>zscore <span class="sc">&gt;=</span> <span class="fl">4.863456</span>), <span class="dv">17</span>, <span class="fu">ifelse</span>(gwas_data<span class="sc">$</span>zscore <span class="sc">&lt;=</span> <span class="sc">-</span><span class="fl">4.863456</span>, <span class="dv">25</span>, <span class="dv">19</span>)), <span class="at">fill =</span> <span class="st">"dodgerblue4"</span>) <span class="sc">+</span></span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label_repel</span>(<span class="fu">aes</span>(<span class="at">label=</span><span class="fu">ifelse</span>((pvalue <span class="sc">&lt;=</span>  sig <span class="sc">&amp;</span> gene_name <span class="sc">%in%</span> human_height_genes<span class="sc">$</span>rat_gene), gene_name, <span class="st">""</span>)), <span class="at">size =</span> <span class="dv">6</span>) <span class="sc">+</span> </span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">10</span>)) <span class="sc">+</span> </span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">label =</span> axis_set<span class="sc">$</span>chr, <span class="at">breaks =</span> axis_set<span class="sc">$</span>center) <span class="sc">+</span></span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">"dodgerblue4"</span>, <span class="st">"midnightblue"</span>), <span class="fu">unique</span>(<span class="fu">length</span>(axis_set<span class="sc">$</span>chr)))) <span class="sc">+</span></span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size_continuous</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="fl">0.5</span>,<span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, </span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="sc">-</span>log[<span class="dv">10</span>](<span class="fu">italic</span>(p)))) <span class="sc">+</span> </span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>( </span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb15-38"><a href="#cb15-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb15-39"><a href="#cb15-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">size =</span> <span class="dv">12</span>), </span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">12</span>,  <span class="at">vjust =</span> <span class="dv">0</span>), </span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="run-multixcan" class="level2">
<h2 class="anchored" data-anchor-id="run-multixcan">Run MultiXcan</h2>
</section>
</section>
<section id="generate-folder-with-predicted-expression-data-for-each-tissue" class="level1">
<h1>Generate Folder with Predicted Expression Data for each Tissue</h1>
<p>First have to remove potential overlap between genotypes used in predicted expression and those in phenotype file. Should only be around 60ish so not too big a deal</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/expression"</span>, <span class="at">pattern =</span> <span class="st">".RDS"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>all_names <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ID =</span> <span class="fu">as.character</span>())</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(fila)</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">rownames</span>(tempo)) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">ID =</span> <span class="st">`</span><span class="at">rownames(tempo)</span><span class="st">`</span>)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  all_names <span class="ot">&lt;-</span> <span class="fu">full_join</span>(tempo, all_names, <span class="at">by =</span> <span class="st">"ID"</span>)</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/processed_obesity_rat_Palmer_phenotypes.csv"</span>, <span class="at">col_names=</span><span class="cn">TRUE</span>)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">ID =</span> rat_rfid) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>ID <span class="sc">%in%</span> all_names<span class="sc">$</span>ID)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Next have to remove overlap rats predicted expression as well</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/metabolic_traits/"</span>, <span class="at">pattern =</span> <span class="st">"__predict.txt"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">fread</span>(fila, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  name <span class="ot">&lt;-</span> <span class="fu">substr</span>(fila, <span class="dv">90</span>,<span class="dv">91</span>)</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> tempo <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>FID <span class="sc">%in%</span> all_names<span class="sc">$</span>ID)</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> tempo[<span class="fu">match</span>(pheno<span class="sc">$</span>ID, tempo<span class="sc">$</span>FID),]</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_tsv</span>(tempo, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/MultiXcan/expr/"</span> <span class="sc">%&amp;%</span> name <span class="sc">%&amp;%</span> <span class="st">".txt"</span>)</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Run MultiXcan using the predicted expression from prediXcan across all 5 tissues to boost power</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="co">#PBS -N multixcan</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="co">#PBS -S /bin/bash</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="co">#PBS -l walltime=4:00:00</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="co">#PBS -l mem=4gb</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="co">#PBS -l nodes=1:ppn=1</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a><span class="co"># SPECIFY LOGGING BEHAVIOR</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a><span class="co">#PBS -o /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/MultiXcan/logs/${PBS_JOBNAME}.${PBS_JOBID}.log</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a><span class="co">#PBS -e /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/MultiXcan/logs/${PBS_JOBNAME}.${PBS_JOBID}.err</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load gcc/6.2.0</span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> ~/.bashrc </span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate /gpfs/data/im-lab/nas40t2/bin/envs/tensorqtl/</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"MultiXcan running on epifat"</span></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a><span class="ex">python</span> /gpfs/data/im-lab/nas40t2/natasha/GTEX_Analysis/MetaXcan/software/MulTiXcan.py <span class="dt">\</span></span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a>        <span class="at">--expression_folder</span> /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/MultiXcan/expr <span class="dt">\</span></span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a>        <span class="at">--expression_pattern</span> <span class="st">"(.*).txt"</span> <span class="dt">\</span></span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">--input_phenos_file</span> /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/MultiXcan/metabolic_trait_phenos_MultiXcan.txt <span class="dt">\</span></span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">--input_phenos_column</span> bmi_bodylength_wo_tail <span class="dt">\</span></span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">--output</span> /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/MultiXcan/results/bmi_bodylength_wo_tail_predict_assoc.txt <span class="dt">\</span></span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true" tabindex="-1"></a>        <span class="at">--pc_condition_number</span> 10 <span class="dt">\</span></span>
<span id="cb19-27"><a href="#cb19-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">--mode</span> linear <span class="dt">\</span></span>
<span id="cb19-28"><a href="#cb19-28" aria-hidden="true" tabindex="-1"></a>        <span class="at">--verbosity</span> 8 <span class="dt">\</span></span>
<span id="cb19-29"><a href="#cb19-29" aria-hidden="true" tabindex="-1"></a>        <span class="at">--throw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="add-zscore-to-mutlixcan" class="level1">
<h1>Add Zscore to MutliXcan</h1>
<p>Calculate most significant Zscore across all tisuses For each trait find most significant pvalue and take sign of that effect</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/processed_obesity_rat_Palmer_phenotypes.csv"</span>, <span class="at">col_names=</span><span class="cn">TRUE</span>)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(pheno)) {</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>  trait <span class="ot">&lt;-</span> <span class="fu">colnames</span>(pheno)[i]</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>  filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(data.dir <span class="sc">%&amp;%</span> <span class="st">"prediXcan/metabolic_traits/associations/"</span>, <span class="at">pattern =</span> trait <span class="sc">%&amp;%</span> <span class="st">".txt"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene=</span> <span class="fu">as.character</span>())</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>    tis <span class="ot">&lt;-</span> <span class="fu">substr</span>(fila, <span class="dv">89</span>, <span class="dv">90</span>)</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(fila) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, effect, pvalue)) </span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>    new_eff <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"effect"</span>, tis, <span class="at">sep =</span> <span class="st">"_"</span>)</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>    new_pval <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"pvalue"</span>, tis, <span class="at">sep =</span> <span class="st">"_"</span>)</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(df)[<span class="dv">2</span>] <span class="ot">&lt;-</span> new_eff</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(df)[<span class="dv">3</span>] <span class="ot">&lt;-</span> new_pval </span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>   tempo <span class="ot">&lt;-</span> <span class="fu">full_join</span>(tempo, df,  <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>  most_sig <span class="ot">=</span> <span class="fu">rowMins</span>(<span class="fu">as.matrix</span>(tempo[,<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">7</span>,<span class="dv">9</span>,<span class="dv">11</span>)]))</span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>  Ac <span class="ot">&lt;-</span> tempo[<span class="fu">na.omit</span>(<span class="fu">match</span>(most_sig, tempo<span class="sc">$</span>pvalue_Ac)), <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)] <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">effect =</span> effect_Ac )</span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>  Il <span class="ot">&lt;-</span> tempo[<span class="fu">na.omit</span>(<span class="fu">match</span>(most_sig, tempo<span class="sc">$</span>pvalue_Il)), <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span>)] <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">effect =</span> effect_Il )</span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>  Lh <span class="ot">&lt;-</span> tempo[<span class="fu">na.omit</span>(<span class="fu">match</span>(most_sig, tempo<span class="sc">$</span>pvalue_Lh)), <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">6</span>)] <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">effect =</span> effect_Lh )</span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>  Pl <span class="ot">&lt;-</span> tempo[<span class="fu">na.omit</span>(<span class="fu">match</span>(most_sig, tempo<span class="sc">$</span>pvalue_Pl)), <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">8</span>)] <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">effect =</span> effect_Pl )</span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a>  Vo <span class="ot">&lt;-</span> tempo[<span class="fu">na.omit</span>(<span class="fu">match</span>(most_sig, tempo<span class="sc">$</span>pvalue_Vo)), <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">10</span>)] <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">effect =</span> effect_Vo )</span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(Ac, Il, Lh, Pl, Vo) </span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">sign =</span> <span class="fu">sign</span>(effect))</span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_tsv</span>(df, data.dir <span class="sc">%&amp;%</span> <span class="st">"prediXcan/metabolic_traits/associations/most_sig_zscores/"</span> <span class="sc">%&amp;%</span> trait <span class="sc">%&amp;%</span> <span class="st">"_avg_zscore.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Loci Analysis</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">source_gist</span>(<span class="st">"50a2bdc64e103e8321fefb9e712aa137"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>gene_annot <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"/Users/natashasanthanam/Github/rat-genomic-analysis/data/gene_annotation.RDS"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(chr, gene_id, gene_name, start, end))</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>height_loci <span class="ot">&lt;-</span> full_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(metabolic_trait <span class="sc">==</span> <span class="st">"Body length including tail"</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pvalue <span class="sc">&lt;=</span> <span class="fl">9.279881e-06</span>)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>height_loci <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(gene_annot, height_loci <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, pvalue)) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">gene_id =</span> gene), <span class="at">by =</span> <span class="st">"gene_id"</span>)</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>height_loci<span class="sc">$</span>chr <span class="ot">=</span> <span class="fu">as.numeric</span>(height_loci<span class="sc">$</span>chr)</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>height_loci <span class="ot">&lt;-</span>  height_loci[<span class="fu">order</span>(height_loci<span class="sc">$</span>chr),]</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>height_distinct_loci <span class="ot">&lt;-</span> <span class="fu">fn_count_distinct_loci</span>(height_loci)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>bmi_loci <span class="ot">&lt;-</span> full_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(metabolic_trait <span class="sc">==</span> <span class="st">"Body Mass Index (BMI) with tail"</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pvalue <span class="sc">&lt;=</span> <span class="fl">9.279881e-06</span>)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>bmi_loci <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(gene_annot, bmi_loci <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, pvalue)) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">gene_id =</span> gene), <span class="at">by =</span> <span class="st">"gene_id"</span>)</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>bmi_loci<span class="sc">$</span>chr <span class="ot">=</span> <span class="fu">as.numeric</span>(bmi_loci<span class="sc">$</span>chr)</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>bmi_loci <span class="ot">&lt;-</span>  bmi_loci[<span class="fu">order</span>(bmi_loci<span class="sc">$</span>chr),]</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>bmi_distinct_loci <span class="ot">&lt;-</span> <span class="fu">fn_count_distinct_loci</span>(bmi_loci)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>



<!-- -->

</section>

<div class="quarto-listing quarto-listing-container-default" id="listing-listing">
<div class="list quarto-listing-default">

</div>
<div class="listing-no-matching d-none">
No matching items
</div>
</div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb24" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "03.PrediXcan_MultiXcan"</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Natasha Santhanam"</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2/7/2022"</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="an">output:</span><span class="co"> html_document</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup, eval=FALSE}</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(broom)</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RSQLite)</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(qqman)</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">source_gist</span>(<span class="st">"0ddc9c0ea03245bb30efbe3e899897be"</span>)</span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a><span class="st">"%&amp;%"</span> <span class="ot">=</span> <span class="cf">function</span>(a,b) <span class="fu">paste</span>(a,b,<span class="at">sep=</span><span class="st">""</span>)</span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>args <span class="ot">=</span> <span class="fu">commandArgs</span>(<span class="at">trailingOnly=</span><span class="cn">TRUE</span>)</span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>geno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/rat_genotypes_LD_pruned_0.95/"</span></span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a><span class="fu">## Run PrediXcan with Metabolic Phenotype Data</span></span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a><span class="fu"># Create model with right column names for PrediXcan (do for all 5 tissues)</span></span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>Sabrina's note: I have no idea what is going on here, I don't know where bimbam file came from. I tried running this script from genotype files generated from previous steps in the pipeline, but it seems like a different file was inputed below. We may be able to ignore this step for now, since we already have PrediXcan formatted genotype from an earlier step in the pipeline.</span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a><span class="in">```{r create txt genotypes, eval=FALSE}</span></span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(geno.dir, <span class="at">pattern =</span> <span class="st">".bimbam"</span>)</span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a><span class="co">#ids for rats are in the phenotype file under rat_rfid</span></span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(file <span class="cf">in</span> filelist) {</span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">fread</span>(geno.dir <span class="sc">%&amp;%</span> file)</span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> tempo <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">chr =</span>  <span class="fu">numextract</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(tempo<span class="sc">$</span>V1, <span class="st">":"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)), <span class="at">.before =</span> V1) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">pos =</span> <span class="fu">numextract</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(tempo<span class="sc">$</span>V1, <span class="st">":"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">2</span>)), <span class="at">.before =</span> V2) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">maf =</span> <span class="dv">0</span>, <span class="at">.before =</span> V4)</span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_tsv</span>(tempo, geno.dir <span class="sc">%&amp;%</span> <span class="fu">substr</span>(file, <span class="dv">1</span>, <span class="fu">nchar</span>(file) <span class="sc">-</span> <span class="dv">7</span>) <span class="sc">%&amp;%</span> <span class="st">".txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-38"><a href="#cb24-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-39"><a href="#cb24-39" aria-hidden="true" tabindex="-1"></a><span class="in">```{r change colnames of Ac prediction model, eval=FALSE}</span></span>
<span id="cb24-40"><a href="#cb24-40" aria-hidden="true" tabindex="-1"></a>tis <span class="ot">=</span> <span class="st">"Ac"</span></span>
<span id="cb24-41"><a href="#cb24-41" aria-hidden="true" tabindex="-1"></a>data.dir <span class="ot">&lt;-</span> <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/Results/"</span></span>
<span id="cb24-42"><a href="#cb24-42" aria-hidden="true" tabindex="-1"></a>filename <span class="ot">&lt;-</span>  data.dir <span class="sc">%&amp;%</span> <span class="st">"sql/"</span> <span class="sc">%&amp;%</span> tis <span class="sc">%&amp;%</span> <span class="st">"_output_db.db"</span></span>
<span id="cb24-43"><a href="#cb24-43" aria-hidden="true" tabindex="-1"></a>sqlite.driver <span class="ot">&lt;-</span> <span class="fu">dbDriver</span>(<span class="st">"SQLite"</span>)</span>
<span id="cb24-44"><a href="#cb24-44" aria-hidden="true" tabindex="-1"></a>conn <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(), filename)</span>
<span id="cb24-45"><a href="#cb24-45" aria-hidden="true" tabindex="-1"></a>extra <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>(conn, <span class="st">'select * from extra'</span>)</span>
<span id="cb24-46"><a href="#cb24-46" aria-hidden="true" tabindex="-1"></a>weights <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>(conn, <span class="st">'select * from weights'</span>)</span>
<span id="cb24-47"><a href="#cb24-47" aria-hidden="true" tabindex="-1"></a><span class="fu">dbDisconnect</span>(conn)</span>
<span id="cb24-48"><a href="#cb24-48" aria-hidden="true" tabindex="-1"></a>extra <span class="ot">&lt;-</span> extra <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, genename, n.snps, R2, pval)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">pred.perf.qval =</span> <span class="cn">NA</span>) </span>
<span id="cb24-49"><a href="#cb24-49" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(extra) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"gene"</span>, <span class="st">"genename"</span>, <span class="st">"n.snps.in.model"</span>, <span class="st">"pred.perf.R2"</span>, <span class="st">"pred.perf.pval"</span>, <span class="st">"pred.perf.qval"</span>)</span>
<span id="cb24-50"><a href="#cb24-50" aria-hidden="true" tabindex="-1"></a>cor_df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(data.dir <span class="sc">%&amp;%</span> <span class="st">"all_results_"</span> <span class="sc">%&amp;%</span> tis, <span class="at">col_names =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, cvm))</span>
<span id="cb24-51"><a href="#cb24-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-52"><a href="#cb24-52" aria-hidden="true" tabindex="-1"></a>extra <span class="ot">&lt;-</span> extra <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pred.perf.R2 <span class="sc">&gt;</span> <span class="fl">0.01</span>)</span>
<span id="cb24-53"><a href="#cb24-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-54"><a href="#cb24-54" aria-hidden="true" tabindex="-1"></a>extra <span class="ot">&lt;-</span> <span class="fu">full_join</span>(extra, cor_df, <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb24-55"><a href="#cb24-55" aria-hidden="true" tabindex="-1"></a>extra <span class="ot">&lt;-</span> extra <span class="sc">%&gt;%</span> <span class="fu">filter</span>(cvm <span class="sc">&gt;=</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">is.na</span>(cvm)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(cvm)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(pred.perf.R2))</span>
<span id="cb24-56"><a href="#cb24-56" aria-hidden="true" tabindex="-1"></a>weights <span class="ot">&lt;-</span> weights <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene <span class="sc">%in%</span> extra<span class="sc">$</span>gene)</span>
<span id="cb24-57"><a href="#cb24-57" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-58"><a href="#cb24-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-59"><a href="#cb24-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-60"><a href="#cb24-60" aria-hidden="true" tabindex="-1"></a><span class="in">```{r create database connection, eval=FALSE}</span></span>
<span id="cb24-61"><a href="#cb24-61" aria-hidden="true" tabindex="-1"></a>model_db <span class="ot">=</span> data.dir <span class="sc">%&amp;%</span> <span class="st">"sql/"</span> <span class="sc">%&amp;%</span> tis <span class="sc">%&amp;%</span> <span class="st">"_best_prediXcan_db.db"</span></span>
<span id="cb24-62"><a href="#cb24-62" aria-hidden="true" tabindex="-1"></a>conn <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(), model_db)</span>
<span id="cb24-63"><a href="#cb24-63" aria-hidden="true" tabindex="-1"></a><span class="fu">dbWriteTable</span>(conn, <span class="st">"weights"</span>, weights)</span>
<span id="cb24-64"><a href="#cb24-64" aria-hidden="true" tabindex="-1"></a><span class="fu">dbWriteTable</span>(conn, <span class="st">"extra"</span>, extra)</span>
<span id="cb24-65"><a href="#cb24-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-66"><a href="#cb24-66" aria-hidden="true" tabindex="-1"></a><span class="co">#check to see model is set up </span></span>
<span id="cb24-67"><a href="#cb24-67" aria-hidden="true" tabindex="-1"></a><span class="fu">dbListTables</span>(conn)</span>
<span id="cb24-68"><a href="#cb24-68" aria-hidden="true" tabindex="-1"></a><span class="fu">dbGetQuery</span>(conn, <span class="st">'SELECT * FROM weights'</span>) <span class="sc">%&gt;%</span> head</span>
<span id="cb24-69"><a href="#cb24-69" aria-hidden="true" tabindex="-1"></a><span class="fu">dbGetQuery</span>(conn, <span class="st">'SELECT * FROM extra'</span>) <span class="sc">%&gt;%</span> head</span>
<span id="cb24-70"><a href="#cb24-70" aria-hidden="true" tabindex="-1"></a><span class="fu">dbDisconnect</span>(conn)</span>
<span id="cb24-71"><a href="#cb24-71" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-72"><a href="#cb24-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-73"><a href="#cb24-73" aria-hidden="true" tabindex="-1"></a>Note: I don't know how these files were generated.</span>
<span id="cb24-74"><a href="#cb24-74" aria-hidden="true" tabindex="-1"></a><span class="in">```{r filter samples for no overlap of rats, eval=FALSE}</span></span>
<span id="cb24-75"><a href="#cb24-75" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(GENO <span class="sc">%&amp;%</span> <span class="st">"samples_Rat_metab_phenos_file"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb24-76"><a href="#cb24-76" aria-hidden="true" tabindex="-1"></a>all_rats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(PHENO <span class="sc">%&amp;%</span> <span class="st">"all_names.txt"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-77"><a href="#cb24-77" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> samples <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>(X1  <span class="sc">%in%</span> all_rats<span class="sc">$</span>ID))</span>
<span id="cb24-78"><a href="#cb24-78" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(samples, GENO <span class="sc">%&amp;%</span> <span class="st">"samples_Rat_metab_abrv_phenos_file"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb24-79"><a href="#cb24-79" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-80"><a href="#cb24-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-81"><a href="#cb24-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-82"><a href="#cb24-82" aria-hidden="true" tabindex="-1"></a><span class="fu"># Do for all 5 tissues </span></span>
<span id="cb24-83"><a href="#cb24-83" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash run prediXcan, eval = FALSE}</span></span>
<span id="cb24-84"><a href="#cb24-84" aria-hidden="true" tabindex="-1"></a><span class="co">#run prediXcan</span></span>
<span id="cb24-85"><a href="#cb24-85" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate imlabtools</span>
<span id="cb24-86"><a href="#cb24-86" aria-hidden="true" tabindex="-1"></a><span class="va">METAXCAN</span><span class="op">=</span>/Users/sabrinami/Github/MetaXcan/software</span>
<span id="cb24-87"><a href="#cb24-87" aria-hidden="true" tabindex="-1"></a><span class="va">GENO</span><span class="op">=</span>/Users/sabrinami/Box/imlab-data/data-Github/Rat_Genomics_Paper_Pipeline/data/rat_genotypes_LD_pruned_0.95</span>
<span id="cb24-88"><a href="#cb24-88" aria-hidden="true" tabindex="-1"></a><span class="va">MODEL</span><span class="op">=</span>/Users/sabrinami/Box/imlab-data/data-Github/Rat_Genomics_Paper_Pipeline/Results/sql</span>
<span id="cb24-89"><a href="#cb24-89" aria-hidden="true" tabindex="-1"></a><span class="va">OUTPUT</span><span class="op">=</span>/Users/sabrinami/Box/imlab-data/data-Github/Rat_Genomics_Paper_Pipeline/Results/PrediXcan/metabolic_traits/</span>
<span id="cb24-90"><a href="#cb24-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-91"><a href="#cb24-91" aria-hidden="true" tabindex="-1"></a><span class="ex">python</span> <span class="va">$METAXCAN</span>/Predict.py <span class="dt">\</span></span>
<span id="cb24-92"><a href="#cb24-92" aria-hidden="true" tabindex="-1"></a>--model_db_path <span class="va">$MODEL</span>/Ac_best_prediXcan_db.db <span class="dt">\</span></span>
<span id="cb24-93"><a href="#cb24-93" aria-hidden="true" tabindex="-1"></a>--text_genotypes <span class="dt">\</span></span>
<span id="cb24-94"><a href="#cb24-94" aria-hidden="true" tabindex="-1"></a> <span class="va">$GENO</span>/chr<span class="pp">*</span>.round2_impute2_3473.txt  <span class="dt">\</span></span>
<span id="cb24-95"><a href="#cb24-95" aria-hidden="true" tabindex="-1"></a>--on_the_fly_mapping METADATA <span class="st">"{}_{}_{}_{}"</span> <span class="dt">\</span></span>
<span id="cb24-96"><a href="#cb24-96" aria-hidden="true" tabindex="-1"></a>--text_sample_ids <span class="va">$GENO</span>/samples_Rat_metab_phenos_file <span class="dt">\</span></span>
<span id="cb24-97"><a href="#cb24-97" aria-hidden="true" tabindex="-1"></a>--prediction_output <span class="va">$OUTPUT</span>/rat_metabolic_Ac_best__predict.txt  <span class="dt">\</span></span>
<span id="cb24-98"><a href="#cb24-98" aria-hidden="true" tabindex="-1"></a>--prediction_summary_output <span class="va">$OUTPUT</span>/rat_metabolic_Ac_best__summary.txt <span class="dt">\</span></span>
<span id="cb24-99"><a href="#cb24-99" aria-hidden="true" tabindex="-1"></a>--throw</span>
<span id="cb24-100"><a href="#cb24-100" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-101"><a href="#cb24-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-102"><a href="#cb24-102" aria-hidden="true" tabindex="-1"></a>Note: Stop here, we have all the results needed to continue to PTRS analysis.</span>
<span id="cb24-103"><a href="#cb24-103" aria-hidden="true" tabindex="-1"></a><span class="in">```{r }</span></span>
<span id="cb24-104"><a href="#cb24-104" aria-hidden="true" tabindex="-1"></a>pred_expr <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/metabolic_traits/rat_metabolic_"</span> <span class="sc">%&amp;%</span> tis <span class="sc">%&amp;%</span> <span class="st">"_best__predict.txt"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-105"><a href="#cb24-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-106"><a href="#cb24-106" aria-hidden="true" tabindex="-1"></a>all_rats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(PHENO <span class="sc">%&amp;%</span> <span class="st">"all_names.txt"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-107"><a href="#cb24-107" aria-hidden="true" tabindex="-1"></a>pred_expr <span class="ot">&lt;-</span> pred_expr <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>(FID  <span class="sc">%in%</span> all_rats<span class="sc">$</span>ID))</span>
<span id="cb24-108"><a href="#cb24-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-109"><a href="#cb24-109" aria-hidden="true" tabindex="-1"></a><span class="co">#write_tsv(pred_expr, "/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/metabolic_traits/rat_metabolic_" %&amp;% tis %&amp;% "_best__predict.txt", col_names = TRUE)</span></span>
<span id="cb24-110"><a href="#cb24-110" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-111"><a href="#cb24-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-112"><a href="#cb24-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-113"><a href="#cb24-113" aria-hidden="true" tabindex="-1"></a><span class="in">```{r run associations, eval=FALSE}</span></span>
<span id="cb24-114"><a href="#cb24-114" aria-hidden="true" tabindex="-1"></a><span class="co">#run asssociation in prediXcan</span></span>
<span id="cb24-115"><a href="#cb24-115" aria-hidden="true" tabindex="-1"></a>PHENO <span class="ot">=</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/"</span></span>
<span id="cb24-116"><a href="#cb24-116" aria-hidden="true" tabindex="-1"></a>RESULTS <span class="ot">=</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/metabolic_traits/associations/"</span></span>
<span id="cb24-117"><a href="#cb24-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-118"><a href="#cb24-118" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(PHENO <span class="sc">%&amp;%</span> <span class="st">"processed_obesity_rat_Palmer_phenotypes.csv"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>(rat_rfid  <span class="sc">%in%</span> all_rats<span class="sc">$</span>ID))</span>
<span id="cb24-119"><a href="#cb24-119" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(pheno, PHENO <span class="sc">%&amp;%</span> <span class="st">"processed_obesity_rat_Palmer_phenotypes_target_set.tsv"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-120"><a href="#cb24-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-121"><a href="#cb24-121" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(pheno))){</span>
<span id="cb24-122"><a href="#cb24-122" aria-hidden="true" tabindex="-1"></a>trait <span class="ot">&lt;-</span> <span class="fu">colnames</span>(pheno)[i]</span>
<span id="cb24-123"><a href="#cb24-123" aria-hidden="true" tabindex="-1"></a>runLOC <span class="ot">&lt;-</span> <span class="st">"python3 "</span> <span class="sc">%&amp;%</span> METAXCAN <span class="sc">%&amp;%</span> <span class="st">"/PrediXcanAssociation.py "</span> <span class="sc">%&amp;%</span> <span class="st">"--expression_file "</span> <span class="sc">%&amp;%</span>  <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/metabolic_traits/rat_metabolic_"</span> <span class="sc">%&amp;%</span> <span class="st">"Ac"</span> <span class="sc">%&amp;%</span> <span class="st">"__predict.txt --input_phenos_file "</span> <span class="sc">%&amp;%</span> PHENO <span class="sc">%&amp;%</span> <span class="st">"processed_obesity_rat_Palmer_phenotypes_target_set.tsv "</span> <span class="sc">%&amp;%</span> <span class="st">"--input_phenos_column "</span> <span class="sc">%&amp;%</span> trait <span class="sc">%&amp;%</span>  <span class="st">" --output "</span> <span class="sc">%&amp;%</span> RESULTS <span class="sc">%&amp;%</span> <span class="st">"associations/"</span> <span class="sc">%&amp;%</span> <span class="st">"Ac"</span> <span class="sc">%&amp;%</span> <span class="st">"__association_"</span> <span class="sc">%&amp;%</span> trait <span class="sc">%&amp;%</span> <span class="st">".txt --verbosity 9 --throw"</span></span>
<span id="cb24-124"><a href="#cb24-124" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(runLOC)</span>
<span id="cb24-125"><a href="#cb24-125" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-126"><a href="#cb24-126" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-127"><a href="#cb24-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-128"><a href="#cb24-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-129"><a href="#cb24-129" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash submit script for all tissues}</span></span>
<span id="cb24-130"><a href="#cb24-130" aria-hidden="true" tabindex="-1"></a><span class="ex">Rscript</span> <span class="at">--vanilla</span> /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/metabolic_trait_assoc_all_tis.R <span class="va">$tissue</span></span>
<span id="cb24-131"><a href="#cb24-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-132"><a href="#cb24-132" aria-hidden="true" tabindex="-1"></a><span class="ex">qsub</span> <span class="at">-v</span> tissue=<span class="va">$tis</span> metabolic_assoc_all_tissues.pbs</span>
<span id="cb24-133"><a href="#cb24-133" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-134"><a href="#cb24-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-135"><a href="#cb24-135" aria-hidden="true" tabindex="-1"></a>Format and Plot PrediXcan Results</span>
<span id="cb24-136"><a href="#cb24-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-137"><a href="#cb24-137" aria-hidden="true" tabindex="-1"></a><span class="in">```{r format prediXcan assocs, eval=FALSE}</span></span>
<span id="cb24-138"><a href="#cb24-138" aria-hidden="true" tabindex="-1"></a>results.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/metabolic_traits/associations"</span></span>
<span id="cb24-139"><a href="#cb24-139" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(results.dir, <span class="at">pattern =</span> <span class="st">"Ac__association_"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-140"><a href="#cb24-140" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb24-141"><a href="#cb24-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-142"><a href="#cb24-142" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb24-143"><a href="#cb24-143" aria-hidden="true" tabindex="-1"></a>  assoc_fila <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(fila, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-144"><a href="#cb24-144" aria-hidden="true" tabindex="-1"></a>  pheno_id <span class="ot">&lt;-</span> <span class="fu">substr</span>(fila, <span class="dv">104</span>, (<span class="fu">str_length</span>(fila) <span class="sc">-</span> <span class="dv">4</span>))</span>
<span id="cb24-145"><a href="#cb24-145" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">cbind</span>(assoc_fila, <span class="at">metabolic_trait=</span>pheno_id) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(status))</span>
<span id="cb24-146"><a href="#cb24-146" aria-hidden="true" tabindex="-1"></a>  full_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(full_df, tempo)</span>
<span id="cb24-147"><a href="#cb24-147" aria-hidden="true" tabindex="-1"></a>} </span>
<span id="cb24-148"><a href="#cb24-148" aria-hidden="true" tabindex="-1"></a><span class="co">#full_df &lt;- read_tsv("/Users/natashasanthanam/Github/rat-genomic-analysis/data/rat_metabolic_traits_best_Ac_full_assocs.txt", col_names = TRUE)</span></span>
<span id="cb24-149"><a href="#cb24-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-150"><a href="#cb24-150" aria-hidden="true" tabindex="-1"></a>tempo_df <span class="ot">&lt;-</span> full_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pvalue <span class="sc">&lt;</span>  <span class="fl">9.279881e-06</span>)</span>
<span id="cb24-151"><a href="#cb24-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-152"><a href="#cb24-152" aria-hidden="true" tabindex="-1"></a><span class="co">#428 sig genes</span></span>
<span id="cb24-153"><a href="#cb24-153" aria-hidden="true" tabindex="-1"></a>tempo_df <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>())</span>
<span id="cb24-154"><a href="#cb24-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-155"><a href="#cb24-155" aria-hidden="true" tabindex="-1"></a><span class="co">#all 11 traits</span></span>
<span id="cb24-156"><a href="#cb24-156" aria-hidden="true" tabindex="-1"></a>tempo_df <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(metabolic_trait) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>())</span>
<span id="cb24-157"><a href="#cb24-157" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-158"><a href="#cb24-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-159"><a href="#cb24-159" aria-hidden="true" tabindex="-1"></a>Filter prediXcan results for Supplementary Table</span>
<span id="cb24-162"><a href="#cb24-162" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb24-163"><a href="#cb24-163" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> full_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(metabolic_trait <span class="sc">==</span> <span class="st">"bmi_bodylength_w_tail"</span><span class="sc">|</span>metabolic_trait <span class="sc">==</span>  <span class="st">"bodylength_w_tail"</span><span class="sc">|</span> metabolic_trait <span class="sc">==</span> <span class="st">"bodyweight"</span> <span class="sc">|</span> metabolic_trait <span class="sc">==</span> <span class="st">"fasting_glucose"</span> <span class="sc">|</span> metabolic_trait <span class="sc">==</span> <span class="st">"epifat"</span> <span class="sc">|</span>  metabolic_trait <span class="sc">==</span> <span class="st">"retrofat"</span> <span class="sc">|</span> metabolic_trait <span class="sc">==</span> <span class="st">"parafat"</span>)</span>
<span id="cb24-164"><a href="#cb24-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-165"><a href="#cb24-165" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> full_df <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">gene_name =</span> orth.rats[<span class="fu">match</span>(full_df<span class="sc">$</span>gene, orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene),<span class="dv">4</span>]<span class="sc">$</span>rnorvegicus_homolog_associated_gene_name, <span class="at">.before =</span> effect)</span>
<span id="cb24-166"><a href="#cb24-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-167"><a href="#cb24-167" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>metabolic_trait[full_df<span class="sc">$</span>metabolic_trait <span class="sc">==</span> <span class="st">"bmi_bodylength_w_tail"</span> ] <span class="ot">&lt;-</span> <span class="st">"Body Mass Index (BMI) with tail"</span></span>
<span id="cb24-168"><a href="#cb24-168" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>metabolic_trait[full_df<span class="sc">$</span>metabolic_trait <span class="sc">==</span> <span class="st">"bodylength_w_tail"</span> ] <span class="ot">&lt;-</span> <span class="st">"Body length including tail"</span></span>
<span id="cb24-169"><a href="#cb24-169" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>metabolic_trait[full_df<span class="sc">$</span>metabolic_trait <span class="sc">==</span> <span class="st">"bodyweight"</span> ] <span class="ot">&lt;-</span> <span class="st">"Body weight"</span></span>
<span id="cb24-170"><a href="#cb24-170" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>metabolic_trait[full_df<span class="sc">$</span>metabolic_trait <span class="sc">==</span> <span class="st">"fasting_glucose"</span> ] <span class="ot">&lt;-</span> <span class="st">"Fasting Glucose"</span></span>
<span id="cb24-171"><a href="#cb24-171" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>metabolic_trait[full_df<span class="sc">$</span>metabolic_trait <span class="sc">==</span> <span class="st">"epifat"</span> ] <span class="ot">&lt;-</span> <span class="st">"Epididymal fat"</span></span>
<span id="cb24-172"><a href="#cb24-172" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>metabolic_trait[full_df<span class="sc">$</span>metabolic_trait <span class="sc">==</span> <span class="st">"retrofat"</span> ] <span class="ot">&lt;-</span> <span class="st">"Retroperitoneal fat"</span></span>
<span id="cb24-173"><a href="#cb24-173" aria-hidden="true" tabindex="-1"></a>full_df<span class="sc">$</span>metabolic_trait[full_df<span class="sc">$</span>metabolic_trait <span class="sc">==</span> <span class="st">"parafat"</span> ] <span class="ot">&lt;-</span> <span class="st">"Parametrial fat"</span></span>
<span id="cb24-174"><a href="#cb24-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-175"><a href="#cb24-175" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> full_df <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">bf_sig =</span> <span class="fu">ifelse</span>(full_df<span class="sc">$</span>pvalue <span class="sc">&lt;=</span> <span class="fl">9.279881e-06</span>, <span class="st">"Yes"</span>, <span class="st">"No"</span>))</span>
<span id="cb24-176"><a href="#cb24-176" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-177"><a href="#cb24-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-178"><a href="#cb24-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-179"><a href="#cb24-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-180"><a href="#cb24-180" aria-hidden="true" tabindex="-1"></a><span class="in">```{r find genes that overlap in Humans, eval=FALSE}</span></span>
<span id="cb24-181"><a href="#cb24-181" aria-hidden="true" tabindex="-1"></a>human_height_genes <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Downloads/Human_phenomeXcan_all_traits.txt"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-182"><a href="#cb24-182" aria-hidden="true" tabindex="-1"></a>human_height_genes <span class="ot">&lt;-</span> human_height_genes <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">rat_gene =</span> orth.rats[<span class="fu">match</span>(human_height_genes<span class="sc">$</span>gene_name, orth.rats<span class="sc">$</span>external_gene_name), <span class="dv">4</span>]<span class="sc">$</span>rnorvegicus_homolog_associated_gene_name) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pvalue_Height <span class="sc">&lt;=</span> <span class="fl">0.01</span>)</span>
<span id="cb24-183"><a href="#cb24-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-184"><a href="#cb24-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-185"><a href="#cb24-185" aria-hidden="true" tabindex="-1"></a>human_bmi_genes <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Downloads/Human_phenomeXcan_all_traits.txt"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>) </span>
<span id="cb24-186"><a href="#cb24-186" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(human_bmi_genes)[<span class="dv">2</span>] <span class="ot">=</span> <span class="st">"pvalue_BMI"</span></span>
<span id="cb24-187"><a href="#cb24-187" aria-hidden="true" tabindex="-1"></a>human_bmi_genes <span class="ot">&lt;-</span> human_bmi_genes <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">rat_gene =</span> orth.rats[<span class="fu">match</span>(human_bmi_genes<span class="sc">$</span>gene_name, orth.rats<span class="sc">$</span>external_gene_name), <span class="dv">4</span>]<span class="sc">$</span>rnorvegicus_homolog_associated_gene_name) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pvalue_BMI <span class="sc">&lt;=</span> <span class="fl">0.01</span> )</span>
<span id="cb24-188"><a href="#cb24-188" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-189"><a href="#cb24-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-190"><a href="#cb24-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-191"><a href="#cb24-191" aria-hidden="true" tabindex="-1"></a><span class="in">```{r plot prediXcan results as miami plot divided per trait}</span></span>
<span id="cb24-192"><a href="#cb24-192" aria-hidden="true" tabindex="-1"></a>gene_annot <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"/Users/natashasanthanam/Github/rat-genomic-analysis/data/gene_annotation.RDS"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(<span class="st">"chr"</span>, <span class="st">"gene_id"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>)) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">gene =</span> gene_id)</span>
<span id="cb24-193"><a href="#cb24-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-194"><a href="#cb24-194" aria-hidden="true" tabindex="-1"></a>tempo_manhatt <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(gene_annot, full_df, <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb24-195"><a href="#cb24-195" aria-hidden="true" tabindex="-1"></a>tempo_manhatt<span class="sc">$</span>chr <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(tempo_manhatt<span class="sc">$</span>chr)</span>
<span id="cb24-196"><a href="#cb24-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-197"><a href="#cb24-197" aria-hidden="true" tabindex="-1"></a>bmi_manhat <span class="ot">&lt;-</span> tempo_manhatt <span class="sc">%&gt;%</span> <span class="fu">filter</span>(metabolic_trait <span class="sc">==</span> <span class="st">"Body Mass Index (BMI) with tail"</span>) </span>
<span id="cb24-198"><a href="#cb24-198" aria-hidden="true" tabindex="-1"></a>bmi_manhat <span class="ot">&lt;-</span> bmi_manhat <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">gene_name =</span> orth.rats[<span class="fu">match</span>(bmi_manhat<span class="sc">$</span>gene, orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene), <span class="dv">4</span>]<span class="sc">$</span>rnorvegicus_homolog_associated_gene_name)</span>
<span id="cb24-199"><a href="#cb24-199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-200"><a href="#cb24-200" aria-hidden="true" tabindex="-1"></a>height_manhat <span class="ot">&lt;-</span> tempo_manhatt <span class="sc">%&gt;%</span> <span class="fu">filter</span>(metabolic_trait <span class="sc">==</span> <span class="st">"Body length including tail"</span>) </span>
<span id="cb24-201"><a href="#cb24-201" aria-hidden="true" tabindex="-1"></a>height_manhat <span class="ot">&lt;-</span> height_manhat <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">gene_name =</span> orth.rats[<span class="fu">match</span>(height_manhat<span class="sc">$</span>gene, orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene), <span class="dv">4</span>]<span class="sc">$</span>rnorvegicus_homolog_associated_gene_name)</span>
<span id="cb24-202"><a href="#cb24-202" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-203"><a href="#cb24-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-204"><a href="#cb24-204" aria-hidden="true" tabindex="-1"></a><span class="in">```{r plot manhattan for BMI}</span></span>
<span id="cb24-205"><a href="#cb24-205" aria-hidden="true" tabindex="-1"></a>data_cum <span class="ot">&lt;-</span> bmi_manhat <span class="sc">%&gt;%</span> </span>
<span id="cb24-206"><a href="#cb24-206" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(chr) <span class="sc">%&gt;%</span> </span>
<span id="cb24-207"><a href="#cb24-207" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">max_bp =</span> <span class="fu">as.numeric</span>(<span class="fu">max</span>(start))) <span class="sc">%&gt;%</span> </span>
<span id="cb24-208"><a href="#cb24-208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">bp_add =</span> <span class="fu">lag</span>(<span class="fu">cumsum</span>(max_bp), <span class="at">default =</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-209"><a href="#cb24-209" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(chr, bp_add)</span>
<span id="cb24-210"><a href="#cb24-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-211"><a href="#cb24-211" aria-hidden="true" tabindex="-1"></a>gwas_data <span class="ot">&lt;-</span> bmi_manhat <span class="sc">%&gt;%</span> </span>
<span id="cb24-212"><a href="#cb24-212" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(data_cum, <span class="at">by =</span> <span class="st">"chr"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-213"><a href="#cb24-213" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">bp_cum =</span> start <span class="sc">+</span> bp_add)</span>
<span id="cb24-214"><a href="#cb24-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-215"><a href="#cb24-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-216"><a href="#cb24-216" aria-hidden="true" tabindex="-1"></a>axis_set <span class="ot">&lt;-</span> gwas_data <span class="sc">%&gt;%</span> </span>
<span id="cb24-217"><a href="#cb24-217" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(chr) <span class="sc">%&gt;%</span> </span>
<span id="cb24-218"><a href="#cb24-218" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">center =</span> <span class="fu">mean</span>(bp_cum))</span>
<span id="cb24-219"><a href="#cb24-219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-220"><a href="#cb24-220" aria-hidden="true" tabindex="-1"></a>ylim <span class="ot">&lt;-</span> gwas_data <span class="sc">%&gt;%</span> </span>
<span id="cb24-221"><a href="#cb24-221" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pvalue <span class="sc">==</span> <span class="fu">min</span>(pvalue)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-222"><a href="#cb24-222" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ylim =</span> <span class="fu">abs</span>(<span class="fu">floor</span>(<span class="fu">log10</span>(pvalue))) <span class="sc">+</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-223"><a href="#cb24-223" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(ylim)</span>
<span id="cb24-224"><a href="#cb24-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-225"><a href="#cb24-225" aria-hidden="true" tabindex="-1"></a>sig <span class="ot">&lt;-</span>  <span class="fl">0.05</span><span class="sc">/</span>(<span class="dv">5388</span>)</span>
<span id="cb24-226"><a href="#cb24-226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-227"><a href="#cb24-227" aria-hidden="true" tabindex="-1"></a>bmi_manhplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(gwas_data, <span class="fu">aes</span>(<span class="at">x =</span> bp_cum, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pvalue), </span>
<span id="cb24-228"><a href="#cb24-228" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">color =</span> <span class="fu">as_factor</span>(chr), <span class="at">size =</span> <span class="sc">-</span><span class="fu">log10</span>(pvalue))) <span class="sc">+</span></span>
<span id="cb24-229"><a href="#cb24-229" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(sig), <span class="at">color =</span> <span class="st">"grey40"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span> </span>
<span id="cb24-230"><a href="#cb24-230" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.0001</span>), <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span> </span>
<span id="cb24-231"><a href="#cb24-231" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.75</span>, <span class="at">shape =</span> <span class="fu">ifelse</span>((gwas_data<span class="sc">$</span>zscore <span class="sc">&gt;=</span> <span class="fl">4.863456</span>), <span class="dv">17</span>, <span class="fu">ifelse</span>(gwas_data<span class="sc">$</span>zscore <span class="sc">&lt;=</span> <span class="sc">-</span><span class="fl">4.863456</span>, <span class="dv">25</span>, <span class="dv">19</span>)), <span class="at">fill =</span> <span class="st">"dodgerblue4"</span>) <span class="sc">+</span></span>
<span id="cb24-232"><a href="#cb24-232" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label_repel</span>(<span class="fu">aes</span>(<span class="at">label=</span><span class="fu">ifelse</span>((pvalue <span class="sc">&lt;=</span>  sig <span class="sc">&amp;</span> gene_name <span class="sc">%in%</span> human_bmi_genes<span class="sc">$</span>rat_gene), gene_name, <span class="st">""</span>)), <span class="at">size =</span> <span class="dv">6</span>) <span class="sc">+</span> </span>
<span id="cb24-233"><a href="#cb24-233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">8</span>)) <span class="sc">+</span> </span>
<span id="cb24-234"><a href="#cb24-234" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">label =</span> axis_set<span class="sc">$</span>chr, <span class="at">breaks =</span> axis_set<span class="sc">$</span>center) <span class="sc">+</span></span>
<span id="cb24-235"><a href="#cb24-235" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">"dodgerblue4"</span>, <span class="st">"midnightblue"</span>), <span class="fu">unique</span>(<span class="fu">length</span>(axis_set<span class="sc">$</span>chr)))) <span class="sc">+</span></span>
<span id="cb24-236"><a href="#cb24-236" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size_continuous</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="fl">0.5</span>,<span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb24-237"><a href="#cb24-237" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, </span>
<span id="cb24-238"><a href="#cb24-238" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="sc">-</span>log[<span class="dv">10</span>](<span class="fu">italic</span>(p)))) <span class="sc">+</span> </span>
<span id="cb24-239"><a href="#cb24-239" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb24-240"><a href="#cb24-240" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>( </span>
<span id="cb24-241"><a href="#cb24-241" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb24-242"><a href="#cb24-242" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb24-243"><a href="#cb24-243" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb24-244"><a href="#cb24-244" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb24-245"><a href="#cb24-245" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">size =</span> <span class="dv">12</span>), </span>
<span id="cb24-246"><a href="#cb24-246" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">12</span>,  <span class="at">vjust =</span> <span class="dv">0</span>), </span>
<span id="cb24-247"><a href="#cb24-247" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>))</span>
<span id="cb24-248"><a href="#cb24-248" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-249"><a href="#cb24-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-250"><a href="#cb24-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-251"><a href="#cb24-251" aria-hidden="true" tabindex="-1"></a><span class="in">```{r plot manhattan for height}</span></span>
<span id="cb24-252"><a href="#cb24-252" aria-hidden="true" tabindex="-1"></a>data_cum <span class="ot">&lt;-</span> height_manhat <span class="sc">%&gt;%</span> </span>
<span id="cb24-253"><a href="#cb24-253" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(chr) <span class="sc">%&gt;%</span> </span>
<span id="cb24-254"><a href="#cb24-254" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">max_bp =</span> <span class="fu">as.numeric</span>(<span class="fu">max</span>(start))) <span class="sc">%&gt;%</span> </span>
<span id="cb24-255"><a href="#cb24-255" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">bp_add =</span> <span class="fu">lag</span>(<span class="fu">cumsum</span>(max_bp), <span class="at">default =</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-256"><a href="#cb24-256" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(chr, bp_add)</span>
<span id="cb24-257"><a href="#cb24-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-258"><a href="#cb24-258" aria-hidden="true" tabindex="-1"></a>gwas_data <span class="ot">&lt;-</span> height_manhat <span class="sc">%&gt;%</span> </span>
<span id="cb24-259"><a href="#cb24-259" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(data_cum, <span class="at">by =</span> <span class="st">"chr"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-260"><a href="#cb24-260" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">bp_cum =</span> start <span class="sc">+</span> bp_add)</span>
<span id="cb24-261"><a href="#cb24-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-262"><a href="#cb24-262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-263"><a href="#cb24-263" aria-hidden="true" tabindex="-1"></a>axis_set <span class="ot">&lt;-</span> gwas_data <span class="sc">%&gt;%</span> </span>
<span id="cb24-264"><a href="#cb24-264" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(chr) <span class="sc">%&gt;%</span> </span>
<span id="cb24-265"><a href="#cb24-265" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">center =</span> <span class="fu">mean</span>(bp_cum))</span>
<span id="cb24-266"><a href="#cb24-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-267"><a href="#cb24-267" aria-hidden="true" tabindex="-1"></a>ylim <span class="ot">&lt;-</span> gwas_data <span class="sc">%&gt;%</span> </span>
<span id="cb24-268"><a href="#cb24-268" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(pvalue <span class="sc">==</span> <span class="fu">min</span>(pvalue)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-269"><a href="#cb24-269" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ylim =</span> <span class="fu">abs</span>(<span class="fu">floor</span>(<span class="fu">log10</span>(pvalue))) <span class="sc">+</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-270"><a href="#cb24-270" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(ylim)</span>
<span id="cb24-271"><a href="#cb24-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-272"><a href="#cb24-272" aria-hidden="true" tabindex="-1"></a>sig <span class="ot">&lt;-</span>   <span class="fl">0.05</span><span class="sc">/</span>(<span class="dv">5388</span>)</span>
<span id="cb24-273"><a href="#cb24-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-274"><a href="#cb24-274" aria-hidden="true" tabindex="-1"></a>height_manhplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(gwas_data, <span class="fu">aes</span>(<span class="at">x =</span> bp_cum, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pvalue), </span>
<span id="cb24-275"><a href="#cb24-275" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">color =</span> <span class="fu">as_factor</span>(chr), <span class="at">size =</span> <span class="sc">-</span><span class="fu">log10</span>(pvalue))) <span class="sc">+</span></span>
<span id="cb24-276"><a href="#cb24-276" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(sig), <span class="at">color =</span> <span class="st">"grey40"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span> </span>
<span id="cb24-277"><a href="#cb24-277" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.0001</span>), <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span> </span>
<span id="cb24-278"><a href="#cb24-278" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.75</span>, <span class="at">shape =</span> <span class="fu">ifelse</span>((gwas_data<span class="sc">$</span>zscore <span class="sc">&gt;=</span> <span class="fl">4.863456</span>), <span class="dv">17</span>, <span class="fu">ifelse</span>(gwas_data<span class="sc">$</span>zscore <span class="sc">&lt;=</span> <span class="sc">-</span><span class="fl">4.863456</span>, <span class="dv">25</span>, <span class="dv">19</span>)), <span class="at">fill =</span> <span class="st">"dodgerblue4"</span>) <span class="sc">+</span></span>
<span id="cb24-279"><a href="#cb24-279" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label_repel</span>(<span class="fu">aes</span>(<span class="at">label=</span><span class="fu">ifelse</span>((pvalue <span class="sc">&lt;=</span>  sig <span class="sc">&amp;</span> gene_name <span class="sc">%in%</span> human_height_genes<span class="sc">$</span>rat_gene), gene_name, <span class="st">""</span>)), <span class="at">size =</span> <span class="dv">6</span>) <span class="sc">+</span> </span>
<span id="cb24-280"><a href="#cb24-280" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">10</span>)) <span class="sc">+</span> </span>
<span id="cb24-281"><a href="#cb24-281" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">label =</span> axis_set<span class="sc">$</span>chr, <span class="at">breaks =</span> axis_set<span class="sc">$</span>center) <span class="sc">+</span></span>
<span id="cb24-282"><a href="#cb24-282" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">"dodgerblue4"</span>, <span class="st">"midnightblue"</span>), <span class="fu">unique</span>(<span class="fu">length</span>(axis_set<span class="sc">$</span>chr)))) <span class="sc">+</span></span>
<span id="cb24-283"><a href="#cb24-283" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size_continuous</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="fl">0.5</span>,<span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb24-284"><a href="#cb24-284" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, </span>
<span id="cb24-285"><a href="#cb24-285" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="sc">-</span>log[<span class="dv">10</span>](<span class="fu">italic</span>(p)))) <span class="sc">+</span> </span>
<span id="cb24-286"><a href="#cb24-286" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb24-287"><a href="#cb24-287" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>( </span>
<span id="cb24-288"><a href="#cb24-288" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb24-289"><a href="#cb24-289" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb24-290"><a href="#cb24-290" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb24-291"><a href="#cb24-291" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb24-292"><a href="#cb24-292" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">size =</span> <span class="dv">12</span>), </span>
<span id="cb24-293"><a href="#cb24-293" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">12</span>,  <span class="at">vjust =</span> <span class="dv">0</span>), </span>
<span id="cb24-294"><a href="#cb24-294" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>))</span>
<span id="cb24-295"><a href="#cb24-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-296"><a href="#cb24-296" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-297"><a href="#cb24-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-298"><a href="#cb24-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-299"><a href="#cb24-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-300"><a href="#cb24-300" aria-hidden="true" tabindex="-1"></a><span class="fu">## Run MultiXcan</span></span>
<span id="cb24-301"><a href="#cb24-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-302"><a href="#cb24-302" aria-hidden="true" tabindex="-1"></a><span class="fu"># Generate Folder with Predicted Expression Data for each Tissue</span></span>
<span id="cb24-303"><a href="#cb24-303" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-304"><a href="#cb24-304" aria-hidden="true" tabindex="-1"></a>First have to remove potential overlap between genotypes used in predicted expression and those in phenotype file. Should only be around 60ish so not too big a deal</span>
<span id="cb24-305"><a href="#cb24-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-306"><a href="#cb24-306" aria-hidden="true" tabindex="-1"></a><span class="in">```{r generate list of ids across all tissues, eval=FALSE}</span></span>
<span id="cb24-307"><a href="#cb24-307" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/expression"</span>, <span class="at">pattern =</span> <span class="st">".RDS"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-308"><a href="#cb24-308" aria-hidden="true" tabindex="-1"></a>all_names <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ID =</span> <span class="fu">as.character</span>())</span>
<span id="cb24-309"><a href="#cb24-309" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb24-310"><a href="#cb24-310" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(fila)</span>
<span id="cb24-311"><a href="#cb24-311" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">rownames</span>(tempo)) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">ID =</span> <span class="st">`</span><span class="at">rownames(tempo)</span><span class="st">`</span>)</span>
<span id="cb24-312"><a href="#cb24-312" aria-hidden="true" tabindex="-1"></a>  all_names <span class="ot">&lt;-</span> <span class="fu">full_join</span>(tempo, all_names, <span class="at">by =</span> <span class="st">"ID"</span>)</span>
<span id="cb24-313"><a href="#cb24-313" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-314"><a href="#cb24-314" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-315"><a href="#cb24-315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-316"><a href="#cb24-316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-317"><a href="#cb24-317" aria-hidden="true" tabindex="-1"></a><span class="in">```{r clean up pheno file, eval=FALSE}</span></span>
<span id="cb24-318"><a href="#cb24-318" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/processed_obesity_rat_Palmer_phenotypes.csv"</span>, <span class="at">col_names=</span><span class="cn">TRUE</span>)</span>
<span id="cb24-319"><a href="#cb24-319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-320"><a href="#cb24-320" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">ID =</span> rat_rfid) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>ID <span class="sc">%in%</span> all_names<span class="sc">$</span>ID)</span>
<span id="cb24-321"><a href="#cb24-321" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-322"><a href="#cb24-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-323"><a href="#cb24-323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-324"><a href="#cb24-324" aria-hidden="true" tabindex="-1"></a>Next have to remove overlap rats predicted expression as well</span>
<span id="cb24-325"><a href="#cb24-325" aria-hidden="true" tabindex="-1"></a><span class="in">```{r  generate expr data, eval=FALSE}</span></span>
<span id="cb24-326"><a href="#cb24-326" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/metabolic_traits/"</span>, <span class="at">pattern =</span> <span class="st">"__predict.txt"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-327"><a href="#cb24-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-328"><a href="#cb24-328" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb24-329"><a href="#cb24-329" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">fread</span>(fila, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb24-330"><a href="#cb24-330" aria-hidden="true" tabindex="-1"></a>  name <span class="ot">&lt;-</span> <span class="fu">substr</span>(fila, <span class="dv">90</span>,<span class="dv">91</span>)</span>
<span id="cb24-331"><a href="#cb24-331" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> tempo <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>FID <span class="sc">%in%</span> all_names<span class="sc">$</span>ID)</span>
<span id="cb24-332"><a href="#cb24-332" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> tempo[<span class="fu">match</span>(pheno<span class="sc">$</span>ID, tempo<span class="sc">$</span>FID),]</span>
<span id="cb24-333"><a href="#cb24-333" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_tsv</span>(tempo, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/MultiXcan/expr/"</span> <span class="sc">%&amp;%</span> name <span class="sc">%&amp;%</span> <span class="st">".txt"</span>)</span>
<span id="cb24-334"><a href="#cb24-334" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-335"><a href="#cb24-335" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-336"><a href="#cb24-336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-337"><a href="#cb24-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-338"><a href="#cb24-338" aria-hidden="true" tabindex="-1"></a>Run MultiXcan using the predicted expression from prediXcan across all 5 tissues to boost power</span>
<span id="cb24-339"><a href="#cb24-339" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash run MultiXcan}</span></span>
<span id="cb24-340"><a href="#cb24-340" aria-hidden="true" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb24-341"><a href="#cb24-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-342"><a href="#cb24-342" aria-hidden="true" tabindex="-1"></a><span class="co">#PBS -N multixcan</span></span>
<span id="cb24-343"><a href="#cb24-343" aria-hidden="true" tabindex="-1"></a><span class="co">#PBS -S /bin/bash</span></span>
<span id="cb24-344"><a href="#cb24-344" aria-hidden="true" tabindex="-1"></a><span class="co">#PBS -l walltime=4:00:00</span></span>
<span id="cb24-345"><a href="#cb24-345" aria-hidden="true" tabindex="-1"></a><span class="co">#PBS -l mem=4gb</span></span>
<span id="cb24-346"><a href="#cb24-346" aria-hidden="true" tabindex="-1"></a><span class="co">#PBS -l nodes=1:ppn=1</span></span>
<span id="cb24-347"><a href="#cb24-347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-348"><a href="#cb24-348" aria-hidden="true" tabindex="-1"></a><span class="co"># SPECIFY LOGGING BEHAVIOR</span></span>
<span id="cb24-349"><a href="#cb24-349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-350"><a href="#cb24-350" aria-hidden="true" tabindex="-1"></a><span class="co">#PBS -o /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/MultiXcan/logs/${PBS_JOBNAME}.${PBS_JOBID}.log</span></span>
<span id="cb24-351"><a href="#cb24-351" aria-hidden="true" tabindex="-1"></a><span class="co">#PBS -e /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/MultiXcan/logs/${PBS_JOBNAME}.${PBS_JOBID}.err</span></span>
<span id="cb24-352"><a href="#cb24-352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-353"><a href="#cb24-353" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load gcc/6.2.0</span>
<span id="cb24-354"><a href="#cb24-354" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> ~/.bashrc </span>
<span id="cb24-355"><a href="#cb24-355" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate /gpfs/data/im-lab/nas40t2/bin/envs/tensorqtl/</span>
<span id="cb24-356"><a href="#cb24-356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-357"><a href="#cb24-357" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"MultiXcan running on epifat"</span></span>
<span id="cb24-358"><a href="#cb24-358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-359"><a href="#cb24-359" aria-hidden="true" tabindex="-1"></a><span class="ex">python</span> /gpfs/data/im-lab/nas40t2/natasha/GTEX_Analysis/MetaXcan/software/MulTiXcan.py <span class="dt">\</span></span>
<span id="cb24-360"><a href="#cb24-360" aria-hidden="true" tabindex="-1"></a>        <span class="at">--expression_folder</span> /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/MultiXcan/expr <span class="dt">\</span></span>
<span id="cb24-361"><a href="#cb24-361" aria-hidden="true" tabindex="-1"></a>        <span class="at">--expression_pattern</span> <span class="st">"(.*).txt"</span> <span class="dt">\</span></span>
<span id="cb24-362"><a href="#cb24-362" aria-hidden="true" tabindex="-1"></a>        <span class="at">--input_phenos_file</span> /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/MultiXcan/metabolic_trait_phenos_MultiXcan.txt <span class="dt">\</span></span>
<span id="cb24-363"><a href="#cb24-363" aria-hidden="true" tabindex="-1"></a>        <span class="at">--input_phenos_column</span> bmi_bodylength_wo_tail <span class="dt">\</span></span>
<span id="cb24-364"><a href="#cb24-364" aria-hidden="true" tabindex="-1"></a>        <span class="at">--output</span> /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/MultiXcan/results/bmi_bodylength_wo_tail_predict_assoc.txt <span class="dt">\</span></span>
<span id="cb24-365"><a href="#cb24-365" aria-hidden="true" tabindex="-1"></a>        <span class="at">--pc_condition_number</span> 10 <span class="dt">\</span></span>
<span id="cb24-366"><a href="#cb24-366" aria-hidden="true" tabindex="-1"></a>        <span class="at">--mode</span> linear <span class="dt">\</span></span>
<span id="cb24-367"><a href="#cb24-367" aria-hidden="true" tabindex="-1"></a>        <span class="at">--verbosity</span> 8 <span class="dt">\</span></span>
<span id="cb24-368"><a href="#cb24-368" aria-hidden="true" tabindex="-1"></a>        <span class="at">--throw</span></span>
<span id="cb24-369"><a href="#cb24-369" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-370"><a href="#cb24-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-371"><a href="#cb24-371" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-372"><a href="#cb24-372" aria-hidden="true" tabindex="-1"></a><span class="fu"># Add Zscore to MutliXcan</span></span>
<span id="cb24-373"><a href="#cb24-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-374"><a href="#cb24-374" aria-hidden="true" tabindex="-1"></a>Calculate most significant  Zscore across all tisuses</span>
<span id="cb24-375"><a href="#cb24-375" aria-hidden="true" tabindex="-1"></a>For each trait find most significant pvalue and take sign of that effect</span>
<span id="cb24-376"><a href="#cb24-376" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-377"><a href="#cb24-377" aria-hidden="true" tabindex="-1"></a><span class="in">```{r find mean Z, eval=FALSE}</span></span>
<span id="cb24-378"><a href="#cb24-378" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/processed_obesity_rat_Palmer_phenotypes.csv"</span>, <span class="at">col_names=</span><span class="cn">TRUE</span>)</span>
<span id="cb24-379"><a href="#cb24-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-380"><a href="#cb24-380" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(pheno)) {</span>
<span id="cb24-381"><a href="#cb24-381" aria-hidden="true" tabindex="-1"></a>  trait <span class="ot">&lt;-</span> <span class="fu">colnames</span>(pheno)[i]</span>
<span id="cb24-382"><a href="#cb24-382" aria-hidden="true" tabindex="-1"></a>  filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(data.dir <span class="sc">%&amp;%</span> <span class="st">"prediXcan/metabolic_traits/associations/"</span>, <span class="at">pattern =</span> trait <span class="sc">%&amp;%</span> <span class="st">".txt"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-383"><a href="#cb24-383" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene=</span> <span class="fu">as.character</span>())</span>
<span id="cb24-384"><a href="#cb24-384" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb24-385"><a href="#cb24-385" aria-hidden="true" tabindex="-1"></a>    tis <span class="ot">&lt;-</span> <span class="fu">substr</span>(fila, <span class="dv">89</span>, <span class="dv">90</span>)</span>
<span id="cb24-386"><a href="#cb24-386" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(fila) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, effect, pvalue)) </span>
<span id="cb24-387"><a href="#cb24-387" aria-hidden="true" tabindex="-1"></a>    new_eff <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"effect"</span>, tis, <span class="at">sep =</span> <span class="st">"_"</span>)</span>
<span id="cb24-388"><a href="#cb24-388" aria-hidden="true" tabindex="-1"></a>    new_pval <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"pvalue"</span>, tis, <span class="at">sep =</span> <span class="st">"_"</span>)</span>
<span id="cb24-389"><a href="#cb24-389" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(df)[<span class="dv">2</span>] <span class="ot">&lt;-</span> new_eff</span>
<span id="cb24-390"><a href="#cb24-390" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(df)[<span class="dv">3</span>] <span class="ot">&lt;-</span> new_pval </span>
<span id="cb24-391"><a href="#cb24-391" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb24-392"><a href="#cb24-392" aria-hidden="true" tabindex="-1"></a>   tempo <span class="ot">&lt;-</span> <span class="fu">full_join</span>(tempo, df,  <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb24-393"><a href="#cb24-393" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb24-394"><a href="#cb24-394" aria-hidden="true" tabindex="-1"></a>  most_sig <span class="ot">=</span> <span class="fu">rowMins</span>(<span class="fu">as.matrix</span>(tempo[,<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">7</span>,<span class="dv">9</span>,<span class="dv">11</span>)]))</span>
<span id="cb24-395"><a href="#cb24-395" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-396"><a href="#cb24-396" aria-hidden="true" tabindex="-1"></a>  Ac <span class="ot">&lt;-</span> tempo[<span class="fu">na.omit</span>(<span class="fu">match</span>(most_sig, tempo<span class="sc">$</span>pvalue_Ac)), <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)] <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">effect =</span> effect_Ac )</span>
<span id="cb24-397"><a href="#cb24-397" aria-hidden="true" tabindex="-1"></a>  Il <span class="ot">&lt;-</span> tempo[<span class="fu">na.omit</span>(<span class="fu">match</span>(most_sig, tempo<span class="sc">$</span>pvalue_Il)), <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span>)] <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">effect =</span> effect_Il )</span>
<span id="cb24-398"><a href="#cb24-398" aria-hidden="true" tabindex="-1"></a>  Lh <span class="ot">&lt;-</span> tempo[<span class="fu">na.omit</span>(<span class="fu">match</span>(most_sig, tempo<span class="sc">$</span>pvalue_Lh)), <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">6</span>)] <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">effect =</span> effect_Lh )</span>
<span id="cb24-399"><a href="#cb24-399" aria-hidden="true" tabindex="-1"></a>  Pl <span class="ot">&lt;-</span> tempo[<span class="fu">na.omit</span>(<span class="fu">match</span>(most_sig, tempo<span class="sc">$</span>pvalue_Pl)), <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">8</span>)] <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">effect =</span> effect_Pl )</span>
<span id="cb24-400"><a href="#cb24-400" aria-hidden="true" tabindex="-1"></a>  Vo <span class="ot">&lt;-</span> tempo[<span class="fu">na.omit</span>(<span class="fu">match</span>(most_sig, tempo<span class="sc">$</span>pvalue_Vo)), <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">10</span>)] <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">effect =</span> effect_Vo )</span>
<span id="cb24-401"><a href="#cb24-401" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-402"><a href="#cb24-402" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(Ac, Il, Lh, Pl, Vo) </span>
<span id="cb24-403"><a href="#cb24-403" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">sign =</span> <span class="fu">sign</span>(effect))</span>
<span id="cb24-404"><a href="#cb24-404" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_tsv</span>(df, data.dir <span class="sc">%&amp;%</span> <span class="st">"prediXcan/metabolic_traits/associations/most_sig_zscores/"</span> <span class="sc">%&amp;%</span> trait <span class="sc">%&amp;%</span> <span class="st">"_avg_zscore.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb24-405"><a href="#cb24-405" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-406"><a href="#cb24-406" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-407"><a href="#cb24-407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-408"><a href="#cb24-408" aria-hidden="true" tabindex="-1"></a>Loci Analysis</span>
<span id="cb24-409"><a href="#cb24-409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-410"><a href="#cb24-410" aria-hidden="true" tabindex="-1"></a><span class="in">```{r function to count distinct genes}</span></span>
<span id="cb24-411"><a href="#cb24-411" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">source_gist</span>(<span class="st">"50a2bdc64e103e8321fefb9e712aa137"</span>)</span>
<span id="cb24-412"><a href="#cb24-412" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-413"><a href="#cb24-413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-414"><a href="#cb24-414" aria-hidden="true" tabindex="-1"></a><span class="in">```{r find distinct loci related to height}</span></span>
<span id="cb24-415"><a href="#cb24-415" aria-hidden="true" tabindex="-1"></a>gene_annot <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"/Users/natashasanthanam/Github/rat-genomic-analysis/data/gene_annotation.RDS"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(chr, gene_id, gene_name, start, end))</span>
<span id="cb24-416"><a href="#cb24-416" aria-hidden="true" tabindex="-1"></a>height_loci <span class="ot">&lt;-</span> full_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(metabolic_trait <span class="sc">==</span> <span class="st">"Body length including tail"</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pvalue <span class="sc">&lt;=</span> <span class="fl">9.279881e-06</span>)</span>
<span id="cb24-417"><a href="#cb24-417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-418"><a href="#cb24-418" aria-hidden="true" tabindex="-1"></a>height_loci <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(gene_annot, height_loci <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, pvalue)) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">gene_id =</span> gene), <span class="at">by =</span> <span class="st">"gene_id"</span>)</span>
<span id="cb24-419"><a href="#cb24-419" aria-hidden="true" tabindex="-1"></a>height_loci<span class="sc">$</span>chr <span class="ot">=</span> <span class="fu">as.numeric</span>(height_loci<span class="sc">$</span>chr)</span>
<span id="cb24-420"><a href="#cb24-420" aria-hidden="true" tabindex="-1"></a>height_loci <span class="ot">&lt;-</span>  height_loci[<span class="fu">order</span>(height_loci<span class="sc">$</span>chr),]</span>
<span id="cb24-421"><a href="#cb24-421" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-422"><a href="#cb24-422" aria-hidden="true" tabindex="-1"></a>height_distinct_loci <span class="ot">&lt;-</span> <span class="fu">fn_count_distinct_loci</span>(height_loci)</span>
<span id="cb24-423"><a href="#cb24-423" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb24-424"><a href="#cb24-424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-425"><a href="#cb24-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-426"><a href="#cb24-426" aria-hidden="true" tabindex="-1"></a><span class="in">```{r find distinct loci related to BMI}</span></span>
<span id="cb24-427"><a href="#cb24-427" aria-hidden="true" tabindex="-1"></a>bmi_loci <span class="ot">&lt;-</span> full_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(metabolic_trait <span class="sc">==</span> <span class="st">"Body Mass Index (BMI) with tail"</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pvalue <span class="sc">&lt;=</span> <span class="fl">9.279881e-06</span>)</span>
<span id="cb24-428"><a href="#cb24-428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-429"><a href="#cb24-429" aria-hidden="true" tabindex="-1"></a>bmi_loci <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(gene_annot, bmi_loci <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, pvalue)) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">gene_id =</span> gene), <span class="at">by =</span> <span class="st">"gene_id"</span>)</span>
<span id="cb24-430"><a href="#cb24-430" aria-hidden="true" tabindex="-1"></a>bmi_loci<span class="sc">$</span>chr <span class="ot">=</span> <span class="fu">as.numeric</span>(bmi_loci<span class="sc">$</span>chr)</span>
<span id="cb24-431"><a href="#cb24-431" aria-hidden="true" tabindex="-1"></a>bmi_loci <span class="ot">&lt;-</span>  bmi_loci[<span class="fu">order</span>(bmi_loci<span class="sc">$</span>chr),]</span>
<span id="cb24-432"><a href="#cb24-432" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-433"><a href="#cb24-433" aria-hidden="true" tabindex="-1"></a>bmi_distinct_loci <span class="ot">&lt;-</span> <span class="fu">fn_count_distinct_loci</span>(bmi_loci)</span>
<span id="cb24-434"><a href="#cb24-434" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>