<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Natasha Santhanam">
<meta name="dcterms.date" content="2022-06-26">

<title>Rat Genomics Pipeline Rendered - 10.PTRS_simulation_weights</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-listing/list.min.js"></script>
<script src="../site_libs/quarto-listing/quarto-listing.js"></script>
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<script>

  window.document.addEventListener("DOMContentLoaded", function (_event) {
    const listingTargetEl = window.document.querySelector('#listing-listing .list');
    if (!listingTargetEl) {
      // No listing discovered, do not attach.
      return; 
    }

    const options = {
      valueNames: [{ data: ['index'] },{ data: ['categories'] },{ data: ['listing-date-sort'] },{ data: ['listing-file-modified-sort'] }],
      
      searchColumns: [],
    };

    window['quarto-listings'] = window['quarto-listings'] || {};
    window['quarto-listings']['listing-listing'] = new List('listing-listing', options);

    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  });

  window.addEventListener('hashchange',() => {
    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  })
  </script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Rat Genomics Pipeline Rendered</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html">
 <span class="menu-text">Home</span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#check-simulation-of-different-weights-for-ptrs-correlation-with-observed-trait" id="toc-check-simulation-of-different-weights-for-ptrs-correlation-with-observed-trait" class="nav-link active" data-scroll-target="#check-simulation-of-different-weights-for-ptrs-correlation-with-observed-trait">Check Simulation of Different Weights for PTRS correlation with observed trait</a></li>
  <li><a href="#first-create-data-frame-of-weights-for-100-variations" id="toc-first-create-data-frame-of-weights-for-100-variations" class="nav-link" data-scroll-target="#first-create-data-frame-of-weights-for-100-variations">First create data frame of weights for 100 variations</a></li>
  <li><a href="#create-predicted-trait-for-sign-permuted-weights" id="toc-create-predicted-trait-for-sign-permuted-weights" class="nav-link" data-scroll-target="#create-predicted-trait-for-sign-permuted-weights">Create predicted trait for sign-permuted weights</a></li>
  <li><a href="#compare-all-values-across-different-weight-sign-permuations-to-actual-bmi-and-height" id="toc-compare-all-values-across-different-weight-sign-permuations-to-actual-bmi-and-height" class="nav-link" data-scroll-target="#compare-all-values-across-different-weight-sign-permuations-to-actual-bmi-and-height">Compare all values across different weight sign permuations to actual BMI and Height</a></li>
  <li><a href="#check-correlation-between-observed-bodylength-and-permuted-sign-weights" id="toc-check-correlation-between-observed-bodylength-and-permuted-sign-weights" class="nav-link" data-scroll-target="#check-correlation-between-observed-bodylength-and-permuted-sign-weights">Check correlation between observed bodylength and permuted sign weights</a></li>
  <li><a href="#compare-performance-of-permuted-weights-with-true-weight" id="toc-compare-performance-of-permuted-weights-with-true-weight" class="nav-link" data-scroll-target="#compare-performance-of-permuted-weights-with-true-weight">Compare performance of permuted weights with true weight</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">10.PTRS_simulation_weights</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Natasha Santhanam </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">June 26, 2022</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="check-simulation-of-different-weights-for-ptrs-correlation-with-observed-trait" class="level2">
<h2 class="anchored" data-anchor-id="check-simulation-of-different-weights-for-ptrs-correlation-with-observed-trait">Check Simulation of Different Weights for PTRS correlation with observed trait</h2>
<p>Read in files</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>weights <span class="ot">&lt;-</span>  <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Downloads/weights.height.annot.tsv"</span>)  </span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>weights <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Downloads/weights.bmi.annot.tsv"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>pred_expr <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/Users/natashasanthanam/Box/imlab-data/data-Github/web-data/2022-06-23-improving-figure-for-rat-ptrs/rat_metabolic_Ac_best__predict.txt"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(FID))</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>pred_expr <span class="ot">&lt;-</span> pred_expr <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(IID, <span class="fu">intersect</span>(<span class="fu">colnames</span>(pred_expr), orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene) ))</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co">#change name to human ensembl id in humans</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(pred_expr)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(pred_expr)] <span class="ot">&lt;-</span> orth.rats[<span class="fu">match</span>(<span class="fu">colnames</span>(pred_expr)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(pred_expr)], orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene), <span class="dv">1</span>] <span class="sc">%&gt;%</span> .[[<span class="st">"ensembl_gene_id"</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="first-create-data-frame-of-weights-for-100-variations" class="level1">
<h1>First create data frame of weights for 100 variations</h1>
<p>We will use model 35 for the simulation for height</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>perm_weights_height <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="at">nrow =</span> <span class="fu">nrow</span>(weights), <span class="at">ncol =</span> <span class="dv">100</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(perm_weights_height)) {</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>  j <span class="ot">=</span> <span class="fu">sample</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>),<span class="fu">nrow</span>(weights),<span class="at">replace=</span>T)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> weights[,<span class="dv">38</span>]<span class="sc">*</span>j</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  perm_weights_height[,i] <span class="ot">=</span> df<span class="sc">$</span>model_36</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>perm_weights_height <span class="ot">&lt;-</span> <span class="fu">cbind</span>(weights<span class="sc">$</span>model_36, perm_weights_height)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>perm_weights_height <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(perm_weights_height) </span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>perm_weights_height <span class="ot">&lt;-</span> perm_weights_height <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">gene_name =</span> weights<span class="sc">$</span>gene_name, <span class="at">.before =</span> <span class="fu">colnames</span>(perm_weights_height)[<span class="dv">1</span>])</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(perm_weights_height)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"gene_name"</span>, <span class="st">"obs_weights"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We will also use model 17 for BMI - since it produced the largest correlation</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>perm_weights_bmi <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="at">nrow =</span> <span class="fu">nrow</span>(weights), <span class="at">ncol =</span> <span class="dv">100</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(perm_weights_bmi)) {</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  j <span class="ot">=</span> <span class="fu">sample</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>),<span class="fu">nrow</span>(weights),<span class="at">replace=</span>T)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> weights[,<span class="dv">19</span>]<span class="sc">*</span>j</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  perm_weights_bmi[,i] <span class="ot">=</span> df<span class="sc">$</span>model_17</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>perm_weights_bmi <span class="ot">&lt;-</span> <span class="fu">cbind</span>(weights<span class="sc">$</span>model_17, perm_weights_bmi)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>perm_weights_bmi <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(perm_weights_bmi) </span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>perm_weights_bmi <span class="ot">&lt;-</span> perm_weights_bmi <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">gene_name =</span> weights<span class="sc">$</span>gene_name, <span class="at">.before =</span> <span class="fu">colnames</span>(perm_weights_bmi)[<span class="dv">1</span>])</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(perm_weights_bmi)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"gene_name"</span>, <span class="st">"obs_weights"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="create-predicted-trait-for-sign-permuted-weights" class="level1">
<h1>Create predicted trait for sign-permuted weights</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>pred_expr <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pred_expr)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>pred_height <span class="ot">&lt;-</span> <span class="fu">fn_generate_trait</span>(pred_expr, perm_weights_height)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>pred_bmi <span class="ot">&lt;-</span> <span class="fu">fn_generate_trait</span>(pred_expr, perm_weights_bmi)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="compare-all-values-across-different-weight-sign-permuations-to-actual-bmi-and-height" class="level1">
<h1>Compare all values across different weight sign permuations to actual BMI and Height</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>pred_height <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"/Users/natashasanthanam/Downloads/PTRS_simulation_random_sign_weights_height.RDS"</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>pred_bmi <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"/Users/natashasanthanam/Downloads/PTRS_simulation_random_sign_weights_bmi.RDS"</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>all_rats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(dir <span class="sc">%&amp;%</span> <span class="st">"all_names.txt"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(dir <span class="sc">%&amp;%</span> <span class="st">"processed_obesity_rat_Palmer_phenotypes.csv"</span>)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>(rat_rfid  <span class="sc">%in%</span> all_rats<span class="sc">$</span>ID))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="check-correlation-between-observed-bodylength-and-permuted-sign-weights" class="level1">
<h1>Check correlation between observed bodylength and permuted sign weights</h1>
<p>Bodylength with tail</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Bodylength wit Tail vs predicted Height from Human PTRS weights</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>bodylength_w_tail <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(rat_rfid, bodylength_w_tail)) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>tempo <span class="ot">&lt;-</span> pred_height[<span class="fu">na.omit</span>(<span class="fu">match</span>(bodylength_w_tail<span class="sc">$</span>rat_rfid, <span class="fu">rownames</span>(pred_height))), ]</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>bodylength_w_tail_cor_perm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">cor =</span> <span class="fu">numeric</span>(), <span class="at">pvalue =</span> <span class="fu">numeric</span>(), <span class="at">perm =</span> <span class="fu">character</span>(), <span class="at">conf.int.min =</span> <span class="fu">numeric</span>(), <span class="at">conf.int.max =</span> <span class="fu">numeric</span>())</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(tempo)){</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  bodylength_w_tail_cor_perm[i,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(bodylength_w_tail<span class="sc">$</span>bodylength_w_tail, tempo[,i])<span class="sc">$</span>estimate</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>  bodylength_w_tail_cor_perm[i,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(bodylength_w_tail<span class="sc">$</span>bodylength_w_tail, tempo[,i])<span class="sc">$</span>p.value</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  bodylength_w_tail_cor_perm[i,<span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(i <span class="sc">==</span><span class="dv">1</span>, <span class="st">"true_weight"</span>, <span class="fu">paste</span>(<span class="st">"perm"</span>, i<span class="dv">-1</span>, <span class="at">sep =</span> <span class="st">"_"</span>))</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>  bodylength_w_tail_cor_perm[i,<span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(bodylength_w_tail<span class="sc">$</span>bodylength_w_tail, tempo[,i])<span class="sc">$</span>conf.int[<span class="dv">1</span>]</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>  bodylength_w_tail_cor_perm[i,<span class="dv">5</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(bodylength_w_tail<span class="sc">$</span>bodylength_w_tail, tempo[,i])<span class="sc">$</span>conf.int[<span class="dv">2</span>]</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>BMI</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>bmi_with_tail <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(rat_rfid, bmi_bodylength_w_tail)) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>tempo <span class="ot">&lt;-</span> pred_bmi[<span class="fu">na.omit</span>(<span class="fu">match</span>(bmi_with_tail<span class="sc">$</span>rat_rfid, <span class="fu">rownames</span>(pred_bmi))), ]</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>bmi_w_tail_cor <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">cor =</span> <span class="fu">numeric</span>(), <span class="at">pvalue =</span> <span class="fu">numeric</span>(), <span class="at">perm =</span> <span class="fu">character</span>(), <span class="at">conf.int.min =</span> <span class="fu">numeric</span>(), <span class="at">conf.int.max =</span> <span class="fu">numeric</span>())</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(tempo)){</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  bmi_w_tail_cor[i,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(bmi_with_tail<span class="sc">$</span>bmi_bodylength_w_tail, tempo[,i])<span class="sc">$</span>estimate</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  bmi_w_tail_cor[i,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(bmi_with_tail<span class="sc">$</span>bmi_bodylength_w_tail, tempo[,i])<span class="sc">$</span>p.value</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  bmi_w_tail_cor[i,<span class="dv">3</span>] <span class="ot">&lt;-</span>  <span class="fu">ifelse</span>(i <span class="sc">==</span><span class="dv">1</span>, <span class="st">"true_weight"</span>, <span class="fu">paste</span>(<span class="st">"perm"</span>, i<span class="dv">-1</span>, <span class="at">sep =</span> <span class="st">"_"</span>))</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  bmi_w_tail_cor[i,<span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(bmi_with_tail<span class="sc">$</span>bmi_bodylength_w_tail, tempo[,i])<span class="sc">$</span>conf.int[<span class="dv">1</span>]</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>  bmi_w_tail_cor[i,<span class="dv">5</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(bmi_with_tail<span class="sc">$</span>bmi_bodylength_w_tail, tempo[,i])<span class="sc">$</span>conf.int[<span class="dv">2</span>]</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="compare-performance-of-permuted-weights-with-true-weight" class="level1">
<h1>Compare performance of permuted weights with true weight</h1>
<p>Height</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>bodylength_w_tail_cor_perm <span class="ot">&lt;-</span> bodylength_w_tail_cor_perm <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">iteration =</span> <span class="fu">seq</span>(<span class="dv">1</span>,<span class="dv">101</span>,<span class="dv">1</span>))</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(bodylength_w_tail_cor_perm, <span class="fu">aes</span>(iteration, cor))  <span class="sc">+</span> <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin =</span> conf.int.min, <span class="at">ymax =</span> conf.int.max ), <span class="at">width=</span><span class="fl">0.2</span>,  <span class="at">color=</span><span class="st">"gray48"</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">col =</span> <span class="fu">ifelse</span>(bodylength_w_tail_cor_perm<span class="sc">$</span>perm <span class="sc">==</span> <span class="st">"true_weight"</span>, <span class="st">"dodgerblue2"</span>, <span class="st">"black"</span>)) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Correlation Coefficient (r)"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Permutation with Random Sign Weights for Height"</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>bodylength_w_tail_cor_perm <span class="ot">&lt;-</span> bodylength_w_tail_cor_perm <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">weight_type =</span> <span class="fu">ifelse</span>(perm <span class="sc">==</span> <span class="st">"true_weight"</span>, <span class="st">"true weight"</span>, <span class="st">"permutation"</span>))</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(bodylength_w_tail_cor_perm, <span class="fu">aes</span>(<span class="at">x=</span>cor, <span class="at">color=</span>weight_type)) <span class="sc">+</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">fill =</span> <span class="st">"white"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Correlation coefficient for bodylength"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>BMI</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>bmi_w_tail_cor <span class="ot">&lt;-</span> bmi_w_tail_cor <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">iteration =</span> <span class="fu">seq</span>(<span class="dv">1</span>,<span class="dv">101</span>,<span class="dv">1</span>))</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(bmi_w_tail_cor, <span class="fu">aes</span>(iteration, cor))  <span class="sc">+</span> <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin =</span> conf.int.min, <span class="at">ymax =</span> conf.int.max ), <span class="at">width=</span><span class="fl">0.2</span>,  <span class="at">color=</span><span class="st">"gray48"</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">col =</span> <span class="fu">ifelse</span>(bodylength_w_tail_cor_perm<span class="sc">$</span>perm <span class="sc">==</span> <span class="st">"true_weight"</span>, <span class="st">"dodgerblue2"</span>, <span class="st">"black"</span>)) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Correlation Coefficient (r)"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Permutation with Random Sign Weights for BMI"</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>bmi_w_tail_cor <span class="ot">&lt;-</span> bmi_w_tail_cor <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">weight_type =</span> <span class="fu">ifelse</span>(perm <span class="sc">==</span> <span class="st">"true_weight"</span>, <span class="st">"true weight"</span>, <span class="st">"permutation"</span>))</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(bmi_w_tail_cor, <span class="fu">aes</span>(<span class="at">x=</span>cor, <span class="at">color=</span>weight_type)) <span class="sc">+</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">fill =</span> <span class="st">"white"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Correlation coefficient for BMI"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>



<!-- -->

</section>

<div class="quarto-listing quarto-listing-container-default" id="listing-listing">
<div class="list quarto-listing-default">

</div>
<div class="listing-no-matching d-none">
No matching items
</div>
</div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb10" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "10.PTRS_simulation_weights"</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Natasha Santhanam"</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "6/26/2022"</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="an">output:</span><span class="co"> html_document</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup, include=FALSE}</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RSQLite)</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a><span class="st">"%&amp;%"</span> <span class="ot">=</span> <span class="cf">function</span>(a,b) <span class="fu">paste</span>(a,b,<span class="at">sep=</span><span class="st">""</span>)</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>dir <span class="ot">&lt;-</span> <span class="st">"/Users/natashasanthanam/Github/rat-genomic-analysis/data/"</span></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">source_gist</span>(<span class="st">"ee5f67abddd0b761ee24410ea71c41aa"</span>)</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">source_gist</span>(<span class="st">"38431b74c6c0bf90c12f"</span>)</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">source_gist</span>(<span class="st">"1e9053c8f35c30396429350a08f33ea7"</span>)</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a><span class="fu">## Check Simulation of Different Weights for PTRS correlation with observed trait</span></span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>Read in files</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a><span class="in">```{r read in weights, phenotype and predicted expression}</span></span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a>weights <span class="ot">&lt;-</span>  <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Downloads/weights.height.annot.tsv"</span>)  </span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a>weights <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Downloads/weights.bmi.annot.tsv"</span>)</span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a>pred_expr <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/Users/natashasanthanam/Box/imlab-data/data-Github/web-data/2022-06-23-improving-figure-for-rat-ptrs/rat_metabolic_Ac_best__predict.txt"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(FID))</span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a>pred_expr <span class="ot">&lt;-</span> pred_expr <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(IID, <span class="fu">intersect</span>(<span class="fu">colnames</span>(pred_expr), orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene) ))</span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a><span class="co">#change name to human ensembl id in humans</span></span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(pred_expr)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(pred_expr)] <span class="ot">&lt;-</span> orth.rats[<span class="fu">match</span>(<span class="fu">colnames</span>(pred_expr)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(pred_expr)], orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene), <span class="dv">1</span>] <span class="sc">%&gt;%</span> .[[<span class="st">"ensembl_gene_id"</span>]]</span>
<span id="cb10-34"><a href="#cb10-34" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-35"><a href="#cb10-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-36"><a href="#cb10-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-37"><a href="#cb10-37" aria-hidden="true" tabindex="-1"></a><span class="fu"># First create data frame of weights for 100 variations</span></span>
<span id="cb10-38"><a href="#cb10-38" aria-hidden="true" tabindex="-1"></a>We will use model 35 for the simulation for height</span>
<span id="cb10-41"><a href="#cb10-41" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-42"><a href="#cb10-42" aria-hidden="true" tabindex="-1"></a>perm_weights_height <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="at">nrow =</span> <span class="fu">nrow</span>(weights), <span class="at">ncol =</span> <span class="dv">100</span>)</span>
<span id="cb10-43"><a href="#cb10-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-44"><a href="#cb10-44" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(perm_weights_height)) {</span>
<span id="cb10-45"><a href="#cb10-45" aria-hidden="true" tabindex="-1"></a>  j <span class="ot">=</span> <span class="fu">sample</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>),<span class="fu">nrow</span>(weights),<span class="at">replace=</span>T)</span>
<span id="cb10-46"><a href="#cb10-46" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> weights[,<span class="dv">38</span>]<span class="sc">*</span>j</span>
<span id="cb10-47"><a href="#cb10-47" aria-hidden="true" tabindex="-1"></a>  perm_weights_height[,i] <span class="ot">=</span> df<span class="sc">$</span>model_36</span>
<span id="cb10-48"><a href="#cb10-48" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-49"><a href="#cb10-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-50"><a href="#cb10-50" aria-hidden="true" tabindex="-1"></a>perm_weights_height <span class="ot">&lt;-</span> <span class="fu">cbind</span>(weights<span class="sc">$</span>model_36, perm_weights_height)</span>
<span id="cb10-51"><a href="#cb10-51" aria-hidden="true" tabindex="-1"></a>perm_weights_height <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(perm_weights_height) </span>
<span id="cb10-52"><a href="#cb10-52" aria-hidden="true" tabindex="-1"></a>perm_weights_height <span class="ot">&lt;-</span> perm_weights_height <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">gene_name =</span> weights<span class="sc">$</span>gene_name, <span class="at">.before =</span> <span class="fu">colnames</span>(perm_weights_height)[<span class="dv">1</span>])</span>
<span id="cb10-53"><a href="#cb10-53" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(perm_weights_height)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"gene_name"</span>, <span class="st">"obs_weights"</span>)</span>
<span id="cb10-54"><a href="#cb10-54" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-55"><a href="#cb10-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-56"><a href="#cb10-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-57"><a href="#cb10-57" aria-hidden="true" tabindex="-1"></a>We will also use model 17 for BMI - since it produced the largest correlation</span>
<span id="cb10-58"><a href="#cb10-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-61"><a href="#cb10-61" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-62"><a href="#cb10-62" aria-hidden="true" tabindex="-1"></a>perm_weights_bmi <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="at">nrow =</span> <span class="fu">nrow</span>(weights), <span class="at">ncol =</span> <span class="dv">100</span>)</span>
<span id="cb10-63"><a href="#cb10-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-64"><a href="#cb10-64" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(perm_weights_bmi)) {</span>
<span id="cb10-65"><a href="#cb10-65" aria-hidden="true" tabindex="-1"></a>  j <span class="ot">=</span> <span class="fu">sample</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>),<span class="fu">nrow</span>(weights),<span class="at">replace=</span>T)</span>
<span id="cb10-66"><a href="#cb10-66" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> weights[,<span class="dv">19</span>]<span class="sc">*</span>j</span>
<span id="cb10-67"><a href="#cb10-67" aria-hidden="true" tabindex="-1"></a>  perm_weights_bmi[,i] <span class="ot">=</span> df<span class="sc">$</span>model_17</span>
<span id="cb10-68"><a href="#cb10-68" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-69"><a href="#cb10-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-70"><a href="#cb10-70" aria-hidden="true" tabindex="-1"></a>perm_weights_bmi <span class="ot">&lt;-</span> <span class="fu">cbind</span>(weights<span class="sc">$</span>model_17, perm_weights_bmi)</span>
<span id="cb10-71"><a href="#cb10-71" aria-hidden="true" tabindex="-1"></a>perm_weights_bmi <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(perm_weights_bmi) </span>
<span id="cb10-72"><a href="#cb10-72" aria-hidden="true" tabindex="-1"></a>perm_weights_bmi <span class="ot">&lt;-</span> perm_weights_bmi <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">gene_name =</span> weights<span class="sc">$</span>gene_name, <span class="at">.before =</span> <span class="fu">colnames</span>(perm_weights_bmi)[<span class="dv">1</span>])</span>
<span id="cb10-73"><a href="#cb10-73" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(perm_weights_bmi)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"gene_name"</span>, <span class="st">"obs_weights"</span>)</span>
<span id="cb10-74"><a href="#cb10-74" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-75"><a href="#cb10-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-76"><a href="#cb10-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-77"><a href="#cb10-77" aria-hidden="true" tabindex="-1"></a><span class="fu"># Create predicted trait for sign-permuted weights</span></span>
<span id="cb10-78"><a href="#cb10-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-79"><a href="#cb10-79" aria-hidden="true" tabindex="-1"></a><span class="in">```{r create predicted trait}</span></span>
<span id="cb10-80"><a href="#cb10-80" aria-hidden="true" tabindex="-1"></a>pred_expr <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pred_expr)</span>
<span id="cb10-81"><a href="#cb10-81" aria-hidden="true" tabindex="-1"></a>pred_height <span class="ot">&lt;-</span> <span class="fu">fn_generate_trait</span>(pred_expr, perm_weights_height)</span>
<span id="cb10-82"><a href="#cb10-82" aria-hidden="true" tabindex="-1"></a>pred_bmi <span class="ot">&lt;-</span> <span class="fu">fn_generate_trait</span>(pred_expr, perm_weights_bmi)</span>
<span id="cb10-83"><a href="#cb10-83" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-84"><a href="#cb10-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-85"><a href="#cb10-85" aria-hidden="true" tabindex="-1"></a><span class="fu"># Compare all values across different weight sign permuations to actual BMI and Height</span></span>
<span id="cb10-86"><a href="#cb10-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-89"><a href="#cb10-89" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-90"><a href="#cb10-90" aria-hidden="true" tabindex="-1"></a>pred_height <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"/Users/natashasanthanam/Downloads/PTRS_simulation_random_sign_weights_height.RDS"</span>)</span>
<span id="cb10-91"><a href="#cb10-91" aria-hidden="true" tabindex="-1"></a>pred_bmi <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"/Users/natashasanthanam/Downloads/PTRS_simulation_random_sign_weights_bmi.RDS"</span>)</span>
<span id="cb10-92"><a href="#cb10-92" aria-hidden="true" tabindex="-1"></a>all_rats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(dir <span class="sc">%&amp;%</span> <span class="st">"all_names.txt"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-93"><a href="#cb10-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-94"><a href="#cb10-94" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(dir <span class="sc">%&amp;%</span> <span class="st">"processed_obesity_rat_Palmer_phenotypes.csv"</span>)</span>
<span id="cb10-95"><a href="#cb10-95" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>(rat_rfid  <span class="sc">%in%</span> all_rats<span class="sc">$</span>ID))</span>
<span id="cb10-96"><a href="#cb10-96" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-97"><a href="#cb10-97" aria-hidden="true" tabindex="-1"></a><span class="fu"># Check correlation between observed bodylength and permuted sign weights</span></span>
<span id="cb10-98"><a href="#cb10-98" aria-hidden="true" tabindex="-1"></a>Bodylength with tail</span>
<span id="cb10-99"><a href="#cb10-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-102"><a href="#cb10-102" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-103"><a href="#cb10-103" aria-hidden="true" tabindex="-1"></a><span class="co">#Bodylength wit Tail vs predicted Height from Human PTRS weights</span></span>
<span id="cb10-104"><a href="#cb10-104" aria-hidden="true" tabindex="-1"></a>bodylength_w_tail <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(rat_rfid, bodylength_w_tail)) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb10-105"><a href="#cb10-105" aria-hidden="true" tabindex="-1"></a>tempo <span class="ot">&lt;-</span> pred_height[<span class="fu">na.omit</span>(<span class="fu">match</span>(bodylength_w_tail<span class="sc">$</span>rat_rfid, <span class="fu">rownames</span>(pred_height))), ]</span>
<span id="cb10-106"><a href="#cb10-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-107"><a href="#cb10-107" aria-hidden="true" tabindex="-1"></a>bodylength_w_tail_cor_perm <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">cor =</span> <span class="fu">numeric</span>(), <span class="at">pvalue =</span> <span class="fu">numeric</span>(), <span class="at">perm =</span> <span class="fu">character</span>(), <span class="at">conf.int.min =</span> <span class="fu">numeric</span>(), <span class="at">conf.int.max =</span> <span class="fu">numeric</span>())</span>
<span id="cb10-108"><a href="#cb10-108" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(tempo)){</span>
<span id="cb10-109"><a href="#cb10-109" aria-hidden="true" tabindex="-1"></a>  bodylength_w_tail_cor_perm[i,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(bodylength_w_tail<span class="sc">$</span>bodylength_w_tail, tempo[,i])<span class="sc">$</span>estimate</span>
<span id="cb10-110"><a href="#cb10-110" aria-hidden="true" tabindex="-1"></a>  bodylength_w_tail_cor_perm[i,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(bodylength_w_tail<span class="sc">$</span>bodylength_w_tail, tempo[,i])<span class="sc">$</span>p.value</span>
<span id="cb10-111"><a href="#cb10-111" aria-hidden="true" tabindex="-1"></a>  bodylength_w_tail_cor_perm[i,<span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(i <span class="sc">==</span><span class="dv">1</span>, <span class="st">"true_weight"</span>, <span class="fu">paste</span>(<span class="st">"perm"</span>, i<span class="dv">-1</span>, <span class="at">sep =</span> <span class="st">"_"</span>))</span>
<span id="cb10-112"><a href="#cb10-112" aria-hidden="true" tabindex="-1"></a>  bodylength_w_tail_cor_perm[i,<span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(bodylength_w_tail<span class="sc">$</span>bodylength_w_tail, tempo[,i])<span class="sc">$</span>conf.int[<span class="dv">1</span>]</span>
<span id="cb10-113"><a href="#cb10-113" aria-hidden="true" tabindex="-1"></a>  bodylength_w_tail_cor_perm[i,<span class="dv">5</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(bodylength_w_tail<span class="sc">$</span>bodylength_w_tail, tempo[,i])<span class="sc">$</span>conf.int[<span class="dv">2</span>]</span>
<span id="cb10-114"><a href="#cb10-114" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-115"><a href="#cb10-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-116"><a href="#cb10-116" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-117"><a href="#cb10-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-118"><a href="#cb10-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-119"><a href="#cb10-119" aria-hidden="true" tabindex="-1"></a>BMI</span>
<span id="cb10-122"><a href="#cb10-122" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-123"><a href="#cb10-123" aria-hidden="true" tabindex="-1"></a>bmi_with_tail <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(rat_rfid, bmi_bodylength_w_tail)) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb10-124"><a href="#cb10-124" aria-hidden="true" tabindex="-1"></a>tempo <span class="ot">&lt;-</span> pred_bmi[<span class="fu">na.omit</span>(<span class="fu">match</span>(bmi_with_tail<span class="sc">$</span>rat_rfid, <span class="fu">rownames</span>(pred_bmi))), ]</span>
<span id="cb10-125"><a href="#cb10-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-126"><a href="#cb10-126" aria-hidden="true" tabindex="-1"></a>bmi_w_tail_cor <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">cor =</span> <span class="fu">numeric</span>(), <span class="at">pvalue =</span> <span class="fu">numeric</span>(), <span class="at">perm =</span> <span class="fu">character</span>(), <span class="at">conf.int.min =</span> <span class="fu">numeric</span>(), <span class="at">conf.int.max =</span> <span class="fu">numeric</span>())</span>
<span id="cb10-127"><a href="#cb10-127" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(tempo)){</span>
<span id="cb10-128"><a href="#cb10-128" aria-hidden="true" tabindex="-1"></a>  bmi_w_tail_cor[i,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(bmi_with_tail<span class="sc">$</span>bmi_bodylength_w_tail, tempo[,i])<span class="sc">$</span>estimate</span>
<span id="cb10-129"><a href="#cb10-129" aria-hidden="true" tabindex="-1"></a>  bmi_w_tail_cor[i,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(bmi_with_tail<span class="sc">$</span>bmi_bodylength_w_tail, tempo[,i])<span class="sc">$</span>p.value</span>
<span id="cb10-130"><a href="#cb10-130" aria-hidden="true" tabindex="-1"></a>  bmi_w_tail_cor[i,<span class="dv">3</span>] <span class="ot">&lt;-</span>  <span class="fu">ifelse</span>(i <span class="sc">==</span><span class="dv">1</span>, <span class="st">"true_weight"</span>, <span class="fu">paste</span>(<span class="st">"perm"</span>, i<span class="dv">-1</span>, <span class="at">sep =</span> <span class="st">"_"</span>))</span>
<span id="cb10-131"><a href="#cb10-131" aria-hidden="true" tabindex="-1"></a>  bmi_w_tail_cor[i,<span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(bmi_with_tail<span class="sc">$</span>bmi_bodylength_w_tail, tempo[,i])<span class="sc">$</span>conf.int[<span class="dv">1</span>]</span>
<span id="cb10-132"><a href="#cb10-132" aria-hidden="true" tabindex="-1"></a>  bmi_w_tail_cor[i,<span class="dv">5</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(bmi_with_tail<span class="sc">$</span>bmi_bodylength_w_tail, tempo[,i])<span class="sc">$</span>conf.int[<span class="dv">2</span>]</span>
<span id="cb10-133"><a href="#cb10-133" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-134"><a href="#cb10-134" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-135"><a href="#cb10-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-136"><a href="#cb10-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-137"><a href="#cb10-137" aria-hidden="true" tabindex="-1"></a><span class="fu"># Compare performance of permuted weights with true weight</span></span>
<span id="cb10-138"><a href="#cb10-138" aria-hidden="true" tabindex="-1"></a>Height</span>
<span id="cb10-139"><a href="#cb10-139" aria-hidden="true" tabindex="-1"></a><span class="in">```{r graph permuted weights vs observed for bodylength}</span></span>
<span id="cb10-140"><a href="#cb10-140" aria-hidden="true" tabindex="-1"></a>bodylength_w_tail_cor_perm <span class="ot">&lt;-</span> bodylength_w_tail_cor_perm <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">iteration =</span> <span class="fu">seq</span>(<span class="dv">1</span>,<span class="dv">101</span>,<span class="dv">1</span>))</span>
<span id="cb10-141"><a href="#cb10-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-142"><a href="#cb10-142" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(bodylength_w_tail_cor_perm, <span class="fu">aes</span>(iteration, cor))  <span class="sc">+</span> <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin =</span> conf.int.min, <span class="at">ymax =</span> conf.int.max ), <span class="at">width=</span><span class="fl">0.2</span>,  <span class="at">color=</span><span class="st">"gray48"</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">col =</span> <span class="fu">ifelse</span>(bodylength_w_tail_cor_perm<span class="sc">$</span>perm <span class="sc">==</span> <span class="st">"true_weight"</span>, <span class="st">"dodgerblue2"</span>, <span class="st">"black"</span>)) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Correlation Coefficient (r)"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Permutation with Random Sign Weights for Height"</span>)</span>
<span id="cb10-143"><a href="#cb10-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-144"><a href="#cb10-144" aria-hidden="true" tabindex="-1"></a>bodylength_w_tail_cor_perm <span class="ot">&lt;-</span> bodylength_w_tail_cor_perm <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">weight_type =</span> <span class="fu">ifelse</span>(perm <span class="sc">==</span> <span class="st">"true_weight"</span>, <span class="st">"true weight"</span>, <span class="st">"permutation"</span>))</span>
<span id="cb10-145"><a href="#cb10-145" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(bodylength_w_tail_cor_perm, <span class="fu">aes</span>(<span class="at">x=</span>cor, <span class="at">color=</span>weight_type)) <span class="sc">+</span></span>
<span id="cb10-146"><a href="#cb10-146" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">fill =</span> <span class="st">"white"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Correlation coefficient for bodylength"</span>) </span>
<span id="cb10-147"><a href="#cb10-147" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-148"><a href="#cb10-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-149"><a href="#cb10-149" aria-hidden="true" tabindex="-1"></a>BMI</span>
<span id="cb10-150"><a href="#cb10-150" aria-hidden="true" tabindex="-1"></a><span class="in">```{r compare permuted vs observed version for BMI}</span></span>
<span id="cb10-151"><a href="#cb10-151" aria-hidden="true" tabindex="-1"></a>bmi_w_tail_cor <span class="ot">&lt;-</span> bmi_w_tail_cor <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">iteration =</span> <span class="fu">seq</span>(<span class="dv">1</span>,<span class="dv">101</span>,<span class="dv">1</span>))</span>
<span id="cb10-152"><a href="#cb10-152" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(bmi_w_tail_cor, <span class="fu">aes</span>(iteration, cor))  <span class="sc">+</span> <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin =</span> conf.int.min, <span class="at">ymax =</span> conf.int.max ), <span class="at">width=</span><span class="fl">0.2</span>,  <span class="at">color=</span><span class="st">"gray48"</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">col =</span> <span class="fu">ifelse</span>(bodylength_w_tail_cor_perm<span class="sc">$</span>perm <span class="sc">==</span> <span class="st">"true_weight"</span>, <span class="st">"dodgerblue2"</span>, <span class="st">"black"</span>)) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Correlation Coefficient (r)"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Permutation with Random Sign Weights for BMI"</span>)</span>
<span id="cb10-153"><a href="#cb10-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-154"><a href="#cb10-154" aria-hidden="true" tabindex="-1"></a>bmi_w_tail_cor <span class="ot">&lt;-</span> bmi_w_tail_cor <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">weight_type =</span> <span class="fu">ifelse</span>(perm <span class="sc">==</span> <span class="st">"true_weight"</span>, <span class="st">"true weight"</span>, <span class="st">"permutation"</span>))</span>
<span id="cb10-155"><a href="#cb10-155" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(bmi_w_tail_cor, <span class="fu">aes</span>(<span class="at">x=</span>cor, <span class="at">color=</span>weight_type)) <span class="sc">+</span></span>
<span id="cb10-156"><a href="#cb10-156" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">fill =</span> <span class="st">"white"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Correlation coefficient for BMI"</span>) </span>
<span id="cb10-157"><a href="#cb10-157" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>