<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.361">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Natasha Santhanam">
<meta name="dcterms.date" content="2022-02-14">

<title>Rat Genomics Pipeline Rendered - Coregulation_across_Tissues_Species</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-listing/list.min.js"></script>
<script src="../site_libs/quarto-listing/quarto-listing.js"></script>
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script>

  window.document.addEventListener("DOMContentLoaded", function (_event) {
    const listingTargetEl = window.document.querySelector('#listing-listing .list');
    if (!listingTargetEl) {
      // No listing discovered, do not attach.
      return; 
    }

    const options = {
      valueNames: [{ data: ['index'] },{ data: ['categories'] },{ data: ['listing-date-sort'] },{ data: ['listing-file-modified-sort'] }],
      
      searchColumns: [],
    };

    window['quarto-listings'] = window['quarto-listings'] || {};
    window['quarto-listings']['listing-listing'] = new List('listing-listing', options);

    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  });

  window.addEventListener('hashchange',() => {
    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  })
  </script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Rat Genomics Pipeline Rendered</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html" rel="" target="">
 <span class="menu-text">paper_code</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../other_code.html" rel="" target="">
 <span class="menu-text">more_code</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Coregulation_across_Tissues_Species</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Natasha Santhanam </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 14, 2022</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(broom)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RSQLite)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Hmisc)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="st">"%&amp;%"</span> <span class="ot">=</span> <span class="cf">function</span>(a,b) <span class="fu">paste</span>(a,b,<span class="at">sep=</span><span class="st">""</span>)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/"</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">777</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol type="1">
<li>Correlation between genes within each tissue and then calculate correlation of the correlation between genes across tissues</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/metabolic_traits"</span>, <span class="at">pattern =</span> <span class="st">"predict.txt"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>orth.rats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(dir <span class="sc">%&amp;%</span> <span class="st">"expression/ortholog_genes_rats_humans.tsv"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Generate Correlation Matrices for 100 rats - compare predicted Expression across tissues</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>names <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(filelist[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(FID))</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>) {</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  id <span class="ot">=</span> <span class="fu">sample</span>(names<span class="sc">$</span>FID, <span class="at">size =</span> <span class="dv">1</span>)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">as.character</span>())</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>    name <span class="ot">&lt;-</span> <span class="fu">substr</span>(fila, <span class="dv">89</span>,<span class="dv">90</span>)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>    tis <span class="ot">&lt;-</span> <span class="fu">fread</span>(fila) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(FID <span class="sc">==</span> id) <span class="sc">%&gt;%</span> <span class="fu">pivot_longer</span>(<span class="sc">!</span><span class="fu">c</span>(FID, IID), <span class="at">names_to =</span> <span class="st">"gene"</span>, <span class="at">values_to =</span> name) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(FID, IID))</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>    tempo <span class="ot">&lt;-</span> <span class="fu">full_join</span>(tempo, tis,  <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  } </span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> tempo <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">var =</span> <span class="fu">apply</span>(tempo[,<span class="sc">-</span><span class="dv">1</span>], <span class="dv">1</span>, var)) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(tempo, dir <span class="sc">%&amp;%</span> <span class="st">"prediXcan/GREx_comp/cor_tis_per_ind/"</span> <span class="sc">%&amp;%</span> id <span class="sc">%&amp;%</span> <span class="st">".GREx.mat.RDS"</span>)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Check heatmap of some individuals</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>i1 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"/Users/natashasanthanam/Downloads/00077E6712.cor.mat.RDS"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>i2 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"/Users/natashasanthanam/Downloads/00077E7788.cor.mat.RDS"</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>melted_i1 <span class="ot">&lt;-</span> <span class="fu">melt</span>(i1, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>melted_i2 <span class="ot">&lt;-</span> <span class="fu">melt</span>(i2, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>p1<span class="ot">=</span> <span class="fu">ggplot</span>(<span class="at">data =</span> melted_i1, <span class="fu">aes</span>(<span class="at">x=</span>Var1, <span class="at">y=</span>Var2, <span class="at">fill=</span>value)) <span class="sc">+</span> </span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"00077E6712"</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="dv">1</span>)</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>p2<span class="ot">=</span> <span class="fu">ggplot</span>(<span class="at">data =</span> melted_i2, <span class="fu">aes</span>(<span class="at">x=</span>Var1, <span class="at">y=</span>Var2, <span class="at">fill=</span>value)) <span class="sc">+</span> </span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"00077E7788"</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="dv">1</span>)</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a><span class="fu">ggarrange</span>(p1, p2, <span class="at">ncol=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Look at all genes across tissues in one individual Evidence of shared regulation across tissues</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>i3 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">"/Users/natashasanthanam/Downloads/00077E83E3.GREx.mat.RDS"</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(gene, var))</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pairs</span>(i3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="2" type="1">
<li>Co-regulation Shared Across Species</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>gene_ids <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">id =</span> <span class="fu">fread</span>(filelist[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(FID, IID)) <span class="sc">%&gt;%</span>  <span class="fu">colnames</span>()</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(filelist)]) {</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">id =</span> <span class="fu">fread</span>(fila) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(FID, IID)) <span class="sc">%&gt;%</span>  <span class="fu">colnames</span>())</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  gene_ids <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(gene_ids, df, <span class="at">by =</span> <span class="st">"id"</span>)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>gtf <span class="ot">&lt;-</span> <span class="fu">fread</span>(dir <span class="sc">%&amp;%</span> <span class="st">"Box_files/gtf.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>gtf <span class="ot">&lt;-</span>  gtf[<span class="fu">match</span>(tempo<span class="sc">$</span>id, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>) {</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> gtf <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Chr <span class="sc">==</span> i) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(Gene))</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(tempo) <span class="sc">==</span> <span class="dv">0</span> ) {</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span>  </span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> {</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">row =</span> <span class="fu">as.character</span>(), <span class="at">column =</span> <span class="fu">as.character</span>())</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>    tis <span class="ot">&lt;-</span> <span class="fu">substr</span>(fila, <span class="dv">58</span>,<span class="dv">59</span>)</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>    expr <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">fread</span>(fila) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(FID, IID)))</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>    expr <span class="ot">&lt;-</span> expr[,<span class="fu">intersect</span>(tempo<span class="sc">$</span>Gene, <span class="fu">colnames</span>(expr))]</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>    res2<span class="ot">&lt;-</span><span class="fu">rcorr</span>(<span class="fu">as.matrix</span>(expr[,]))</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>  d2 <span class="ot">&lt;-</span> <span class="fu">flattenCorrMatrix</span>(res2<span class="sc">$</span>r, res2<span class="sc">$</span>P)</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(d2)[<span class="dv">3</span>] <span class="ot">=</span> tis</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(d2)[<span class="dv">4</span>] <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">"p"</span>, tis, <span class="at">sep =</span> <span class="st">"_"</span>)</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">full_join</span>(df, d2, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"row"</span>, <span class="st">"column"</span>))</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(df, dir <span class="sc">%&amp;%</span> <span class="st">"prediXcan/GREx_comp/cor_genes_per_chr/chr"</span> <span class="sc">%&amp;%</span> i <span class="sc">%&amp;%</span> <span class="st">".RDS"</span> )</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Save correlation of coregulation across tissues</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>coreg.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/GREx_comp/cor_genes_per_chr"</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(coreg.dir, <span class="at">pattern =</span> <span class="st">".RDS"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(fila) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(row, column, Ac, Il, Pl, Lh, Vo)) <span class="sc">%&gt;%</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  i <span class="ot">&lt;-</span> <span class="fu">substr</span>(fila, <span class="dv">89</span>, <span class="fu">str_length</span>(fila)<span class="sc">-</span> <span class="dv">4</span>)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>  cor.mat <span class="ot">&lt;-</span>  <span class="fu">cor</span>(tempo[,<span class="dv">3</span><span class="sc">:</span><span class="dv">7</span>])</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(cor.mat, dir <span class="sc">%&amp;%</span> <span class="st">"prediXcan/GREx_comp/cor_genes_per_chr/cor_coreg_chr"</span> <span class="sc">%&amp;%</span> i <span class="sc">%&amp;%</span> <span class="st">".RDS"</span>)</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Graph of Correlation of Coregulation across tissues</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>data.dir <span class="ot">&lt;-</span> <span class="st">"/Users/natashasanthanam/CRI/"</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(data.dir, <span class="at">pattern=</span><span class="st">"cor_coreg"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>corr_coreg <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  i <span class="ot">&lt;-</span> <span class="fu">match</span>(fila, filelist)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>  corr_coreg[[i]] <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(fila)</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Generate Coregulation in Humans (GTEx)</p>
<p>First generate predicted expression in G1000 using GTEx models</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate imlabtools</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">METAXCAN</span><span class="op">=</span>/gpfs/data/im-lab/nas40t2/natasha/GTEX_Analysis/MetaXcan/software</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">GENO</span><span class="op">=</span>/gpfs/data/im-lab/nas40t2/Data/dbGaP/Transcriptome/G1000/imputed_hrc1.1</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">MODEL</span><span class="op">=</span>/gpfs/data/im-lab/nas40t2/Data/PredictDB/GTEx_v8/models_v1/eqtl/ctimp</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">RESULTS</span><span class="op">=</span>/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/PTRS_weights/PGP</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">DATA</span><span class="op">=</span>/gpfs/data/im-lab/nas40t2/natasha/GTEX_Analysis/predixcan</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="bu">printf</span> <span class="st">"Predict expression\n\n"</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="ex">python3</span> <span class="va">$METAXCAN</span>/Predict.py <span class="dt">\</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>--model_db_path <span class="va">$MODEL</span>/ctimp_<span class="va">$TISSUE</span>.db <span class="dt">\</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>--model_db_snp_key varID <span class="dt">\</span></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>--vcf_genotypes <span class="va">$GENO</span>/chr<span class="pp">*</span>.dose.vcf.gz  <span class="dt">\</span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>--vcf_mode genotyped <span class="dt">\</span></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>--liftover <span class="va">$DATA</span>/hg19ToHg38.over.chain.gz  <span class="dt">\</span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>--on_the_fly_mapping METADATA <span class="st">"chr{}_{}_{}_{}_b38"</span> <span class="dt">\</span></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>--prediction_output <span class="va">$RESULTS</span>/G1000__<span class="va">$TISSUE</span>.predict.txt <span class="dt">\</span></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>--prediction_summary_output <span class="va">$RESULTS</span>/G1000__<span class="va">$TISSUE</span>.summary.txt <span class="dt">\</span></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>--verbosity 9 <span class="dt">\</span></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>--throw</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a><span class="va">TISSUE</span><span class="op">=</span>Brain_Cerebellum</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>flattenCorrMatrix <span class="ot">&lt;-</span> <span class="cf">function</span>(cormat, pmat) {</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  ut <span class="ot">&lt;-</span> <span class="fu">upper.tri</span>(cormat)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">row =</span> <span class="fu">rownames</span>(cormat)[<span class="fu">row</span>(cormat)[ut]],</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">column =</span> <span class="fu">rownames</span>(cormat)[<span class="fu">col</span>(cormat)[ut]],</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">cor  =</span>(cormat)[ut],</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">p =</span> pmat[ut]</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Calculate Coregulation between genes in GTEx</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>gtf <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/GTEX_Analysis/annotations_gencode_v26.tsv"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a> <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>) {</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> gtf <span class="sc">%&gt;%</span> <span class="fu">filter</span>(chromosome <span class="sc">==</span> <span class="fu">paste</span>(<span class="st">"chr"</span>, i, <span class="at">sep=</span><span class="st">""</span>)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene_id))</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>  expr <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>( <span class="fu">fread</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/GREx_comp/G1000_less_mem__Brain_Cerebellum.predict.txt"</span>)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(FID, IID)) </span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a> <span class="fu">colnames</span>(expr) <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">colnames</span>(expr), <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>    expr <span class="ot">&lt;-</span> expr[, <span class="fu">intersect</span>(tempo<span class="sc">$</span>gene_id, <span class="fu">colnames</span>(expr))]</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>   res2<span class="ot">&lt;-</span><span class="fu">rcorr</span>(<span class="fu">as.matrix</span>(expr[,]))</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  d2 <span class="ot">&lt;-</span> <span class="fu">flattenCorrMatrix</span>(res2<span class="sc">$</span>r, res2<span class="sc">$</span>P)</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(d2, dir <span class="sc">%&amp;%</span> <span class="st">"prediXcan/GREx_comp/cor_GTEx_genes_per_chr/GTEx_chr"</span> <span class="sc">%&amp;%</span> i <span class="sc">%&amp;%</span> <span class="st">".RDS"</span> )</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a> }</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a><span class="co">#Can also calculate correlation between genes in GTEx for all genes</span></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>expr <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>( <span class="fu">fread</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/GREx_comp/G1000_less_mem__Brain_Cerebellum.predict.txt"</span>)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(FID, IID)) </span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a> <span class="fu">colnames</span>(expr) <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">colnames</span>(expr), <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a> res2<span class="ot">&lt;-</span><span class="fu">rcorr</span>(<span class="fu">as.matrix</span>(expr[,]))</span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>  d2 <span class="ot">&lt;-</span> <span class="fu">flattenCorrMatrix</span>(res2<span class="sc">$</span>r, res2<span class="sc">$</span>P)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Check if Coregualtion is preserved across species</p>
<p>Plot Coregulation across species</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>p.dir <span class="ot">&lt;-</span> <span class="st">"/Users/natashasanthanam/CRI/"</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(p.dir, <span class="at">pattern =</span> <span class="st">"all"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(fila)</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pairs</span>(df)</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Heatmap ordered with TSS</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>only_GTEx <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"/Users/natashasanthanam/Downloads/cor_pred_expr_GTEx_all_genes.RDS"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(p)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">start =</span> gtf[<span class="fu">match</span>(only_GTEx<span class="sc">$</span>row, gtf<span class="sc">$</span>gene_id), <span class="dv">5</span>]<span class="sc">$</span>start) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(row, column, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>GTEx_ordered <span class="ot">&lt;-</span> only_GTEx[<span class="fu">sort</span>(only_GTEx<span class="sc">$</span>start),]</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>GTEx_chr2_genes <span class="ot">&lt;-</span> only_GTEx[<span class="fu">na.omit</span>(<span class="fu">match</span>(gtf<span class="sc">$</span>gene_id, only_GTEx<span class="sc">$</span>row)), ]</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>p3<span class="ot">=</span> <span class="fu">ggplot</span>(<span class="at">data =</span> GTEx_ordered, <span class="fu">aes</span>(<span class="at">x=</span>row, <span class="at">y=</span>column, <span class="at">fill=</span>cor)) <span class="sc">+</span> </span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">2</span>)) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Heatmap for predicted expression in GTEx Cerebellum"</span>)</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>p3</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>



<!-- -->


<div class="quarto-listing quarto-listing-container-default" id="listing-listing">
<div class="list quarto-listing-default">

</div>
<div class="listing-no-matching d-none">
No matching items
</div>
</div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb15" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Coregulation_across_Tissues_Species"</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Natasha Santhanam"</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2/14/2022"</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="an">output:</span><span class="co"> html_document</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="in">````{r}</span></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(broom)</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RSQLite)</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Hmisc)</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a><span class="st">"%&amp;%"</span> <span class="ot">=</span> <span class="cf">function</span>(a,b) <span class="fu">paste</span>(a,b,<span class="at">sep=</span><span class="st">""</span>)</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/"</span></span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">777</span>)</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a><span class="st">```</span></span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a><span class="at">1. Correlation between genes within each tissue and then</span></span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a><span class="at">calculate correlation of the correlation between genes across tissues</span></span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/metabolic_traits"</span>, <span class="at">pattern =</span> <span class="st">"predict.txt"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a>orth.rats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(dir <span class="sc">%&amp;%</span> <span class="st">"expression/ortholog_genes_rats_humans.tsv"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a><span class="st">```</span></span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a><span class="at">Generate Correlation Matrices for 100 rats - compare predicted Expression across tissues</span></span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a>names <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(filelist[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(FID))</span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>) {</span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a>  id <span class="ot">=</span> <span class="fu">sample</span>(names<span class="sc">$</span>FID, <span class="at">size =</span> <span class="dv">1</span>)</span>
<span id="cb15-38"><a href="#cb15-38" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">as.character</span>())</span>
<span id="cb15-39"><a href="#cb15-39" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a>    name <span class="ot">&lt;-</span> <span class="fu">substr</span>(fila, <span class="dv">89</span>,<span class="dv">90</span>)</span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a>    tis <span class="ot">&lt;-</span> <span class="fu">fread</span>(fila) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(FID <span class="sc">==</span> id) <span class="sc">%&gt;%</span> <span class="fu">pivot_longer</span>(<span class="sc">!</span><span class="fu">c</span>(FID, IID), <span class="at">names_to =</span> <span class="st">"gene"</span>, <span class="at">values_to =</span> name) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(FID, IID))</span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a>    tempo <span class="ot">&lt;-</span> <span class="fu">full_join</span>(tempo, tis,  <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a>  } </span>
<span id="cb15-44"><a href="#cb15-44" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> tempo <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">var =</span> <span class="fu">apply</span>(tempo[,<span class="sc">-</span><span class="dv">1</span>], <span class="dv">1</span>, var)) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb15-45"><a href="#cb15-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(tempo, dir <span class="sc">%&amp;%</span> <span class="st">"prediXcan/GREx_comp/cor_tis_per_ind/"</span> <span class="sc">%&amp;%</span> id <span class="sc">%&amp;%</span> <span class="st">".GREx.mat.RDS"</span>)</span>
<span id="cb15-46"><a href="#cb15-46" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-47"><a href="#cb15-47" aria-hidden="true" tabindex="-1"></a><span class="st">```</span></span>
<span id="cb15-48"><a href="#cb15-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-49"><a href="#cb15-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-50"><a href="#cb15-50" aria-hidden="true" tabindex="-1"></a><span class="at">Check heatmap of some individuals</span></span>
<span id="cb15-53"><a href="#cb15-53" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb15-54"><a href="#cb15-54" aria-hidden="true" tabindex="-1"></a>i1 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"/Users/natashasanthanam/Downloads/00077E6712.cor.mat.RDS"</span>)</span>
<span id="cb15-55"><a href="#cb15-55" aria-hidden="true" tabindex="-1"></a>i2 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"/Users/natashasanthanam/Downloads/00077E7788.cor.mat.RDS"</span>)</span>
<span id="cb15-56"><a href="#cb15-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-57"><a href="#cb15-57" aria-hidden="true" tabindex="-1"></a>melted_i1 <span class="ot">&lt;-</span> <span class="fu">melt</span>(i1, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-58"><a href="#cb15-58" aria-hidden="true" tabindex="-1"></a>melted_i2 <span class="ot">&lt;-</span> <span class="fu">melt</span>(i2, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-59"><a href="#cb15-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-60"><a href="#cb15-60" aria-hidden="true" tabindex="-1"></a>p1<span class="ot">=</span> <span class="fu">ggplot</span>(<span class="at">data =</span> melted_i1, <span class="fu">aes</span>(<span class="at">x=</span>Var1, <span class="at">y=</span>Var2, <span class="at">fill=</span>value)) <span class="sc">+</span> </span>
<span id="cb15-61"><a href="#cb15-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb15-62"><a href="#cb15-62" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb15-63"><a href="#cb15-63" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"00077E6712"</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="dv">1</span>)</span>
<span id="cb15-64"><a href="#cb15-64" aria-hidden="true" tabindex="-1"></a>p2<span class="ot">=</span> <span class="fu">ggplot</span>(<span class="at">data =</span> melted_i2, <span class="fu">aes</span>(<span class="at">x=</span>Var1, <span class="at">y=</span>Var2, <span class="at">fill=</span>value)) <span class="sc">+</span> </span>
<span id="cb15-65"><a href="#cb15-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb15-66"><a href="#cb15-66" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb15-67"><a href="#cb15-67" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"00077E7788"</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="dv">1</span>)</span>
<span id="cb15-68"><a href="#cb15-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-69"><a href="#cb15-69" aria-hidden="true" tabindex="-1"></a><span class="fu">ggarrange</span>(p1, p2, <span class="at">ncol=</span><span class="dv">2</span>)</span>
<span id="cb15-70"><a href="#cb15-70" aria-hidden="true" tabindex="-1"></a><span class="st">```</span></span>
<span id="cb15-71"><a href="#cb15-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-72"><a href="#cb15-72" aria-hidden="true" tabindex="-1"></a><span class="at">Look at all genes across tissues in one individual</span></span>
<span id="cb15-73"><a href="#cb15-73" aria-hidden="true" tabindex="-1"></a><span class="at">Evidence of shared regulation across tissues</span></span>
<span id="cb15-74"><a href="#cb15-74" aria-hidden="true" tabindex="-1"></a><span class="st">```</span>{r read <span class="cf">in</span> one ind cor}</span>
<span id="cb15-75"><a href="#cb15-75" aria-hidden="true" tabindex="-1"></a>i3 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">"/Users/natashasanthanam/Downloads/00077E83E3.GREx.mat.RDS"</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(gene, var))</span>
<span id="cb15-76"><a href="#cb15-76" aria-hidden="true" tabindex="-1"></a><span class="fu">pairs</span>(i3)</span>
<span id="cb15-77"><a href="#cb15-77" aria-hidden="true" tabindex="-1"></a><span class="st">```</span></span>
<span id="cb15-78"><a href="#cb15-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-79"><a href="#cb15-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-80"><a href="#cb15-80" aria-hidden="true" tabindex="-1"></a><span class="at">2.  Co-regulation Shared Across Species</span></span>
<span id="cb15-81"><a href="#cb15-81" aria-hidden="true" tabindex="-1"></a><span class="st">```</span>{r find gene ids <span class="cf">in</span> all rat tissues, eval<span class="ot">=</span><span class="cn">FALSE</span>}</span>
<span id="cb15-82"><a href="#cb15-82" aria-hidden="true" tabindex="-1"></a>gene_ids <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">id =</span> <span class="fu">fread</span>(filelist[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(FID, IID)) <span class="sc">%&gt;%</span>  <span class="fu">colnames</span>()</span>
<span id="cb15-83"><a href="#cb15-83" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(filelist)]) {</span>
<span id="cb15-84"><a href="#cb15-84" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">id =</span> <span class="fu">fread</span>(fila) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(FID, IID)) <span class="sc">%&gt;%</span>  <span class="fu">colnames</span>())</span>
<span id="cb15-85"><a href="#cb15-85" aria-hidden="true" tabindex="-1"></a>  gene_ids <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(gene_ids, df, <span class="at">by =</span> <span class="st">"id"</span>)</span>
<span id="cb15-86"><a href="#cb15-86" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-87"><a href="#cb15-87" aria-hidden="true" tabindex="-1"></a><span class="st">```</span></span>
<span id="cb15-88"><a href="#cb15-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-89"><a href="#cb15-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-90"><a href="#cb15-90" aria-hidden="true" tabindex="-1"></a><span class="st">```</span>{r find cor of pred expression matrix <span class="cf">for</span> all rat tissues, eval<span class="ot">=</span><span class="cn">FALSE</span>}</span>
<span id="cb15-91"><a href="#cb15-91" aria-hidden="true" tabindex="-1"></a>gtf <span class="ot">&lt;-</span> <span class="fu">fread</span>(dir <span class="sc">%&amp;%</span> <span class="st">"Box_files/gtf.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-92"><a href="#cb15-92" aria-hidden="true" tabindex="-1"></a>gtf <span class="ot">&lt;-</span>  gtf[<span class="fu">match</span>(tempo<span class="sc">$</span>id, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb15-93"><a href="#cb15-93" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb15-94"><a href="#cb15-94" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>) {</span>
<span id="cb15-95"><a href="#cb15-95" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> gtf <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Chr <span class="sc">==</span> i) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(Gene))</span>
<span id="cb15-96"><a href="#cb15-96" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(tempo) <span class="sc">==</span> <span class="dv">0</span> ) {</span>
<span id="cb15-97"><a href="#cb15-97" aria-hidden="true" tabindex="-1"></a>  i <span class="ot">=</span> i<span class="sc">+</span><span class="dv">1</span>  </span>
<span id="cb15-98"><a href="#cb15-98" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb15-99"><a href="#cb15-99" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> {</span>
<span id="cb15-100"><a href="#cb15-100" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">row =</span> <span class="fu">as.character</span>(), <span class="at">column =</span> <span class="fu">as.character</span>())</span>
<span id="cb15-101"><a href="#cb15-101" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb15-102"><a href="#cb15-102" aria-hidden="true" tabindex="-1"></a>    tis <span class="ot">&lt;-</span> <span class="fu">substr</span>(fila, <span class="dv">58</span>,<span class="dv">59</span>)</span>
<span id="cb15-103"><a href="#cb15-103" aria-hidden="true" tabindex="-1"></a>    expr <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">fread</span>(fila) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(FID, IID)))</span>
<span id="cb15-104"><a href="#cb15-104" aria-hidden="true" tabindex="-1"></a>    expr <span class="ot">&lt;-</span> expr[,<span class="fu">intersect</span>(tempo<span class="sc">$</span>Gene, <span class="fu">colnames</span>(expr))]</span>
<span id="cb15-105"><a href="#cb15-105" aria-hidden="true" tabindex="-1"></a>    res2<span class="ot">&lt;-</span><span class="fu">rcorr</span>(<span class="fu">as.matrix</span>(expr[,]))</span>
<span id="cb15-106"><a href="#cb15-106" aria-hidden="true" tabindex="-1"></a>  d2 <span class="ot">&lt;-</span> <span class="fu">flattenCorrMatrix</span>(res2<span class="sc">$</span>r, res2<span class="sc">$</span>P)</span>
<span id="cb15-107"><a href="#cb15-107" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(d2)[<span class="dv">3</span>] <span class="ot">=</span> tis</span>
<span id="cb15-108"><a href="#cb15-108" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(d2)[<span class="dv">4</span>] <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">"p"</span>, tis, <span class="at">sep =</span> <span class="st">"_"</span>)</span>
<span id="cb15-109"><a href="#cb15-109" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">full_join</span>(df, d2, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"row"</span>, <span class="st">"column"</span>))</span>
<span id="cb15-110"><a href="#cb15-110" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb15-111"><a href="#cb15-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(df, dir <span class="sc">%&amp;%</span> <span class="st">"prediXcan/GREx_comp/cor_genes_per_chr/chr"</span> <span class="sc">%&amp;%</span> i <span class="sc">%&amp;%</span> <span class="st">".RDS"</span> )</span>
<span id="cb15-112"><a href="#cb15-112" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb15-113"><a href="#cb15-113" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-114"><a href="#cb15-114" aria-hidden="true" tabindex="-1"></a><span class="st">```</span></span>
<span id="cb15-115"><a href="#cb15-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-116"><a href="#cb15-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-117"><a href="#cb15-117" aria-hidden="true" tabindex="-1"></a><span class="at">Save correlation of coregulation across tissues</span></span>
<span id="cb15-120"><a href="#cb15-120" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb15-121"><a href="#cb15-121" aria-hidden="true" tabindex="-1"></a>coreg.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/GREx_comp/cor_genes_per_chr"</span></span>
<span id="cb15-122"><a href="#cb15-122" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(coreg.dir, <span class="at">pattern =</span> <span class="st">".RDS"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-123"><a href="#cb15-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-124"><a href="#cb15-124" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb15-125"><a href="#cb15-125" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(fila) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(row, column, Ac, Il, Pl, Lh, Vo)) <span class="sc">%&gt;%</span></span>
<span id="cb15-126"><a href="#cb15-126" aria-hidden="true" tabindex="-1"></a>  i <span class="ot">&lt;-</span> <span class="fu">substr</span>(fila, <span class="dv">89</span>, <span class="fu">str_length</span>(fila)<span class="sc">-</span> <span class="dv">4</span>)</span>
<span id="cb15-127"><a href="#cb15-127" aria-hidden="true" tabindex="-1"></a>  cor.mat <span class="ot">&lt;-</span>  <span class="fu">cor</span>(tempo[,<span class="dv">3</span><span class="sc">:</span><span class="dv">7</span>])</span>
<span id="cb15-128"><a href="#cb15-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(cor.mat, dir <span class="sc">%&amp;%</span> <span class="st">"prediXcan/GREx_comp/cor_genes_per_chr/cor_coreg_chr"</span> <span class="sc">%&amp;%</span> i <span class="sc">%&amp;%</span> <span class="st">".RDS"</span>)</span>
<span id="cb15-129"><a href="#cb15-129" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-130"><a href="#cb15-130" aria-hidden="true" tabindex="-1"></a><span class="st">```</span></span>
<span id="cb15-131"><a href="#cb15-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-132"><a href="#cb15-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-133"><a href="#cb15-133" aria-hidden="true" tabindex="-1"></a><span class="at">Graph of Correlation of Coregulation across tissues</span></span>
<span id="cb15-134"><a href="#cb15-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-137"><a href="#cb15-137" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb15-138"><a href="#cb15-138" aria-hidden="true" tabindex="-1"></a>data.dir <span class="ot">&lt;-</span> <span class="st">"/Users/natashasanthanam/CRI/"</span></span>
<span id="cb15-139"><a href="#cb15-139" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(data.dir, <span class="at">pattern=</span><span class="st">"cor_coreg"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-140"><a href="#cb15-140" aria-hidden="true" tabindex="-1"></a>corr_coreg <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb15-141"><a href="#cb15-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-142"><a href="#cb15-142" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb15-143"><a href="#cb15-143" aria-hidden="true" tabindex="-1"></a>  i <span class="ot">&lt;-</span> <span class="fu">match</span>(fila, filelist)</span>
<span id="cb15-144"><a href="#cb15-144" aria-hidden="true" tabindex="-1"></a>  corr_coreg[[i]] <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(fila)</span>
<span id="cb15-145"><a href="#cb15-145" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-146"><a href="#cb15-146" aria-hidden="true" tabindex="-1"></a><span class="st">```</span></span>
<span id="cb15-147"><a href="#cb15-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-148"><a href="#cb15-148" aria-hidden="true" tabindex="-1"></a><span class="at">Generate Coregulation in Humans (GTEx)</span></span>
<span id="cb15-149"><a href="#cb15-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-150"><a href="#cb15-150" aria-hidden="true" tabindex="-1"></a><span class="at">First generate predicted expression in G1000 using GTEx models</span></span>
<span id="cb15-153"><a href="#cb15-153" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb15-154"><a href="#cb15-154" aria-hidden="true" tabindex="-1"></a>conda activate imlabtools</span>
<span id="cb15-155"><a href="#cb15-155" aria-hidden="true" tabindex="-1"></a>export <span class="at">METAXCAN=</span><span class="sc">/</span>gpfs<span class="sc">/</span>data<span class="sc">/</span>im<span class="sc">-</span>lab<span class="sc">/</span>nas40t2<span class="sc">/</span>natasha<span class="sc">/</span>GTEX_Analysis<span class="sc">/</span>MetaXcan<span class="sc">/</span>software</span>
<span id="cb15-156"><a href="#cb15-156" aria-hidden="true" tabindex="-1"></a>export <span class="at">GENO=</span><span class="sc">/</span>gpfs<span class="sc">/</span>data<span class="sc">/</span>im<span class="sc">-</span>lab<span class="sc">/</span>nas40t2<span class="sc">/</span>Data<span class="sc">/</span>dbGaP<span class="sc">/</span>Transcriptome<span class="sc">/</span>G1000<span class="sc">/</span>imputed_hrc1<span class="fl">.1</span></span>
<span id="cb15-157"><a href="#cb15-157" aria-hidden="true" tabindex="-1"></a>export <span class="at">MODEL=</span><span class="sc">/</span>gpfs<span class="sc">/</span>data<span class="sc">/</span>im<span class="sc">-</span>lab<span class="sc">/</span>nas40t2<span class="sc">/</span>Data<span class="sc">/</span>PredictDB<span class="sc">/</span>GTEx_v8<span class="sc">/</span>models_v1<span class="sc">/</span>eqtl<span class="sc">/</span>ctimp</span>
<span id="cb15-158"><a href="#cb15-158" aria-hidden="true" tabindex="-1"></a>export <span class="at">RESULTS=</span><span class="sc">/</span>gpfs<span class="sc">/</span>data<span class="sc">/</span>im<span class="sc">-</span>lab<span class="sc">/</span>nas40t2<span class="sc">/</span>natasha<span class="sc">/</span>rat_genomics<span class="sc">/</span>PTRS_weights<span class="sc">/</span>PGP</span>
<span id="cb15-159"><a href="#cb15-159" aria-hidden="true" tabindex="-1"></a>export <span class="at">DATA=</span><span class="sc">/</span>gpfs<span class="sc">/</span>data<span class="sc">/</span>im<span class="sc">-</span>lab<span class="sc">/</span>nas40t2<span class="sc">/</span>natasha<span class="sc">/</span>GTEX_Analysis<span class="sc">/</span>predixcan</span>
<span id="cb15-160"><a href="#cb15-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-161"><a href="#cb15-161" aria-hidden="true" tabindex="-1"></a>printf <span class="st">"Predict expression</span><span class="sc">\n\n</span><span class="st">"</span></span>
<span id="cb15-162"><a href="#cb15-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-163"><a href="#cb15-163" aria-hidden="true" tabindex="-1"></a>python3 <span class="sc">$</span>METAXCAN<span class="sc">/</span>Predict.py <span class="sc">\</span></span>
<span id="cb15-164"><a href="#cb15-164" aria-hidden="true" tabindex="-1"></a><span class="sc">--</span>model_db_path <span class="sc">$</span>MODEL<span class="sc">/</span>ctimp_<span class="sc">$</span>TISSUE.db <span class="sc">\</span></span>
<span id="cb15-165"><a href="#cb15-165" aria-hidden="true" tabindex="-1"></a><span class="sc">--</span>model_db_snp_key varID <span class="sc">\</span></span>
<span id="cb15-166"><a href="#cb15-166" aria-hidden="true" tabindex="-1"></a><span class="sc">--</span>vcf_genotypes <span class="sc">$</span>GENO<span class="sc">/</span>chr<span class="sc">*</span>.dose.vcf.gz  <span class="sc">\</span></span>
<span id="cb15-167"><a href="#cb15-167" aria-hidden="true" tabindex="-1"></a><span class="sc">--</span>vcf_mode genotyped <span class="sc">\</span></span>
<span id="cb15-168"><a href="#cb15-168" aria-hidden="true" tabindex="-1"></a><span class="sc">--</span>liftover <span class="sc">$</span>DATA<span class="sc">/</span>hg19ToHg38.over.chain.gz  <span class="sc">\</span></span>
<span id="cb15-169"><a href="#cb15-169" aria-hidden="true" tabindex="-1"></a><span class="sc">--</span>on_the_fly_mapping METADATA <span class="st">"chr{}_{}_{}_{}_b38"</span> <span class="sc">\</span></span>
<span id="cb15-170"><a href="#cb15-170" aria-hidden="true" tabindex="-1"></a><span class="sc">--</span>prediction_output <span class="sc">$</span>RESULTS<span class="sc">/</span>G1000__<span class="sc">$</span>TISSUE.predict.txt <span class="sc">\</span></span>
<span id="cb15-171"><a href="#cb15-171" aria-hidden="true" tabindex="-1"></a><span class="sc">--</span>prediction_summary_output <span class="sc">$</span>RESULTS<span class="sc">/</span>G1000__<span class="sc">$</span>TISSUE.summary.txt <span class="sc">\</span></span>
<span id="cb15-172"><a href="#cb15-172" aria-hidden="true" tabindex="-1"></a><span class="sc">--</span>verbosity <span class="dv">9</span> <span class="sc">\</span></span>
<span id="cb15-173"><a href="#cb15-173" aria-hidden="true" tabindex="-1"></a><span class="sc">--</span>throw</span>
<span id="cb15-174"><a href="#cb15-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-175"><a href="#cb15-175" aria-hidden="true" tabindex="-1"></a><span class="at">TISSUE=</span>Brain_Cerebellum</span>
<span id="cb15-176"><a href="#cb15-176" aria-hidden="true" tabindex="-1"></a><span class="st">```</span></span>
<span id="cb15-177"><a href="#cb15-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-178"><a href="#cb15-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-179"><a href="#cb15-179" aria-hidden="true" tabindex="-1"></a><span class="st">```</span>{r flatten corr mat <span class="cf">function</span>, eval<span class="ot">=</span><span class="cn">FALSE</span>}</span>
<span id="cb15-180"><a href="#cb15-180" aria-hidden="true" tabindex="-1"></a>flattenCorrMatrix <span class="ot">&lt;-</span> <span class="cf">function</span>(cormat, pmat) {</span>
<span id="cb15-181"><a href="#cb15-181" aria-hidden="true" tabindex="-1"></a>  ut <span class="ot">&lt;-</span> <span class="fu">upper.tri</span>(cormat)</span>
<span id="cb15-182"><a href="#cb15-182" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb15-183"><a href="#cb15-183" aria-hidden="true" tabindex="-1"></a>    <span class="at">row =</span> <span class="fu">rownames</span>(cormat)[<span class="fu">row</span>(cormat)[ut]],</span>
<span id="cb15-184"><a href="#cb15-184" aria-hidden="true" tabindex="-1"></a>    <span class="at">column =</span> <span class="fu">rownames</span>(cormat)[<span class="fu">col</span>(cormat)[ut]],</span>
<span id="cb15-185"><a href="#cb15-185" aria-hidden="true" tabindex="-1"></a>    <span class="at">cor  =</span>(cormat)[ut],</span>
<span id="cb15-186"><a href="#cb15-186" aria-hidden="true" tabindex="-1"></a>    <span class="at">p =</span> pmat[ut]</span>
<span id="cb15-187"><a href="#cb15-187" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb15-188"><a href="#cb15-188" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-189"><a href="#cb15-189" aria-hidden="true" tabindex="-1"></a><span class="st">```</span></span>
<span id="cb15-190"><a href="#cb15-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-191"><a href="#cb15-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-192"><a href="#cb15-192" aria-hidden="true" tabindex="-1"></a><span class="at">Calculate Coregulation between genes in GTEx</span></span>
<span id="cb15-195"><a href="#cb15-195" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb15-196"><a href="#cb15-196" aria-hidden="true" tabindex="-1"></a>gtf <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/GTEX_Analysis/annotations_gencode_v26.tsv"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-197"><a href="#cb15-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-198"><a href="#cb15-198" aria-hidden="true" tabindex="-1"></a> <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>) {</span>
<span id="cb15-199"><a href="#cb15-199" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> gtf <span class="sc">%&gt;%</span> <span class="fu">filter</span>(chromosome <span class="sc">==</span> <span class="fu">paste</span>(<span class="st">"chr"</span>, i, <span class="at">sep=</span><span class="st">""</span>)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene_id))</span>
<span id="cb15-200"><a href="#cb15-200" aria-hidden="true" tabindex="-1"></a>  expr <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>( <span class="fu">fread</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/GREx_comp/G1000_less_mem__Brain_Cerebellum.predict.txt"</span>)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(FID, IID)) </span>
<span id="cb15-201"><a href="#cb15-201" aria-hidden="true" tabindex="-1"></a> <span class="fu">colnames</span>(expr) <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">colnames</span>(expr), <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb15-202"><a href="#cb15-202" aria-hidden="true" tabindex="-1"></a>    expr <span class="ot">&lt;-</span> expr[, <span class="fu">intersect</span>(tempo<span class="sc">$</span>gene_id, <span class="fu">colnames</span>(expr))]</span>
<span id="cb15-203"><a href="#cb15-203" aria-hidden="true" tabindex="-1"></a>   res2<span class="ot">&lt;-</span><span class="fu">rcorr</span>(<span class="fu">as.matrix</span>(expr[,]))</span>
<span id="cb15-204"><a href="#cb15-204" aria-hidden="true" tabindex="-1"></a>  d2 <span class="ot">&lt;-</span> <span class="fu">flattenCorrMatrix</span>(res2<span class="sc">$</span>r, res2<span class="sc">$</span>P)</span>
<span id="cb15-205"><a href="#cb15-205" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(d2, dir <span class="sc">%&amp;%</span> <span class="st">"prediXcan/GREx_comp/cor_GTEx_genes_per_chr/GTEx_chr"</span> <span class="sc">%&amp;%</span> i <span class="sc">%&amp;%</span> <span class="st">".RDS"</span> )</span>
<span id="cb15-206"><a href="#cb15-206" aria-hidden="true" tabindex="-1"></a> }</span>
<span id="cb15-207"><a href="#cb15-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-208"><a href="#cb15-208" aria-hidden="true" tabindex="-1"></a><span class="co">#Can also calculate correlation between genes in GTEx for all genes</span></span>
<span id="cb15-209"><a href="#cb15-209" aria-hidden="true" tabindex="-1"></a>expr <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>( <span class="fu">fread</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/GREx_comp/G1000_less_mem__Brain_Cerebellum.predict.txt"</span>)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(FID, IID)) </span>
<span id="cb15-210"><a href="#cb15-210" aria-hidden="true" tabindex="-1"></a> <span class="fu">colnames</span>(expr) <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">colnames</span>(expr), <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb15-211"><a href="#cb15-211" aria-hidden="true" tabindex="-1"></a> res2<span class="ot">&lt;-</span><span class="fu">rcorr</span>(<span class="fu">as.matrix</span>(expr[,]))</span>
<span id="cb15-212"><a href="#cb15-212" aria-hidden="true" tabindex="-1"></a>  d2 <span class="ot">&lt;-</span> <span class="fu">flattenCorrMatrix</span>(res2<span class="sc">$</span>r, res2<span class="sc">$</span>P)</span>
<span id="cb15-213"><a href="#cb15-213" aria-hidden="true" tabindex="-1"></a><span class="st">```</span></span>
<span id="cb15-214"><a href="#cb15-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-215"><a href="#cb15-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-216"><a href="#cb15-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-217"><a href="#cb15-217" aria-hidden="true" tabindex="-1"></a><span class="at">Check if Coregualtion is preserved across species</span></span>
<span id="cb15-218"><a href="#cb15-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-219"><a href="#cb15-219" aria-hidden="true" tabindex="-1"></a><span class="st">```</span>{r, include<span class="ot">=</span><span class="cn">FALSE</span>, eval<span class="ot">=</span><span class="cn">FALSE</span>}</span>
<span id="cb15-220"><a href="#cb15-220" aria-hidden="true" tabindex="-1"></a>human.mat <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(dir <span class="sc">%&amp;%</span> <span class="st">"prediXcan/GREx_comp/cor_GTEx_genes_per_chr/cor_pred_expr_GTEx_all_genes.RDS"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(p)) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">GTEx =</span> cor)</span>
<span id="cb15-221"><a href="#cb15-221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-222"><a href="#cb15-222" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">6</span><span class="sc">:</span><span class="dv">20</span>) {</span>
<span id="cb15-223"><a href="#cb15-223" aria-hidden="true" tabindex="-1"></a>  rat.mat <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/GREx_comp/cor_genes_per_chr/chr"</span> <span class="sc">%&amp;%</span> i <span class="sc">%&amp;%</span> <span class="st">".RDS"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(p_Ac, p_Il, p_Lh, p_Vo, p_Pl))</span>
<span id="cb15-224"><a href="#cb15-224" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-225"><a href="#cb15-225" aria-hidden="true" tabindex="-1"></a>  rat.mat <span class="ot">&lt;-</span> rat.mat <span class="sc">%&gt;%</span> <span class="fu">filter</span>(row <span class="sc">%in%</span> orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene <span class="sc">&amp;</span> column <span class="sc">%in%</span> orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene) </span>
<span id="cb15-226"><a href="#cb15-226" aria-hidden="true" tabindex="-1"></a>  rat.mat<span class="sc">$</span>row <span class="ot">=</span> orth.rats[<span class="fu">match</span>(rat.mat<span class="sc">$</span>row, orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene),<span class="dv">1</span>]<span class="sc">$</span>ensembl_gene_id</span>
<span id="cb15-227"><a href="#cb15-227" aria-hidden="true" tabindex="-1"></a>  rat.mat<span class="sc">$</span>column <span class="ot">=</span> orth.rats[<span class="fu">match</span>(rat.mat<span class="sc">$</span>column, orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene),<span class="dv">1</span>]<span class="sc">$</span>ensembl_gene_id</span>
<span id="cb15-228"><a href="#cb15-228" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-229"><a href="#cb15-229" aria-hidden="true" tabindex="-1"></a>  all_genes <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(rat.mat, human.mat, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"row"</span>, <span class="st">"column"</span>)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(row, column)) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb15-230"><a href="#cb15-230" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(all_genes, dir <span class="sc">%&amp;%</span> <span class="st">"prediXcan/GREx_comp/coregulation_across_species/all_chr"</span> <span class="sc">%&amp;%</span> i <span class="sc">%&amp;%</span> <span class="st">".RDS"</span> )</span>
<span id="cb15-231"><a href="#cb15-231" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-232"><a href="#cb15-232" aria-hidden="true" tabindex="-1"></a><span class="st">```</span></span>
<span id="cb15-233"><a href="#cb15-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-234"><a href="#cb15-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-235"><a href="#cb15-235" aria-hidden="true" tabindex="-1"></a><span class="at">Plot Coregulation across species</span></span>
<span id="cb15-236"><a href="#cb15-236" aria-hidden="true" tabindex="-1"></a><span class="st">```</span>{r plot coregulation across species, eval<span class="ot">=</span><span class="cn">FALSE</span>}</span>
<span id="cb15-237"><a href="#cb15-237" aria-hidden="true" tabindex="-1"></a>p.dir <span class="ot">&lt;-</span> <span class="st">"/Users/natashasanthanam/CRI/"</span></span>
<span id="cb15-238"><a href="#cb15-238" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(p.dir, <span class="at">pattern =</span> <span class="st">"all"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-239"><a href="#cb15-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-240"><a href="#cb15-240" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb15-241"><a href="#cb15-241" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(fila)</span>
<span id="cb15-242"><a href="#cb15-242" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pairs</span>(df)</span>
<span id="cb15-243"><a href="#cb15-243" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-244"><a href="#cb15-244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-245"><a href="#cb15-245" aria-hidden="true" tabindex="-1"></a><span class="st">```</span></span>
<span id="cb15-246"><a href="#cb15-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-247"><a href="#cb15-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-248"><a href="#cb15-248" aria-hidden="true" tabindex="-1"></a><span class="at">Heatmap ordered with TSS</span></span>
<span id="cb15-249"><a href="#cb15-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-250"><a href="#cb15-250" aria-hidden="true" tabindex="-1"></a><span class="st">```</span>{r ordered heatmap, eval<span class="ot">=</span><span class="cn">FALSE</span>}</span>
<span id="cb15-251"><a href="#cb15-251" aria-hidden="true" tabindex="-1"></a>only_GTEx <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"/Users/natashasanthanam/Downloads/cor_pred_expr_GTEx_all_genes.RDS"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(p)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">start =</span> gtf[<span class="fu">match</span>(only_GTEx<span class="sc">$</span>row, gtf<span class="sc">$</span>gene_id), <span class="dv">5</span>]<span class="sc">$</span>start) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(row, column, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-252"><a href="#cb15-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-253"><a href="#cb15-253" aria-hidden="true" tabindex="-1"></a>GTEx_ordered <span class="ot">&lt;-</span> only_GTEx[<span class="fu">sort</span>(only_GTEx<span class="sc">$</span>start),]</span>
<span id="cb15-254"><a href="#cb15-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-255"><a href="#cb15-255" aria-hidden="true" tabindex="-1"></a>GTEx_chr2_genes <span class="ot">&lt;-</span> only_GTEx[<span class="fu">na.omit</span>(<span class="fu">match</span>(gtf<span class="sc">$</span>gene_id, only_GTEx<span class="sc">$</span>row)), ]</span>
<span id="cb15-256"><a href="#cb15-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-257"><a href="#cb15-257" aria-hidden="true" tabindex="-1"></a><span class="at">p3=</span> <span class="fu">ggplot</span>(<span class="at">data =</span> GTEx_ordered, <span class="fu">aes</span>(<span class="at">x=</span>row, <span class="at">y=</span>column, <span class="at">fill=</span>cor)) <span class="sc">+</span> </span>
<span id="cb15-258"><a href="#cb15-258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span> <span class="fu">theme</span>(</span>
<span id="cb15-259"><a href="#cb15-259" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb15-260"><a href="#cb15-260" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">2</span>)) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Heatmap for predicted expression in GTEx Cerebellum"</span>)</span>
<span id="cb15-261"><a href="#cb15-261" aria-hidden="true" tabindex="-1"></a>p3</span>
<span id="cb15-262"><a href="#cb15-262" aria-hidden="true" tabindex="-1"></a><span class="st">```</span></span>
<span id="cb15-263"><a href="#cb15-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-264"><a href="#cb15-264" aria-hidden="true" tabindex="-1"></a><span class="at"> </span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>