<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Natasha Santhanam">
<meta name="dcterms.date" content="2022-02-14">

<title>Rat Genomics Pipeline Rendered - Generate_plink_format_genotypes</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-listing/list.min.js"></script>
<script src="../site_libs/quarto-listing/quarto-listing.js"></script>
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script>

  window.document.addEventListener("DOMContentLoaded", function (_event) {
    const listingTargetEl = window.document.querySelector('#listing-listing .list');
    if (!listingTargetEl) {
      // No listing discovered, do not attach.
      return; 
    }

    const options = {
      valueNames: [{ data: ['index'] },{ data: ['categories'] },{ data: ['listing-date-sort'] },{ data: ['listing-file-modified-sort'] }],
      
      searchColumns: [],
    };

    window['quarto-listings'] = window['quarto-listings'] || {};
    window['quarto-listings']['listing-listing'] = new List('listing-listing', options);

    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  });

  window.addEventListener('hashchange',() => {
    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  })
  </script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Rat Genomics Pipeline Rendered</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html" rel="" target="">
 <span class="menu-text">paper_code</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../other_code.html" rel="" target="">
 <span class="menu-text">more_code</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#definitions" id="toc-definitions" class="nav-link active" data-scroll-target="#definitions">Definitions</a></li>
  <li><a href="#generating-raw-genotype-files-from-tysons-original-analysis" id="toc-generating-raw-genotype-files-from-tysons-original-analysis" class="nav-link" data-scroll-target="#generating-raw-genotype-files-from-tysons-original-analysis">Generating Raw Genotype files from Tyson’s original analysis</a></li>
  <li><a href="#generating-mapped-files-for-plink" id="toc-generating-mapped-files-for-plink" class="nav-link" data-scroll-target="#generating-mapped-files-for-plink">Generating MAP/PED files for plink</a></li>
  <li><a href="#convert-mapped-files-to-plink" id="toc-convert-mapped-files-to-plink" class="nav-link" data-scroll-target="#convert-mapped-files-to-plink">Convert MAP/PED files to plink</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Generate_plink_format_genotypes</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Natasha Santhanam </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 14, 2022</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="definitions" class="level2">
<h2 class="anchored" data-anchor-id="definitions">Definitions</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(workflowr)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glue)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="st">"%&amp;%"</span> <span class="ot">=</span> <span class="cf">function</span>(a,b) <span class="fu">paste</span>(a,b,<span class="at">sep=</span><span class="st">""</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>wd <span class="ot">=</span>  <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/data/"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="generating-raw-genotype-files-from-tysons-original-analysis" class="level2">
<h2 class="anchored" data-anchor-id="generating-raw-genotype-files-from-tysons-original-analysis">Generating Raw Genotype files from Tyson’s original analysis</h2>
<p>Data from <a href="https://uchicago.app.box.com/folder/102043737114">here</a> - genoGex.Rdata has everything we need in it There are 5 ‘gex’ RDS files which are the gene expressions for the 5 different tissues, the ‘gtf’ is the gene annotation, ‘phyMap’ is the snp annotation, and ‘geno’ is the genotype matrix</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"~/Box/imlab-data/Projects/PTRS-PGRS-Rosetta/Data-From-Abe-Palmer-Lab/Rdata/genoGex.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>n <span class="ot">=</span> gexAc<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>gexAc_transpose <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(gexAc[,<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gexAc_transpose) <span class="ot">&lt;-</span> n</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>n <span class="ot">=</span> gexIl<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>gexIl_transpose <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(gexIl[,<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gexIl_transpose) <span class="ot">&lt;-</span> n</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>n <span class="ot">=</span> gexLh<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>gexLh_transpose <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(gexLh[,<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gexLh_transpose) <span class="ot">&lt;-</span> n</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>n <span class="ot">=</span> gexPl<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>gexPl_transpose <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(gexPl[,<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gexPl_transpose) <span class="ot">&lt;-</span> n</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>n <span class="ot">=</span> gexVo<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>gexVo_transpose <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(gexVo[,<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gexVo_transpose) <span class="ot">&lt;-</span> n</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>invnorm <span class="ot">=</span> <span class="cf">function</span>(x) {</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">is.null</span>(<span class="fu">dim</span>(x))) res <span class="ot">=</span> <span class="fu">invnorm.vector</span>(x) <span class="cf">else</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  res<span class="ot">=</span><span class="fu">apply</span>(x,<span class="dv">2</span>,invnorm.vector)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  res</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>invnorm.vector <span class="ot">=</span> <span class="cf">function</span>(x) {yy <span class="ot">=</span> <span class="fu">rank</span>(x)<span class="sc">/</span>(<span class="fu">length</span>(x)<span class="sc">+</span><span class="dv">1</span>); <span class="fu">qnorm</span>(yy)}</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>gexAc_transpose <span class="ot">=</span> <span class="fu">invnorm</span>(gexAc_transpose)</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>gexIl_transpose <span class="ot">=</span> <span class="fu">invnorm</span>(gexIl_transpose)</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>gexLh_transpose <span class="ot">=</span> <span class="fu">invnorm</span>(gexLh_transpose)</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>gexPl_transpose <span class="ot">=</span> <span class="fu">invnorm</span>(gexPl_transpose)</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>gexVo_transpose <span class="ot">=</span> <span class="fu">invnorm</span>(gexVo_transpose)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We first transposed each gene expression file as a matrix with individuals as the columns and genes for the rows. Now we add gene annotations.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>gtf<span class="sc">$</span>gene_type <span class="ot">=</span> <span class="fu">sub</span>(<span class="st">".*?gene_biotype(.*?);.*"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, gtf<span class="sc">$</span>Attr)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>gtf<span class="sc">$</span>gene_name <span class="ot">=</span> <span class="fu">sub</span>(<span class="st">".*?gene_name(.*?);.*"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, gtf<span class="sc">$</span>Attr)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>gene_annotation <span class="ot">=</span> <span class="fu">subset</span>(gtf, <span class="at">select =</span> <span class="sc">-</span><span class="fu">c</span>(Source, Feature, Score, Strand, Attr, Frame) )</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>gene_annotation <span class="ot">=</span> gene_annotation[, <span class="fu">c</span>(<span class="st">"Chr"</span>,<span class="st">"Gene"</span>, <span class="st">"gene_name"</span>, <span class="st">"Start"</span>, <span class="st">"End"</span>, <span class="st">"gene_type"</span> )]</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gene_annotation) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"chr"</span>, <span class="st">"gene_id"</span>, <span class="st">"gene_name"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(gene_annotation) <span class="ot">=</span> gene_annotation<span class="sc">$</span>gene_id</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>phyMap <span class="ot">&lt;-</span> <span class="fu">within</span>(phyMap,  varID <span class="ot">&lt;-</span> <span class="fu">paste</span>(Chr, Pos, Ref, Alt, <span class="at">sep=</span><span class="st">"_"</span>))</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(phyMap) <span class="ot">=</span> phyMap<span class="sc">$</span>varID</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>phyMap<span class="sc">$</span>rsid <span class="ot">=</span> phyMap<span class="sc">$</span>varID</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(phyMap) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"snp"</span>, <span class="st">"chr"</span>, <span class="st">"pos"</span>, <span class="st">"refAllele"</span>, <span class="st">"effectAllele"</span>, <span class="st">'varID'</span>, <span class="st">"rsid"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Next, we subset genotype files to the rats with gene expression data from respective tissues.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geno) <span class="ot">=</span> <span class="fu">rownames</span>(phyMap)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co">#write.table(geno, file = "./data/genotype.txt", sep = "\t", col.names = TRUE, row.names = TRUE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>geno_Ac <span class="ot">=</span> geno[,<span class="fu">match</span>(<span class="fu">rownames</span>(gexAc_transpose), <span class="fu">colnames</span>(geno))]</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>geno_Il <span class="ot">=</span> geno[,<span class="fu">match</span>(<span class="fu">rownames</span>(gexIl_transpose), <span class="fu">colnames</span>(geno))]</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>geno_Lh <span class="ot">=</span> geno[,<span class="fu">match</span>(<span class="fu">rownames</span>(gexLh_transpose), <span class="fu">colnames</span>(geno))]</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>geno_Pl <span class="ot">=</span> geno[,<span class="fu">match</span>(<span class="fu">rownames</span>(gexPl_transpose), <span class="fu">colnames</span>(geno))]</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>geno_Vo <span class="ot">=</span> geno[,<span class="fu">match</span>(<span class="fu">rownames</span>(gexVo_transpose), <span class="fu">colnames</span>(geno))]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(geno_Ac, <span class="at">file =</span> wd <span class="sc">%&amp;%</span> <span class="st">"genotype_Ac.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(geno_Il, <span class="at">file =</span> wd <span class="sc">%&amp;%</span> <span class="st">"genotype_Il.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(geno_Lh, <span class="at">file =</span> wd <span class="sc">%&amp;%</span> <span class="st">"genotype_Lh.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(geno_Pl, <span class="at">file =</span> wd <span class="sc">%&amp;%</span> <span class="st">"genotype_Pl.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(geno_Vo, <span class="at">file =</span> wd <span class="sc">%&amp;%</span> <span class="st">"genotype_Vo.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="generating-mapped-files-for-plink" class="level2">
<h2 class="anchored" data-anchor-id="generating-mapped-files-for-plink">Generating MAP/PED files for plink</h2>
<p>MAP</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>nIndividuals_Ac <span class="ot">&lt;-</span> <span class="fu">ncol</span>(geno_Ac)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>nSNPs_Ac <span class="ot">&lt;-</span> <span class="fu">nrow</span>(geno_Ac)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>MAP <span class="ot">&lt;-</span> phyMap <span class="sc">%&gt;%</span> <span class="fu">rename</span>( <span class="at">SNP =</span> rsid, <span class="at">CHR =</span> chr, <span class="at">BP =</span> pos) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">CM =</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(CHR, SNP, CM, BP))</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(MAP, wd <span class="sc">%&amp;%</span> <span class="st">"rat_genotype.map"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>PED</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>PED6 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">FID =</span> <span class="fu">colnames</span>(geno_Ac),</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">IID =</span> <span class="fu">colnames</span>(geno_Ac),</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">FatherID =</span> <span class="dv">0</span>,</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">MotherID =</span> <span class="dv">0</span>,</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">Sex =</span> <span class="dv">1</span>,</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">Phenotype =</span> <span class="dv">1</span>)</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="co"># convert 0,1,2 to genotypes, a bit tricky</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="co"># make helper dataframe for matching alleles</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>myAlleles <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">AA =</span> <span class="fu">paste</span>(phyMap<span class="sc">$</span>refAllele,phyMap<span class="sc">$</span>refAllele),</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">AB =</span> <span class="fu">paste</span>(phyMap<span class="sc">$</span>refAllele, phyMap<span class="sc">$</span>effectAllele),</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">BB =</span> <span class="fu">paste</span>(phyMap<span class="sc">$</span>effectAllele,phyMap<span class="sc">$</span>effectAllele) )</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>PEDsnps <span class="ot">&lt;-</span> geno_Ac[,] <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>PEDsnpsAB <span class="ot">&lt;-</span> </span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(<span class="fu">seq</span>(nSNPs_Ac), <span class="cf">function</span>(snp)</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sapply</span>(PEDsnps[snp, ], <span class="cf">function</span>(ind) myAlleles[snp, ind]))</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>PED <span class="ot">&lt;-</span> <span class="fu">cbind</span>(PED6, PEDsnpsAB)</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PED, wd <span class="sc">%&amp;%</span> <span class="st">"rat_genotype.ped"</span>,<span class="at">col_names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Repeat for other tissues ~ only need to change the rows</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>PED_Il <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">FID =</span> <span class="fu">colnames</span>(geno_Il), <span class="at">IID =</span> <span class="fu">colnames</span>(geno_Il), <span class="at">FatherID =</span> <span class="dv">0</span>, <span class="at">MotherID =</span> <span class="dv">0</span>, <span class="at">Sex =</span> <span class="dv">1</span>, <span class="at">Phenotype =</span> <span class="dv">1</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>PED_Lh <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">FID =</span> <span class="fu">colnames</span>(geno_Lh), <span class="at">IID =</span> <span class="fu">colnames</span>(geno_Lh), <span class="at">FatherID =</span> <span class="dv">0</span>, <span class="at">MotherID =</span> <span class="dv">0</span>, <span class="at">Sex =</span> <span class="dv">1</span>, <span class="at">Phenotype =</span> <span class="dv">1</span>)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>PED_Pl <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">FID =</span> <span class="fu">colnames</span>(geno_Pl), <span class="at">IID =</span> <span class="fu">colnames</span>(geno_Pl), <span class="at">FatherID =</span> <span class="dv">0</span>, <span class="at">MotherID =</span> <span class="dv">0</span>, <span class="at">Sex =</span> <span class="dv">1</span>, <span class="at">Phenotype =</span> <span class="dv">1</span>)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>PED_Vo <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">FID =</span> <span class="fu">colnames</span>(geno_Vo), <span class="at">IID =</span> <span class="fu">colnames</span>(geno_Vo), <span class="at">FatherID =</span> <span class="dv">0</span>, <span class="at">MotherID =</span> <span class="dv">0</span>, <span class="at">Sex =</span> <span class="dv">1</span>, <span class="at">Phenotype =</span> <span class="dv">1</span>)</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>PEDsnps <span class="ot">&lt;-</span> geno_Il[,] <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>PEDsnpsAB <span class="ot">&lt;-</span> </span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(<span class="fu">seq</span>(<span class="fu">nrow</span>(geno_Il)), <span class="cf">function</span>(snp)</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sapply</span>(PEDsnps[snp, ], <span class="cf">function</span>(ind) myAlleles[snp, ind]))</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>PED <span class="ot">&lt;-</span> <span class="fu">cbind</span>(PED_Il, PEDsnpsAB)</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PED, wd <span class="sc">%&amp;%</span> <span class="st">"rat_genotype_Il.ped"</span>,<span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>PEDsnps <span class="ot">&lt;-</span> geno_Lh[,] <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>PEDsnpsAB <span class="ot">&lt;-</span> </span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(<span class="fu">seq</span>(<span class="fu">nrow</span>(geno_Lh)), <span class="cf">function</span>(snp)</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sapply</span>(PEDsnps[snp, ], <span class="cf">function</span>(ind) myAlleles[snp, ind]))</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>PED <span class="ot">&lt;-</span> <span class="fu">cbind</span>(PED_Lh, PEDsnpsAB)</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PED, wd <span class="sc">%&amp;%</span> <span class="st">"rat_genotype_Lh.ped"</span>,<span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>PEDsnps <span class="ot">&lt;-</span> geno_Pl[,] <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>PEDsnpsAB <span class="ot">&lt;-</span> </span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(<span class="fu">seq</span>(<span class="fu">nrow</span>(geno_Pl)), <span class="cf">function</span>(snp)</span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sapply</span>(PEDsnps[snp, ], <span class="cf">function</span>(ind) myAlleles[snp, ind]))</span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>PED <span class="ot">&lt;-</span> <span class="fu">cbind</span>(PED_Pl, PEDsnpsAB)</span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PED, wd <span class="sc">%&amp;%</span> <span class="st">"rat_genotype_Pl.ped"</span>,<span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a>PEDsnps <span class="ot">&lt;-</span> geno_Vo[,] <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a>PEDsnpsAB <span class="ot">&lt;-</span> </span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(<span class="fu">seq</span>(<span class="fu">nrow</span>(geno_Vo)), <span class="cf">function</span>(snp)</span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sapply</span>(PEDsnps[snp, ], <span class="cf">function</span>(ind) myAlleles[snp, ind]))</span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>PED <span class="ot">&lt;-</span> <span class="fu">cbind</span>(PED_Vo, PEDsnpsAB)</span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PED, wd <span class="sc">%&amp;%</span> <span class="st">"rat_genotype_Vo.ped"</span>,<span class="at">col_names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Generating Phenotype file ~ We’re using gene expression levels</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>pheno_Ac <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(<span class="at">FID =</span> <span class="fu">rownames</span>(gexAc_transpose), <span class="at">IID =</span> <span class="fu">rownames</span>(gexAc_transpose), gexAc_transpose))</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>pheno_Il <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(<span class="at">FID =</span> <span class="fu">rownames</span>(gexIl_transpose), <span class="at">IID =</span> <span class="fu">rownames</span>(gexIl_transpose), gexIl_transpose))</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>pheno_Lh <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(<span class="at">FID =</span> <span class="fu">rownames</span>(gexLh_transpose), <span class="at">IID =</span> <span class="fu">rownames</span>(gexLh_transpose), gexLh_transpose))</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>pheno_Pl <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(<span class="at">FID =</span> <span class="fu">rownames</span>(gexPl_transpose), <span class="at">IID =</span> <span class="fu">rownames</span>(gexPl_transpose), gexPl_transpose))</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>pheno_Vo <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(<span class="at">FID =</span> <span class="fu">rownames</span>(gexVo_transpose), <span class="at">IID =</span> <span class="fu">rownames</span>(gexVo_transpose), gexVo_transpose))</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(pheno_Ac, wd <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/rat_pheno_Ac.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(pheno_Il, wd <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/rat_pheno_Il.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(pheno_Lh, wd <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/rat_pheno_Lh.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(pheno_Pl, wd <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/rat_pheno_Pl.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(pheno_Vo, wd <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/rat_pheno_Vo.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="convert-mapped-files-to-plink" class="level2">
<h2 class="anchored" data-anchor-id="convert-mapped-files-to-plink">Convert MAP/PED files to plink</h2>
<p>All conversion to plink format use the same MAP file, but because of plink conventions, the MAP file must match the name of the PED file, so we rename the map file each time we convert genotypes for a different tissue.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /gpfs/data/im-lab/nas40t2/sabrina/Rat_Genomics_Paper_Pipeline/data</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="ex">plink</span> <span class="at">--file</span> rat_genotype <span class="at">--make-bed</span> <span class="at">--out</span> plink_files/rat_genome_Ac</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> rat_genotype.map rat_genotype_Il.map</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="ex">plink</span> <span class="at">--file</span> rat_genotype_Il <span class="at">--make-bed</span> <span class="at">--out</span> plink_files/rat_genome_Il</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> rat_genotype_Il.map rat_genotype_Lh.map</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="ex">plink</span> <span class="at">--file</span> rat_genotype_Lh <span class="at">--make-bed</span> <span class="at">--out</span> plink_files/rat_genome_Lh</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> rat_genotype_Lh.map rat_genotype_Pl.map</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a><span class="ex">plink</span> <span class="at">--file</span> rat_genotype_Pl <span class="at">--make-bed</span> <span class="at">--out</span> plink_files/rat_genome_Pl</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> rat_genotype_Pl.map rat_genotype_Vo.map</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a><span class="ex">plink</span> <span class="at">--file</span> rat_genotype_Vo <span class="at">--make-bed</span> <span class="at">--out</span> plink_files/rat_genome_Vo</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>



<!-- -->

</section>

<div class="quarto-listing quarto-listing-container-default" id="listing-listing">
<div class="list quarto-listing-default">

</div>
<div class="listing-no-matching d-none">
No matching items
</div>
</div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb14" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Generate_plink_format_genotypes"</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Natasha Santhanam"</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2/14/2022"</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="an">output:</span><span class="co"> html_document</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="fu">## Definitions</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(workflowr)</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glue)</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a><span class="st">"%&amp;%"</span> <span class="ot">=</span> <span class="cf">function</span>(a,b) <span class="fu">paste</span>(a,b,<span class="at">sep=</span><span class="st">""</span>)</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>wd <span class="ot">=</span>  <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/data/"</span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a><span class="fu">## Generating Raw Genotype files from Tyson's original analysis</span></span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>Data from <span class="co">[</span><span class="ot">here</span><span class="co">](https://uchicago.app.box.com/folder/102043737114)</span> - genoGex.Rdata has everything we need in it</span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>There are 5 'gex' RDS files which are the gene expressions for the 5 different tissues, the 'gtf' is the gene annotation, 'phyMap' is the snp annotation, and 'geno' is the genotype matrix</span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"~/Box/imlab-data/Projects/PTRS-PGRS-Rosetta/Data-From-Abe-Palmer-Lab/Rdata/genoGex.RData"</span>)</span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>n <span class="ot">=</span> gexAc<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>gexAc_transpose <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(gexAc[,<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gexAc_transpose) <span class="ot">&lt;-</span> n</span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a>n <span class="ot">=</span> gexIl<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>gexIl_transpose <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(gexIl[,<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gexIl_transpose) <span class="ot">&lt;-</span> n</span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a>n <span class="ot">=</span> gexLh<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a>gexLh_transpose <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(gexLh[,<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gexLh_transpose) <span class="ot">&lt;-</span> n</span>
<span id="cb14-48"><a href="#cb14-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-49"><a href="#cb14-49" aria-hidden="true" tabindex="-1"></a>n <span class="ot">=</span> gexPl<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb14-50"><a href="#cb14-50" aria-hidden="true" tabindex="-1"></a>gexPl_transpose <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(gexPl[,<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb14-51"><a href="#cb14-51" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gexPl_transpose) <span class="ot">&lt;-</span> n</span>
<span id="cb14-52"><a href="#cb14-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-53"><a href="#cb14-53" aria-hidden="true" tabindex="-1"></a>n <span class="ot">=</span> gexVo<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb14-54"><a href="#cb14-54" aria-hidden="true" tabindex="-1"></a>gexVo_transpose <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(gexVo[,<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb14-55"><a href="#cb14-55" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gexVo_transpose) <span class="ot">&lt;-</span> n</span>
<span id="cb14-56"><a href="#cb14-56" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-57"><a href="#cb14-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-60"><a href="#cb14-60" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb14-61"><a href="#cb14-61" aria-hidden="true" tabindex="-1"></a>invnorm <span class="ot">=</span> <span class="cf">function</span>(x) {</span>
<span id="cb14-62"><a href="#cb14-62" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">is.null</span>(<span class="fu">dim</span>(x))) res <span class="ot">=</span> <span class="fu">invnorm.vector</span>(x) <span class="cf">else</span></span>
<span id="cb14-63"><a href="#cb14-63" aria-hidden="true" tabindex="-1"></a>  res<span class="ot">=</span><span class="fu">apply</span>(x,<span class="dv">2</span>,invnorm.vector)</span>
<span id="cb14-64"><a href="#cb14-64" aria-hidden="true" tabindex="-1"></a>  res</span>
<span id="cb14-65"><a href="#cb14-65" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-66"><a href="#cb14-66" aria-hidden="true" tabindex="-1"></a>invnorm.vector <span class="ot">=</span> <span class="cf">function</span>(x) {yy <span class="ot">=</span> <span class="fu">rank</span>(x)<span class="sc">/</span>(<span class="fu">length</span>(x)<span class="sc">+</span><span class="dv">1</span>); <span class="fu">qnorm</span>(yy)}</span>
<span id="cb14-67"><a href="#cb14-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-68"><a href="#cb14-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-69"><a href="#cb14-69" aria-hidden="true" tabindex="-1"></a>gexAc_transpose <span class="ot">=</span> <span class="fu">invnorm</span>(gexAc_transpose)</span>
<span id="cb14-70"><a href="#cb14-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-71"><a href="#cb14-71" aria-hidden="true" tabindex="-1"></a>gexIl_transpose <span class="ot">=</span> <span class="fu">invnorm</span>(gexIl_transpose)</span>
<span id="cb14-72"><a href="#cb14-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-73"><a href="#cb14-73" aria-hidden="true" tabindex="-1"></a>gexLh_transpose <span class="ot">=</span> <span class="fu">invnorm</span>(gexLh_transpose)</span>
<span id="cb14-74"><a href="#cb14-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-75"><a href="#cb14-75" aria-hidden="true" tabindex="-1"></a>gexPl_transpose <span class="ot">=</span> <span class="fu">invnorm</span>(gexPl_transpose)</span>
<span id="cb14-76"><a href="#cb14-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-77"><a href="#cb14-77" aria-hidden="true" tabindex="-1"></a>gexVo_transpose <span class="ot">=</span> <span class="fu">invnorm</span>(gexVo_transpose)</span>
<span id="cb14-78"><a href="#cb14-78" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-79"><a href="#cb14-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-80"><a href="#cb14-80" aria-hidden="true" tabindex="-1"></a>We first transposed each gene expression file as a matrix with individuals as the columns and genes for the rows. Now we add gene annotations.</span>
<span id="cb14-81"><a href="#cb14-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-84"><a href="#cb14-84" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb14-85"><a href="#cb14-85" aria-hidden="true" tabindex="-1"></a>gtf<span class="sc">$</span>gene_type <span class="ot">=</span> <span class="fu">sub</span>(<span class="st">".*?gene_biotype(.*?);.*"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, gtf<span class="sc">$</span>Attr)</span>
<span id="cb14-86"><a href="#cb14-86" aria-hidden="true" tabindex="-1"></a>gtf<span class="sc">$</span>gene_name <span class="ot">=</span> <span class="fu">sub</span>(<span class="st">".*?gene_name(.*?);.*"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, gtf<span class="sc">$</span>Attr)</span>
<span id="cb14-87"><a href="#cb14-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-88"><a href="#cb14-88" aria-hidden="true" tabindex="-1"></a>gene_annotation <span class="ot">=</span> <span class="fu">subset</span>(gtf, <span class="at">select =</span> <span class="sc">-</span><span class="fu">c</span>(Source, Feature, Score, Strand, Attr, Frame) )</span>
<span id="cb14-89"><a href="#cb14-89" aria-hidden="true" tabindex="-1"></a>gene_annotation <span class="ot">=</span> gene_annotation[, <span class="fu">c</span>(<span class="st">"Chr"</span>,<span class="st">"Gene"</span>, <span class="st">"gene_name"</span>, <span class="st">"Start"</span>, <span class="st">"End"</span>, <span class="st">"gene_type"</span> )]</span>
<span id="cb14-90"><a href="#cb14-90" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gene_annotation) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"chr"</span>, <span class="st">"gene_id"</span>, <span class="st">"gene_name"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>)</span>
<span id="cb14-91"><a href="#cb14-91" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(gene_annotation) <span class="ot">=</span> gene_annotation<span class="sc">$</span>gene_id</span>
<span id="cb14-92"><a href="#cb14-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-93"><a href="#cb14-93" aria-hidden="true" tabindex="-1"></a>phyMap <span class="ot">&lt;-</span> <span class="fu">within</span>(phyMap,  varID <span class="ot">&lt;-</span> <span class="fu">paste</span>(Chr, Pos, Ref, Alt, <span class="at">sep=</span><span class="st">"_"</span>))</span>
<span id="cb14-94"><a href="#cb14-94" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(phyMap) <span class="ot">=</span> phyMap<span class="sc">$</span>varID</span>
<span id="cb14-95"><a href="#cb14-95" aria-hidden="true" tabindex="-1"></a>phyMap<span class="sc">$</span>rsid <span class="ot">=</span> phyMap<span class="sc">$</span>varID</span>
<span id="cb14-96"><a href="#cb14-96" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(phyMap) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"snp"</span>, <span class="st">"chr"</span>, <span class="st">"pos"</span>, <span class="st">"refAllele"</span>, <span class="st">"effectAllele"</span>, <span class="st">'varID'</span>, <span class="st">"rsid"</span>)</span>
<span id="cb14-97"><a href="#cb14-97" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-98"><a href="#cb14-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-99"><a href="#cb14-99" aria-hidden="true" tabindex="-1"></a>Next, we subset genotype files to the rats with gene expression data from respective tissues.</span>
<span id="cb14-102"><a href="#cb14-102" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb14-103"><a href="#cb14-103" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(geno) <span class="ot">=</span> <span class="fu">rownames</span>(phyMap)</span>
<span id="cb14-104"><a href="#cb14-104" aria-hidden="true" tabindex="-1"></a><span class="co">#write.table(geno, file = "./data/genotype.txt", sep = "\t", col.names = TRUE, row.names = TRUE)</span></span>
<span id="cb14-105"><a href="#cb14-105" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-106"><a href="#cb14-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-109"><a href="#cb14-109" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb14-110"><a href="#cb14-110" aria-hidden="true" tabindex="-1"></a>geno_Ac <span class="ot">=</span> geno[,<span class="fu">match</span>(<span class="fu">rownames</span>(gexAc_transpose), <span class="fu">colnames</span>(geno))]</span>
<span id="cb14-111"><a href="#cb14-111" aria-hidden="true" tabindex="-1"></a>geno_Il <span class="ot">=</span> geno[,<span class="fu">match</span>(<span class="fu">rownames</span>(gexIl_transpose), <span class="fu">colnames</span>(geno))]</span>
<span id="cb14-112"><a href="#cb14-112" aria-hidden="true" tabindex="-1"></a>geno_Lh <span class="ot">=</span> geno[,<span class="fu">match</span>(<span class="fu">rownames</span>(gexLh_transpose), <span class="fu">colnames</span>(geno))]</span>
<span id="cb14-113"><a href="#cb14-113" aria-hidden="true" tabindex="-1"></a>geno_Pl <span class="ot">=</span> geno[,<span class="fu">match</span>(<span class="fu">rownames</span>(gexPl_transpose), <span class="fu">colnames</span>(geno))]</span>
<span id="cb14-114"><a href="#cb14-114" aria-hidden="true" tabindex="-1"></a>geno_Vo <span class="ot">=</span> geno[,<span class="fu">match</span>(<span class="fu">rownames</span>(gexVo_transpose), <span class="fu">colnames</span>(geno))]</span>
<span id="cb14-115"><a href="#cb14-115" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-116"><a href="#cb14-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-119"><a href="#cb14-119" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb14-120"><a href="#cb14-120" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(geno_Ac, <span class="at">file =</span> wd <span class="sc">%&amp;%</span> <span class="st">"genotype_Ac.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-121"><a href="#cb14-121" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(geno_Il, <span class="at">file =</span> wd <span class="sc">%&amp;%</span> <span class="st">"genotype_Il.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-122"><a href="#cb14-122" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(geno_Lh, <span class="at">file =</span> wd <span class="sc">%&amp;%</span> <span class="st">"genotype_Lh.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-123"><a href="#cb14-123" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(geno_Pl, <span class="at">file =</span> wd <span class="sc">%&amp;%</span> <span class="st">"genotype_Pl.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-124"><a href="#cb14-124" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(geno_Vo, <span class="at">file =</span> wd <span class="sc">%&amp;%</span> <span class="st">"genotype_Vo.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-125"><a href="#cb14-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-126"><a href="#cb14-126" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-127"><a href="#cb14-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-128"><a href="#cb14-128" aria-hidden="true" tabindex="-1"></a><span class="fu">## Generating MAP/PED files for plink</span></span>
<span id="cb14-129"><a href="#cb14-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-130"><a href="#cb14-130" aria-hidden="true" tabindex="-1"></a>MAP</span>
<span id="cb14-133"><a href="#cb14-133" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb14-134"><a href="#cb14-134" aria-hidden="true" tabindex="-1"></a>nIndividuals_Ac <span class="ot">&lt;-</span> <span class="fu">ncol</span>(geno_Ac)</span>
<span id="cb14-135"><a href="#cb14-135" aria-hidden="true" tabindex="-1"></a>nSNPs_Ac <span class="ot">&lt;-</span> <span class="fu">nrow</span>(geno_Ac)</span>
<span id="cb14-136"><a href="#cb14-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-137"><a href="#cb14-137" aria-hidden="true" tabindex="-1"></a>MAP <span class="ot">&lt;-</span> phyMap <span class="sc">%&gt;%</span> <span class="fu">rename</span>( <span class="at">SNP =</span> rsid, <span class="at">CHR =</span> chr, <span class="at">BP =</span> pos) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">CM =</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(CHR, SNP, CM, BP))</span>
<span id="cb14-138"><a href="#cb14-138" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(MAP, wd <span class="sc">%&amp;%</span> <span class="st">"rat_genotype.map"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb14-139"><a href="#cb14-139" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-140"><a href="#cb14-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-141"><a href="#cb14-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-142"><a href="#cb14-142" aria-hidden="true" tabindex="-1"></a>PED</span>
<span id="cb14-145"><a href="#cb14-145" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb14-146"><a href="#cb14-146" aria-hidden="true" tabindex="-1"></a>PED6 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb14-147"><a href="#cb14-147" aria-hidden="true" tabindex="-1"></a>  <span class="at">FID =</span> <span class="fu">colnames</span>(geno_Ac),</span>
<span id="cb14-148"><a href="#cb14-148" aria-hidden="true" tabindex="-1"></a>  <span class="at">IID =</span> <span class="fu">colnames</span>(geno_Ac),</span>
<span id="cb14-149"><a href="#cb14-149" aria-hidden="true" tabindex="-1"></a>  <span class="at">FatherID =</span> <span class="dv">0</span>,</span>
<span id="cb14-150"><a href="#cb14-150" aria-hidden="true" tabindex="-1"></a>  <span class="at">MotherID =</span> <span class="dv">0</span>,</span>
<span id="cb14-151"><a href="#cb14-151" aria-hidden="true" tabindex="-1"></a>  <span class="at">Sex =</span> <span class="dv">1</span>,</span>
<span id="cb14-152"><a href="#cb14-152" aria-hidden="true" tabindex="-1"></a>  <span class="at">Phenotype =</span> <span class="dv">1</span>)</span>
<span id="cb14-153"><a href="#cb14-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-154"><a href="#cb14-154" aria-hidden="true" tabindex="-1"></a><span class="co"># convert 0,1,2 to genotypes, a bit tricky</span></span>
<span id="cb14-155"><a href="#cb14-155" aria-hidden="true" tabindex="-1"></a><span class="co"># make helper dataframe for matching alleles</span></span>
<span id="cb14-156"><a href="#cb14-156" aria-hidden="true" tabindex="-1"></a>myAlleles <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb14-157"><a href="#cb14-157" aria-hidden="true" tabindex="-1"></a>  <span class="at">AA =</span> <span class="fu">paste</span>(phyMap<span class="sc">$</span>refAllele,phyMap<span class="sc">$</span>refAllele),</span>
<span id="cb14-158"><a href="#cb14-158" aria-hidden="true" tabindex="-1"></a>  <span class="at">AB =</span> <span class="fu">paste</span>(phyMap<span class="sc">$</span>refAllele, phyMap<span class="sc">$</span>effectAllele),</span>
<span id="cb14-159"><a href="#cb14-159" aria-hidden="true" tabindex="-1"></a>  <span class="at">BB =</span> <span class="fu">paste</span>(phyMap<span class="sc">$</span>effectAllele,phyMap<span class="sc">$</span>effectAllele) )</span>
<span id="cb14-160"><a href="#cb14-160" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb14-161"><a href="#cb14-161" aria-hidden="true" tabindex="-1"></a>PEDsnps <span class="ot">&lt;-</span> geno_Ac[,] <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb14-162"><a href="#cb14-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-163"><a href="#cb14-163" aria-hidden="true" tabindex="-1"></a>PEDsnpsAB <span class="ot">&lt;-</span> </span>
<span id="cb14-164"><a href="#cb14-164" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(<span class="fu">seq</span>(nSNPs_Ac), <span class="cf">function</span>(snp)</span>
<span id="cb14-165"><a href="#cb14-165" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sapply</span>(PEDsnps[snp, ], <span class="cf">function</span>(ind) myAlleles[snp, ind]))</span>
<span id="cb14-166"><a href="#cb14-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-167"><a href="#cb14-167" aria-hidden="true" tabindex="-1"></a>PED <span class="ot">&lt;-</span> <span class="fu">cbind</span>(PED6, PEDsnpsAB)</span>
<span id="cb14-168"><a href="#cb14-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-169"><a href="#cb14-169" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PED, wd <span class="sc">%&amp;%</span> <span class="st">"rat_genotype.ped"</span>,<span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb14-170"><a href="#cb14-170" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-171"><a href="#cb14-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-172"><a href="#cb14-172" aria-hidden="true" tabindex="-1"></a>Repeat for other tissues ~ only need to change the rows</span>
<span id="cb14-175"><a href="#cb14-175" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb14-176"><a href="#cb14-176" aria-hidden="true" tabindex="-1"></a>PED_Il <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">FID =</span> <span class="fu">colnames</span>(geno_Il), <span class="at">IID =</span> <span class="fu">colnames</span>(geno_Il), <span class="at">FatherID =</span> <span class="dv">0</span>, <span class="at">MotherID =</span> <span class="dv">0</span>, <span class="at">Sex =</span> <span class="dv">1</span>, <span class="at">Phenotype =</span> <span class="dv">1</span>)</span>
<span id="cb14-177"><a href="#cb14-177" aria-hidden="true" tabindex="-1"></a>PED_Lh <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">FID =</span> <span class="fu">colnames</span>(geno_Lh), <span class="at">IID =</span> <span class="fu">colnames</span>(geno_Lh), <span class="at">FatherID =</span> <span class="dv">0</span>, <span class="at">MotherID =</span> <span class="dv">0</span>, <span class="at">Sex =</span> <span class="dv">1</span>, <span class="at">Phenotype =</span> <span class="dv">1</span>)</span>
<span id="cb14-178"><a href="#cb14-178" aria-hidden="true" tabindex="-1"></a>PED_Pl <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">FID =</span> <span class="fu">colnames</span>(geno_Pl), <span class="at">IID =</span> <span class="fu">colnames</span>(geno_Pl), <span class="at">FatherID =</span> <span class="dv">0</span>, <span class="at">MotherID =</span> <span class="dv">0</span>, <span class="at">Sex =</span> <span class="dv">1</span>, <span class="at">Phenotype =</span> <span class="dv">1</span>)</span>
<span id="cb14-179"><a href="#cb14-179" aria-hidden="true" tabindex="-1"></a>PED_Vo <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">FID =</span> <span class="fu">colnames</span>(geno_Vo), <span class="at">IID =</span> <span class="fu">colnames</span>(geno_Vo), <span class="at">FatherID =</span> <span class="dv">0</span>, <span class="at">MotherID =</span> <span class="dv">0</span>, <span class="at">Sex =</span> <span class="dv">1</span>, <span class="at">Phenotype =</span> <span class="dv">1</span>)</span>
<span id="cb14-180"><a href="#cb14-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-181"><a href="#cb14-181" aria-hidden="true" tabindex="-1"></a>PEDsnps <span class="ot">&lt;-</span> geno_Il[,] <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb14-182"><a href="#cb14-182" aria-hidden="true" tabindex="-1"></a>PEDsnpsAB <span class="ot">&lt;-</span> </span>
<span id="cb14-183"><a href="#cb14-183" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(<span class="fu">seq</span>(<span class="fu">nrow</span>(geno_Il)), <span class="cf">function</span>(snp)</span>
<span id="cb14-184"><a href="#cb14-184" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sapply</span>(PEDsnps[snp, ], <span class="cf">function</span>(ind) myAlleles[snp, ind]))</span>
<span id="cb14-185"><a href="#cb14-185" aria-hidden="true" tabindex="-1"></a>PED <span class="ot">&lt;-</span> <span class="fu">cbind</span>(PED_Il, PEDsnpsAB)</span>
<span id="cb14-186"><a href="#cb14-186" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PED, wd <span class="sc">%&amp;%</span> <span class="st">"rat_genotype_Il.ped"</span>,<span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb14-187"><a href="#cb14-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-188"><a href="#cb14-188" aria-hidden="true" tabindex="-1"></a>PEDsnps <span class="ot">&lt;-</span> geno_Lh[,] <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb14-189"><a href="#cb14-189" aria-hidden="true" tabindex="-1"></a>PEDsnpsAB <span class="ot">&lt;-</span> </span>
<span id="cb14-190"><a href="#cb14-190" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(<span class="fu">seq</span>(<span class="fu">nrow</span>(geno_Lh)), <span class="cf">function</span>(snp)</span>
<span id="cb14-191"><a href="#cb14-191" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sapply</span>(PEDsnps[snp, ], <span class="cf">function</span>(ind) myAlleles[snp, ind]))</span>
<span id="cb14-192"><a href="#cb14-192" aria-hidden="true" tabindex="-1"></a>PED <span class="ot">&lt;-</span> <span class="fu">cbind</span>(PED_Lh, PEDsnpsAB)</span>
<span id="cb14-193"><a href="#cb14-193" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PED, wd <span class="sc">%&amp;%</span> <span class="st">"rat_genotype_Lh.ped"</span>,<span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb14-194"><a href="#cb14-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-195"><a href="#cb14-195" aria-hidden="true" tabindex="-1"></a>PEDsnps <span class="ot">&lt;-</span> geno_Pl[,] <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb14-196"><a href="#cb14-196" aria-hidden="true" tabindex="-1"></a>PEDsnpsAB <span class="ot">&lt;-</span> </span>
<span id="cb14-197"><a href="#cb14-197" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(<span class="fu">seq</span>(<span class="fu">nrow</span>(geno_Pl)), <span class="cf">function</span>(snp)</span>
<span id="cb14-198"><a href="#cb14-198" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sapply</span>(PEDsnps[snp, ], <span class="cf">function</span>(ind) myAlleles[snp, ind]))</span>
<span id="cb14-199"><a href="#cb14-199" aria-hidden="true" tabindex="-1"></a>PED <span class="ot">&lt;-</span> <span class="fu">cbind</span>(PED_Pl, PEDsnpsAB)</span>
<span id="cb14-200"><a href="#cb14-200" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PED, wd <span class="sc">%&amp;%</span> <span class="st">"rat_genotype_Pl.ped"</span>,<span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb14-201"><a href="#cb14-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-202"><a href="#cb14-202" aria-hidden="true" tabindex="-1"></a>PEDsnps <span class="ot">&lt;-</span> geno_Vo[,] <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb14-203"><a href="#cb14-203" aria-hidden="true" tabindex="-1"></a>PEDsnpsAB <span class="ot">&lt;-</span> </span>
<span id="cb14-204"><a href="#cb14-204" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(<span class="fu">seq</span>(<span class="fu">nrow</span>(geno_Vo)), <span class="cf">function</span>(snp)</span>
<span id="cb14-205"><a href="#cb14-205" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sapply</span>(PEDsnps[snp, ], <span class="cf">function</span>(ind) myAlleles[snp, ind]))</span>
<span id="cb14-206"><a href="#cb14-206" aria-hidden="true" tabindex="-1"></a>PED <span class="ot">&lt;-</span> <span class="fu">cbind</span>(PED_Vo, PEDsnpsAB)</span>
<span id="cb14-207"><a href="#cb14-207" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(PED, wd <span class="sc">%&amp;%</span> <span class="st">"rat_genotype_Vo.ped"</span>,<span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb14-208"><a href="#cb14-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-209"><a href="#cb14-209" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-210"><a href="#cb14-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-211"><a href="#cb14-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-212"><a href="#cb14-212" aria-hidden="true" tabindex="-1"></a>Generating Phenotype file ~ We're using gene expression levels</span>
<span id="cb14-215"><a href="#cb14-215" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb14-216"><a href="#cb14-216" aria-hidden="true" tabindex="-1"></a>pheno_Ac <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(<span class="at">FID =</span> <span class="fu">rownames</span>(gexAc_transpose), <span class="at">IID =</span> <span class="fu">rownames</span>(gexAc_transpose), gexAc_transpose))</span>
<span id="cb14-217"><a href="#cb14-217" aria-hidden="true" tabindex="-1"></a>pheno_Il <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(<span class="at">FID =</span> <span class="fu">rownames</span>(gexIl_transpose), <span class="at">IID =</span> <span class="fu">rownames</span>(gexIl_transpose), gexIl_transpose))</span>
<span id="cb14-218"><a href="#cb14-218" aria-hidden="true" tabindex="-1"></a>pheno_Lh <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(<span class="at">FID =</span> <span class="fu">rownames</span>(gexLh_transpose), <span class="at">IID =</span> <span class="fu">rownames</span>(gexLh_transpose), gexLh_transpose))</span>
<span id="cb14-219"><a href="#cb14-219" aria-hidden="true" tabindex="-1"></a>pheno_Pl <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(<span class="at">FID =</span> <span class="fu">rownames</span>(gexPl_transpose), <span class="at">IID =</span> <span class="fu">rownames</span>(gexPl_transpose), gexPl_transpose))</span>
<span id="cb14-220"><a href="#cb14-220" aria-hidden="true" tabindex="-1"></a>pheno_Vo <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(<span class="at">FID =</span> <span class="fu">rownames</span>(gexVo_transpose), <span class="at">IID =</span> <span class="fu">rownames</span>(gexVo_transpose), gexVo_transpose))</span>
<span id="cb14-221"><a href="#cb14-221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-222"><a href="#cb14-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-223"><a href="#cb14-223" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(pheno_Ac, wd <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/rat_pheno_Ac.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb14-224"><a href="#cb14-224" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(pheno_Il, wd <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/rat_pheno_Il.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb14-225"><a href="#cb14-225" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(pheno_Lh, wd <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/rat_pheno_Lh.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb14-226"><a href="#cb14-226" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(pheno_Pl, wd <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/rat_pheno_Pl.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb14-227"><a href="#cb14-227" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(pheno_Vo, wd <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/rat_pheno_Vo.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb14-228"><a href="#cb14-228" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-229"><a href="#cb14-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-230"><a href="#cb14-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-231"><a href="#cb14-231" aria-hidden="true" tabindex="-1"></a><span class="fu">## Convert MAP/PED files to plink </span></span>
<span id="cb14-232"><a href="#cb14-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-233"><a href="#cb14-233" aria-hidden="true" tabindex="-1"></a>All conversion to plink format use the same MAP file, but because of plink conventions, the MAP file must match the name of the PED file, so we rename the map file each time we convert genotypes for a different tissue.</span>
<span id="cb14-234"><a href="#cb14-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-235"><a href="#cb14-235" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash, eval=FALSE}</span></span>
<span id="cb14-236"><a href="#cb14-236" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /gpfs/data/im-lab/nas40t2/sabrina/Rat_Genomics_Paper_Pipeline/data</span>
<span id="cb14-237"><a href="#cb14-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-238"><a href="#cb14-238" aria-hidden="true" tabindex="-1"></a><span class="ex">plink</span> <span class="at">--file</span> rat_genotype <span class="at">--make-bed</span> <span class="at">--out</span> plink_files/rat_genome_Ac</span>
<span id="cb14-239"><a href="#cb14-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-240"><a href="#cb14-240" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> rat_genotype.map rat_genotype_Il.map</span>
<span id="cb14-241"><a href="#cb14-241" aria-hidden="true" tabindex="-1"></a><span class="ex">plink</span> <span class="at">--file</span> rat_genotype_Il <span class="at">--make-bed</span> <span class="at">--out</span> plink_files/rat_genome_Il</span>
<span id="cb14-242"><a href="#cb14-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-243"><a href="#cb14-243" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> rat_genotype_Il.map rat_genotype_Lh.map</span>
<span id="cb14-244"><a href="#cb14-244" aria-hidden="true" tabindex="-1"></a><span class="ex">plink</span> <span class="at">--file</span> rat_genotype_Lh <span class="at">--make-bed</span> <span class="at">--out</span> plink_files/rat_genome_Lh</span>
<span id="cb14-245"><a href="#cb14-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-246"><a href="#cb14-246" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> rat_genotype_Lh.map rat_genotype_Pl.map</span>
<span id="cb14-247"><a href="#cb14-247" aria-hidden="true" tabindex="-1"></a><span class="ex">plink</span> <span class="at">--file</span> rat_genotype_Pl <span class="at">--make-bed</span> <span class="at">--out</span> plink_files/rat_genome_Pl</span>
<span id="cb14-248"><a href="#cb14-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-249"><a href="#cb14-249" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> rat_genotype_Pl.map rat_genotype_Vo.map</span>
<span id="cb14-250"><a href="#cb14-250" aria-hidden="true" tabindex="-1"></a><span class="ex">plink</span> <span class="at">--file</span> rat_genotype_Vo <span class="at">--make-bed</span> <span class="at">--out</span> plink_files/rat_genome_Vo</span>
<span id="cb14-251"><a href="#cb14-251" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>