<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Natasha Santhanam">
<meta name="dcterms.date" content="2022-02-14">

<title>Rat Genomics Pipeline Rendered - GTEx_comparison_rat_n_genes</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-listing/list.min.js"></script>
<script src="../site_libs/quarto-listing/quarto-listing.js"></script>
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script>

  window.document.addEventListener("DOMContentLoaded", function (_event) {
    const listingTargetEl = window.document.querySelector('#listing-listing .list');
    if (!listingTargetEl) {
      // No listing discovered, do not attach.
      return; 
    }

    const options = {
      valueNames: [{ data: ['index'] },{ data: ['categories'] },{ data: ['listing-date-sort'] },{ data: ['listing-file-modified-sort'] }],
      
      searchColumns: [],
    };

    window['quarto-listings'] = window['quarto-listings'] || {};
    window['quarto-listings']['listing-listing'] = new List('listing-listing', options);

    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  });

  window.addEventListener('hashchange',() => {
    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  })
  </script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Rat Genomics Pipeline Rendered</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html" rel="" target="">
 <span class="menu-text">paper_code</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../other_code.html" rel="" target="">
 <span class="menu-text">more_code</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#compare-number-of-genes-predicted-between-rat-and-gtex-prediction-models" id="toc-compare-number-of-genes-predicted-between-rat-and-gtex-prediction-models" class="nav-link active" data-scroll-target="#compare-number-of-genes-predicted-between-rat-and-gtex-prediction-models">Compare number of genes predicted between Rat and GTEx prediction models</a></li>
  <li><a href="#generate-gtex-prediction-models-using-the-same-pipeline-as-in-rats" id="toc-generate-gtex-prediction-models-using-the-same-pipeline-as-in-rats" class="nav-link" data-scroll-target="#generate-gtex-prediction-models-using-the-same-pipeline-as-in-rats">Generate GTEx prediction models using the same pipeline as in Rats</a></li>
  <li><a href="#compare-number-of-genes-predicted-in-gtex-to-those-in-rats" id="toc-compare-number-of-genes-predicted-in-gtex-to-those-in-rats" class="nav-link" data-scroll-target="#compare-number-of-genes-predicted-in-gtex-to-those-in-rats">Compare number of genes predicted in GTEx to those in Rats</a></li>
  <li><a href="#plot-sparsity-for-gtex" id="toc-plot-sparsity-for-gtex" class="nav-link" data-scroll-target="#plot-sparsity-for-gtex">Plot Sparsity for GTEx</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">GTEx_comparison_rat_n_genes</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Natasha Santhanam </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 14, 2022</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(broom)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RSQLite)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="st">"%&amp;%"</span> <span class="ot">=</span> <span class="cf">function</span>(a,b) <span class="fu">paste</span>(a,b,<span class="at">sep=</span><span class="st">""</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/Data/GTEx/V8/GTEx_Analysis_v8_eQTL_expression_matrices/"</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>geno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/comp_to_GTEx/"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="compare-number-of-genes-predicted-between-rat-and-gtex-prediction-models" class="level2">
<h2 class="anchored" data-anchor-id="compare-number-of-genes-predicted-between-rat-and-gtex-prediction-models">Compare number of genes predicted between Rat and GTEx prediction models</h2>
</section>
<section id="generate-gtex-prediction-models-using-the-same-pipeline-as-in-rats" class="level1">
<h1>Generate GTEx prediction models using the same pipeline as in Rats</h1>
<p>Gather Genotype, Gene epression data, snp annotation file and gtf (made with ensembl notation)</p>
<p>First convert genotypes to single coded format to be read by pipeline</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="ex">plink</span> <span class="at">--bfile</span> /gpfs/data/im-lab/nas40t2/Data/GTEx/V8/genotype/plink_files/GTEx_maf_0.01 <span class="at">--geno</span> 0.02 <span class="at">--mind</span> 0.02 <span class="at">--maf</span> 0.05 <span class="at">--make-bed</span> <span class="at">--out</span> GTEx_comp</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="ex">plink</span> <span class="at">--bfile</span> GTEx_comp <span class="at">--recode</span> A-transpose <span class="at">--out</span> GTEx_single_code</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>geno <span class="ot">&lt;-</span> <span class="fu">fread</span>(geno.dir <span class="sc">%&amp;%</span> <span class="st">"genos/GTEx_single_code.traw"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>gex <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(dir <span class="sc">%&amp;%</span> TISSUE <span class="sc">%&amp;%</span> <span class="st">".v8.normalized_expression.bed.gz"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">`</span><span class="at">#chr</span><span class="st">`</span>, start, end))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Genotype has to have SNP in first column</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>geno <span class="ot">&lt;-</span> geno <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(SNP, CHR, <span class="fu">colnames</span>(geno)[<span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(geno)]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>transposing gene expression files and inverse normalize</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>gex <span class="ot">&lt;-</span> gex <span class="sc">%&gt;%</span> <span class="fu">pivot_longer</span>(<span class="sc">!</span>gene_id, <span class="at">names_to =</span> <span class="st">"IID"</span>, <span class="at">values_to =</span> <span class="st">"count"</span>) <span class="sc">%&gt;%</span> <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> gene_id, <span class="at">values_from =</span> count)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>invnorm <span class="ot">=</span> <span class="cf">function</span>(x) {</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">is.null</span>(<span class="fu">dim</span>(x))) res <span class="ot">=</span> <span class="fu">invnorm.vector</span>(x) <span class="cf">else</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  res<span class="ot">=</span><span class="fu">apply</span>(x,<span class="dv">2</span>,invnorm.vector)</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  res</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>invnorm.vector <span class="ot">=</span> <span class="cf">function</span>(x) {yy <span class="ot">=</span> <span class="fu">rank</span>(x)<span class="sc">/</span>(<span class="fu">length</span>(x)<span class="sc">+</span><span class="dv">1</span>); <span class="fu">qnorm</span>(yy)}</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>gex_transpose <span class="ot">&lt;-</span> gex <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(IID))</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>gex_transpose <span class="ot">=</span> <span class="fu">invnorm</span>(gex_transpose)</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(gex_transpose) <span class="ot">=</span> gex<span class="sc">$</span>IID</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(gex_transpose, <span class="at">file =</span> <span class="st">'/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/comp_to_GTEx/expression/gexWB.csv'</span>, <span class="at">sep =</span> <span class="st">","</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Format gene annotation</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>gtf <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/GTEX_Analysis/annotations_gencode_v26.tsv"</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>gene_annotation <span class="ot">&lt;-</span> gtf <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(chromosome, gene_id, gene_name, start, end))</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(gene_annotation) <span class="ot">=</span> gtf<span class="sc">$</span>gene_id</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Format snp annotation</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>phyMap <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(geno.dir <span class="sc">%&amp;%</span> <span class="st">"genos/GTEx_comp.bim"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)  <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">varID =</span> X2) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(X2, X1, X4, X5, X6, varID))</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>phyMap<span class="sc">$</span>rsid <span class="ot">=</span> phyMap<span class="sc">$</span>varID</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(phyMap) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"snp"</span>, <span class="st">"chr"</span>, <span class="st">"pos"</span>, <span class="st">"refAllele"</span>, <span class="st">"effectAllele"</span>, <span class="st">"varID"</span>, <span class="st">"rsid"</span>)  </span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(phyMap) <span class="ot">=</span> phyMap<span class="sc">$</span>varID</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Splitting the snp annotation file by chromosome</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>s <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">split</span>(phyMap, phyMap<span class="sc">$</span>chr), <span class="fu">paste0</span>(<span class="st">"snp_annot.chr"</span>, <span class="fu">unique</span>(phyMap<span class="sc">$</span>chr)))</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="fu">list2env</span>(s, <span class="fu">globalenv</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Splitting the genotype file by chromosome - run this from the rat_genomic_alaysis directory</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="ex">python</span> /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/rat-genomic-analysis/scripts/split_genotype_by_chr.py GTEx_genotype.txt  /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/comp_to_GTEx/genos/geno_by_chr/<span class="st">'genotype'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Now we are performing PEER factor analysis on each tissue choosing 7 factors</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="ex">peertool</span> <span class="at">-f</span> expression/<span class="st">"gexWB.csv"</span> <span class="at">-n</span> 7 <span class="at">-o</span> peer_GTEx <span class="at">--has_header</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Read in Peer Factors</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>peer_factors<span class="ot">=</span> <span class="fu">read.csv</span>(<span class="at">file =</span> geno.dir <span class="sc">%&amp;%</span> <span class="st">"peer_GTEx/X.csv"</span>, <span class="at">header =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Manipulating the PEER factor files so we can append to covariate file</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(peer_factors) <span class="ot">=</span> gex<span class="sc">$</span>IID</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(peer_factors) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'PF1'</span>, <span class="st">'PF2'</span>, <span class="st">'PF3'</span>, <span class="st">'PF4'</span>, <span class="st">'PF5'</span>, <span class="st">'PF6'</span>, <span class="st">'PF7'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co">#initializing matrices to be filled with t-stats, p_vals, and residuals of the regression of each gene vs. the covariates for each tissue. t-stat and p-val matrices are just for diagnostics</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="co"># t_stats = matrix(nrow = 13, ncol = length(colnames(gex)))</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="co"># p_vals = matrix(nrow = 13, ncol = length(colnames(gex)))</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>expression <span class="ot">=</span> gex_transpose</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Regressing out the covariates and saving the residuals as the new expression for each tissue</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(gex_transpose))) {</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>    fit <span class="ot">=</span> <span class="fu">lm</span>(gex_transpose[,i] <span class="sc">~</span>  peer_factors<span class="sc">$</span>PF1 <span class="sc">+</span> peer_factors<span class="sc">$</span>PF2 <span class="sc">+</span> peer_factors<span class="sc">$</span>PF3 <span class="sc">+</span> peer_factors<span class="sc">$</span>PF4 <span class="sc">+</span> peer_factors<span class="sc">$</span>PF5 <span class="sc">+</span> peer_factors<span class="sc">$</span>PF6 <span class="sc">+</span> peer_factors<span class="sc">$</span>PF7)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>    expression[,i] <span class="ot">&lt;-</span> fit<span class="sc">$</span>residuals</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the expression RDS objects to be used as arguments in the script</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(<span class="fu">as.matrix</span>(expression), geno.dir <span class="sc">%&amp;%</span> TISSUE <span class="sc">%&amp;%</span> <span class="st">"_expression_transformed.RDS"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Find intersection of genes across all GTEx tissues and Rats</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">=</span> <span class="fu">list.files</span>(geno.dir <span class="sc">%&amp;%</span> <span class="st">"expression"</span> , <span class="at">pattern =</span> <span class="st">"expression_transformed.RDS"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>names <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">readRDS</span>(filelist[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> <span class="fu">colnames</span>())</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(filelist)]) {</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">readRDS</span>(fila) <span class="sc">%&gt;%</span> <span class="fu">colnames</span>())</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  names <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(names, tempo, <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>names<span class="sc">$</span>gene <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(names<span class="sc">$</span>gene, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a><span class="co">#do the same for all 5 rat tissues </span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">=</span> <span class="fu">list.files</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/expression/"</span>, <span class="at">pattern =</span> <span class="st">"expression_transformed.RDS"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>rat_names <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">readRDS</span>(filelist[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> <span class="fu">colnames</span>())</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(filelist)]) {</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">readRDS</span>(fila) <span class="sc">%&gt;%</span> <span class="fu">colnames</span>())</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>  rat_names <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(rat_names, tempo, <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a><span class="co">#change gene id in rats to human notation</span></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>orth.rats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/expression/ortholog_genes_rats_humans.tsv"</span>)</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>rat_names <span class="ot">&lt;-</span> rat_names <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">human_gene =</span> orth.rats[<span class="fu">match</span>(rat_names<span class="sc">$</span>gene, orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene), <span class="dv">1</span>]<span class="sc">$</span>ensembl_gene_id)</span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a>overlap <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">human_gene =</span> <span class="fu">intersect</span>(rat_names<span class="sc">$</span>human_gene, names<span class="sc">$</span>gene)) <span class="co">#12,230 overlap </span></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>overlap <span class="ot">&lt;-</span> overlap <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">rat_gene =</span> orth.rats[<span class="fu">match</span>(overlap<span class="sc">$</span>human_gene, orth.rats<span class="sc">$</span>ensembl_gene_id), <span class="dv">3</span>]<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Filter for only overlapped genes in expression files</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">=</span> <span class="fu">list.files</span>(geno.dir <span class="sc">%&amp;%</span> <span class="st">"expression"</span>, <span class="at">pattern =</span> <span class="st">"expression_transformed.RDS"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(fila)</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tempo) <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">colnames</span>(tempo), <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> tempo[, overlap<span class="sc">$</span>human_gene]</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  tis <span class="ot">=</span> <span class="fu">substr</span>(fila, <span class="dv">73</span>, <span class="fu">str_length</span>(fila) <span class="sc">-</span> <span class="dv">27</span>)</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(tempo, geno.dir <span class="sc">%&amp;%</span> txis <span class="sc">%&amp;%</span> <span class="st">"_expression_abrv.RDS"</span>)</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co">#set snp annotation rownames </span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr1) <span class="ot">=</span> snp_annot.chr1<span class="sc">$</span>varID</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr2) <span class="ot">=</span> snp_annot.chr2<span class="sc">$</span>varID</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr3) <span class="ot">=</span> snp_annot.chr3<span class="sc">$</span>varID</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr4) <span class="ot">=</span> snp_annot.chr4<span class="sc">$</span>varID</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr5) <span class="ot">=</span> snp_annot.chr5<span class="sc">$</span>varID</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr6) <span class="ot">=</span> snp_annot.chr6<span class="sc">$</span>varID</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr7) <span class="ot">=</span> snp_annot.chr7<span class="sc">$</span>varID</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr8) <span class="ot">=</span> snp_annot.chr8<span class="sc">$</span>varID</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr9) <span class="ot">=</span> snp_annot.chr9<span class="sc">$</span>varID</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr10) <span class="ot">=</span> snp_annot.chr10<span class="sc">$</span>varID</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr11) <span class="ot">=</span> snp_annot.chr11<span class="sc">$</span>varID</span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr12) <span class="ot">=</span> snp_annot.chr12<span class="sc">$</span>varID</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr13) <span class="ot">=</span> snp_annot.chr13<span class="sc">$</span>varID</span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr14) <span class="ot">=</span> snp_annot.chr14<span class="sc">$</span>varID</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr15) <span class="ot">=</span> snp_annot.chr15<span class="sc">$</span>varID</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr16) <span class="ot">=</span> snp_annot.chr16<span class="sc">$</span>varID</span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr17) <span class="ot">=</span> snp_annot.chr17<span class="sc">$</span>varID</span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr18) <span class="ot">=</span> snp_annot.chr18<span class="sc">$</span>varID</span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr19) <span class="ot">=</span> snp_annot.chr19<span class="sc">$</span>varID</span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr20) <span class="ot">=</span> snp_annot.chr20<span class="sc">$</span>varID</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the SNP annotation RDS objects to be used as arguments in the script - too lazy to write a for loop </span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr1, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr1.RDS"</span>)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr2, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr2.RDS"</span>)</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr3, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr3.RDS"</span>)</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr4, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr4.RDS"</span>)</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr5, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr5.RDS"</span>)</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr6, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr6.RDS"</span>)</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr7, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr7.RDS"</span>)</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr8, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr8.RDS"</span>)</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr9, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr9.RDS"</span>)</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr10, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr10.RDS"</span>)</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr11, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr11.RDS"</span>)</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr12, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr12.RDS"</span>)</span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr13, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr13.RDS"</span>)</span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr14, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr14.RDS"</span>)</span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr15, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr15.RDS"</span>)</span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr16, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr16.RDS"</span>)</span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr17, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr17.RDS"</span>)</span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr18, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr18.RDS"</span>)</span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr19, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr19.RDS"</span>)</span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr20, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr20.RDS"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the meta data file for each tissue </span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="ex">python</span> /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/rat-genomic-analysis/scripts/create_meta_data.py <span class="at">--geno</span> <span class="st">"genos/GTEx_genotype.txt"</span> <span class="at">--expr</span> <span class="st">"Ovary_expression_transformed.RDS"</span> <span class="at">--snpset</span> <span class="st">"1KG"</span> <span class="at">--rsid_label</span> 1 <span class="at">--window</span> 1000000 <span class="at">--out_prefix</span> <span class="st">"Results/allMetaData/GTEx_ovary"</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Running the model training script for each tissue/chromosome pair</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="ex">Rscript</span> <span class="at">--vanilla</span> /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/comp_to_GTEx/GTEx_TW_CV_elasticNet.R  <span class="va">$chr</span> <span class="va">$alpha</span> </span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="va">a</span><span class="op">=</span>0.01</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">20</span><span class="dt">}</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a><span class="ex">qsub</span> <span class="at">-v</span> chr=<span class="va">$i</span>,alpha=<span class="va">$a</span> GTEx_single_nested_EN.pbs  </span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Concatenating all of the results files for each tissue</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fu">bash</span> /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/rat-genomic-analysis/scripts/make_all_results.sh  <span class="st">'Whole_Blood'</span> <span class="st">'/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/comp_to_GTEx/Results/all_Results_Whole_Blood_0.01'</span> 0.01 <span class="st">'1KG_snps'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="compare-number-of-genes-predicted-in-gtex-to-those-in-rats" class="level1">
<h1>Compare number of genes predicted in GTEx to those in Rats</h1>
<p>Now we have predictability for all parameters of alpha. We can now iterate through all alphas and create the long data format. We also only select for genes that have an average R2 &gt; 0.3 and subsample 20 genes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/comp_to_GTEx/Results/chr_Results"</span>, <span class="at">pattern =</span> <span class="st">"working_TW_"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>tempo <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(filelist[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(R2 <span class="sc">&lt;=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, cor, R2)) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(gene, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>tempo<span class="sc">$</span>cor <span class="ot">=</span> <span class="fu">as.numeric</span>(tempo<span class="sc">$</span>cor)</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>tempo<span class="sc">$</span>R2 <span class="ot">=</span> tempo<span class="sc">$</span>R2<span class="sc">*</span><span class="fu">sign</span>(tempo<span class="sc">$</span>cor)</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tempo)[<span class="dv">2</span>] <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">"cor"</span>, <span class="st">"0.01"</span>, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tempo)[<span class="dv">3</span>] <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">"R2"</span>, <span class="st">"0.01"</span>, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (k <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span>(<span class="fu">length</span>(filelist))){</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a> df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(filelist[k], <span class="at">col_names =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(gene, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a> alpha <span class="ot">&lt;-</span> <span class="fu">substr</span>(filelist[k], <span class="dv">135</span>, <span class="fu">str_length</span>(filelist[k]) <span class="sc">-</span> <span class="dv">8</span>)</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a> df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(df) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, cor, R2)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(R2 <span class="sc">&lt;=</span> <span class="dv">1</span>)</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a> df<span class="sc">$</span>cor <span class="ot">=</span> <span class="fu">as.numeric</span>(df<span class="sc">$</span>cor)</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a> df<span class="sc">$</span>R2 <span class="ot">=</span> <span class="fu">as.numeric</span>(df<span class="sc">$</span>R2)<span class="sc">*</span><span class="fu">sign</span>(df<span class="sc">$</span>cor)</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a> <span class="fu">colnames</span>(df)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"cor"</span>, alpha, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a> <span class="fu">colnames</span>(df)[<span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"R2"</span>, alpha, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a> tempo <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(tempo, df, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene"</span>))</span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a><span class="co">#tempo &lt;- read_tsv("/Users/natashasanthanam/Downloads/GTEx_Whole_Blood_sparsity_all_parameters.txt", col_names = TRUE)</span></span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>tempo_cor_df <span class="ot">&lt;-</span> tempo <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, <span class="fu">starts_with</span>(<span class="st">"cor"</span>))) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(cor_0<span class="fl">.01</span>))</span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a>tempo_R2_df <span class="ot">&lt;-</span> tempo <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, <span class="fu">starts_with</span>(<span class="st">"R"</span>))) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(R_0<span class="fl">.01</span>))</span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>tempo_R2_df <span class="ot">&lt;-</span> tempo_R2_df[<span class="fu">sample</span>(<span class="fu">nrow</span>(tempo_R2_df), <span class="dv">20</span>), ]</span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>tempo_cor_df <span class="ot">&lt;-</span> tempo_cor_df[<span class="fu">sample</span>(<span class="fu">nrow</span>(tempo_cor_df), <span class="dv">20</span>), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-sparsity-for-gtex" class="level1">
<h1>Plot Sparsity for GTEx</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>data_long <span class="ot">&lt;-</span> tempo_R2_df  <span class="sc">%&gt;%</span>  <span class="fu">pivot_longer</span>(<span class="sc">!</span>gene, <span class="at">names_to =</span> <span class="st">"value"</span>, <span class="at">values_to =</span> <span class="st">"count"</span>) </span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>data_long<span class="sc">$</span>value <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="fu">substr</span>(data_long<span class="sc">$</span>value, <span class="dv">4</span>, <span class="fu">str_length</span>(data_long<span class="sc">$</span>value)))</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span>  <span class="fu">ggplot</span>(data_long, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.numeric</span>(value), <span class="at">y =</span> count)) <span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"loess"</span>, <span class="at">span =</span> <span class="dv">2</span>, <span class="at">show_guide =</span> <span class="cn">FALSE</span>, <span class="at">se=</span>T, <span class="at">size =</span> .<span class="dv">5</span>, <span class="at">col =</span> <span class="st">"maroon1"</span>)  <span class="sc">+</span>  <span class="fu">xlab</span>(<span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">"Elastic net mixing parameter ("</span>,alpha, <span class="st">")"</span>))) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">"10-fold cross-validated R2"</span>)))</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>p1</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>cor_long <span class="ot">&lt;-</span> tempo_cor_df  <span class="sc">%&gt;%</span>  <span class="fu">pivot_longer</span>(<span class="sc">!</span>gene, <span class="at">names_to =</span> <span class="st">"value"</span>, <span class="at">values_to =</span> <span class="st">"count"</span>) </span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>cor_long<span class="sc">$</span>value <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="fu">substr</span>(cor_long<span class="sc">$</span>value, <span class="dv">5</span>, <span class="fu">str_length</span>(cor_long<span class="sc">$</span>value)))</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span>  <span class="fu">ggplot</span>(cor_long, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.numeric</span>(value), <span class="at">y =</span> count)) <span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="at">show_guide =</span> <span class="cn">FALSE</span>, <span class="at">se=</span>T, <span class="at">size =</span> .<span class="dv">5</span>, <span class="at">col =</span> <span class="st">"maroon1"</span>)  <span class="sc">+</span>  <span class="fu">xlab</span>(<span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">"Elastic net mixing parameter ("</span>,alpha, <span class="st">")"</span>))) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">"10-fold cross-validated cor"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Create Number of Genes for Rat Tissues</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>rat.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Tyson_Results/"</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(rat.dir, <span class="at">pattern =</span> <span class="st">"all_results"</span>, <span class="at">full.names=</span><span class="cn">TRUE</span>)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>rat_n_genes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">tis=</span> <span class="fu">as.character</span>(), <span class="at">n.genes =</span> <span class="fu">numeric</span>(), <span class="at">species =</span> <span class="fu">as.character</span>())</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  i <span class="ot">=</span> <span class="fu">match</span>(fila, filelist)</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(fila, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> tempo <span class="sc">%&gt;%</span> <span class="fu">filter</span>(R2 <span class="sc">&gt;=</span> <span class="dv">0</span>)</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>  rat_n_genes[i,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">n_distinct</span>(tempo<span class="sc">$</span>gene) </span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>  rat_n_genes[i,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">substr</span>(fila, <span class="dv">75</span>, <span class="dv">76</span>)</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>  rat_n_genes[i,<span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="st">"Rat"</span></span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>rat_n_genes <span class="ot">&lt;-</span> rat_n_genes <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">n.samples =</span> <span class="fu">c</span>(<span class="dv">78</span>, <span class="dv">83</span>, <span class="dv">83</span>, <span class="dv">81</span>, <span class="dv">82</span>)) </span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>total_genes <span class="ot">&lt;-</span> <span class="fu">rbind</span>(GWAS_n_genes, rat_n_genes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Create Plot Comparing Tissues</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>total_genes <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Github/rat-genomic-analysis/data/n_genes_comparison_GTEx_subset_genes.txt"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(total_genes, <span class="fu">aes</span>(n.samples, n.genes)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span><span class="fu">geom_smooth</span>(<span class="at">data=</span><span class="fu">subset</span>(total_genes,species<span class="sc">==</span><span class="st">"Human"</span>),</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="fu">aes</span>(n.samples, n.genes), <span class="at">method=</span>lm) <span class="sc">+</span> <span class="fu">geom_label_repel</span>( <span class="at">label =</span> total_genes<span class="sc">$</span>tis, <span class="at">box.padding   =</span> <span class="fl">0.35</span>, <span class="at">point.padding =</span> <span class="fl">0.5</span>) <span class="sc">+</span>  <span class="fu">xlab</span>(<span class="st">"Number of Individuals"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Number of Genes Predicted"</span>)  <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"None"</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>()</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"/Users/natashasanthanam/Downloads/GTEx_rats_n_genes.pdf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>



<!-- -->

</section>

<div class="quarto-listing quarto-listing-container-default" id="listing-listing">
<div class="list quarto-listing-default">

</div>
<div class="listing-no-matching d-none">
No matching items
</div>
</div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb28" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "GTEx_comparison_rat_n_genes"</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Natasha Santhanam"</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2/14/2022"</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="an">output:</span><span class="co"> html_document</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(broom)</span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RSQLite)</span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a><span class="st">"%&amp;%"</span> <span class="ot">=</span> <span class="cf">function</span>(a,b) <span class="fu">paste</span>(a,b,<span class="at">sep=</span><span class="st">""</span>)</span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/Data/GTEx/V8/GTEx_Analysis_v8_eQTL_expression_matrices/"</span></span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>geno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/comp_to_GTEx/"</span></span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a><span class="fu">## Compare number of genes predicted between Rat and GTEx prediction models </span></span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a><span class="fu"># Generate GTEx prediction models using the same pipeline as in Rats</span></span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a>Gather Genotype, Gene epression data, snp annotation file and gtf (made with ensembl notation)</span>
<span id="cb28-27"><a href="#cb28-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-28"><a href="#cb28-28" aria-hidden="true" tabindex="-1"></a>First convert genotypes to single coded format to be read by pipeline</span>
<span id="cb28-29"><a href="#cb28-29" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash convert genos to bimbam fromat, eval=FALSE}</span></span>
<span id="cb28-30"><a href="#cb28-30" aria-hidden="true" tabindex="-1"></a><span class="ex">plink</span> <span class="at">--bfile</span> /gpfs/data/im-lab/nas40t2/Data/GTEx/V8/genotype/plink_files/GTEx_maf_0.01 <span class="at">--geno</span> 0.02 <span class="at">--mind</span> 0.02 <span class="at">--maf</span> 0.05 <span class="at">--make-bed</span> <span class="at">--out</span> GTEx_comp</span>
<span id="cb28-31"><a href="#cb28-31" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-32"><a href="#cb28-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-33"><a href="#cb28-33" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash creates PED format with people as rows, eval=FALSE}</span></span>
<span id="cb28-34"><a href="#cb28-34" aria-hidden="true" tabindex="-1"></a><span class="ex">plink</span> <span class="at">--bfile</span> GTEx_comp <span class="at">--recode</span> A-transpose <span class="at">--out</span> GTEx_single_code</span>
<span id="cb28-35"><a href="#cb28-35" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-36"><a href="#cb28-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-37"><a href="#cb28-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-38"><a href="#cb28-38" aria-hidden="true" tabindex="-1"></a><span class="in">```{r read in genotypes and expression data, eval=FALSE}</span></span>
<span id="cb28-39"><a href="#cb28-39" aria-hidden="true" tabindex="-1"></a>geno <span class="ot">&lt;-</span> <span class="fu">fread</span>(geno.dir <span class="sc">%&amp;%</span> <span class="st">"genos/GTEx_single_code.traw"</span>)</span>
<span id="cb28-40"><a href="#cb28-40" aria-hidden="true" tabindex="-1"></a>gex <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(dir <span class="sc">%&amp;%</span> TISSUE <span class="sc">%&amp;%</span> <span class="st">".v8.normalized_expression.bed.gz"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">`</span><span class="at">#chr</span><span class="st">`</span>, start, end))</span>
<span id="cb28-41"><a href="#cb28-41" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-42"><a href="#cb28-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-43"><a href="#cb28-43" aria-hidden="true" tabindex="-1"></a>Genotype has to have SNP in first column </span>
<span id="cb28-44"><a href="#cb28-44" aria-hidden="true" tabindex="-1"></a><span class="in">```{r format genotype, eval=FALSE}</span></span>
<span id="cb28-45"><a href="#cb28-45" aria-hidden="true" tabindex="-1"></a>geno <span class="ot">&lt;-</span> geno <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(SNP, CHR, <span class="fu">colnames</span>(geno)[<span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(geno)]))</span>
<span id="cb28-46"><a href="#cb28-46" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-47"><a href="#cb28-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-48"><a href="#cb28-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-49"><a href="#cb28-49" aria-hidden="true" tabindex="-1"></a>transposing gene expression files and inverse normalize</span>
<span id="cb28-50"><a href="#cb28-50" aria-hidden="true" tabindex="-1"></a><span class="in">```{r format expression, eval=FALSE}</span></span>
<span id="cb28-51"><a href="#cb28-51" aria-hidden="true" tabindex="-1"></a>gex <span class="ot">&lt;-</span> gex <span class="sc">%&gt;%</span> <span class="fu">pivot_longer</span>(<span class="sc">!</span>gene_id, <span class="at">names_to =</span> <span class="st">"IID"</span>, <span class="at">values_to =</span> <span class="st">"count"</span>) <span class="sc">%&gt;%</span> <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> gene_id, <span class="at">values_from =</span> count)</span>
<span id="cb28-52"><a href="#cb28-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-53"><a href="#cb28-53" aria-hidden="true" tabindex="-1"></a>invnorm <span class="ot">=</span> <span class="cf">function</span>(x) {</span>
<span id="cb28-54"><a href="#cb28-54" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">is.null</span>(<span class="fu">dim</span>(x))) res <span class="ot">=</span> <span class="fu">invnorm.vector</span>(x) <span class="cf">else</span></span>
<span id="cb28-55"><a href="#cb28-55" aria-hidden="true" tabindex="-1"></a>  res<span class="ot">=</span><span class="fu">apply</span>(x,<span class="dv">2</span>,invnorm.vector)</span>
<span id="cb28-56"><a href="#cb28-56" aria-hidden="true" tabindex="-1"></a>  res</span>
<span id="cb28-57"><a href="#cb28-57" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-58"><a href="#cb28-58" aria-hidden="true" tabindex="-1"></a>invnorm.vector <span class="ot">=</span> <span class="cf">function</span>(x) {yy <span class="ot">=</span> <span class="fu">rank</span>(x)<span class="sc">/</span>(<span class="fu">length</span>(x)<span class="sc">+</span><span class="dv">1</span>); <span class="fu">qnorm</span>(yy)}</span>
<span id="cb28-59"><a href="#cb28-59" aria-hidden="true" tabindex="-1"></a>gex_transpose <span class="ot">&lt;-</span> gex <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(IID))</span>
<span id="cb28-60"><a href="#cb28-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-61"><a href="#cb28-61" aria-hidden="true" tabindex="-1"></a>gex_transpose <span class="ot">=</span> <span class="fu">invnorm</span>(gex_transpose)</span>
<span id="cb28-62"><a href="#cb28-62" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(gex_transpose) <span class="ot">=</span> gex<span class="sc">$</span>IID</span>
<span id="cb28-63"><a href="#cb28-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-64"><a href="#cb28-64" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(gex_transpose, <span class="at">file =</span> <span class="st">'/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/comp_to_GTEx/expression/gexWB.csv'</span>, <span class="at">sep =</span> <span class="st">","</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb28-65"><a href="#cb28-65" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-66"><a href="#cb28-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-67"><a href="#cb28-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-68"><a href="#cb28-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-69"><a href="#cb28-69" aria-hidden="true" tabindex="-1"></a>Format gene annotation</span>
<span id="cb28-70"><a href="#cb28-70" aria-hidden="true" tabindex="-1"></a><span class="in">```{r format gene annotation, eval=FALSE}</span></span>
<span id="cb28-71"><a href="#cb28-71" aria-hidden="true" tabindex="-1"></a>gtf <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/GTEX_Analysis/annotations_gencode_v26.tsv"</span>)</span>
<span id="cb28-72"><a href="#cb28-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-73"><a href="#cb28-73" aria-hidden="true" tabindex="-1"></a>gene_annotation <span class="ot">&lt;-</span> gtf <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(chromosome, gene_id, gene_name, start, end))</span>
<span id="cb28-74"><a href="#cb28-74" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(gene_annotation) <span class="ot">=</span> gtf<span class="sc">$</span>gene_id</span>
<span id="cb28-75"><a href="#cb28-75" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-76"><a href="#cb28-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-77"><a href="#cb28-77" aria-hidden="true" tabindex="-1"></a>Format snp annotation</span>
<span id="cb28-78"><a href="#cb28-78" aria-hidden="true" tabindex="-1"></a><span class="in">```{r format snp annotation, eval=FALSE}</span></span>
<span id="cb28-79"><a href="#cb28-79" aria-hidden="true" tabindex="-1"></a>phyMap <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(geno.dir <span class="sc">%&amp;%</span> <span class="st">"genos/GTEx_comp.bim"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)  <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">varID =</span> X2) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(X2, X1, X4, X5, X6, varID))</span>
<span id="cb28-80"><a href="#cb28-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-81"><a href="#cb28-81" aria-hidden="true" tabindex="-1"></a>phyMap<span class="sc">$</span>rsid <span class="ot">=</span> phyMap<span class="sc">$</span>varID</span>
<span id="cb28-82"><a href="#cb28-82" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(phyMap) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"snp"</span>, <span class="st">"chr"</span>, <span class="st">"pos"</span>, <span class="st">"refAllele"</span>, <span class="st">"effectAllele"</span>, <span class="st">"varID"</span>, <span class="st">"rsid"</span>)  </span>
<span id="cb28-83"><a href="#cb28-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-84"><a href="#cb28-84" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(phyMap) <span class="ot">=</span> phyMap<span class="sc">$</span>varID</span>
<span id="cb28-85"><a href="#cb28-85" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-86"><a href="#cb28-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-87"><a href="#cb28-87" aria-hidden="true" tabindex="-1"></a><span class="in">```{r split snp annotation by chr, eval=FALSE}</span></span>
<span id="cb28-88"><a href="#cb28-88" aria-hidden="true" tabindex="-1"></a><span class="co"># Splitting the snp annotation file by chromosome</span></span>
<span id="cb28-89"><a href="#cb28-89" aria-hidden="true" tabindex="-1"></a>s <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">split</span>(phyMap, phyMap<span class="sc">$</span>chr), <span class="fu">paste0</span>(<span class="st">"snp_annot.chr"</span>, <span class="fu">unique</span>(phyMap<span class="sc">$</span>chr)))</span>
<span id="cb28-90"><a href="#cb28-90" aria-hidden="true" tabindex="-1"></a><span class="fu">list2env</span>(s, <span class="fu">globalenv</span>())</span>
<span id="cb28-91"><a href="#cb28-91" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-92"><a href="#cb28-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-93"><a href="#cb28-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-94"><a href="#cb28-94" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash split genotype by chr, eval=FALSE}</span></span>
<span id="cb28-95"><a href="#cb28-95" aria-hidden="true" tabindex="-1"></a><span class="co">#Splitting the genotype file by chromosome - run this from the rat_genomic_alaysis directory</span></span>
<span id="cb28-96"><a href="#cb28-96" aria-hidden="true" tabindex="-1"></a><span class="ex">python</span> /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/rat-genomic-analysis/scripts/split_genotype_by_chr.py GTEx_genotype.txt  /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/comp_to_GTEx/genos/geno_by_chr/<span class="st">'genotype'</span></span>
<span id="cb28-97"><a href="#cb28-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-98"><a href="#cb28-98" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-99"><a href="#cb28-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-100"><a href="#cb28-100" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash run PEER factor, eval=FALSE}</span></span>
<span id="cb28-101"><a href="#cb28-101" aria-hidden="true" tabindex="-1"></a><span class="co"># Now we are performing PEER factor analysis on each tissue choosing 7 factors</span></span>
<span id="cb28-102"><a href="#cb28-102" aria-hidden="true" tabindex="-1"></a><span class="ex">peertool</span> <span class="at">-f</span> expression/<span class="st">"gexWB.csv"</span> <span class="at">-n</span> 7 <span class="at">-o</span> peer_GTEx <span class="at">--has_header</span></span>
<span id="cb28-103"><a href="#cb28-103" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-104"><a href="#cb28-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-105"><a href="#cb28-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-106"><a href="#cb28-106" aria-hidden="true" tabindex="-1"></a>Read in Peer Factors</span>
<span id="cb28-107"><a href="#cb28-107" aria-hidden="true" tabindex="-1"></a><span class="in">```{r read in peer, eval=FALSE}</span></span>
<span id="cb28-108"><a href="#cb28-108" aria-hidden="true" tabindex="-1"></a>peer_factors<span class="ot">=</span> <span class="fu">read.csv</span>(<span class="at">file =</span> geno.dir <span class="sc">%&amp;%</span> <span class="st">"peer_GTEx/X.csv"</span>, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb28-109"><a href="#cb28-109" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-110"><a href="#cb28-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-111"><a href="#cb28-111" aria-hidden="true" tabindex="-1"></a><span class="in">```{r format PEER, eval=FALSE}</span></span>
<span id="cb28-112"><a href="#cb28-112" aria-hidden="true" tabindex="-1"></a><span class="co"># Manipulating the PEER factor files so we can append to covariate file</span></span>
<span id="cb28-113"><a href="#cb28-113" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(peer_factors) <span class="ot">=</span> gex<span class="sc">$</span>IID</span>
<span id="cb28-114"><a href="#cb28-114" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(peer_factors) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'PF1'</span>, <span class="st">'PF2'</span>, <span class="st">'PF3'</span>, <span class="st">'PF4'</span>, <span class="st">'PF5'</span>, <span class="st">'PF6'</span>, <span class="st">'PF7'</span>)</span>
<span id="cb28-115"><a href="#cb28-115" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-116"><a href="#cb28-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-117"><a href="#cb28-117" aria-hidden="true" tabindex="-1"></a><span class="in">```{r set up expression matrices, eval=FALSE}</span></span>
<span id="cb28-118"><a href="#cb28-118" aria-hidden="true" tabindex="-1"></a><span class="co">#initializing matrices to be filled with t-stats, p_vals, and residuals of the regression of each gene vs. the covariates for each tissue. t-stat and p-val matrices are just for diagnostics</span></span>
<span id="cb28-119"><a href="#cb28-119" aria-hidden="true" tabindex="-1"></a><span class="co"># t_stats = matrix(nrow = 13, ncol = length(colnames(gex)))</span></span>
<span id="cb28-120"><a href="#cb28-120" aria-hidden="true" tabindex="-1"></a><span class="co"># p_vals = matrix(nrow = 13, ncol = length(colnames(gex)))</span></span>
<span id="cb28-121"><a href="#cb28-121" aria-hidden="true" tabindex="-1"></a>expression <span class="ot">=</span> gex_transpose</span>
<span id="cb28-122"><a href="#cb28-122" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-123"><a href="#cb28-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-124"><a href="#cb28-124" aria-hidden="true" tabindex="-1"></a><span class="in">```{r regress out covariates, eval=FALSE}</span></span>
<span id="cb28-125"><a href="#cb28-125" aria-hidden="true" tabindex="-1"></a><span class="co"># Regressing out the covariates and saving the residuals as the new expression for each tissue</span></span>
<span id="cb28-126"><a href="#cb28-126" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">colnames</span>(gex_transpose))) {</span>
<span id="cb28-127"><a href="#cb28-127" aria-hidden="true" tabindex="-1"></a>    fit <span class="ot">=</span> <span class="fu">lm</span>(gex_transpose[,i] <span class="sc">~</span>  peer_factors<span class="sc">$</span>PF1 <span class="sc">+</span> peer_factors<span class="sc">$</span>PF2 <span class="sc">+</span> peer_factors<span class="sc">$</span>PF3 <span class="sc">+</span> peer_factors<span class="sc">$</span>PF4 <span class="sc">+</span> peer_factors<span class="sc">$</span>PF5 <span class="sc">+</span> peer_factors<span class="sc">$</span>PF6 <span class="sc">+</span> peer_factors<span class="sc">$</span>PF7)</span>
<span id="cb28-128"><a href="#cb28-128" aria-hidden="true" tabindex="-1"></a>    expression[,i] <span class="ot">&lt;-</span> fit<span class="sc">$</span>residuals</span>
<span id="cb28-129"><a href="#cb28-129" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-130"><a href="#cb28-130" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-131"><a href="#cb28-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-132"><a href="#cb28-132" aria-hidden="true" tabindex="-1"></a><span class="in">```{r save expression, eval=FALSE}</span></span>
<span id="cb28-133"><a href="#cb28-133" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the expression RDS objects to be used as arguments in the script</span></span>
<span id="cb28-134"><a href="#cb28-134" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(<span class="fu">as.matrix</span>(expression), geno.dir <span class="sc">%&amp;%</span> TISSUE <span class="sc">%&amp;%</span> <span class="st">"_expression_transformed.RDS"</span>)</span>
<span id="cb28-135"><a href="#cb28-135" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-136"><a href="#cb28-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-137"><a href="#cb28-137" aria-hidden="true" tabindex="-1"></a>Find intersection of genes across all GTEx tissues and Rats</span>
<span id="cb28-138"><a href="#cb28-138" aria-hidden="true" tabindex="-1"></a><span class="in">```{r find common genes, eval=FALSE}</span></span>
<span id="cb28-139"><a href="#cb28-139" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">=</span> <span class="fu">list.files</span>(geno.dir <span class="sc">%&amp;%</span> <span class="st">"expression"</span> , <span class="at">pattern =</span> <span class="st">"expression_transformed.RDS"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-140"><a href="#cb28-140" aria-hidden="true" tabindex="-1"></a>names <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">readRDS</span>(filelist[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> <span class="fu">colnames</span>())</span>
<span id="cb28-141"><a href="#cb28-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-142"><a href="#cb28-142" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(filelist)]) {</span>
<span id="cb28-143"><a href="#cb28-143" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">readRDS</span>(fila) <span class="sc">%&gt;%</span> <span class="fu">colnames</span>())</span>
<span id="cb28-144"><a href="#cb28-144" aria-hidden="true" tabindex="-1"></a>  names <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(names, tempo, <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb28-145"><a href="#cb28-145" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-146"><a href="#cb28-146" aria-hidden="true" tabindex="-1"></a>names<span class="sc">$</span>gene <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(names<span class="sc">$</span>gene, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb28-147"><a href="#cb28-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-148"><a href="#cb28-148" aria-hidden="true" tabindex="-1"></a><span class="co">#do the same for all 5 rat tissues </span></span>
<span id="cb28-149"><a href="#cb28-149" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">=</span> <span class="fu">list.files</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/expression/"</span>, <span class="at">pattern =</span> <span class="st">"expression_transformed.RDS"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-150"><a href="#cb28-150" aria-hidden="true" tabindex="-1"></a>rat_names <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">readRDS</span>(filelist[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> <span class="fu">colnames</span>())</span>
<span id="cb28-151"><a href="#cb28-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-152"><a href="#cb28-152" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(filelist)]) {</span>
<span id="cb28-153"><a href="#cb28-153" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">readRDS</span>(fila) <span class="sc">%&gt;%</span> <span class="fu">colnames</span>())</span>
<span id="cb28-154"><a href="#cb28-154" aria-hidden="true" tabindex="-1"></a>  rat_names <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(rat_names, tempo, <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb28-155"><a href="#cb28-155" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-156"><a href="#cb28-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-157"><a href="#cb28-157" aria-hidden="true" tabindex="-1"></a><span class="co">#change gene id in rats to human notation</span></span>
<span id="cb28-158"><a href="#cb28-158" aria-hidden="true" tabindex="-1"></a>orth.rats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/expression/ortholog_genes_rats_humans.tsv"</span>)</span>
<span id="cb28-159"><a href="#cb28-159" aria-hidden="true" tabindex="-1"></a>rat_names <span class="ot">&lt;-</span> rat_names <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">human_gene =</span> orth.rats[<span class="fu">match</span>(rat_names<span class="sc">$</span>gene, orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene), <span class="dv">1</span>]<span class="sc">$</span>ensembl_gene_id)</span>
<span id="cb28-160"><a href="#cb28-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-161"><a href="#cb28-161" aria-hidden="true" tabindex="-1"></a>overlap <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">human_gene =</span> <span class="fu">intersect</span>(rat_names<span class="sc">$</span>human_gene, names<span class="sc">$</span>gene)) <span class="co">#12,230 overlap </span></span>
<span id="cb28-162"><a href="#cb28-162" aria-hidden="true" tabindex="-1"></a>overlap <span class="ot">&lt;-</span> overlap <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">rat_gene =</span> orth.rats[<span class="fu">match</span>(overlap<span class="sc">$</span>human_gene, orth.rats<span class="sc">$</span>ensembl_gene_id), <span class="dv">3</span>]<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene)</span>
<span id="cb28-163"><a href="#cb28-163" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-164"><a href="#cb28-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-165"><a href="#cb28-165" aria-hidden="true" tabindex="-1"></a>Filter for only overlapped genes in expression files</span>
<span id="cb28-166"><a href="#cb28-166" aria-hidden="true" tabindex="-1"></a><span class="in">```{r filter for common genes, eval=FALSE}</span></span>
<span id="cb28-167"><a href="#cb28-167" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">=</span> <span class="fu">list.files</span>(geno.dir <span class="sc">%&amp;%</span> <span class="st">"expression"</span>, <span class="at">pattern =</span> <span class="st">"expression_transformed.RDS"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-168"><a href="#cb28-168" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb28-169"><a href="#cb28-169" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(fila)</span>
<span id="cb28-170"><a href="#cb28-170" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tempo) <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">colnames</span>(tempo), <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb28-171"><a href="#cb28-171" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> tempo[, overlap<span class="sc">$</span>human_gene]</span>
<span id="cb28-172"><a href="#cb28-172" aria-hidden="true" tabindex="-1"></a>  tis <span class="ot">=</span> <span class="fu">substr</span>(fila, <span class="dv">73</span>, <span class="fu">str_length</span>(fila) <span class="sc">-</span> <span class="dv">27</span>)</span>
<span id="cb28-173"><a href="#cb28-173" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(tempo, geno.dir <span class="sc">%&amp;%</span> txis <span class="sc">%&amp;%</span> <span class="st">"_expression_abrv.RDS"</span>)</span>
<span id="cb28-174"><a href="#cb28-174" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-175"><a href="#cb28-175" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-176"><a href="#cb28-176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-177"><a href="#cb28-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-178"><a href="#cb28-178" aria-hidden="true" tabindex="-1"></a><span class="in">```{r set snp annotation as rownames, eval=FALSE}</span></span>
<span id="cb28-179"><a href="#cb28-179" aria-hidden="true" tabindex="-1"></a><span class="co">#set snp annotation rownames </span></span>
<span id="cb28-180"><a href="#cb28-180" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr1) <span class="ot">=</span> snp_annot.chr1<span class="sc">$</span>varID</span>
<span id="cb28-181"><a href="#cb28-181" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr2) <span class="ot">=</span> snp_annot.chr2<span class="sc">$</span>varID</span>
<span id="cb28-182"><a href="#cb28-182" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr3) <span class="ot">=</span> snp_annot.chr3<span class="sc">$</span>varID</span>
<span id="cb28-183"><a href="#cb28-183" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr4) <span class="ot">=</span> snp_annot.chr4<span class="sc">$</span>varID</span>
<span id="cb28-184"><a href="#cb28-184" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr5) <span class="ot">=</span> snp_annot.chr5<span class="sc">$</span>varID</span>
<span id="cb28-185"><a href="#cb28-185" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr6) <span class="ot">=</span> snp_annot.chr6<span class="sc">$</span>varID</span>
<span id="cb28-186"><a href="#cb28-186" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr7) <span class="ot">=</span> snp_annot.chr7<span class="sc">$</span>varID</span>
<span id="cb28-187"><a href="#cb28-187" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr8) <span class="ot">=</span> snp_annot.chr8<span class="sc">$</span>varID</span>
<span id="cb28-188"><a href="#cb28-188" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr9) <span class="ot">=</span> snp_annot.chr9<span class="sc">$</span>varID</span>
<span id="cb28-189"><a href="#cb28-189" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr10) <span class="ot">=</span> snp_annot.chr10<span class="sc">$</span>varID</span>
<span id="cb28-190"><a href="#cb28-190" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr11) <span class="ot">=</span> snp_annot.chr11<span class="sc">$</span>varID</span>
<span id="cb28-191"><a href="#cb28-191" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr12) <span class="ot">=</span> snp_annot.chr12<span class="sc">$</span>varID</span>
<span id="cb28-192"><a href="#cb28-192" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr13) <span class="ot">=</span> snp_annot.chr13<span class="sc">$</span>varID</span>
<span id="cb28-193"><a href="#cb28-193" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr14) <span class="ot">=</span> snp_annot.chr14<span class="sc">$</span>varID</span>
<span id="cb28-194"><a href="#cb28-194" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr15) <span class="ot">=</span> snp_annot.chr15<span class="sc">$</span>varID</span>
<span id="cb28-195"><a href="#cb28-195" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr16) <span class="ot">=</span> snp_annot.chr16<span class="sc">$</span>varID</span>
<span id="cb28-196"><a href="#cb28-196" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr17) <span class="ot">=</span> snp_annot.chr17<span class="sc">$</span>varID</span>
<span id="cb28-197"><a href="#cb28-197" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr18) <span class="ot">=</span> snp_annot.chr18<span class="sc">$</span>varID</span>
<span id="cb28-198"><a href="#cb28-198" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr19) <span class="ot">=</span> snp_annot.chr19<span class="sc">$</span>varID</span>
<span id="cb28-199"><a href="#cb28-199" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(snp_annot.chr20) <span class="ot">=</span> snp_annot.chr20<span class="sc">$</span>varID</span>
<span id="cb28-200"><a href="#cb28-200" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-201"><a href="#cb28-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-202"><a href="#cb28-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-203"><a href="#cb28-203" aria-hidden="true" tabindex="-1"></a><span class="in">```{r save SNP annot, eval=FALSE}</span></span>
<span id="cb28-204"><a href="#cb28-204" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving the SNP annotation RDS objects to be used as arguments in the script - too lazy to write a for loop </span></span>
<span id="cb28-205"><a href="#cb28-205" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr1, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr1.RDS"</span>)</span>
<span id="cb28-206"><a href="#cb28-206" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr2, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr2.RDS"</span>)</span>
<span id="cb28-207"><a href="#cb28-207" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr3, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr3.RDS"</span>)</span>
<span id="cb28-208"><a href="#cb28-208" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr4, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr4.RDS"</span>)</span>
<span id="cb28-209"><a href="#cb28-209" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr5, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr5.RDS"</span>)</span>
<span id="cb28-210"><a href="#cb28-210" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr6, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr6.RDS"</span>)</span>
<span id="cb28-211"><a href="#cb28-211" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr7, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr7.RDS"</span>)</span>
<span id="cb28-212"><a href="#cb28-212" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr8, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr8.RDS"</span>)</span>
<span id="cb28-213"><a href="#cb28-213" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr9, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr9.RDS"</span>)</span>
<span id="cb28-214"><a href="#cb28-214" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr10, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr10.RDS"</span>)</span>
<span id="cb28-215"><a href="#cb28-215" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr11, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr11.RDS"</span>)</span>
<span id="cb28-216"><a href="#cb28-216" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr12, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr12.RDS"</span>)</span>
<span id="cb28-217"><a href="#cb28-217" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr13, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr13.RDS"</span>)</span>
<span id="cb28-218"><a href="#cb28-218" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr14, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr14.RDS"</span>)</span>
<span id="cb28-219"><a href="#cb28-219" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr15, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr15.RDS"</span>)</span>
<span id="cb28-220"><a href="#cb28-220" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr16, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr16.RDS"</span>)</span>
<span id="cb28-221"><a href="#cb28-221" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr17, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr17.RDS"</span>)</span>
<span id="cb28-222"><a href="#cb28-222" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr18, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr18.RDS"</span>)</span>
<span id="cb28-223"><a href="#cb28-223" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr19, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr19.RDS"</span>)</span>
<span id="cb28-224"><a href="#cb28-224" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(snp_annot.chr20, geno.dir <span class="sc">%&amp;%</span> <span class="st">"snp_annot/snp_annot.chr20.RDS"</span>)</span>
<span id="cb28-225"><a href="#cb28-225" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-226"><a href="#cb28-226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-227"><a href="#cb28-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-228"><a href="#cb28-228" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash create meta file for each tissue, eval=FALSE}</span></span>
<span id="cb28-229"><a href="#cb28-229" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the meta data file for each tissue </span></span>
<span id="cb28-230"><a href="#cb28-230" aria-hidden="true" tabindex="-1"></a><span class="ex">python</span> /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/rat-genomic-analysis/scripts/create_meta_data.py <span class="at">--geno</span> <span class="st">"genos/GTEx_genotype.txt"</span> <span class="at">--expr</span> <span class="st">"Ovary_expression_transformed.RDS"</span> <span class="at">--snpset</span> <span class="st">"1KG"</span> <span class="at">--rsid_label</span> 1 <span class="at">--window</span> 1000000 <span class="at">--out_prefix</span> <span class="st">"Results/allMetaData/GTEx_ovary"</span> </span>
<span id="cb28-231"><a href="#cb28-231" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-232"><a href="#cb28-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-233"><a href="#cb28-233" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash run model script, eval=FALSE}</span></span>
<span id="cb28-234"><a href="#cb28-234" aria-hidden="true" tabindex="-1"></a><span class="co"># Running the model training script for each tissue/chromosome pair</span></span>
<span id="cb28-235"><a href="#cb28-235" aria-hidden="true" tabindex="-1"></a><span class="ex">Rscript</span> <span class="at">--vanilla</span> /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/comp_to_GTEx/GTEx_TW_CV_elasticNet.R  <span class="va">$chr</span> <span class="va">$alpha</span> </span>
<span id="cb28-236"><a href="#cb28-236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-237"><a href="#cb28-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-238"><a href="#cb28-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-239"><a href="#cb28-239" aria-hidden="true" tabindex="-1"></a><span class="va">a</span><span class="op">=</span>0.01</span>
<span id="cb28-240"><a href="#cb28-240" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">20</span><span class="dt">}</span></span>
<span id="cb28-241"><a href="#cb28-241" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb28-242"><a href="#cb28-242" aria-hidden="true" tabindex="-1"></a><span class="ex">qsub</span> <span class="at">-v</span> chr=<span class="va">$i</span>,alpha=<span class="va">$a</span> GTEx_single_nested_EN.pbs  </span>
<span id="cb28-243"><a href="#cb28-243" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb28-244"><a href="#cb28-244" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-245"><a href="#cb28-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-246"><a href="#cb28-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-247"><a href="#cb28-247" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash concatenate all results, eval=FALSE}</span></span>
<span id="cb28-248"><a href="#cb28-248" aria-hidden="true" tabindex="-1"></a><span class="co"># Concatenating all of the results files for each tissue</span></span>
<span id="cb28-249"><a href="#cb28-249" aria-hidden="true" tabindex="-1"></a><span class="fu">bash</span> /gpfs/data/im-lab/nas40t2/natasha/rat_genomics/rat-genomic-analysis/scripts/make_all_results.sh  <span class="st">'Whole_Blood'</span> <span class="st">'/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/comp_to_GTEx/Results/all_Results_Whole_Blood_0.01'</span> 0.01 <span class="st">'1KG_snps'</span></span>
<span id="cb28-250"><a href="#cb28-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-251"><a href="#cb28-251" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-252"><a href="#cb28-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-253"><a href="#cb28-253" aria-hidden="true" tabindex="-1"></a><span class="fu"># Compare number of genes predicted in GTEx to those in Rats</span></span>
<span id="cb28-254"><a href="#cb28-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-255"><a href="#cb28-255" aria-hidden="true" tabindex="-1"></a>Now we have predictability for all parameters of alpha. We can now iterate through all alphas and create the long data format. We also only select for genes that have an average R2 &gt; 0.3 and subsample 20 genes. </span>
<span id="cb28-256"><a href="#cb28-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-257"><a href="#cb28-257" aria-hidden="true" tabindex="-1"></a><span class="in">```{r create GTEx n genes matrix, eval=FALSE}</span></span>
<span id="cb28-258"><a href="#cb28-258" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/comp_to_GTEx/Results/chr_Results"</span>, <span class="at">pattern =</span> <span class="st">"working_TW_"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-259"><a href="#cb28-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-260"><a href="#cb28-260" aria-hidden="true" tabindex="-1"></a>tempo <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(filelist[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(R2 <span class="sc">&lt;=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, cor, R2)) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(gene, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-261"><a href="#cb28-261" aria-hidden="true" tabindex="-1"></a>tempo<span class="sc">$</span>cor <span class="ot">=</span> <span class="fu">as.numeric</span>(tempo<span class="sc">$</span>cor)</span>
<span id="cb28-262"><a href="#cb28-262" aria-hidden="true" tabindex="-1"></a>tempo<span class="sc">$</span>R2 <span class="ot">=</span> tempo<span class="sc">$</span>R2<span class="sc">*</span><span class="fu">sign</span>(tempo<span class="sc">$</span>cor)</span>
<span id="cb28-263"><a href="#cb28-263" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tempo)[<span class="dv">2</span>] <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">"cor"</span>, <span class="st">"0.01"</span>, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb28-264"><a href="#cb28-264" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tempo)[<span class="dv">3</span>] <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">"R2"</span>, <span class="st">"0.01"</span>, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb28-265"><a href="#cb28-265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-266"><a href="#cb28-266" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb28-267"><a href="#cb28-267" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (k <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span>(<span class="fu">length</span>(filelist))){</span>
<span id="cb28-268"><a href="#cb28-268" aria-hidden="true" tabindex="-1"></a> df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(filelist[k], <span class="at">col_names =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(gene, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-269"><a href="#cb28-269" aria-hidden="true" tabindex="-1"></a> alpha <span class="ot">&lt;-</span> <span class="fu">substr</span>(filelist[k], <span class="dv">135</span>, <span class="fu">str_length</span>(filelist[k]) <span class="sc">-</span> <span class="dv">8</span>)</span>
<span id="cb28-270"><a href="#cb28-270" aria-hidden="true" tabindex="-1"></a> df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(df) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, cor, R2)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(R2 <span class="sc">&lt;=</span> <span class="dv">1</span>)</span>
<span id="cb28-271"><a href="#cb28-271" aria-hidden="true" tabindex="-1"></a> df<span class="sc">$</span>cor <span class="ot">=</span> <span class="fu">as.numeric</span>(df<span class="sc">$</span>cor)</span>
<span id="cb28-272"><a href="#cb28-272" aria-hidden="true" tabindex="-1"></a> df<span class="sc">$</span>R2 <span class="ot">=</span> <span class="fu">as.numeric</span>(df<span class="sc">$</span>R2)<span class="sc">*</span><span class="fu">sign</span>(df<span class="sc">$</span>cor)</span>
<span id="cb28-273"><a href="#cb28-273" aria-hidden="true" tabindex="-1"></a> <span class="fu">colnames</span>(df)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"cor"</span>, alpha, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb28-274"><a href="#cb28-274" aria-hidden="true" tabindex="-1"></a> <span class="fu">colnames</span>(df)[<span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"R2"</span>, alpha, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb28-275"><a href="#cb28-275" aria-hidden="true" tabindex="-1"></a> tempo <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(tempo, df, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene"</span>))</span>
<span id="cb28-276"><a href="#cb28-276" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-277"><a href="#cb28-277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-278"><a href="#cb28-278" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-279"><a href="#cb28-279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-280"><a href="#cb28-280" aria-hidden="true" tabindex="-1"></a><span class="co">#tempo &lt;- read_tsv("/Users/natashasanthanam/Downloads/GTEx_Whole_Blood_sparsity_all_parameters.txt", col_names = TRUE)</span></span>
<span id="cb28-281"><a href="#cb28-281" aria-hidden="true" tabindex="-1"></a>tempo_cor_df <span class="ot">&lt;-</span> tempo <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, <span class="fu">starts_with</span>(<span class="st">"cor"</span>))) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(cor_0<span class="fl">.01</span>))</span>
<span id="cb28-282"><a href="#cb28-282" aria-hidden="true" tabindex="-1"></a>tempo_R2_df <span class="ot">&lt;-</span> tempo <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, <span class="fu">starts_with</span>(<span class="st">"R"</span>))) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(R_0<span class="fl">.01</span>))</span>
<span id="cb28-283"><a href="#cb28-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-284"><a href="#cb28-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-285"><a href="#cb28-285" aria-hidden="true" tabindex="-1"></a>tempo_R2_df <span class="ot">&lt;-</span> tempo_R2_df[<span class="fu">sample</span>(<span class="fu">nrow</span>(tempo_R2_df), <span class="dv">20</span>), ]</span>
<span id="cb28-286"><a href="#cb28-286" aria-hidden="true" tabindex="-1"></a>tempo_cor_df <span class="ot">&lt;-</span> tempo_cor_df[<span class="fu">sample</span>(<span class="fu">nrow</span>(tempo_cor_df), <span class="dv">20</span>), ]</span>
<span id="cb28-287"><a href="#cb28-287" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-288"><a href="#cb28-288" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-289"><a href="#cb28-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-290"><a href="#cb28-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-291"><a href="#cb28-291" aria-hidden="true" tabindex="-1"></a><span class="fu"># Plot Sparsity for GTEx</span></span>
<span id="cb28-292"><a href="#cb28-292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-295"><a href="#cb28-295" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-296"><a href="#cb28-296" aria-hidden="true" tabindex="-1"></a>data_long <span class="ot">&lt;-</span> tempo_R2_df  <span class="sc">%&gt;%</span>  <span class="fu">pivot_longer</span>(<span class="sc">!</span>gene, <span class="at">names_to =</span> <span class="st">"value"</span>, <span class="at">values_to =</span> <span class="st">"count"</span>) </span>
<span id="cb28-297"><a href="#cb28-297" aria-hidden="true" tabindex="-1"></a>data_long<span class="sc">$</span>value <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="fu">substr</span>(data_long<span class="sc">$</span>value, <span class="dv">4</span>, <span class="fu">str_length</span>(data_long<span class="sc">$</span>value)))</span>
<span id="cb28-298"><a href="#cb28-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-299"><a href="#cb28-299" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span>  <span class="fu">ggplot</span>(data_long, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.numeric</span>(value), <span class="at">y =</span> count)) <span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"loess"</span>, <span class="at">span =</span> <span class="dv">2</span>, <span class="at">show_guide =</span> <span class="cn">FALSE</span>, <span class="at">se=</span>T, <span class="at">size =</span> .<span class="dv">5</span>, <span class="at">col =</span> <span class="st">"maroon1"</span>)  <span class="sc">+</span>  <span class="fu">xlab</span>(<span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">"Elastic net mixing parameter ("</span>,alpha, <span class="st">")"</span>))) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">"10-fold cross-validated R2"</span>)))</span>
<span id="cb28-300"><a href="#cb28-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-301"><a href="#cb28-301" aria-hidden="true" tabindex="-1"></a>p1</span>
<span id="cb28-302"><a href="#cb28-302" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-303"><a href="#cb28-303" aria-hidden="true" tabindex="-1"></a>cor_long <span class="ot">&lt;-</span> tempo_cor_df  <span class="sc">%&gt;%</span>  <span class="fu">pivot_longer</span>(<span class="sc">!</span>gene, <span class="at">names_to =</span> <span class="st">"value"</span>, <span class="at">values_to =</span> <span class="st">"count"</span>) </span>
<span id="cb28-304"><a href="#cb28-304" aria-hidden="true" tabindex="-1"></a>cor_long<span class="sc">$</span>value <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="fu">substr</span>(cor_long<span class="sc">$</span>value, <span class="dv">5</span>, <span class="fu">str_length</span>(cor_long<span class="sc">$</span>value)))</span>
<span id="cb28-305"><a href="#cb28-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-306"><a href="#cb28-306" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span>  <span class="fu">ggplot</span>(cor_long, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.numeric</span>(value), <span class="at">y =</span> count)) <span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="at">show_guide =</span> <span class="cn">FALSE</span>, <span class="at">se=</span>T, <span class="at">size =</span> .<span class="dv">5</span>, <span class="at">col =</span> <span class="st">"maroon1"</span>)  <span class="sc">+</span>  <span class="fu">xlab</span>(<span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">"Elastic net mixing parameter ("</span>,alpha, <span class="st">")"</span>))) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">"10-fold cross-validated cor"</span>)))</span>
<span id="cb28-307"><a href="#cb28-307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-308"><a href="#cb28-308" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-309"><a href="#cb28-309" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-310"><a href="#cb28-310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-311"><a href="#cb28-311" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-312"><a href="#cb28-312" aria-hidden="true" tabindex="-1"></a>Create Number of Genes for Rat Tissues</span>
<span id="cb28-313"><a href="#cb28-313" aria-hidden="true" tabindex="-1"></a><span class="in">```{r create rat tissue matrix, eval=FALSE}</span></span>
<span id="cb28-314"><a href="#cb28-314" aria-hidden="true" tabindex="-1"></a>rat.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Tyson_Results/"</span></span>
<span id="cb28-315"><a href="#cb28-315" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(rat.dir, <span class="at">pattern =</span> <span class="st">"all_results"</span>, <span class="at">full.names=</span><span class="cn">TRUE</span>)</span>
<span id="cb28-316"><a href="#cb28-316" aria-hidden="true" tabindex="-1"></a>rat_n_genes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">tis=</span> <span class="fu">as.character</span>(), <span class="at">n.genes =</span> <span class="fu">numeric</span>(), <span class="at">species =</span> <span class="fu">as.character</span>())</span>
<span id="cb28-317"><a href="#cb28-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-318"><a href="#cb28-318" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb28-319"><a href="#cb28-319" aria-hidden="true" tabindex="-1"></a>  i <span class="ot">=</span> <span class="fu">match</span>(fila, filelist)</span>
<span id="cb28-320"><a href="#cb28-320" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(fila, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-321"><a href="#cb28-321" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> tempo <span class="sc">%&gt;%</span> <span class="fu">filter</span>(R2 <span class="sc">&gt;=</span> <span class="dv">0</span>)</span>
<span id="cb28-322"><a href="#cb28-322" aria-hidden="true" tabindex="-1"></a>  rat_n_genes[i,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">n_distinct</span>(tempo<span class="sc">$</span>gene) </span>
<span id="cb28-323"><a href="#cb28-323" aria-hidden="true" tabindex="-1"></a>  rat_n_genes[i,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">substr</span>(fila, <span class="dv">75</span>, <span class="dv">76</span>)</span>
<span id="cb28-324"><a href="#cb28-324" aria-hidden="true" tabindex="-1"></a>  rat_n_genes[i,<span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="st">"Rat"</span></span>
<span id="cb28-325"><a href="#cb28-325" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-326"><a href="#cb28-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-327"><a href="#cb28-327" aria-hidden="true" tabindex="-1"></a>rat_n_genes <span class="ot">&lt;-</span> rat_n_genes <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">n.samples =</span> <span class="fu">c</span>(<span class="dv">78</span>, <span class="dv">83</span>, <span class="dv">83</span>, <span class="dv">81</span>, <span class="dv">82</span>)) </span>
<span id="cb28-328"><a href="#cb28-328" aria-hidden="true" tabindex="-1"></a>total_genes <span class="ot">&lt;-</span> <span class="fu">rbind</span>(GWAS_n_genes, rat_n_genes)</span>
<span id="cb28-329"><a href="#cb28-329" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb28-330"><a href="#cb28-330" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-331"><a href="#cb28-331" aria-hidden="true" tabindex="-1"></a>Create Plot Comparing Tissues</span>
<span id="cb28-334"><a href="#cb28-334" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb28-335"><a href="#cb28-335" aria-hidden="true" tabindex="-1"></a>total_genes <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Github/rat-genomic-analysis/data/n_genes_comparison_GTEx_subset_genes.txt"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-336"><a href="#cb28-336" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-337"><a href="#cb28-337" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(total_genes, <span class="fu">aes</span>(n.samples, n.genes)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span><span class="fu">geom_smooth</span>(<span class="at">data=</span><span class="fu">subset</span>(total_genes,species<span class="sc">==</span><span class="st">"Human"</span>),</span>
<span id="cb28-338"><a href="#cb28-338" aria-hidden="true" tabindex="-1"></a><span class="fu">aes</span>(n.samples, n.genes), <span class="at">method=</span>lm) <span class="sc">+</span> <span class="fu">geom_label_repel</span>( <span class="at">label =</span> total_genes<span class="sc">$</span>tis, <span class="at">box.padding   =</span> <span class="fl">0.35</span>, <span class="at">point.padding =</span> <span class="fl">0.5</span>) <span class="sc">+</span>  <span class="fu">xlab</span>(<span class="st">"Number of Individuals"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Number of Genes Predicted"</span>)  <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"None"</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>()</span>
<span id="cb28-339"><a href="#cb28-339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-340"><a href="#cb28-340" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"/Users/natashasanthanam/Downloads/GTEx_rats_n_genes.pdf"</span>)</span>
<span id="cb28-341"><a href="#cb28-341" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>