<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Natasha Santhanam">
<meta name="dcterms.date" content="2022-02-14">

<title>Rat Genomics Pipeline Rendered - Rat_Predictability_comp_Humans</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-listing/list.min.js"></script>
<script src="../site_libs/quarto-listing/quarto-listing.js"></script>
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script>

  window.document.addEventListener("DOMContentLoaded", function (_event) {
    const listingTargetEl = window.document.querySelector('#listing-listing .list');
    if (!listingTargetEl) {
      // No listing discovered, do not attach.
      return; 
    }

    const options = {
      valueNames: [{ data: ['index'] },{ data: ['categories'] },{ data: ['listing-date-sort'] },{ data: ['listing-file-modified-sort'] }],
      
      searchColumns: [],
    };

    window['quarto-listings'] = window['quarto-listings'] || {};
    window['quarto-listings']['listing-listing'] = new List('listing-listing', options);

    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  });

  window.addEventListener('hashchange',() => {
    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  })
  </script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Rat Genomics Pipeline Rendered</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#script-to-compare-heritability-and-predictability-between-rats-and-humans" id="toc-script-to-compare-heritability-and-predictability-between-rats-and-humans" class="nav-link active" data-scroll-target="#script-to-compare-heritability-and-predictability-between-rats-and-humans">Script to compare Heritability and Predictability Between Rats and Humans</a></li>
  <li><a href="#correlation-of-predicted-r2-between-tissues-in-gtex" id="toc-correlation-of-predicted-r2-between-tissues-in-gtex" class="nav-link" data-scroll-target="#correlation-of-predicted-r2-between-tissues-in-gtex">Correlation of predicted R2 between tissues in GTEx</a></li>
  <li><a href="#compare-heritability-and-predictability-in-humans-and-rats" id="toc-compare-heritability-and-predictability-in-humans-and-rats" class="nav-link" data-scroll-target="#compare-heritability-and-predictability-in-humans-and-rats">Compare Heritability and Predictability in Humans and Rats</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Rat_Predictability_comp_Humans</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Natasha Santhanam </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 14, 2022</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="script-to-compare-heritability-and-predictability-between-rats-and-humans" class="level2">
<h2 class="anchored" data-anchor-id="script-to-compare-heritability-and-predictability-between-rats-and-humans">Script to compare Heritability and Predictability Between Rats and Humans</h2>
<p>#Figure comparing pred R2 vs obs R2 in Ac tissue in Rats</p>
<p>First find well expressed genes in Rats</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>fila <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/models//Ac_output_db.db"</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>sqlite.driver <span class="ot">&lt;-</span> <span class="fu">dbDriver</span>(<span class="st">"SQLite"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>conn <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(), fila)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>tempo <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>(conn, <span class="st">'select * from extra'</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>tempo <span class="ot">&lt;-</span> tempo[<span class="fu">order</span>(<span class="sc">-</span>tempo<span class="sc">$</span>R2),]    </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Find Predicted Expression with separate Rat Genotypes</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate imlabtools </span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">METAXCAN</span><span class="op">=</span>/gpfs/data/im-lab/nas40t2/natasha/GTEX_Analysis/MetaXcan/software</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">GENO</span><span class="op">=</span>/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">MODEL</span><span class="op">=</span>/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/models</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="ex">python3</span> <span class="va">$METAXCAN</span>/Predict.py <span class="dt">\</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>--model_db_path <span class="va">$MODEL</span>/Ac_output_db.db  <span class="dt">\</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>--model_db_snp_key varID <span class="dt">\</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>--vcf_genotypes <span class="va">$GENO</span>/BLA_NAcc2_PL2.vcf.gz <span class="dt">\</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>--vcf_mode genotyped <span class="dt">\</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>--on_the_fly_mapping METADATA <span class="st">"{}_{}_{}_{}"</span> <span class="dt">\</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>--prediction_output Ac_NAcc2__predict.txt  <span class="dt">\</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>--prediction_summary_output Ac_NAcc2__summary.txt <span class="dt">\</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>--verbosity 9 <span class="dt">\</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>--throw</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Mgmt</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>pred_expr <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/Ac_NAcc2__predict.txt"</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>obs_expr <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/NAcc2.expr.iqn.bed.gz"</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>obs_expr <span class="ot">&lt;-</span> obs_expr <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">`</span><span class="at">#chr</span><span class="st">`</span>, start, end)) <span class="sc">%&gt;%</span> <span class="fu">pivot_longer</span>(<span class="sc">!</span>gene_id, <span class="at">names_to =</span> <span class="st">"FID"</span>, <span class="at">values_to =</span> <span class="st">"count"</span>) <span class="sc">%&gt;%</span> <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> gene_id, <span class="at">values_from =</span> count)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>Mgmt_pred <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pred_expr[, <span class="fu">c</span>(<span class="st">"FID"</span>, <span class="st">"ENSRNOG00000016038"</span>)])</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>Mgmt_obs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(obs_expr[,  <span class="fu">c</span>(<span class="st">"FID"</span>, <span class="st">"ENSRNOG00000016038"</span>)]) </span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>Mgmt_pred<span class="sc">$</span>FID <span class="ot">=</span> <span class="fu">as.character</span>(Mgmt_pred<span class="sc">$</span>FID)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>Mgmt_pred <span class="ot">&lt;-</span> Mgmt_pred[<span class="fu">match</span>(Mgmt_obs<span class="sc">$</span>FID, Mgmt_pred<span class="sc">$</span>FID),]</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>Mgmt_obs <span class="ot">&lt;-</span> Mgmt_obs[<span class="fu">match</span>(Mgmt_pred<span class="sc">$</span>FID, Mgmt_obs<span class="sc">$</span>FID ), ]</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>Mgmt_pred<span class="sc">$</span>FID <span class="ot">=</span> <span class="fu">as.character</span>(Mgmt_pred<span class="sc">$</span>FID)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>Mgmt <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(Mgmt_pred, Mgmt_obs, <span class="at">by =</span> <span class="st">"FID"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Polr3k</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>Polr3k_pred <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pred_expr[, <span class="fu">c</span>(<span class="st">"FID"</span>, <span class="st">"ENSRNOG00000017843"</span>)])</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>Polr3k_obs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(obs_expr[,  <span class="fu">c</span>(<span class="st">"FID"</span>, <span class="st">"ENSRNOG00000017843"</span>)]) </span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>Polr3k_pred<span class="sc">$</span>FID <span class="ot">=</span> <span class="fu">as.character</span>(Polr3k_pred<span class="sc">$</span>FID)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>Polr3k_pred <span class="ot">&lt;-</span> Polr3k_pred[<span class="fu">match</span>(Polr3k_obs<span class="sc">$</span>FID, Polr3k_pred<span class="sc">$</span>FID),]</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>Polr3k_obs <span class="ot">&lt;-</span> Polr3k_obs[<span class="fu">match</span>(Polr3k_pred<span class="sc">$</span>FID, Polr3k_obs<span class="sc">$</span>FID ), ]</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>Polr3k_pred<span class="sc">$</span>FID <span class="ot">=</span> <span class="fu">as.character</span>(Polr3k_pred<span class="sc">$</span>FID)</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>Polr3k <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(Polr3k_pred, Polr3k_obs, <span class="at">by =</span> <span class="st">"FID"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Dhfr</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>Dhfr_pred <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pred_expr[, <span class="fu">c</span>(<span class="st">"FID"</span>, <span class="st">"ENSRNOG00000013521"</span>)])</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>Dhfr_obs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(obs_expr[,  <span class="fu">c</span>(<span class="st">"FID"</span>, <span class="st">"ENSRNOG00000013521"</span>)]) </span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>Dhfr_pred<span class="sc">$</span>FID <span class="ot">=</span> <span class="fu">as.character</span>(Dhfr_pred<span class="sc">$</span>FID)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>Dhfr_pred <span class="ot">&lt;-</span> Dhfr_pred[<span class="fu">match</span>(Dhfr_obs<span class="sc">$</span>FID, Dhfr_pred<span class="sc">$</span>FID),]</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>Dhfr_obs <span class="ot">&lt;-</span> Dhfr_obs[<span class="fu">match</span>(Dhfr_pred<span class="sc">$</span>FID, Dhfr_obs<span class="sc">$</span>FID ), ]</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>Dhfr_pred<span class="sc">$</span>FID <span class="ot">=</span> <span class="fu">as.character</span>(Dhfr_pred<span class="sc">$</span>FID)</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>Dhfr <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(Dhfr_pred, Dhfr_obs, <span class="at">by =</span> <span class="st">"FID"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot pred vs obs expression with our models in NcAcc2 tissue</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>Mgmt <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Downloads/model_perf_Mgmt.txt"</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>Polr3k <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Downloads/model_perf_Polr3k.txt"</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>Dhfr <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Downloads/model_perf_Dhfr.txt"</span>)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>g1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(Mgmt, <span class="fu">aes</span>(pred, obs)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">0.4</span>, <span class="at">y =</span> <span class="fl">2.5</span>, <span class="at">label =</span> <span class="st">"Mgmt"</span>, <span class="at">size =</span> <span class="dv">10</span>, <span class="at">fontface =</span><span class="st">"bold.italic"</span>) <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">0.53</span>, <span class="at">y =</span> <span class="dv">2</span>, <span class="at">label =</span> <span class="st">"R2 =  0.724"</span>, <span class="at">size =</span> <span class="dv">8</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Predicted expression"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Observed Expression"</span>)</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>g2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(Polr3k, <span class="fu">aes</span>(pred, obs)) <span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Polr3k"</span>) <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="sc">-</span><span class="fl">1.8</span>, <span class="at">y =</span> <span class="dv">2</span>, <span class="at">label =</span> <span class="st">"r2 =  0.796"</span>, <span class="at">size =</span> <span class="dv">3</span>)  <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>g3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(Dhfr, <span class="fu">aes</span>(pred, obs))  <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="sc">-</span><span class="fl">0.4</span>, <span class="at">y =</span> <span class="fl">2.5</span>, <span class="at">label =</span> <span class="st">"Dhfr"</span>, <span class="at">size =</span> <span class="dv">10</span>, <span class="at">fontface =</span><span class="dv">2</span>) <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="sc">-</span><span class="fl">0.25</span>, <span class="at">y =</span> <span class="dv">2</span>, <span class="at">label =</span> <span class="st">"r2 =  0.554"</span>, <span class="at">size =</span> <span class="dv">8</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Predicted expression"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Observed Expression"</span>) <span class="sc">+</span> <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="cn">NA</span>, <span class="fl">1.25</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Calculate Predicted vs Observed Expression for Mgmt in Humans using GTEx NA predictDB with Geuvadis genotypes (hg38) as well</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate imlabtools </span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">METAXCAN</span><span class="op">=</span>/gpfs/data/im-lab/nas40t2/natasha/GTEX_Analysis/MetaXcan/software</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">GENO</span><span class="op">=</span>/gpfs/data/im-lab/nas40t2/Data/1000G/vcf_hg38/geuvadis</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">MODEL</span><span class="op">=</span>/gpfs/data/im-lab/nas40t2/Data/PredictDB/GTEx_v8/models_v1/eqtl/elastic_net_models</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">DATA</span><span class="op">=</span>/gpfs/data/im-lab/nas40t2/natasha/GTEX_Analysis/predixcan</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="ex">python3</span> <span class="va">$METAXCAN</span>/Predict.py <span class="dt">\</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>--model_db_path <span class="va">$MODEL</span>/en_Brain_Nucleus_accumbens_basal_ganglia.db  <span class="dt">\</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>--model_db_snp_key varID <span class="dt">\</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>--vcf_genotypes <span class="va">$GENO</span>/ALL.chr<span class="pp">*</span>.shapeit2_integrated_v1a.GRCh38.20181129.phased.vcf.gz <span class="dt">\</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>--vcf_mode genotyped <span class="dt">\</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>--on_the_fly_mapping METADATA <span class="st">"{}_{}_{}_{}_b38"</span> <span class="dt">\</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>--prediction_output 1000G_NAcc2__predict.txt  <span class="dt">\</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>--prediction_summary_output 1000G_NAcc2__summary.txt  <span class="dt">\</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>--verbosity 9 <span class="dt">\</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>--throw</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>RPS26 r2= 0.7352744</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>pred_expr <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/comp_to_GTEx/1000G_NAcc2__predict.txt"</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>obs_expr <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/comp_to_GTEx/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.gz"</span>)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>obs_expr <span class="ot">&lt;-</span> obs_expr <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(Chr, Gene_Symbol, Coord)) <span class="sc">%&gt;%</span> <span class="fu">pivot_longer</span>(<span class="sc">!</span>TargetID, <span class="at">names_to =</span> <span class="st">"FID"</span>, <span class="at">values_to =</span> <span class="st">"count"</span>) <span class="sc">%&gt;%</span> <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> TargetID, <span class="at">values_from =</span> count)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(obs_expr)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(obs_expr)] <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">colnames</span>(obs_expr)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(obs_expr)], <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>RPS26_pred <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pred_expr[, <span class="fu">c</span>(<span class="st">"FID"</span>, <span class="st">"ENSG00000197728.9"</span>)])</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>RPS26_obs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(obs_expr[,  <span class="fu">c</span>(<span class="st">"FID"</span>, <span class="st">"ENSG00000197728"</span>)]) </span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>RPS26_pred <span class="ot">&lt;-</span> RPS26_pred[<span class="fu">match</span>(RPS26_obs<span class="sc">$</span>FID, RPS26_pred<span class="sc">$</span>FID),]</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>RPS26_obs <span class="ot">&lt;-</span> RPS26_obs[<span class="fu">match</span>(RPS26_pred<span class="sc">$</span>FID, RPS26_obs<span class="sc">$</span>FID ), ]</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>RPS26 <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(RPS26_pred, RPS26_obs, <span class="at">by =</span> <span class="st">"FID"</span>)</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(RPS26) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"FID"</span>, <span class="st">"pred"</span>, <span class="st">"obs"</span>)</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>RPS26_GTEx <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Downloads/GTEx_RPS26.txt"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>DHFR_GTEx <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Downloads/GTEx_DHFR_results.txt"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>g4 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(DHFR_GTEx, <span class="fu">aes</span>(pred, obs)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="sc">-</span><span class="fl">0.45</span>, <span class="at">y =</span> <span class="dv">55</span>, <span class="at">label =</span> <span class="st">"DHFR"</span>, <span class="at">size =</span> <span class="dv">10</span>, <span class="at">fontface =</span><span class="dv">2</span>) <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="sc">-</span><span class="fl">0.4</span>, <span class="at">y =</span> <span class="dv">51</span>, <span class="at">label =</span> <span class="st">"r2 =  0.506"</span>, <span class="at">size =</span> <span class="dv">8</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Predicted expression"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Observed Expression"</span>)</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a>g5 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(RPS26_GTEx, <span class="fu">aes</span>(pred, obs)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">y =</span> <span class="dv">385</span>, <span class="at">label =</span> <span class="st">"RPS26"</span>, <span class="at">size =</span> <span class="dv">10</span>, <span class="at">fontface =</span><span class="st">"bold.italic"</span>) <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="sc">-</span><span class="fl">0.45</span>, <span class="at">y =</span> <span class="dv">355</span>, <span class="at">label =</span> <span class="st">"R2 =  0.735"</span>, <span class="at">size =</span> <span class="dv">8</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Predicted expression"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Observed Expression"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="correlation-of-predicted-r2-between-tissues-in-gtex" class="level1">
<h1>Correlation of predicted R2 between tissues in GTEx</h1>
<p>We use Nucleus Accumbens, Hippocampus, Cerebellum, Cortex and Anterior Cingulate Cortex in GTEx</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list</span>( <span class="st">"en_Brain_Nucleus_accumbens_basal_ganglia.db"</span>, <span class="st">"en_Brain_Hippocampus.db"</span>, <span class="st">"en_Brain_Cerebellum.db"</span>, <span class="st">"en_Brain_Cortex.db"</span>, <span class="st">"en_Brain_Anterior_cingulate_cortex_BA24.db"</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/Data/PredictDB/GTEx_v8/models_v1/eqtl/elastic_net_models/"</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>pred_tis <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">as.character</span>())</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> dir <span class="sc">%&amp;%</span> fila</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>  sqlite.driver <span class="ot">&lt;-</span> <span class="fu">dbDriver</span>(<span class="st">"SQLite"</span>)</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>  conn <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(), filename)</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>  tis <span class="ot">&lt;-</span> <span class="fu">substr</span>(fila, <span class="dv">4</span>, <span class="fu">str_length</span>(fila)<span class="sc">-</span><span class="dv">3</span>)</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>  extra <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>(conn, <span class="st">'select * from extra'</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, test_R2_avg)) </span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(extra)[<span class="dv">2</span>] <span class="ot">=</span> tis</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>  pred_tis <span class="ot">&lt;-</span> <span class="fu">full_join</span>(pred_tis, extra, <span class="at">by =</span> <span class="st">"gene"</span>) </span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dbDisconnect</span>(conn)</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(pred_tis) <span class="ot">&lt;-</span> pred_tis<span class="sc">$</span>gene</span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>pred_tis <span class="ot">&lt;-</span> pred_tis <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(gene))</span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(pred_tis, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/comp_to_GTEx/pred_R2_betw_GTEx_brain_tissues.RDS"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>pred_tis <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"/Users/natashasanthanam/Downloads/pred_R2_betw_GTEx_brain_tissues.RDS"</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pairs</span>(pred_tis)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="compare-heritability-and-predictability-in-humans-and-rats" class="level1">
<h1>Compare Heritability and Predictability in Humans and Rats</h1>
<p>Get R2 and H2 from Heather’s Models for Brain</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode sql code-with-copy"><code class="sourceCode sql"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>sqlite3 genarch.db</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>.headers <span class="kw">on</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>.<span class="kw">mode</span> csv</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>.output human_h2.csv</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="kw">select</span> gene, ensid, en_r2 ,h2,h2_ci <span class="kw">from</span> results <span class="kw">where</span> tissue <span class="op">=</span> <span class="ot">"DGN-WB"</span>;</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>.quit</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>rat_h2 <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Github/rat-genomic-analysis/data/Ac_PVE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">rat_herit =</span> X2)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>human_h2 <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"/Users/natashasanthanam/Github/rat-genomic-analysis/data/human_h2.csv"</span>)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>human_h2<span class="sc">$</span>ensid <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(human_h2<span class="sc">$</span>ensid, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(rat_h2<span class="sc">$</span>rat_herit)</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(human_h2<span class="sc">$</span>h2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Change gene names in rats</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>orth.rats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Downloads/ortholog_genes_rats_humans.tsv"</span>)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>rat_h2 <span class="ot">&lt;-</span> rat_h2 <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">ensid =</span> orth.rats[<span class="fu">match</span>(rat_h2<span class="sc">$</span>X1, orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene), <span class="dv">1</span>]<span class="sc">$</span>ensembl_gene_id)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>all_h2 <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(rat_h2, human_h2, <span class="at">by =</span> <span class="st">"ensid"</span>)</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(all_h2<span class="sc">$</span>rat_herit, all_h2<span class="sc">$</span>h2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Compare Predictability (R2) between Humans and Rats</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>fila <span class="ot">&lt;-</span> <span class="st">"/Users/natashasanthanam/Box/imlab-data/data-Github/rat-genomic-analysis/sql/Ac_output_db.db"</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>sqlite.driver <span class="ot">&lt;-</span> <span class="fu">dbDriver</span>(<span class="st">"SQLite"</span>)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>conn <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(), fila)</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>rat_r2 <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>(conn, <span class="st">'select * from extra'</span>)</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dbDisconnect</span>(conn)</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>rat_r2 <span class="ot">&lt;-</span> rat_r2 <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">ensid =</span> orth.rats[<span class="fu">match</span>(rat_r2<span class="sc">$</span>gene, orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene), <span class="dv">1</span>]<span class="sc">$</span>ensembl_gene_id)</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>all_r2 <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(rat_r2, human_h2, <span class="at">by =</span> <span class="st">"ensid"</span>)</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(all_r2<span class="sc">$</span>R2, all_r2<span class="sc">$</span>en_r2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot the Comparison between the two</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>Ac_h2 <span class="ot">&lt;-</span> rat_h2 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene.x =</span> X1) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">rat_ci =</span> <span class="fu">paste</span>(<span class="fu">round</span>(Ac_h2<span class="sc">$</span>X3, <span class="at">digits =</span> <span class="dv">5</span>), <span class="fu">round</span>(Ac_h2<span class="sc">$</span>X4, <span class="at">digits =</span> <span class="dv">5</span>), <span class="at">sep =</span> <span class="st">"-"</span>)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(gene.x, rat_herit, rat_ci))</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>lg_df <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(all_r2, Ac_h2, <span class="at">by =</span> <span class="st">"gene.x"</span>)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>lg_df <span class="ot">&lt;-</span> lg_df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(ensid, gene.x, genename, R2, en_r2,rat_herit,h2,  rat_ci, h2_ci))</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(lg_df) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"ensembl_id"</span>, <span class="st">"rat_ensembl_id"</span>, <span class="st">"genename"</span>, <span class="st">"rat_r2"</span>, <span class="st">"human_r2"</span>, <span class="st">"rat_h2"</span>, <span class="st">"human_h2"</span>, <span class="st">"rat_h2_ci"</span>, <span class="st">"human_h2_ci"</span>)</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot h2 between rats and Humans with line for null </span></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>lg_df <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">human_shuffled_h2 =</span> <span class="fu">sample</span>(human_h2, <span class="fu">nrow</span>(lg_df), <span class="at">replace=</span>F)) <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(rat_h2, human_h2)) <span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="at">col =</span> <span class="st">"darkgreen"</span>) <span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="fu">aes</span>(rat_h2, human_shuffled_h2),<span class="at">col=</span><span class="st">"dark gray"</span>)<span class="sc">+</span><span class="fu">theme_bw</span>()<span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Rat Heritability"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Human Heritability"</span>)</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot R2 between rats and Humans with line for null </span></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>lg_df <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">human_shuffled_r2 =</span> <span class="fu">sample</span>(human_r2, <span class="fu">nrow</span>(lg_df), <span class="at">replace=</span>F)) <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(rat_r2, human_r2)) <span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="at">col =</span> <span class="st">"midnightblue"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Rat Predictability"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Human Predictability"</span>) <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">0.1</span>, <span class="at">y =</span> <span class="fl">0.2</span>, <span class="at">label =</span> <span class="st">"R =  0.061"</span>, <span class="at">size =</span> <span class="dv">8</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Compare Predictability Between Tissues</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>box.dir <span class="ot">&lt;-</span> <span class="st">"/Users/natashasanthanam/Box/imlab-data/data-Github/rat-genomic-analysis/sql/"</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Il_output_db.db"</span>, <span class="st">"Lh_output_db.db"</span>, <span class="st">"Ac_output_db.db"</span>, <span class="st">"Vo_output_db.db"</span>, <span class="st">"Pl_output_db.db"</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>tis.list <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Infralimbic Cortex"</span>, <span class="st">"Lateral Habenula"</span>, <span class="st">"Nucleus accumbens"</span>, <span class="st">"Orbitofrontal Cortex"</span>, <span class="st">"Prelimibic Cortex"</span>)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>pred_tis <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">as.character</span>())</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> box.dir <span class="sc">%&amp;%</span> fila</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  sqlite.driver <span class="ot">&lt;-</span> <span class="fu">dbDriver</span>(<span class="st">"SQLite"</span>)</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>  conn <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(), filename)</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>  i <span class="ot">&lt;-</span> <span class="fu">match</span>(fila, filelist)</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>  tis <span class="ot">&lt;-</span> tis.list[i]</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>  extra <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>(conn, <span class="st">'select * from extra'</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, R2)) </span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(extra)[<span class="dv">2</span>] <span class="ot">=</span> tis</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>  pred_tis <span class="ot">&lt;-</span> <span class="fu">full_join</span>(pred_tis, extra, <span class="at">by =</span> <span class="st">"gene"</span>) </span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dbDisconnect</span>(conn)</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(pred_tis) <span class="ot">&lt;-</span> pred_tis<span class="sc">$</span>gene</span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>pred_tis <span class="ot">&lt;-</span> pred_tis <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(gene))</span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>d1 <span class="ot">&lt;-</span> <span class="fu">cor</span>(pred_tis <span class="sc">%&gt;%</span> na.omit)</span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>d1[<span class="fu">lower.tri</span>(d1)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>d1 <span class="ot">&lt;-</span> d1 <span class="sc">%&gt;%</span></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>  as.data.frame <span class="sc">%&gt;%</span></span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">'var1'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>(var2, value, <span class="sc">-</span>var1) </span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>d1 <span class="ot">&lt;-</span> d1 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">r =</span> value) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>() </span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> d1, <span class="fu">aes</span>(<span class="at">x =</span> var1, <span class="at">y =</span> var2, <span class="at">fill =</span> r)) <span class="sc">+</span></span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_tile</span>(<span class="at">color =</span> <span class="st">"white"</span>)<span class="sc">+</span></span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">"dodgerblue"</span>, <span class="at">high=</span><span class="st">"dodgerblue4"</span>) <span class="sc">+</span></span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">""</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>), <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)   </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Calculate R2 across GTEx Tissues</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>model.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/Data/PredictDB/GTEx_v8/models_v1/eqtl/elastic_net_models/"</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"en_Brain_Amygdala.db"</span>, <span class="st">"en_Brain_Cerebellum.db"</span>, <span class="st">"en_Brain_Nucleus_accumbens_basal_ganglia.db"</span>, <span class="st">"en_Brain_Spinal_cord_cervical_c-1.db"</span>, <span class="st">"en_Brain_Substantia_nigra.db"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co">#pred_tis &lt;- read_tsv(dir %&amp;% "pred_R2_betw_GTEx_brain_tissues.txt", col_names = TRUE) </span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(pred_tis) <span class="ot">&lt;-</span> pred_tis<span class="sc">$</span>gene</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>pred_tis <span class="ot">&lt;-</span> pred_tis <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(gene))</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(pred_tis) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Amygdala"</span>, <span class="st">"Cerebellum"</span>, <span class="st">"Nucleus accumbens"</span>,  <span class="st">"Spinal cord"</span>, <span class="st">"Substantia nigra"</span>)</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>d2 <span class="ot">&lt;-</span> <span class="fu">cor</span>(pred_tis <span class="sc">%&gt;%</span> na.omit)</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>d2[<span class="fu">lower.tri</span>(d2)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>d2 <span class="ot">&lt;-</span> d2 <span class="sc">%&gt;%</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>  as.data.frame <span class="sc">%&gt;%</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">'var1'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>(var2, value, <span class="sc">-</span>var1) </span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>d2 <span class="ot">&lt;-</span> d2 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">r =</span> value) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> d2, <span class="fu">aes</span>(<span class="at">x =</span> var1, <span class="at">y =</span> var2, <span class="at">fill =</span> r)) <span class="sc">+</span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_tile</span>(<span class="at">color =</span> <span class="st">"white"</span>)<span class="sc">+</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">"maroon1"</span>, <span class="at">high=</span><span class="st">"maroon4"</span>) <span class="sc">+</span></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">""</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>), <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)   </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Check Number of SNPs used in Prediction in Rats vs Humans</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(rat_r2<span class="sc">$</span>n.snps)</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="co">#human NAcc models</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>filename <span class="ot">&lt;-</span> <span class="st">"/Users/natashasanthanam/Downloads/en_Brain_Nucleus_accumbens_basal_ganglia.db"</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>sqlite.driver <span class="ot">&lt;-</span> <span class="fu">dbDriver</span>(<span class="st">"SQLite"</span>)</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>conn <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(), filename)</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>human_r2 <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>(conn, <span class="st">'select * from extra'</span>) </span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(human_r2<span class="sc">$</span>n.snps.in.model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>



<!-- -->

</section>

<div class="quarto-listing quarto-listing-container-default" id="listing-listing">
<div class="list quarto-listing-default">

</div>
<div class="listing-no-matching d-none">
No matching items
</div>
</div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb20" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Rat_Predictability_comp_Humans"</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Natasha Santhanam"</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2/14/2022"</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="an">output:</span><span class="co"> html_document</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup, include=FALSE}</span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RSQLite)</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glue)</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(grid)</span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GGally)</span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a><span class="fu">## Script to compare Heritability and Predictability Between Rats and Humans</span></span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>#Figure comparing pred R2 vs obs R2 in Ac tissue in Rats</span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a>First find well expressed genes in Rats</span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a><span class="in">```{r find well expressed genes in rats, eval=FALSE}</span></span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a>fila <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/models//Ac_output_db.db"</span></span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a>sqlite.driver <span class="ot">&lt;-</span> <span class="fu">dbDriver</span>(<span class="st">"SQLite"</span>)</span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a>conn <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(), fila)</span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a>tempo <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>(conn, <span class="st">'select * from extra'</span>)</span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-31"><a href="#cb20-31" aria-hidden="true" tabindex="-1"></a>tempo <span class="ot">&lt;-</span> tempo[<span class="fu">order</span>(<span class="sc">-</span>tempo<span class="sc">$</span>R2),]    </span>
<span id="cb20-32"><a href="#cb20-32" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb20-33"><a href="#cb20-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-34"><a href="#cb20-34" aria-hidden="true" tabindex="-1"></a>Find Predicted Expression with separate Rat Genotypes</span>
<span id="cb20-35"><a href="#cb20-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-36"><a href="#cb20-36" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash calculate pred expr with models, eval=FALSE}</span></span>
<span id="cb20-37"><a href="#cb20-37" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate imlabtools </span>
<span id="cb20-38"><a href="#cb20-38" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">METAXCAN</span><span class="op">=</span>/gpfs/data/im-lab/nas40t2/natasha/GTEX_Analysis/MetaXcan/software</span>
<span id="cb20-39"><a href="#cb20-39" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">GENO</span><span class="op">=</span>/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files</span>
<span id="cb20-40"><a href="#cb20-40" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">MODEL</span><span class="op">=</span>/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/models</span>
<span id="cb20-41"><a href="#cb20-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-42"><a href="#cb20-42" aria-hidden="true" tabindex="-1"></a><span class="ex">python3</span> <span class="va">$METAXCAN</span>/Predict.py <span class="dt">\</span></span>
<span id="cb20-43"><a href="#cb20-43" aria-hidden="true" tabindex="-1"></a>--model_db_path <span class="va">$MODEL</span>/Ac_output_db.db  <span class="dt">\</span></span>
<span id="cb20-44"><a href="#cb20-44" aria-hidden="true" tabindex="-1"></a>--model_db_snp_key varID <span class="dt">\</span></span>
<span id="cb20-45"><a href="#cb20-45" aria-hidden="true" tabindex="-1"></a>--vcf_genotypes <span class="va">$GENO</span>/BLA_NAcc2_PL2.vcf.gz <span class="dt">\</span></span>
<span id="cb20-46"><a href="#cb20-46" aria-hidden="true" tabindex="-1"></a>--vcf_mode genotyped <span class="dt">\</span></span>
<span id="cb20-47"><a href="#cb20-47" aria-hidden="true" tabindex="-1"></a>--on_the_fly_mapping METADATA <span class="st">"{}_{}_{}_{}"</span> <span class="dt">\</span></span>
<span id="cb20-48"><a href="#cb20-48" aria-hidden="true" tabindex="-1"></a>--prediction_output Ac_NAcc2__predict.txt  <span class="dt">\</span></span>
<span id="cb20-49"><a href="#cb20-49" aria-hidden="true" tabindex="-1"></a>--prediction_summary_output Ac_NAcc2__summary.txt <span class="dt">\</span></span>
<span id="cb20-50"><a href="#cb20-50" aria-hidden="true" tabindex="-1"></a>--verbosity 9 <span class="dt">\</span></span>
<span id="cb20-51"><a href="#cb20-51" aria-hidden="true" tabindex="-1"></a>--throw</span>
<span id="cb20-52"><a href="#cb20-52" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb20-53"><a href="#cb20-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-54"><a href="#cb20-54" aria-hidden="true" tabindex="-1"></a>Mgmt</span>
<span id="cb20-55"><a href="#cb20-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-56"><a href="#cb20-56" aria-hidden="true" tabindex="-1"></a><span class="in">```{r check correlation between predicted and observed expression, eval=FALSE}</span></span>
<span id="cb20-57"><a href="#cb20-57" aria-hidden="true" tabindex="-1"></a>pred_expr <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/prediXcan/Ac_NAcc2__predict.txt"</span>)</span>
<span id="cb20-58"><a href="#cb20-58" aria-hidden="true" tabindex="-1"></a>obs_expr <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/Box_files/NAcc2.expr.iqn.bed.gz"</span>)</span>
<span id="cb20-59"><a href="#cb20-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-60"><a href="#cb20-60" aria-hidden="true" tabindex="-1"></a>obs_expr <span class="ot">&lt;-</span> obs_expr <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">`</span><span class="at">#chr</span><span class="st">`</span>, start, end)) <span class="sc">%&gt;%</span> <span class="fu">pivot_longer</span>(<span class="sc">!</span>gene_id, <span class="at">names_to =</span> <span class="st">"FID"</span>, <span class="at">values_to =</span> <span class="st">"count"</span>) <span class="sc">%&gt;%</span> <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> gene_id, <span class="at">values_from =</span> count)</span>
<span id="cb20-61"><a href="#cb20-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-62"><a href="#cb20-62" aria-hidden="true" tabindex="-1"></a>Mgmt_pred <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pred_expr[, <span class="fu">c</span>(<span class="st">"FID"</span>, <span class="st">"ENSRNOG00000016038"</span>)])</span>
<span id="cb20-63"><a href="#cb20-63" aria-hidden="true" tabindex="-1"></a>Mgmt_obs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(obs_expr[,  <span class="fu">c</span>(<span class="st">"FID"</span>, <span class="st">"ENSRNOG00000016038"</span>)]) </span>
<span id="cb20-64"><a href="#cb20-64" aria-hidden="true" tabindex="-1"></a>Mgmt_pred<span class="sc">$</span>FID <span class="ot">=</span> <span class="fu">as.character</span>(Mgmt_pred<span class="sc">$</span>FID)</span>
<span id="cb20-65"><a href="#cb20-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-66"><a href="#cb20-66" aria-hidden="true" tabindex="-1"></a>Mgmt_pred <span class="ot">&lt;-</span> Mgmt_pred[<span class="fu">match</span>(Mgmt_obs<span class="sc">$</span>FID, Mgmt_pred<span class="sc">$</span>FID),]</span>
<span id="cb20-67"><a href="#cb20-67" aria-hidden="true" tabindex="-1"></a>Mgmt_obs <span class="ot">&lt;-</span> Mgmt_obs[<span class="fu">match</span>(Mgmt_pred<span class="sc">$</span>FID, Mgmt_obs<span class="sc">$</span>FID ), ]</span>
<span id="cb20-68"><a href="#cb20-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-69"><a href="#cb20-69" aria-hidden="true" tabindex="-1"></a>Mgmt_pred<span class="sc">$</span>FID <span class="ot">=</span> <span class="fu">as.character</span>(Mgmt_pred<span class="sc">$</span>FID)</span>
<span id="cb20-70"><a href="#cb20-70" aria-hidden="true" tabindex="-1"></a>Mgmt <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(Mgmt_pred, Mgmt_obs, <span class="at">by =</span> <span class="st">"FID"</span>)</span>
<span id="cb20-71"><a href="#cb20-71" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb20-72"><a href="#cb20-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-73"><a href="#cb20-73" aria-hidden="true" tabindex="-1"></a>Polr3k</span>
<span id="cb20-74"><a href="#cb20-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-75"><a href="#cb20-75" aria-hidden="true" tabindex="-1"></a><span class="in">```{r polr3k, eval=FALSE}</span></span>
<span id="cb20-76"><a href="#cb20-76" aria-hidden="true" tabindex="-1"></a>Polr3k_pred <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pred_expr[, <span class="fu">c</span>(<span class="st">"FID"</span>, <span class="st">"ENSRNOG00000017843"</span>)])</span>
<span id="cb20-77"><a href="#cb20-77" aria-hidden="true" tabindex="-1"></a>Polr3k_obs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(obs_expr[,  <span class="fu">c</span>(<span class="st">"FID"</span>, <span class="st">"ENSRNOG00000017843"</span>)]) </span>
<span id="cb20-78"><a href="#cb20-78" aria-hidden="true" tabindex="-1"></a>Polr3k_pred<span class="sc">$</span>FID <span class="ot">=</span> <span class="fu">as.character</span>(Polr3k_pred<span class="sc">$</span>FID)</span>
<span id="cb20-79"><a href="#cb20-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-80"><a href="#cb20-80" aria-hidden="true" tabindex="-1"></a>Polr3k_pred <span class="ot">&lt;-</span> Polr3k_pred[<span class="fu">match</span>(Polr3k_obs<span class="sc">$</span>FID, Polr3k_pred<span class="sc">$</span>FID),]</span>
<span id="cb20-81"><a href="#cb20-81" aria-hidden="true" tabindex="-1"></a>Polr3k_obs <span class="ot">&lt;-</span> Polr3k_obs[<span class="fu">match</span>(Polr3k_pred<span class="sc">$</span>FID, Polr3k_obs<span class="sc">$</span>FID ), ]</span>
<span id="cb20-82"><a href="#cb20-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-83"><a href="#cb20-83" aria-hidden="true" tabindex="-1"></a>Polr3k_pred<span class="sc">$</span>FID <span class="ot">=</span> <span class="fu">as.character</span>(Polr3k_pred<span class="sc">$</span>FID)</span>
<span id="cb20-84"><a href="#cb20-84" aria-hidden="true" tabindex="-1"></a>Polr3k <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(Polr3k_pred, Polr3k_obs, <span class="at">by =</span> <span class="st">"FID"</span>)</span>
<span id="cb20-85"><a href="#cb20-85" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb20-86"><a href="#cb20-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-87"><a href="#cb20-87" aria-hidden="true" tabindex="-1"></a>Dhfr</span>
<span id="cb20-88"><a href="#cb20-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-89"><a href="#cb20-89" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Eno1, eval=FALSE}</span></span>
<span id="cb20-90"><a href="#cb20-90" aria-hidden="true" tabindex="-1"></a>Dhfr_pred <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pred_expr[, <span class="fu">c</span>(<span class="st">"FID"</span>, <span class="st">"ENSRNOG00000013521"</span>)])</span>
<span id="cb20-91"><a href="#cb20-91" aria-hidden="true" tabindex="-1"></a>Dhfr_obs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(obs_expr[,  <span class="fu">c</span>(<span class="st">"FID"</span>, <span class="st">"ENSRNOG00000013521"</span>)]) </span>
<span id="cb20-92"><a href="#cb20-92" aria-hidden="true" tabindex="-1"></a>Dhfr_pred<span class="sc">$</span>FID <span class="ot">=</span> <span class="fu">as.character</span>(Dhfr_pred<span class="sc">$</span>FID)</span>
<span id="cb20-93"><a href="#cb20-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-94"><a href="#cb20-94" aria-hidden="true" tabindex="-1"></a>Dhfr_pred <span class="ot">&lt;-</span> Dhfr_pred[<span class="fu">match</span>(Dhfr_obs<span class="sc">$</span>FID, Dhfr_pred<span class="sc">$</span>FID),]</span>
<span id="cb20-95"><a href="#cb20-95" aria-hidden="true" tabindex="-1"></a>Dhfr_obs <span class="ot">&lt;-</span> Dhfr_obs[<span class="fu">match</span>(Dhfr_pred<span class="sc">$</span>FID, Dhfr_obs<span class="sc">$</span>FID ), ]</span>
<span id="cb20-96"><a href="#cb20-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-97"><a href="#cb20-97" aria-hidden="true" tabindex="-1"></a>Dhfr_pred<span class="sc">$</span>FID <span class="ot">=</span> <span class="fu">as.character</span>(Dhfr_pred<span class="sc">$</span>FID)</span>
<span id="cb20-98"><a href="#cb20-98" aria-hidden="true" tabindex="-1"></a>Dhfr <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(Dhfr_pred, Dhfr_obs, <span class="at">by =</span> <span class="st">"FID"</span>)</span>
<span id="cb20-99"><a href="#cb20-99" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb20-100"><a href="#cb20-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-101"><a href="#cb20-101" aria-hidden="true" tabindex="-1"></a>Plot pred vs obs expression with our models in NcAcc2 tissue</span>
<span id="cb20-102"><a href="#cb20-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-103"><a href="#cb20-103" aria-hidden="true" tabindex="-1"></a><span class="in">```{r compare model performance in NcAcc2}</span></span>
<span id="cb20-104"><a href="#cb20-104" aria-hidden="true" tabindex="-1"></a>Mgmt <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Downloads/model_perf_Mgmt.txt"</span>)</span>
<span id="cb20-105"><a href="#cb20-105" aria-hidden="true" tabindex="-1"></a>Polr3k <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Downloads/model_perf_Polr3k.txt"</span>)</span>
<span id="cb20-106"><a href="#cb20-106" aria-hidden="true" tabindex="-1"></a>Dhfr <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Downloads/model_perf_Dhfr.txt"</span>)</span>
<span id="cb20-107"><a href="#cb20-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-108"><a href="#cb20-108" aria-hidden="true" tabindex="-1"></a>g1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(Mgmt, <span class="fu">aes</span>(pred, obs)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">0.4</span>, <span class="at">y =</span> <span class="fl">2.5</span>, <span class="at">label =</span> <span class="st">"Mgmt"</span>, <span class="at">size =</span> <span class="dv">10</span>, <span class="at">fontface =</span><span class="st">"bold.italic"</span>) <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">0.53</span>, <span class="at">y =</span> <span class="dv">2</span>, <span class="at">label =</span> <span class="st">"R2 =  0.724"</span>, <span class="at">size =</span> <span class="dv">8</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Predicted expression"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Observed Expression"</span>)</span>
<span id="cb20-109"><a href="#cb20-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-110"><a href="#cb20-110" aria-hidden="true" tabindex="-1"></a>g2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(Polr3k, <span class="fu">aes</span>(pred, obs)) <span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Polr3k"</span>) <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="sc">-</span><span class="fl">1.8</span>, <span class="at">y =</span> <span class="dv">2</span>, <span class="at">label =</span> <span class="st">"r2 =  0.796"</span>, <span class="at">size =</span> <span class="dv">3</span>)  <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span>
<span id="cb20-111"><a href="#cb20-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-112"><a href="#cb20-112" aria-hidden="true" tabindex="-1"></a>g3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(Dhfr, <span class="fu">aes</span>(pred, obs))  <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="sc">-</span><span class="fl">0.4</span>, <span class="at">y =</span> <span class="fl">2.5</span>, <span class="at">label =</span> <span class="st">"Dhfr"</span>, <span class="at">size =</span> <span class="dv">10</span>, <span class="at">fontface =</span><span class="dv">2</span>) <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="sc">-</span><span class="fl">0.25</span>, <span class="at">y =</span> <span class="dv">2</span>, <span class="at">label =</span> <span class="st">"r2 =  0.554"</span>, <span class="at">size =</span> <span class="dv">8</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Predicted expression"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Observed Expression"</span>) <span class="sc">+</span> <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="cn">NA</span>, <span class="fl">1.25</span>)) </span>
<span id="cb20-113"><a href="#cb20-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-114"><a href="#cb20-114" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb20-115"><a href="#cb20-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-116"><a href="#cb20-116" aria-hidden="true" tabindex="-1"></a>Calculate Predicted vs Observed Expression for Mgmt in Humans using GTEx NA predictDB with Geuvadis genotypes (hg38) as well</span>
<span id="cb20-117"><a href="#cb20-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-118"><a href="#cb20-118" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash calculate pred expr with models 2, eval=FALSE}</span></span>
<span id="cb20-119"><a href="#cb20-119" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate imlabtools </span>
<span id="cb20-120"><a href="#cb20-120" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">METAXCAN</span><span class="op">=</span>/gpfs/data/im-lab/nas40t2/natasha/GTEX_Analysis/MetaXcan/software</span>
<span id="cb20-121"><a href="#cb20-121" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">GENO</span><span class="op">=</span>/gpfs/data/im-lab/nas40t2/Data/1000G/vcf_hg38/geuvadis</span>
<span id="cb20-122"><a href="#cb20-122" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">MODEL</span><span class="op">=</span>/gpfs/data/im-lab/nas40t2/Data/PredictDB/GTEx_v8/models_v1/eqtl/elastic_net_models</span>
<span id="cb20-123"><a href="#cb20-123" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">DATA</span><span class="op">=</span>/gpfs/data/im-lab/nas40t2/natasha/GTEX_Analysis/predixcan</span>
<span id="cb20-124"><a href="#cb20-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-125"><a href="#cb20-125" aria-hidden="true" tabindex="-1"></a><span class="ex">python3</span> <span class="va">$METAXCAN</span>/Predict.py <span class="dt">\</span></span>
<span id="cb20-126"><a href="#cb20-126" aria-hidden="true" tabindex="-1"></a>--model_db_path <span class="va">$MODEL</span>/en_Brain_Nucleus_accumbens_basal_ganglia.db  <span class="dt">\</span></span>
<span id="cb20-127"><a href="#cb20-127" aria-hidden="true" tabindex="-1"></a>--model_db_snp_key varID <span class="dt">\</span></span>
<span id="cb20-128"><a href="#cb20-128" aria-hidden="true" tabindex="-1"></a>--vcf_genotypes <span class="va">$GENO</span>/ALL.chr<span class="pp">*</span>.shapeit2_integrated_v1a.GRCh38.20181129.phased.vcf.gz <span class="dt">\</span></span>
<span id="cb20-129"><a href="#cb20-129" aria-hidden="true" tabindex="-1"></a>--vcf_mode genotyped <span class="dt">\</span></span>
<span id="cb20-130"><a href="#cb20-130" aria-hidden="true" tabindex="-1"></a>--on_the_fly_mapping METADATA <span class="st">"{}_{}_{}_{}_b38"</span> <span class="dt">\</span></span>
<span id="cb20-131"><a href="#cb20-131" aria-hidden="true" tabindex="-1"></a>--prediction_output 1000G_NAcc2__predict.txt  <span class="dt">\</span></span>
<span id="cb20-132"><a href="#cb20-132" aria-hidden="true" tabindex="-1"></a>--prediction_summary_output 1000G_NAcc2__summary.txt  <span class="dt">\</span></span>
<span id="cb20-133"><a href="#cb20-133" aria-hidden="true" tabindex="-1"></a>--verbosity 9 <span class="dt">\</span></span>
<span id="cb20-134"><a href="#cb20-134" aria-hidden="true" tabindex="-1"></a>--throw</span>
<span id="cb20-135"><a href="#cb20-135" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb20-136"><a href="#cb20-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-137"><a href="#cb20-137" aria-hidden="true" tabindex="-1"></a>RPS26 r2= 0.7352744</span>
<span id="cb20-138"><a href="#cb20-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-139"><a href="#cb20-139" aria-hidden="true" tabindex="-1"></a><span class="in">```{r check correlation between predicted and observed expression 2, eval=FALSE}</span></span>
<span id="cb20-140"><a href="#cb20-140" aria-hidden="true" tabindex="-1"></a>pred_expr <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/comp_to_GTEx/1000G_NAcc2__predict.txt"</span>)</span>
<span id="cb20-141"><a href="#cb20-141" aria-hidden="true" tabindex="-1"></a>obs_expr <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/comp_to_GTEx/GD462.GeneQuantRPKM.50FN.samplename.resk10.txt.gz"</span>)</span>
<span id="cb20-142"><a href="#cb20-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-143"><a href="#cb20-143" aria-hidden="true" tabindex="-1"></a>obs_expr <span class="ot">&lt;-</span> obs_expr <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(Chr, Gene_Symbol, Coord)) <span class="sc">%&gt;%</span> <span class="fu">pivot_longer</span>(<span class="sc">!</span>TargetID, <span class="at">names_to =</span> <span class="st">"FID"</span>, <span class="at">values_to =</span> <span class="st">"count"</span>) <span class="sc">%&gt;%</span> <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> TargetID, <span class="at">values_from =</span> count)</span>
<span id="cb20-144"><a href="#cb20-144" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(obs_expr)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(obs_expr)] <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">colnames</span>(obs_expr)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(obs_expr)], <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb20-145"><a href="#cb20-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-146"><a href="#cb20-146" aria-hidden="true" tabindex="-1"></a>RPS26_pred <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pred_expr[, <span class="fu">c</span>(<span class="st">"FID"</span>, <span class="st">"ENSG00000197728.9"</span>)])</span>
<span id="cb20-147"><a href="#cb20-147" aria-hidden="true" tabindex="-1"></a>RPS26_obs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(obs_expr[,  <span class="fu">c</span>(<span class="st">"FID"</span>, <span class="st">"ENSG00000197728"</span>)]) </span>
<span id="cb20-148"><a href="#cb20-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-149"><a href="#cb20-149" aria-hidden="true" tabindex="-1"></a>RPS26_pred <span class="ot">&lt;-</span> RPS26_pred[<span class="fu">match</span>(RPS26_obs<span class="sc">$</span>FID, RPS26_pred<span class="sc">$</span>FID),]</span>
<span id="cb20-150"><a href="#cb20-150" aria-hidden="true" tabindex="-1"></a>RPS26_obs <span class="ot">&lt;-</span> RPS26_obs[<span class="fu">match</span>(RPS26_pred<span class="sc">$</span>FID, RPS26_obs<span class="sc">$</span>FID ), ]</span>
<span id="cb20-151"><a href="#cb20-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-152"><a href="#cb20-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-153"><a href="#cb20-153" aria-hidden="true" tabindex="-1"></a>RPS26 <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(RPS26_pred, RPS26_obs, <span class="at">by =</span> <span class="st">"FID"</span>)</span>
<span id="cb20-154"><a href="#cb20-154" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(RPS26) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"FID"</span>, <span class="st">"pred"</span>, <span class="st">"obs"</span>)</span>
<span id="cb20-155"><a href="#cb20-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-156"><a href="#cb20-156" aria-hidden="true" tabindex="-1"></a>RPS26_GTEx <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Downloads/GTEx_RPS26.txt"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb20-157"><a href="#cb20-157" aria-hidden="true" tabindex="-1"></a>DHFR_GTEx <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Downloads/GTEx_DHFR_results.txt"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb20-158"><a href="#cb20-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-159"><a href="#cb20-159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-160"><a href="#cb20-160" aria-hidden="true" tabindex="-1"></a>g4 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(DHFR_GTEx, <span class="fu">aes</span>(pred, obs)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="sc">-</span><span class="fl">0.45</span>, <span class="at">y =</span> <span class="dv">55</span>, <span class="at">label =</span> <span class="st">"DHFR"</span>, <span class="at">size =</span> <span class="dv">10</span>, <span class="at">fontface =</span><span class="dv">2</span>) <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="sc">-</span><span class="fl">0.4</span>, <span class="at">y =</span> <span class="dv">51</span>, <span class="at">label =</span> <span class="st">"r2 =  0.506"</span>, <span class="at">size =</span> <span class="dv">8</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Predicted expression"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Observed Expression"</span>)</span>
<span id="cb20-161"><a href="#cb20-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-162"><a href="#cb20-162" aria-hidden="true" tabindex="-1"></a>g5 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(RPS26_GTEx, <span class="fu">aes</span>(pred, obs)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">y =</span> <span class="dv">385</span>, <span class="at">label =</span> <span class="st">"RPS26"</span>, <span class="at">size =</span> <span class="dv">10</span>, <span class="at">fontface =</span><span class="st">"bold.italic"</span>) <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="sc">-</span><span class="fl">0.45</span>, <span class="at">y =</span> <span class="dv">355</span>, <span class="at">label =</span> <span class="st">"R2 =  0.735"</span>, <span class="at">size =</span> <span class="dv">8</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Predicted expression"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Observed Expression"</span>)</span>
<span id="cb20-163"><a href="#cb20-163" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb20-164"><a href="#cb20-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-165"><a href="#cb20-165" aria-hidden="true" tabindex="-1"></a><span class="fu"># Correlation of predicted R2 between tissues in GTEx</span></span>
<span id="cb20-166"><a href="#cb20-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-167"><a href="#cb20-167" aria-hidden="true" tabindex="-1"></a>We use Nucleus Accumbens, Hippocampus, Cerebellum, Cortex and Anterior Cingulate Cortex in GTEx</span>
<span id="cb20-168"><a href="#cb20-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-169"><a href="#cb20-169" aria-hidden="true" tabindex="-1"></a><span class="in">```{r compile all R2 for tissues}</span></span>
<span id="cb20-170"><a href="#cb20-170" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list</span>( <span class="st">"en_Brain_Nucleus_accumbens_basal_ganglia.db"</span>, <span class="st">"en_Brain_Hippocampus.db"</span>, <span class="st">"en_Brain_Cerebellum.db"</span>, <span class="st">"en_Brain_Cortex.db"</span>, <span class="st">"en_Brain_Anterior_cingulate_cortex_BA24.db"</span>)</span>
<span id="cb20-171"><a href="#cb20-171" aria-hidden="true" tabindex="-1"></a>dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/Data/PredictDB/GTEx_v8/models_v1/eqtl/elastic_net_models/"</span></span>
<span id="cb20-172"><a href="#cb20-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-173"><a href="#cb20-173" aria-hidden="true" tabindex="-1"></a>pred_tis <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">as.character</span>())</span>
<span id="cb20-174"><a href="#cb20-174" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb20-175"><a href="#cb20-175" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> dir <span class="sc">%&amp;%</span> fila</span>
<span id="cb20-176"><a href="#cb20-176" aria-hidden="true" tabindex="-1"></a>  sqlite.driver <span class="ot">&lt;-</span> <span class="fu">dbDriver</span>(<span class="st">"SQLite"</span>)</span>
<span id="cb20-177"><a href="#cb20-177" aria-hidden="true" tabindex="-1"></a>  conn <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(), filename)</span>
<span id="cb20-178"><a href="#cb20-178" aria-hidden="true" tabindex="-1"></a>  tis <span class="ot">&lt;-</span> <span class="fu">substr</span>(fila, <span class="dv">4</span>, <span class="fu">str_length</span>(fila)<span class="sc">-</span><span class="dv">3</span>)</span>
<span id="cb20-179"><a href="#cb20-179" aria-hidden="true" tabindex="-1"></a>  extra <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>(conn, <span class="st">'select * from extra'</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, test_R2_avg)) </span>
<span id="cb20-180"><a href="#cb20-180" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(extra)[<span class="dv">2</span>] <span class="ot">=</span> tis</span>
<span id="cb20-181"><a href="#cb20-181" aria-hidden="true" tabindex="-1"></a>  pred_tis <span class="ot">&lt;-</span> <span class="fu">full_join</span>(pred_tis, extra, <span class="at">by =</span> <span class="st">"gene"</span>) </span>
<span id="cb20-182"><a href="#cb20-182" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dbDisconnect</span>(conn)</span>
<span id="cb20-183"><a href="#cb20-183" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb20-184"><a href="#cb20-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-185"><a href="#cb20-185" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(pred_tis) <span class="ot">&lt;-</span> pred_tis<span class="sc">$</span>gene</span>
<span id="cb20-186"><a href="#cb20-186" aria-hidden="true" tabindex="-1"></a>pred_tis <span class="ot">&lt;-</span> pred_tis <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(gene))</span>
<span id="cb20-187"><a href="#cb20-187" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(pred_tis, <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/comp_to_GTEx/pred_R2_betw_GTEx_brain_tissues.RDS"</span>)</span>
<span id="cb20-188"><a href="#cb20-188" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb20-189"><a href="#cb20-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-190"><a href="#cb20-190" aria-hidden="true" tabindex="-1"></a><span class="in">```{r make pairs plot}</span></span>
<span id="cb20-191"><a href="#cb20-191" aria-hidden="true" tabindex="-1"></a>pred_tis <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"/Users/natashasanthanam/Downloads/pred_R2_betw_GTEx_brain_tissues.RDS"</span>)</span>
<span id="cb20-192"><a href="#cb20-192" aria-hidden="true" tabindex="-1"></a><span class="fu">pairs</span>(pred_tis)</span>
<span id="cb20-193"><a href="#cb20-193" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb20-194"><a href="#cb20-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-195"><a href="#cb20-195" aria-hidden="true" tabindex="-1"></a><span class="fu"># Compare Heritability and Predictability in Humans and Rats</span></span>
<span id="cb20-196"><a href="#cb20-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-197"><a href="#cb20-197" aria-hidden="true" tabindex="-1"></a>Get R2 and H2 from Heather's Models for Brain</span>
<span id="cb20-198"><a href="#cb20-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-199"><a href="#cb20-199" aria-hidden="true" tabindex="-1"></a><span class="in">```{sql connection, eval=FALSE}</span></span>
<span id="cb20-200"><a href="#cb20-200" aria-hidden="true" tabindex="-1"></a>sqlite3 genarch<span class="ch">.</span>db</span>
<span id="cb20-201"><a href="#cb20-201" aria-hidden="true" tabindex="-1"></a><span class="ch">.</span>headers <span class="kw">on</span></span>
<span id="cb20-202"><a href="#cb20-202" aria-hidden="true" tabindex="-1"></a><span class="ch">.</span>mode csv</span>
<span id="cb20-203"><a href="#cb20-203" aria-hidden="true" tabindex="-1"></a><span class="ch">.</span>output human_h2<span class="ch">.</span>csv</span>
<span id="cb20-204"><a href="#cb20-204" aria-hidden="true" tabindex="-1"></a><span class="kw">select</span> gene, ensid, en_r2 ,h2,h2_ci <span class="kw">from</span> results <span class="kw">where</span> tissue <span class="op">=</span> <span class="st">"DGN-WB"</span>;</span>
<span id="cb20-205"><a href="#cb20-205" aria-hidden="true" tabindex="-1"></a><span class="ch">.</span>quit</span>
<span id="cb20-206"><a href="#cb20-206" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb20-207"><a href="#cb20-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-208"><a href="#cb20-208" aria-hidden="true" tabindex="-1"></a><span class="in">```{r read in herit for humans and rats }</span></span>
<span id="cb20-209"><a href="#cb20-209" aria-hidden="true" tabindex="-1"></a>rat_h2 <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Github/rat-genomic-analysis/data/Ac_PVE_estimates.txt"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">rat_herit =</span> X2)</span>
<span id="cb20-210"><a href="#cb20-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-211"><a href="#cb20-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-212"><a href="#cb20-212" aria-hidden="true" tabindex="-1"></a>human_h2 <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"/Users/natashasanthanam/Github/rat-genomic-analysis/data/human_h2.csv"</span>)</span>
<span id="cb20-213"><a href="#cb20-213" aria-hidden="true" tabindex="-1"></a>human_h2<span class="sc">$</span>ensid <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(human_h2<span class="sc">$</span>ensid, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb20-214"><a href="#cb20-214" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(rat_h2<span class="sc">$</span>rat_herit)</span>
<span id="cb20-215"><a href="#cb20-215" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(human_h2<span class="sc">$</span>h2)</span>
<span id="cb20-216"><a href="#cb20-216" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb20-217"><a href="#cb20-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-218"><a href="#cb20-218" aria-hidden="true" tabindex="-1"></a>Change gene names in rats</span>
<span id="cb20-219"><a href="#cb20-219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-220"><a href="#cb20-220" aria-hidden="true" tabindex="-1"></a><span class="in">```{r plot heritability in rats and humans}</span></span>
<span id="cb20-221"><a href="#cb20-221" aria-hidden="true" tabindex="-1"></a>orth.rats <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/natashasanthanam/Downloads/ortholog_genes_rats_humans.tsv"</span>)</span>
<span id="cb20-222"><a href="#cb20-222" aria-hidden="true" tabindex="-1"></a>rat_h2 <span class="ot">&lt;-</span> rat_h2 <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">ensid =</span> orth.rats[<span class="fu">match</span>(rat_h2<span class="sc">$</span>X1, orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene), <span class="dv">1</span>]<span class="sc">$</span>ensembl_gene_id)</span>
<span id="cb20-223"><a href="#cb20-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-224"><a href="#cb20-224" aria-hidden="true" tabindex="-1"></a>all_h2 <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(rat_h2, human_h2, <span class="at">by =</span> <span class="st">"ensid"</span>)</span>
<span id="cb20-225"><a href="#cb20-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-226"><a href="#cb20-226" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(all_h2<span class="sc">$</span>rat_herit, all_h2<span class="sc">$</span>h2)</span>
<span id="cb20-227"><a href="#cb20-227" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb20-228"><a href="#cb20-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-229"><a href="#cb20-229" aria-hidden="true" tabindex="-1"></a>Compare Predictability (R2) between Humans and Rats</span>
<span id="cb20-230"><a href="#cb20-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-231"><a href="#cb20-231" aria-hidden="true" tabindex="-1"></a><span class="in">```{r comp r2 between humans and rats}</span></span>
<span id="cb20-232"><a href="#cb20-232" aria-hidden="true" tabindex="-1"></a>fila <span class="ot">&lt;-</span> <span class="st">"/Users/natashasanthanam/Box/imlab-data/data-Github/rat-genomic-analysis/sql/Ac_output_db.db"</span></span>
<span id="cb20-233"><a href="#cb20-233" aria-hidden="true" tabindex="-1"></a>sqlite.driver <span class="ot">&lt;-</span> <span class="fu">dbDriver</span>(<span class="st">"SQLite"</span>)</span>
<span id="cb20-234"><a href="#cb20-234" aria-hidden="true" tabindex="-1"></a>conn <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(), fila)</span>
<span id="cb20-235"><a href="#cb20-235" aria-hidden="true" tabindex="-1"></a>rat_r2 <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>(conn, <span class="st">'select * from extra'</span>)</span>
<span id="cb20-236"><a href="#cb20-236" aria-hidden="true" tabindex="-1"></a><span class="fu">dbDisconnect</span>(conn)</span>
<span id="cb20-237"><a href="#cb20-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-238"><a href="#cb20-238" aria-hidden="true" tabindex="-1"></a>rat_r2 <span class="ot">&lt;-</span> rat_r2 <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">ensid =</span> orth.rats[<span class="fu">match</span>(rat_r2<span class="sc">$</span>gene, orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene), <span class="dv">1</span>]<span class="sc">$</span>ensembl_gene_id)</span>
<span id="cb20-239"><a href="#cb20-239" aria-hidden="true" tabindex="-1"></a>all_r2 <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(rat_r2, human_h2, <span class="at">by =</span> <span class="st">"ensid"</span>)</span>
<span id="cb20-240"><a href="#cb20-240" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(all_r2<span class="sc">$</span>R2, all_r2<span class="sc">$</span>en_r2)</span>
<span id="cb20-241"><a href="#cb20-241" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb20-242"><a href="#cb20-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-243"><a href="#cb20-243" aria-hidden="true" tabindex="-1"></a>Plot the Comparison between the two</span>
<span id="cb20-244"><a href="#cb20-244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-245"><a href="#cb20-245" aria-hidden="true" tabindex="-1"></a><span class="in">```{r save inner join of r2 and h2}</span></span>
<span id="cb20-246"><a href="#cb20-246" aria-hidden="true" tabindex="-1"></a>Ac_h2 <span class="ot">&lt;-</span> rat_h2 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene.x =</span> X1) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">rat_ci =</span> <span class="fu">paste</span>(<span class="fu">round</span>(Ac_h2<span class="sc">$</span>X3, <span class="at">digits =</span> <span class="dv">5</span>), <span class="fu">round</span>(Ac_h2<span class="sc">$</span>X4, <span class="at">digits =</span> <span class="dv">5</span>), <span class="at">sep =</span> <span class="st">"-"</span>)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(gene.x, rat_herit, rat_ci))</span>
<span id="cb20-247"><a href="#cb20-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-248"><a href="#cb20-248" aria-hidden="true" tabindex="-1"></a>lg_df <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(all_r2, Ac_h2, <span class="at">by =</span> <span class="st">"gene.x"</span>)</span>
<span id="cb20-249"><a href="#cb20-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-250"><a href="#cb20-250" aria-hidden="true" tabindex="-1"></a>lg_df <span class="ot">&lt;-</span> lg_df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(ensid, gene.x, genename, R2, en_r2,rat_herit,h2,  rat_ci, h2_ci))</span>
<span id="cb20-251"><a href="#cb20-251" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(lg_df) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"ensembl_id"</span>, <span class="st">"rat_ensembl_id"</span>, <span class="st">"genename"</span>, <span class="st">"rat_r2"</span>, <span class="st">"human_r2"</span>, <span class="st">"rat_h2"</span>, <span class="st">"human_h2"</span>, <span class="st">"rat_h2_ci"</span>, <span class="st">"human_h2_ci"</span>)</span>
<span id="cb20-252"><a href="#cb20-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-253"><a href="#cb20-253" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot h2 between rats and Humans with line for null </span></span>
<span id="cb20-254"><a href="#cb20-254" aria-hidden="true" tabindex="-1"></a>lg_df <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">human_shuffled_h2 =</span> <span class="fu">sample</span>(human_h2, <span class="fu">nrow</span>(lg_df), <span class="at">replace=</span>F)) <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(rat_h2, human_h2)) <span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="at">col =</span> <span class="st">"darkgreen"</span>) <span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="fu">aes</span>(rat_h2, human_shuffled_h2),<span class="at">col=</span><span class="st">"dark gray"</span>)<span class="sc">+</span><span class="fu">theme_bw</span>()<span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Rat Heritability"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Human Heritability"</span>)</span>
<span id="cb20-255"><a href="#cb20-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-256"><a href="#cb20-256" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot R2 between rats and Humans with line for null </span></span>
<span id="cb20-257"><a href="#cb20-257" aria-hidden="true" tabindex="-1"></a>lg_df <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">human_shuffled_r2 =</span> <span class="fu">sample</span>(human_r2, <span class="fu">nrow</span>(lg_df), <span class="at">replace=</span>F)) <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(rat_r2, human_r2)) <span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="at">col =</span> <span class="st">"midnightblue"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Rat Predictability"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Human Predictability"</span>) <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">0.1</span>, <span class="at">y =</span> <span class="fl">0.2</span>, <span class="at">label =</span> <span class="st">"R =  0.061"</span>, <span class="at">size =</span> <span class="dv">8</span>) </span>
<span id="cb20-258"><a href="#cb20-258" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb20-259"><a href="#cb20-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-260"><a href="#cb20-260" aria-hidden="true" tabindex="-1"></a>Compare Predictability Between Tissues</span>
<span id="cb20-261"><a href="#cb20-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-262"><a href="#cb20-262" aria-hidden="true" tabindex="-1"></a><span class="in">```{r compare predictability between rat tissue, eval=FALSE}</span></span>
<span id="cb20-263"><a href="#cb20-263" aria-hidden="true" tabindex="-1"></a>box.dir <span class="ot">&lt;-</span> <span class="st">"/Users/natashasanthanam/Box/imlab-data/data-Github/rat-genomic-analysis/sql/"</span></span>
<span id="cb20-264"><a href="#cb20-264" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Il_output_db.db"</span>, <span class="st">"Lh_output_db.db"</span>, <span class="st">"Ac_output_db.db"</span>, <span class="st">"Vo_output_db.db"</span>, <span class="st">"Pl_output_db.db"</span>)</span>
<span id="cb20-265"><a href="#cb20-265" aria-hidden="true" tabindex="-1"></a>tis.list <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Infralimbic Cortex"</span>, <span class="st">"Lateral Habenula"</span>, <span class="st">"Nucleus accumbens"</span>, <span class="st">"Orbitofrontal Cortex"</span>, <span class="st">"Prelimibic Cortex"</span>)</span>
<span id="cb20-266"><a href="#cb20-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-267"><a href="#cb20-267" aria-hidden="true" tabindex="-1"></a>pred_tis <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene =</span> <span class="fu">as.character</span>())</span>
<span id="cb20-268"><a href="#cb20-268" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb20-269"><a href="#cb20-269" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> box.dir <span class="sc">%&amp;%</span> fila</span>
<span id="cb20-270"><a href="#cb20-270" aria-hidden="true" tabindex="-1"></a>  sqlite.driver <span class="ot">&lt;-</span> <span class="fu">dbDriver</span>(<span class="st">"SQLite"</span>)</span>
<span id="cb20-271"><a href="#cb20-271" aria-hidden="true" tabindex="-1"></a>  conn <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(), filename)</span>
<span id="cb20-272"><a href="#cb20-272" aria-hidden="true" tabindex="-1"></a>  i <span class="ot">&lt;-</span> <span class="fu">match</span>(fila, filelist)</span>
<span id="cb20-273"><a href="#cb20-273" aria-hidden="true" tabindex="-1"></a>  tis <span class="ot">&lt;-</span> tis.list[i]</span>
<span id="cb20-274"><a href="#cb20-274" aria-hidden="true" tabindex="-1"></a>  extra <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>(conn, <span class="st">'select * from extra'</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, R2)) </span>
<span id="cb20-275"><a href="#cb20-275" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(extra)[<span class="dv">2</span>] <span class="ot">=</span> tis</span>
<span id="cb20-276"><a href="#cb20-276" aria-hidden="true" tabindex="-1"></a>  pred_tis <span class="ot">&lt;-</span> <span class="fu">full_join</span>(pred_tis, extra, <span class="at">by =</span> <span class="st">"gene"</span>) </span>
<span id="cb20-277"><a href="#cb20-277" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dbDisconnect</span>(conn)</span>
<span id="cb20-278"><a href="#cb20-278" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb20-279"><a href="#cb20-279" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(pred_tis) <span class="ot">&lt;-</span> pred_tis<span class="sc">$</span>gene</span>
<span id="cb20-280"><a href="#cb20-280" aria-hidden="true" tabindex="-1"></a>pred_tis <span class="ot">&lt;-</span> pred_tis <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(gene))</span>
<span id="cb20-281"><a href="#cb20-281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-282"><a href="#cb20-282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-283"><a href="#cb20-283" aria-hidden="true" tabindex="-1"></a>d1 <span class="ot">&lt;-</span> <span class="fu">cor</span>(pred_tis <span class="sc">%&gt;%</span> na.omit)</span>
<span id="cb20-284"><a href="#cb20-284" aria-hidden="true" tabindex="-1"></a>d1[<span class="fu">lower.tri</span>(d1)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb20-285"><a href="#cb20-285" aria-hidden="true" tabindex="-1"></a>d1 <span class="ot">&lt;-</span> d1 <span class="sc">%&gt;%</span></span>
<span id="cb20-286"><a href="#cb20-286" aria-hidden="true" tabindex="-1"></a>  as.data.frame <span class="sc">%&gt;%</span></span>
<span id="cb20-287"><a href="#cb20-287" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">'var1'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-288"><a href="#cb20-288" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>(var2, value, <span class="sc">-</span>var1) </span>
<span id="cb20-289"><a href="#cb20-289" aria-hidden="true" tabindex="-1"></a>d1 <span class="ot">&lt;-</span> d1 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">r =</span> value) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>() </span>
<span id="cb20-290"><a href="#cb20-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-291"><a href="#cb20-291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-292"><a href="#cb20-292" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> d1, <span class="fu">aes</span>(<span class="at">x =</span> var1, <span class="at">y =</span> var2, <span class="at">fill =</span> r)) <span class="sc">+</span></span>
<span id="cb20-293"><a href="#cb20-293" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_tile</span>(<span class="at">color =</span> <span class="st">"white"</span>)<span class="sc">+</span></span>
<span id="cb20-294"><a href="#cb20-294" aria-hidden="true" tabindex="-1"></a> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">"dodgerblue"</span>, <span class="at">high=</span><span class="st">"dodgerblue4"</span>) <span class="sc">+</span></span>
<span id="cb20-295"><a href="#cb20-295" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb20-296"><a href="#cb20-296" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">""</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>), <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)   </span>
<span id="cb20-297"><a href="#cb20-297" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb20-298"><a href="#cb20-298" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb20-299"><a href="#cb20-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-300"><a href="#cb20-300" aria-hidden="true" tabindex="-1"></a>Calculate R2 across GTEx Tissues</span>
<span id="cb20-301"><a href="#cb20-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-302"><a href="#cb20-302" aria-hidden="true" tabindex="-1"></a><span class="in">```{r iterate GTEx models, eval=FALSE}</span></span>
<span id="cb20-303"><a href="#cb20-303" aria-hidden="true" tabindex="-1"></a>model.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/Data/PredictDB/GTEx_v8/models_v1/eqtl/elastic_net_models/"</span></span>
<span id="cb20-304"><a href="#cb20-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-305"><a href="#cb20-305" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"en_Brain_Amygdala.db"</span>, <span class="st">"en_Brain_Cerebellum.db"</span>, <span class="st">"en_Brain_Nucleus_accumbens_basal_ganglia.db"</span>, <span class="st">"en_Brain_Spinal_cord_cervical_c-1.db"</span>, <span class="st">"en_Brain_Substantia_nigra.db"</span>)</span>
<span id="cb20-306"><a href="#cb20-306" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb20-307"><a href="#cb20-307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-308"><a href="#cb20-308" aria-hidden="true" tabindex="-1"></a><span class="in">```{r plot R2 across GTEx tissues}</span></span>
<span id="cb20-309"><a href="#cb20-309" aria-hidden="true" tabindex="-1"></a><span class="co">#pred_tis &lt;- read_tsv(dir %&amp;% "pred_R2_betw_GTEx_brain_tissues.txt", col_names = TRUE) </span></span>
<span id="cb20-310"><a href="#cb20-310" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(pred_tis) <span class="ot">&lt;-</span> pred_tis<span class="sc">$</span>gene</span>
<span id="cb20-311"><a href="#cb20-311" aria-hidden="true" tabindex="-1"></a>pred_tis <span class="ot">&lt;-</span> pred_tis <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(gene))</span>
<span id="cb20-312"><a href="#cb20-312" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(pred_tis) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Amygdala"</span>, <span class="st">"Cerebellum"</span>, <span class="st">"Nucleus accumbens"</span>,  <span class="st">"Spinal cord"</span>, <span class="st">"Substantia nigra"</span>)</span>
<span id="cb20-313"><a href="#cb20-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-314"><a href="#cb20-314" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-315"><a href="#cb20-315" aria-hidden="true" tabindex="-1"></a>d2 <span class="ot">&lt;-</span> <span class="fu">cor</span>(pred_tis <span class="sc">%&gt;%</span> na.omit)</span>
<span id="cb20-316"><a href="#cb20-316" aria-hidden="true" tabindex="-1"></a>d2[<span class="fu">lower.tri</span>(d2)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb20-317"><a href="#cb20-317" aria-hidden="true" tabindex="-1"></a>d2 <span class="ot">&lt;-</span> d2 <span class="sc">%&gt;%</span></span>
<span id="cb20-318"><a href="#cb20-318" aria-hidden="true" tabindex="-1"></a>  as.data.frame <span class="sc">%&gt;%</span></span>
<span id="cb20-319"><a href="#cb20-319" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">'var1'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-320"><a href="#cb20-320" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>(var2, value, <span class="sc">-</span>var1) </span>
<span id="cb20-321"><a href="#cb20-321" aria-hidden="true" tabindex="-1"></a>d2 <span class="ot">&lt;-</span> d2 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">r =</span> value) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb20-322"><a href="#cb20-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-323"><a href="#cb20-323" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> d2, <span class="fu">aes</span>(<span class="at">x =</span> var1, <span class="at">y =</span> var2, <span class="at">fill =</span> r)) <span class="sc">+</span></span>
<span id="cb20-324"><a href="#cb20-324" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_tile</span>(<span class="at">color =</span> <span class="st">"white"</span>)<span class="sc">+</span></span>
<span id="cb20-325"><a href="#cb20-325" aria-hidden="true" tabindex="-1"></a> <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">"maroon1"</span>, <span class="at">high=</span><span class="st">"maroon4"</span>) <span class="sc">+</span></span>
<span id="cb20-326"><a href="#cb20-326" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb20-327"><a href="#cb20-327" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">""</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>), <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)   </span>
<span id="cb20-328"><a href="#cb20-328" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-329"><a href="#cb20-329" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb20-330"><a href="#cb20-330" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-331"><a href="#cb20-331" aria-hidden="true" tabindex="-1"></a>Check Number of SNPs used in Prediction in Rats vs Humans</span>
<span id="cb20-332"><a href="#cb20-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-333"><a href="#cb20-333" aria-hidden="true" tabindex="-1"></a><span class="in">```{r check n.snps}</span></span>
<span id="cb20-334"><a href="#cb20-334" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(rat_r2<span class="sc">$</span>n.snps)</span>
<span id="cb20-335"><a href="#cb20-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-336"><a href="#cb20-336" aria-hidden="true" tabindex="-1"></a><span class="co">#human NAcc models</span></span>
<span id="cb20-337"><a href="#cb20-337" aria-hidden="true" tabindex="-1"></a>filename <span class="ot">&lt;-</span> <span class="st">"/Users/natashasanthanam/Downloads/en_Brain_Nucleus_accumbens_basal_ganglia.db"</span></span>
<span id="cb20-338"><a href="#cb20-338" aria-hidden="true" tabindex="-1"></a>sqlite.driver <span class="ot">&lt;-</span> <span class="fu">dbDriver</span>(<span class="st">"SQLite"</span>)</span>
<span id="cb20-339"><a href="#cb20-339" aria-hidden="true" tabindex="-1"></a>conn <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(), filename)</span>
<span id="cb20-340"><a href="#cb20-340" aria-hidden="true" tabindex="-1"></a>human_r2 <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>(conn, <span class="st">'select * from extra'</span>) </span>
<span id="cb20-341"><a href="#cb20-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-342"><a href="#cb20-342" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(human_r2<span class="sc">$</span>n.snps.in.model)</span>
<span id="cb20-343"><a href="#cb20-343" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>