<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="natasha.santhanam">
<meta name="author" content="Sabrina">
<meta name="dcterms.date" content="2022-02-07">

<title>Rat Genomics Pipeline Rendered - 01.Heritability_Sparsity Sabrina</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-listing/list.min.js"></script>
<script src="../site_libs/quarto-listing/quarto-listing.js"></script>
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script>

  window.document.addEventListener("DOMContentLoaded", function (_event) {
    const listingTargetEl = window.document.querySelector('#listing-listing .list');
    if (!listingTargetEl) {
      // No listing discovered, do not attach.
      return; 
    }

    const options = {
      valueNames: [{ data: ['index'] },{ data: ['categories'] },{ data: ['listing-date-sort'] },{ data: ['listing-file-modified-sort'] }],
      
      searchColumns: [],
    };

    window['quarto-listings'] = window['quarto-listings'] || {};
    window['quarto-listings']['listing-listing'] = new List('listing-listing', options);

    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  });

  window.addEventListener('hashchange',() => {
    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  })
  </script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Rat Genomics Pipeline Rendered</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../other_code.html" rel="" target="">
 <span class="menu-text">more_code</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#generate-heritability-and-sparsity-estimates-for-all-5-tissues" id="toc-generate-heritability-and-sparsity-estimates-for-all-5-tissues" class="nav-link active" data-scroll-target="#generate-heritability-and-sparsity-estimates-for-all-5-tissues">Generate Heritability and Sparsity Estimates for all 5 tissues</a>
  <ul class="collapse">
  <li><a href="#definitions" id="toc-definitions" class="nav-link" data-scroll-target="#definitions">Definitions</a></li>
  <li><a href="#calculate-cis-heritability-within-1mb" id="toc-calculate-cis-heritability-within-1mb" class="nav-link" data-scroll-target="#calculate-cis-heritability-within-1mb">Calculate Cis Heritability within 1MB</a></li>
  </ul></li>
  <li><a href="#calculate-sparsity-estimates-for-all-5-tissues" id="toc-calculate-sparsity-estimates-for-all-5-tissues" class="nav-link" data-scroll-target="#calculate-sparsity-estimates-for-all-5-tissues">Calculate Sparsity Estimates for all 5 tissues</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">01.Heritability_Sparsity Sabrina</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Authors</div>
    <div class="quarto-title-meta-contents">
             <p>natasha.santhanam </p>
             <p>Sabrina </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 7, 2022</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="generate-heritability-and-sparsity-estimates-for-all-5-tissues" class="level1">
<h1>Generate Heritability and Sparsity Estimates for all 5 tissues</h1>
<section id="definitions" class="level2">
<h2 class="anchored" data-anchor-id="definitions">Definitions</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/data/"</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="st">"%&amp;%"</span> <span class="ot">=</span> <span class="cf">function</span>(a,b) <span class="fu">paste</span>(a,b,<span class="at">sep=</span><span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="calculate-cis-heritability-within-1mb" class="level2">
<h2 class="anchored" data-anchor-id="calculate-cis-heritability-within-1mb">Calculate Cis Heritability within 1MB</h2>
<p>For each gene, we calculate heritability from its local GRM. Start with creating list of genes for each of the gene expression file:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"~/Box/imlab-data/Projects/PTRS-PGRS-Rosetta/Data-From-Abe-Palmer-Lab/Rdata/genoGex.RData"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>ensidlist <span class="ot">&lt;-</span> gexAc<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>ensidlist_Il <span class="ot">&lt;-</span> gexIl<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>ensidlist_Lh <span class="ot">&lt;-</span> gexLh<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>ensidlist_Pl <span class="ot">&lt;-</span> gexPl<span class="sc">$</span>EnsemblGeneID</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>ensidlist_Vo <span class="ot">&lt;-</span> gexVo<span class="sc">$</span>EnsemblGeneID</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in bim files for each tissue</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>bimfile <span class="ot">&lt;-</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Ac.bim"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>bimfile_Lh <span class="ot">&lt;-</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Lh.bim"</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>bimfile_Il <span class="ot">&lt;-</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Il.bim"</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>bimfile_Pl <span class="ot">&lt;-</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Pl.bim"</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>bimfile_Vo <span class="ot">&lt;-</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Vo.bim"</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>bim <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimfile)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>bim_Lh <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimfile_Lh)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>bim_Il <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimfile_Il)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>bim_Pl <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimfile_Pl)</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>bim_Vo <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimfile_Vo)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(bim) <span class="ot">&lt;-</span> bim<span class="sc">$</span>V2</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(bim_Lh) <span class="ot">&lt;-</span> bim_Lh<span class="sc">$</span>V2</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(bim_Il) <span class="ot">&lt;-</span> bim_Il<span class="sc">$</span>V2</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(bim_Pl) <span class="ot">&lt;-</span> bim_Pl<span class="sc">$</span>V2</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(bim_Vo) <span class="ot">&lt;-</span> bim_Vo<span class="sc">$</span>V2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We run gcta Ac plink files and gene annotation to generate local GRMs, then h2 calculations.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>gt.dir <span class="ot">&lt;-</span> <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/output/Ac/"</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/"</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>grm.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"GRMs/"</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>h2.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"h2_output/"</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Make local GRMs for each gene</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span>geneinfo[<span class="dv">1</span>]</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bim,bim[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">4</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">4</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">2</span>]    </span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Ac.SNPlist"</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">col.names=</span><span class="cn">FALSE</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>    runGCTAgrm <span class="ot">&lt;-</span> <span class="st">"gcta --bfile "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Ac --make-grm-bin --extract "</span> <span class="sc">%&amp;%</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Ac.SNPlist"</span> <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span>  gene</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGCTAgrm)</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculate h2</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"of"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>    ensid <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">10</span>])</span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">1</span>])</span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>  individuals <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexAc)[<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)]</span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>  expression <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gexAc[i,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)])</span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>    <span class="co">#output expression pheno for gcta</span></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>      <span class="at">famid =</span> individuals,</span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>      <span class="at">id =</span> individuals,</span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a>      <span class="at">expr =</span> expression</span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid, <span class="at">col.names=</span><span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>) <span class="co">#output pheno for gcta</span></span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Y ~ localGRM</span></span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a>    runLOC <span class="ot">&lt;-</span> <span class="st">"gcta --grm "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --reml --pheno "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> h2.dir <span class="sc">%&amp;%</span> <span class="st">"tmp."</span> <span class="sc">%&amp;%</span> ensid</span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runLOC)</span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Repeat for Il:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>gt.dir <span class="ot">&lt;-</span> <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/output/Il/"</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/"</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>grm.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"GRMs/"</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>h2.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"h2_output/"</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Make local GRMs for each gene</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Il)){</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist_Il),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist_Il[i]</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span>geneinfo[<span class="dv">1</span>]</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bim_Il,bim_Il[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">4</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">4</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">2</span>]    </span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Il.SNPlist"</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">col.names=</span><span class="cn">FALSE</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>    runGCTAgrm <span class="ot">&lt;-</span> <span class="st">"gcta --bfile "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Il --make-grm-bin --extract "</span> <span class="sc">%&amp;%</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Il.SNPlist"</span> <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span>  gene</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGCTAgrm)</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculate h2</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Il)){</span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"of"</span>,<span class="fu">length</span>(ensidlist_Il),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>    ensid <span class="ot">&lt;-</span> ensidlist_Il[i]</span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">10</span>])</span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">1</span>])</span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>  individuals <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexIl)[<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)]</span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a>  expression <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gexIl[i,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)])</span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a>    <span class="co">#output expression pheno for gcta</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>      <span class="at">famid =</span> individuals,</span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a>      <span class="at">id =</span> individuals,</span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a>      <span class="at">expr =</span> expression</span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid, <span class="at">col.names=</span><span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>) <span class="co">#output pheno for gcta</span></span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Y ~ localGRM</span></span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a>    runLOC <span class="ot">&lt;-</span> <span class="st">"gcta --grm "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --reml --pheno "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> h2.dir <span class="sc">%&amp;%</span> <span class="st">"tmp."</span> <span class="sc">%&amp;%</span> ensid</span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runLOC)</span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Lh:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>gt.dir <span class="ot">&lt;-</span> <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/output/Lh/"</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/"</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>grm.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"GRMs/"</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>h2.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"h2_output/"</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Make local GRMs for each gene</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Lh)){</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist_Lh),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist_Lh[i]</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span>geneinfo[<span class="dv">1</span>]</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bim_Lh,bim_Lh[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">4</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">4</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">2</span>]    </span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Lh.SNPlist"</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">col.names=</span><span class="cn">FALSE</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>    runGCTAgrm <span class="ot">&lt;-</span> <span class="st">"gcta --bfile "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Lh --make-grm-bin --extract "</span> <span class="sc">%&amp;%</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Lh.SNPlist"</span> <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span>  gene</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGCTAgrm)</span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculate h2</span></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Lh)){</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"of"</span>,<span class="fu">length</span>(ensidlist_Lh),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>    ensid <span class="ot">&lt;-</span> ensidlist_Lh[i]</span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">10</span>])</span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">1</span>])</span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>  individuals <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexLh)[<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)]</span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>  expression <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gexLh[i,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)])</span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a>    <span class="co">#output expression pheno for gcta</span></span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a>      <span class="at">famid =</span> individuals,</span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a>      <span class="at">id =</span> individuals,</span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a>      <span class="at">expr =</span> expression</span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid, <span class="at">col.names=</span><span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>) <span class="co">#output pheno for gcta</span></span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Y ~ localGRM</span></span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a>    runLOC <span class="ot">&lt;-</span> <span class="st">"gcta --grm "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --reml --pheno "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> h2.dir <span class="sc">%&amp;%</span> <span class="st">"tmp."</span> <span class="sc">%&amp;%</span> ensid</span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runLOC)</span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Pl:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>gt.dir <span class="ot">&lt;-</span> <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/output/Pl/"</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/"</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>grm.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"GRMs/"</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>h2.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"h2_output/"</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Make local GRMs for each gene</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Pl)){</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist_Pl),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist_Pl[i]</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span>geneinfo[<span class="dv">1</span>]</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bim_Pl,bim_Pl[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">4</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">4</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">2</span>]    </span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Pl.SNPlist"</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">col.names=</span><span class="cn">FALSE</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>    runGCTAgrm <span class="ot">&lt;-</span> <span class="st">"gcta --bfile "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Pl --make-grm-bin --extract "</span> <span class="sc">%&amp;%</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Pl.SNPlist"</span> <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span>  gene</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGCTAgrm)</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculate h2</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Pl)){</span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"of"</span>,<span class="fu">length</span>(ensidlist_Pl),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>    ensid <span class="ot">&lt;-</span> ensidlist_Pl[i]</span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">10</span>])</span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">1</span>])</span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>  individuals <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexPl)[<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)]</span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>  expression <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gexPl[i,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)])</span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>    <span class="co">#output expression pheno for gcta</span></span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a>      <span class="at">famid =</span> individuals,</span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a>      <span class="at">id =</span> individuals,</span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>      <span class="at">expr =</span> expression</span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid, <span class="at">col.names=</span><span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>) <span class="co">#output pheno for gcta</span></span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Y ~ localGRM</span></span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a>    runLOC <span class="ot">&lt;-</span> <span class="st">"gcta --grm "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --reml --pheno "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> h2.dir <span class="sc">%&amp;%</span> <span class="st">"tmp."</span> <span class="sc">%&amp;%</span> ensid</span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runLOC)</span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Vo</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>gt.dir <span class="ot">&lt;-</span> <span class="st">"/Users/sabrinami/Github/Rat_Genomics_Paper_Pipeline/output/Vo/"</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"phenotype_files/"</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>grm.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"GRMs/"</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>h2.dir <span class="ot">&lt;-</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"h2_output/"</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Make local GRMs for each gene</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Vo)){</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist_Vo),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist_Vo[i]</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span>geneinfo[<span class="dv">1</span>]</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bim_Vo,bim_Vo[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">4</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">4</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">2</span>]    </span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Vo.SNPlist"</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">col.names=</span><span class="cn">FALSE</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>    runGCTAgrm <span class="ot">&lt;-</span> <span class="st">"gcta --bfile "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"rat_genome_Vo --make-grm-bin --extract "</span> <span class="sc">%&amp;%</span> gt.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Vo.SNPlist"</span> <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span>  gene</span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGCTAgrm)</span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculate h2</span></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist_Vo)){</span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"of"</span>,<span class="fu">length</span>(ensidlist_Vo),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a>    ensid <span class="ot">&lt;-</span> ensidlist_Vo[i]</span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">10</span>])</span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gtf[<span class="fu">match</span>(ensid, gtf<span class="sc">$</span>Gene),<span class="dv">1</span>])</span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a>  individuals <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gexVo)[<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)]</span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a>  expression <span class="ot">&lt;-</span> <span class="fu">as.character</span>(gexVo[i,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)])</span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a>    <span class="co">#output expression pheno for gcta</span></span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a>      <span class="at">famid =</span> individuals,</span>
<span id="cb8-34"><a href="#cb8-34" aria-hidden="true" tabindex="-1"></a>      <span class="at">id =</span> individuals,</span>
<span id="cb8-35"><a href="#cb8-35" aria-hidden="true" tabindex="-1"></a>      <span class="at">expr =</span> expression</span>
<span id="cb8-36"><a href="#cb8-36" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb8-37"><a href="#cb8-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid, <span class="at">col.names=</span><span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>) <span class="co">#output pheno for gcta</span></span>
<span id="cb8-38"><a href="#cb8-38" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Y ~ localGRM</span></span>
<span id="cb8-39"><a href="#cb8-39" aria-hidden="true" tabindex="-1"></a>    runLOC <span class="ot">&lt;-</span> <span class="st">"gcta --grm "</span> <span class="sc">%&amp;%</span> grm.dir <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --reml --pheno "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> ensid <span class="sc">%&amp;%</span> <span class="st">" --out "</span> <span class="sc">%&amp;%</span> h2.dir <span class="sc">%&amp;%</span> <span class="st">"tmp."</span> <span class="sc">%&amp;%</span> ensid</span>
<span id="cb8-40"><a href="#cb8-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runLOC)</span>
<span id="cb8-41"><a href="#cb8-41" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="calculate-sparsity-estimates-for-all-5-tissues" class="level1">
<h1>Calculate Sparsity Estimates for all 5 tissues</h1>
<p>The following code uses <code>bim_bam</code> files generated through Palmer lab data, steps <a href="convert_genotypes_old.Rmd">here</a>. The code</p>
<p>Ac</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/phenotype_files/"</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/genotype_files/"</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/bim_bam/"</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Ac_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Ac/"</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Ac.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexAc_transpose[,i])</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Ac.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Ac_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Il</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/phenotype_files/"</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/genotype_files/"</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/bim_bam/"</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Il_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Il/"</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr </span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]    </span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Il.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexIl_transpose[,i])</span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Il.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Il_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Lh</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/phenotype_files/"</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/genotype_files/"</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/bim_bam/"</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Lh_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Lh/"</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]    </span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Lh.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexLh_transpose[,i])</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Lh.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Lh_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Pl</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/phenotype_files/"</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/genotype_files/"</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/bim_bam/"</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Pl_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Pl/"</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]    </span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Pl.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexPl_transpose[,i])</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Pl.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Pl_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Vo</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>pheno.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/phenotype_files/"</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>ge.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/genotype_files/"</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>bim.dir <span class="ot">&lt;-</span> <span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/bim_bam/"</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in bimbam file </span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>bimbamfile <span class="ot">&lt;-</span> bim.dir <span class="sc">%&amp;%</span> <span class="st">"Vo_bimbam"</span> <span class="do">###get SNP position information###</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>bimbam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(bimbamfile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/GEMMA/Vo/"</span>)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(ensidlist)){</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(i,<span class="st">"/"</span>,<span class="fu">length</span>(ensidlist),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>    gene <span class="ot">&lt;-</span> ensidlist[i]</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>    geneinfo <span class="ot">&lt;-</span> gtf[<span class="fu">match</span>(gene, gtf<span class="sc">$</span>Gene),]</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> geneinfo[<span class="dv">1</span>]</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>    c <span class="ot">&lt;-</span> chr<span class="sc">$</span>Chr</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>Start <span class="sc">-</span> <span class="fl">1e6</span> <span class="do">### 1Mb lower bound for cis-eQTLS</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> geneinfo<span class="sc">$</span>End <span class="sc">+</span> <span class="fl">1e6</span> <span class="do">### 1Mb upper bound for cis-eQTLs</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>    chrsnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(bimbam, bimbam[,<span class="dv">1</span>]<span class="sc">==</span>c) <span class="do">### pull snps on same chr</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>    cissnps <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrsnps,chrsnps[,<span class="dv">2</span>]<span class="sc">&gt;=</span>start <span class="sc">&amp;</span> chrsnps[,<span class="dv">2</span>]<span class="sc">&lt;=</span>end) <span class="do">### pull cis-SNP info</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>    snplist <span class="ot">&lt;-</span> cissnps[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(cissnps)]    </span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(snplist, <span class="at">file=</span> ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Vo.geno"</span> <span class="sc">%&amp;%</span> gene, <span class="at">quote=</span>F,<span class="at">col.names=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>    geneexp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gexVo_transpose[,i])</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(geneexp, <span class="at">file=</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span> gene, <span class="at">col.names=</span>F, <span class="at">row.names =</span> F, <span class="at">quote=</span>F) <span class="co">#output pheno for gemma</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>    runGEMMAgrm <span class="ot">&lt;-</span> <span class="st">"gemma -g "</span> <span class="sc">%&amp;%</span>  ge.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.Vo.geno"</span> <span class="sc">%&amp;%</span> gene <span class="sc">%&amp;%</span> <span class="st">" -p "</span> <span class="sc">%&amp;%</span> pheno.dir <span class="sc">%&amp;%</span> <span class="st">"tmp.pheno."</span> <span class="sc">%&amp;%</span>  gene  <span class="sc">%&amp;%</span>  <span class="st">" -gk -o grm_Vo_"</span> <span class="sc">%&amp;%</span> gene</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system</span>(runGEMMAgrm)</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>



</section>

<div class="quarto-listing quarto-listing-container-default" id="listing-listing">
<div class="list quarto-listing-default">

</div>
<div class="listing-no-matching d-none">
No matching items
</div>
</div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>