<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Haky Im">
<meta name="dcterms.date" content="2023-11-27">
<meta name="description" content="tutorial for running ratxcan given genotype, phenotype, and prediction weights as input">

<title>Rat Genomics Pipeline Rendered - RatXcan analysis Figure 3 to 4</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-listing/list.min.js"></script>
<script src="../site_libs/quarto-listing/quarto-listing.js"></script>
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script>

  window.document.addEventListener("DOMContentLoaded", function (_event) {
    const listingTargetEl = window.document.querySelector('#listing-listing .list');
    if (!listingTargetEl) {
      // No listing discovered, do not attach.
      return; 
    }

    const options = {
      valueNames: [{ data: ['index'] },{ data: ['categories'] },{ data: ['listing-date-sort'] },{ data: ['listing-file-modified-sort'] }],
      
      searchColumns: [],
    };

    window['quarto-listings'] = window['quarto-listings'] || {};
    window['quarto-listings']['listing-listing'] = new List('listing-listing', options);

    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  });

  window.addEventListener('hashchange',() => {
    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  })
  </script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Rat Genomics Pipeline Rendered</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html" rel="" target="">
 <span class="menu-text">paper_code</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../other_code.html" rel="" target="">
 <span class="menu-text">more_code</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#top" id="toc-top" class="nav-link active" data-scroll-target="#top">top</a></li>
  <li><a href="#input-data" id="toc-input-data" class="nav-link" data-scroll-target="#input-data">input data</a></li>
  <li><a href="#load-libraries-and-functions" id="toc-load-libraries-and-functions" class="nav-link" data-scroll-target="#load-libraries-and-functions">load libraries and functions</a></li>
  <li><a href="#download-data-for-the-tutorial" id="toc-download-data-for-the-tutorial" class="nav-link" data-scroll-target="#download-data-for-the-tutorial">download data for the tutorial</a></li>
  <li><a href="#define-data-and-software-paths-for-r" id="toc-define-data-and-software-paths-for-r" class="nav-link" data-scroll-target="#define-data-and-software-paths-for-r">define data and software paths for R</a></li>
  <li><a href="#define-data-and-software-for-the-terminal" id="toc-define-data-and-software-for-the-terminal" class="nav-link" data-scroll-target="#define-data-and-software-for-the-terminal">define data and software for the terminal</a></li>
  <li><a href="#set-up-conda-environment-to-run-predict-expression" id="toc-set-up-conda-environment-to-run-predict-expression" class="nav-link" data-scroll-target="#set-up-conda-environment-to-run-predict-expression">set up conda environment to run predict expression</a></li>
  <li><a href="#predict-expression" id="toc-predict-expression" class="nav-link" data-scroll-target="#predict-expression">predict expression</a></li>
  <li><a href="#compare-with-previous-prediction-of-ac" id="toc-compare-with-previous-prediction-of-ac" class="nav-link" data-scroll-target="#compare-with-previous-prediction-of-ac">compare with previous prediction of Ac</a></li>
  <li><a href="#read-phenotype-and-check-that-ids-are-included-in-genotype-fam-file" id="toc-read-phenotype-and-check-that-ids-are-included-in-genotype-fam-file" class="nav-link" data-scroll-target="#read-phenotype-and-check-that-ids-are-included-in-genotype-fam-file">read phenotype and check that ids are included in genotype fam file</a></li>
  <li><a href="#read-gene-annotation" id="toc-read-gene-annotation" class="nav-link" data-scroll-target="#read-gene-annotation">read gene annotation</a></li>
  <li><a href="#calculate-grm" id="toc-calculate-grm" class="nav-link" data-scroll-target="#calculate-grm">calculate GRM</a></li>
  <li><a href="#calculate-h2" id="toc-calculate-h2" class="nav-link" data-scroll-target="#calculate-h2">calculate h2</a></li>
  <li><a href="#read-h2" id="toc-read-h2" class="nav-link" data-scroll-target="#read-h2">read h2</a></li>
  <li><a href="#read-grm-matrix" id="toc-read-grm-matrix" class="nav-link" data-scroll-target="#read-grm-matrix">read grm matrix</a></li>
  <li><a href="#read-predicted-expression" id="toc-read-predicted-expression" class="nav-link" data-scroll-target="#read-predicted-expression">read predicted expression</a></li>
  <li><a href="#show-calibration-of-type-i-error-remove-from-tutorial" id="toc-show-calibration-of-type-i-error-remove-from-tutorial" class="nav-link" data-scroll-target="#show-calibration-of-type-i-error-remove-from-tutorial">show calibration of type I error REMOVE FROM TUTORIAL</a>
  <ul class="collapse">
  <li><a href="#simulate-genetic-effect-sizes-for-nsnp" id="toc-simulate-genetic-effect-sizes-for-nsnp" class="nav-link" data-scroll-target="#simulate-genetic-effect-sizes-for-nsnp">simulate genetic effect sizes for nsnp</a></li>
  <li><a href="#use-plink-to-calculate-the-sim-phenotype-y_-_k-x_k" id="toc-use-plink-to-calculate-the-sim-phenotype-y_-_k-x_k" class="nav-link" data-scroll-target="#use-plink-to-calculate-the-sim-phenotype-y_-_k-x_k">use plink to calculate the sim phenotype $y_ = _k X_k $</a></li>
  <li><a href="#format-simulated-scores-into-matrix" id="toc-format-simulated-scores-into-matrix" class="nav-link" data-scroll-target="#format-simulated-scores-into-matrix">format simulated scores into matrix</a></li>
  <li><a href="#visualize-raw-and-corrected-pvalues" id="toc-visualize-raw-and-corrected-pvalues" class="nav-link" data-scroll-target="#visualize-raw-and-corrected-pvalues">visualize raw and corrected pvalues</a></li>
  </ul></li>
  <li><a href="#define-function-lmm-with-grm" id="toc-define-function-lmm-with-grm" class="nav-link" data-scroll-target="#define-function-lmm-with-grm">define function lmm with GRM</a>
  <ul class="collapse">
  <li><a href="#run-ratxcan-null-regression" id="toc-run-ratxcan-null-regression" class="nav-link" data-scroll-target="#run-ratxcan-null-regression">run ratxcan null regression</a></li>
  <li><a href="#simulate-y-sigma12epsilon-and-run-assoc-with-expr_mat" id="toc-simulate-y-sigma12epsilon-and-run-assoc-with-expr_mat" class="nav-link" data-scroll-target="#simulate-y-sigma12epsilon-and-run-assoc-with-expr_mat">simulate <span class="math inline">\(Y = Sigma^{1/2}\epsilon\)</span> and run assoc with expr_mat</a></li>
  <li><a href="#simulate-y-grm12-h2-1---h2-epsilon-and-run-assoc-with-expr_mat" id="toc-simulate-y-grm12-h2-1---h2-epsilon-and-run-assoc-with-expr_mat" class="nav-link" data-scroll-target="#simulate-y-grm12-h2-1---h2-epsilon-and-run-assoc-with-expr_mat">simulate <span class="math inline">\(Y = GRM^{1/2} h2 + (1 - h2) \epsilon\)</span> and run assoc with expr_mat</a></li>
  <li><a href="#run-bodylen-regression-with-br-expr-lmmgrm" id="toc-run-bodylen-regression-with-br-expr-lmmgrm" class="nav-link" data-scroll-target="#run-bodylen-regression-with-br-expr-lmmgrm">run bodylen regression with BR expr lmmGRM</a></li>
  <li><a href="#run-bmi-regression-with-lmmgrm" id="toc-run-bmi-regression-with-lmmgrm" class="nav-link" data-scroll-target="#run-bmi-regression-with-lmmgrm">run bmi regression with lmmGRM</a></li>
  <li><a href="#bodylen-and-bmi-vs-5-brain-region-expr-association" id="toc-bodylen-and-bmi-vs-5-brain-region-expr-association" class="nav-link" data-scroll-target="#bodylen-and-bmi-vs-5-brain-region-expr-association">bodylen and bmi vs 5 brain region expr association</a></li>
  <li><a href="#download-rat-gene-annotation-and-human-phenomexcan" id="toc-download-rat-gene-annotation-and-human-phenomexcan" class="nav-link" data-scroll-target="#download-rat-gene-annotation-and-human-phenomexcan">download rat gene annotation and human phenomexcan</a></li>
  </ul></li>
  <li><a href="#create-ratxcan-association-table-to-save-as-csv-files" id="toc-create-ratxcan-association-table-to-save-as-csv-files" class="nav-link" data-scroll-target="#create-ratxcan-association-table-to-save-as-csv-files">create ratxcan association table to save as csv files</a></li>
  <li><a href="#collect-signs" id="toc-collect-signs" class="nav-link" data-scroll-target="#collect-signs">collect signs</a></li>
  <li><a href="#enrichment-of-human-gene-analysis" id="toc-enrichment-of-human-gene-analysis" class="nav-link" data-scroll-target="#enrichment-of-human-gene-analysis">enrichment of human gene analysis</a></li>
  <li><a href="#enrichment-of-crouses-consensus-genes" id="toc-enrichment-of-crouses-consensus-genes" class="nav-link" data-scroll-target="#enrichment-of-crouses-consensus-genes">enrichment of Crouse’s consensus genes</a></li>
  <li><a href="#summarize-results-for-the-paper" id="toc-summarize-results-for-the-paper" class="nav-link" data-scroll-target="#summarize-results-for-the-paper">summarize results for the paper</a></li>
  <li><a href="#gene-set-enrichment" id="toc-gene-set-enrichment" class="nav-link" data-scroll-target="#gene-set-enrichment">gene set enrichment</a></li>
  <li><a href="#define-ggplot-based-manhattan-plotting-function" id="toc-define-ggplot-based-manhattan-plotting-function" class="nav-link" data-scroll-target="#define-ggplot-based-manhattan-plotting-function">define ggplot based manhattan plotting function</a></li>
  <li><a href="#plot-manhattan-plots-of-bmi-and-body-length" id="toc-plot-manhattan-plots-of-bmi-and-body-length" class="nav-link" data-scroll-target="#plot-manhattan-plots-of-bmi-and-body-length">plot manhattan plots of bmi and body length</a></li>
  <li><a href="#double-checking-that-the-saved-qvalues-are-correct" id="toc-double-checking-that-the-saved-qvalues-are-correct" class="nav-link" data-scroll-target="#double-checking-that-the-saved-qvalues-are-correct">double checking that the saved qvalues are correct</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">RatXcan analysis Figure 3 to 4</h1>
</div>

<div>
  <div class="description">
    tutorial for running ratxcan given genotype, phenotype, and prediction weights as input
  </div>
</div>


<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Haky Im </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">November 27, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="top" class="level2">
<h2 class="anchored" data-anchor-id="top">top</h2>
</section>
<section id="input-data" class="level2">
<h2 class="anchored" data-anchor-id="input-data">input data</h2>
<ul>
<li>genotype</li>
<li>phenotype</li>
<li>prediction weights</li>
</ul>
</section>
<section id="load-libraries-and-functions" class="level2">
<h2 class="anchored" data-anchor-id="load-libraries-and-functions">load libraries and functions</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#options(error=recover)</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co">#options(error=browser)</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">error=</span><span class="cn">NULL</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="do">## compare observed correlation with null correlation</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(devtools<span class="sc">::</span><span class="fu">source_gist</span>(<span class="st">"a925fea01b365a8c605e"</span>)) <span class="do">## load qqR fn https://gist.github.com/hakyim/a925fea01b365a8c605e</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(devtools<span class="sc">::</span><span class="fu">source_gist</span>(<span class="st">"38431b74c6c0bf90c12f"</span>)) <span class="do">## qqunif https://gist.github.com/hakyim/38431b74c6c0bf90c12f</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(devtools<span class="sc">::</span><span class="fu">source_gist</span>(<span class="st">"115403f16bec0a0e871f3616d552ce9b"</span>)) <span class="do">## source ratxcan functions https://gist.github.com/hakyim/115403f16bec0a0e871f3616d552ce9b </span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(glue))</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(RSQLite))</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="co">#suppressMessages(library(expm))</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="co">#suppressMessages(library(readxl))</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages("devtools")</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="co"># library("devtools")</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="co"># install_github("jdstorey/qvalue")</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(qvalue))</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="co"># if (!require("BiocManager", quietly = TRUE))</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="co">#     install.packages("BiocManager")</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("biomaRt")</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="do">##suppressMessages(library(biomaRt))</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="do">##install.packages("ggrepel")</span></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(ggrepel))</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>traitlist <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"bodylen"</span>,<span class="st">"bmi"</span>)</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>tissuelist <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"AC"</span>, <span class="st">"IL"</span>, <span class="st">"LH"</span>, <span class="st">"PL"</span>, <span class="st">"VO"</span>)</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>phenolist <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Standing height"</span>,<span class="st">"Body mass index (BMI) (21001_raw)"</span>)</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(phenolist) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"bodylen"</span>,<span class="st">"bmi"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="download-data-for-the-tutorial" class="level2">
<h2 class="anchored" data-anchor-id="download-data-for-the-tutorial">download data for the tutorial</h2>
<ul class="task-list">
<li><input type="checkbox">download files from <a href="https://uchicago.box.com/v/ratxcan-tutorial">here</a> The folder structure should look like the list below</li>
</ul>
<div class="cell">
<pre class="text cell-code"><code>(base) MBP-HKI-22 ratxcan-tutorial $ tree -L 3
.
├── data
│  ├── expression
│   │   └── gene_annotation.RDS
│   ├── genotype
│   │   ├── DELETE
│   │   ├── rat6k_autosome.bed
│   │   ├── rat6k_autosome.bim
│   │   └── rat6k_autosome.fam
│   ├── phenomexcan
│   │   └── phenomexcan_results.RDS
│   └── phenotype
│       └── pheno.fam
├── models
│   ├── Ac_best_prediXcan_db.db
│   ├── Br-hki.db
│   ├── Il_best_prediXcan_db.db
│   ├── Lh_best_prediXcan_db.db
│   ├── Pl_best_prediXcan_db.db
│   └── Vo_best_prediXcan_db.db
└── software
    └── MetaXcan
        ├── CODE_OF_CONDUCT.md
        ├── DevNotes.Rmd
        ├── LICENSE
        ├── README.md
        ├── codemap
        ├── papers
        └── software</code></pre>
</div>
</section>
<section id="define-data-and-software-paths-for-r" class="level2">
<h2 class="anchored" data-anchor-id="define-data-and-software-paths-for-r">define data and software paths for R</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>WEBDATA<span class="ot">=</span><span class="st">"/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data"</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>PLINK<span class="ot">=</span><span class="st">"/Users/haekyungim/bin/plink_mac_20231211/plink"</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>GCTA<span class="ot">=</span><span class="st">"/Users/haekyungim/bin/gcta-1.94.2-MacOS-ARM-x86_64/gcta64"</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>INPUT <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"{WEBDATA}/ratxcan-tutorial"</span>) <span class="do">## this has the input data </span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>OUTPUT <span class="ot">&lt;-</span> <span class="fu">glue</span>(<span class="st">"{WEBDATA}/2023-11-27-ratxcan-tutorial/scratch"</span>) <span class="do">## this has the output data, intermediate results</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="define-data-and-software-for-the-terminal" class="level2">
<h2 class="anchored" data-anchor-id="define-data-and-software-for-the-terminal">define data and software for the terminal</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="va">WEBDATA</span><span class="op">=</span><span class="st">"/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data"</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="va">PLINK</span><span class="op">=</span><span class="st">"/Users/haekyungim/bin/plink_mac_20231211/plink"</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="va">GCTA</span><span class="op">=</span><span class="st">"/Users/haekyungim/bin/gcta-1.94.2-MacOS-ARM-x86_64/gcta64"</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="va">INPUT</span><span class="op">=</span><span class="va">$WEBDATA</span>/ratxcan-tutorial</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="va">OUTPUT</span><span class="op">=</span><span class="va">$WEBDATA</span>/2023-11-27-ratxcan-tutorial/scratch</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="set-up-conda-environment-to-run-predict-expression" class="level2">
<h2 class="anchored" data-anchor-id="set-up-conda-environment-to-run-predict-expression">set up conda environment to run predict expression</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> create <span class="at">-n</span> rat311 python=3.11</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate rat311</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> install pandas scipy numpy statsmodels</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> install h5py</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co">## missing modules probably due to M2 processor</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="co">## NOT NEEDED conda install bgen_reader</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="ex">pip</span> install pyliftover</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="ex">pip</span> install cyvcf2  <span class="co">## pip intall worked on my macbook M2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="predict-expression" class="level2">
<h2 class="anchored" data-anchor-id="predict-expression">predict expression</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co">#WEBDATA="/Users/haekyungim/Library/CloudStorage/Box-Box/LargeFiles/imlab-data/data-Github/web-data"</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co">#PLINK="/Users/haekyungim/bin/plink_mac_20231211/plink"</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co">#GCTA="/Users/haekyungim/bin/gcta-1.94.2-MacOS-ARM-x86_64/gcta64"</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co">#INPUT=$WEBDATA/ratxcan-tutorial</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="co">#OUTPUT=$WEBDATA/2023-11-27-ratxcan-tutorial/scratch</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="co">## convert plink file to vcf</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="co">#$PLINK --bfile $GENO_HEADER --recode vcf --out $GENO_HEADER</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="co">#gzip $GENO_HEADER.vcf ## to keep file small</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate rat311</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="va">METAXCAN</span><span class="op">=</span><span class="va">$INPUT</span>/software/MetaXcan</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a><span class="va">GENO</span><span class="op">=</span><span class="va">$INPUT</span>/data/genotype</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="va">GENO_HEADER</span><span class="op">=</span><span class="va">$GENO</span>/rat6k_autosome</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a><span class="va">MODEL</span><span class="op">=</span><span class="va">$INPUT</span>/models</span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a><span class="co">## predict Ac</span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a><span class="va">MODLEFT</span><span class="op">=</span><span class="st">"VO-filtered"</span></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a><span class="va">MODEL_DB</span><span class="op">=</span><span class="st">"</span><span class="va">$MODLEFT</span><span class="st">.db"</span></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a><span class="ex">python</span> <span class="va">$METAXCAN</span>/software/Predict.py <span class="dt">\</span></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>--model_db_path <span class="va">$MODEL</span>/<span class="va">$MODEL_DB</span> <span class="dt">\</span></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>--model_db_snp_key rsid <span class="dt">\</span></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>--vcf_genotypes <span class="va">${GENO_HEADER}</span>.vcf.gz <span class="dt">\</span></span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>--vcf_mode genotyped <span class="dt">\</span></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>--on_the_fly_mapping METADATA <span class="st">"{}_{}_{}_{}"</span> <span class="dt">\</span></span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>--prediction_output <span class="va">$OUTPUT</span>/<span class="va">${MODLEFT}</span>__predict.txt  <span class="dt">\</span></span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a>--prediction_summary_output <span class="va">$OUTPUT</span>/<span class="va">${MODLEFT}</span>__summary.txt <span class="dt">\</span></span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>--throw</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="compare-with-previous-prediction-of-ac" class="level2">
<h2 class="anchored" data-anchor-id="compare-with-previous-prediction-of-ac">compare with previous prediction of Ac</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>oldexpr <span class="ot">&lt;-</span> vroom<span class="sc">::</span><span class="fu">vroom</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/Ac-hki-large_geno__predict.txt"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>FID) <span class="sc">%&gt;%</span>  <span class="co"># Remove the FID column</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">IID =</span> <span class="fu">str_split</span>(IID, <span class="st">"_"</span>, <span class="at">simplify =</span> <span class="cn">TRUE</span>)[, <span class="dv">1</span>])  <span class="co"># Keep the first part of IID</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(tis <span class="cf">in</span> tissuelist)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  new_pred_expr <span class="ot">&lt;-</span> vroom<span class="sc">::</span><span class="fu">vroom</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/{tis}-filtered__predict.txt"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>FID) <span class="sc">%&gt;%</span>  <span class="co"># Remove the FID column</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">IID =</span> <span class="fu">str_split</span>(IID, <span class="st">"_"</span>, <span class="at">simplify =</span> <span class="cn">TRUE</span>)[, <span class="dv">1</span>])  <span class="co"># Keep the first part of IID</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>  kk<span class="ot">=</span><span class="fu">calc_cor_matched_cols</span>(oldexpr,new_pred_expr)</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hist</span>(kk<span class="sc">$</span>cor,<span class="at">main=</span><span class="fu">paste</span>(tis, <span class="st">"vs. old Ac-hki-large_geno"</span>) )</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mtext</span>(<span class="fu">paste</span>(<span class="fu">sum</span>(kk<span class="sc">$</span>cor<span class="sc">&lt;</span><span class="dv">0</span>), <span class="st">"genes with negative cor"</span>) )</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="read-phenotype-and-check-that-ids-are-included-in-genotype-fam-file" class="level2">
<h2 class="anchored" data-anchor-id="read-phenotype-and-check-that-ids-are-included-in-genotype-fam-file">read phenotype and check that ids are included in genotype fam file</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="do">## DO EVAL #| eval: FALSE</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="do">## check whether ids in the phenotype file are in the genotype file</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="fu">glue</span>(<span class="st">"{INPUT}/data/phenotype/pheno.fam"</span>),<span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>fam <span class="ot">=</span> <span class="fu">read_table</span>(<span class="fu">glue</span>(<span class="st">"{INPUT}/data/genotype/rat6k_autosome.fam"</span>),<span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>( pheno <span class="sc">%&gt;%</span> <span class="fu">filter</span>(X1 <span class="sc">%in%</span> fam<span class="sc">$</span>X1) <span class="sc">%&gt;%</span> nrow <span class="sc">!=</span> <span class="fu">nrow</span>(pheno) ) <span class="fu">message</span>(<span class="st">"WARNING: missing pheno ids with missing genotypes"</span>) </span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(pheno) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"FID"</span>,<span class="st">"IID"</span>,<span class="st">"bodylen"</span>,<span class="st">"bmi"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="read-gene-annotation" class="level2">
<h2 class="anchored" data-anchor-id="read-gene-annotation">read gene annotation</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>gene_annotation <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">glue</span>(<span class="st">"{INPUT}/data/expression/gene_annotation.RDS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="calculate-grm" class="level2">
<h2 class="anchored" data-anchor-id="calculate-grm">calculate GRM</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co">## - [ ] calculate GRM from genotype</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="co">## GCTA="/Users/haekyungim/bin/gcta-1.94.2-MacOS-ARM-x86_64/gcta64" ## defined at the top</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="va">$GCTA</span> <span class="at">--bfile</span> <span class="va">$GENO0</span>/rat6k_autosome <span class="at">--make-grm-bin</span> <span class="at">--out</span> <span class="va">$OUTPUT</span>/rat6k_autosome</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Analysis finished at 00:31:49 CST on Fri Dec 01 2023</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Overall computational time: 1 minute 15 sec</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="calculate-h2" class="level2">
<h2 class="anchored" data-anchor-id="calculate-h2">calculate h2</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co">## 1 after --mpheno will use bodylen as phenotype</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="va">$GCTA</span> <span class="at">--grm</span> <span class="va">$OUTPUT</span>/rat6k_autosome <span class="at">--reml</span> <span class="at">--pheno</span> <span class="va">$INPUT</span>/data/phenotype/pheno.fam <span class="at">--mpheno</span> 1 <span class="at">--out</span> <span class="va">$OUTPUT</span>/bodylen_h2</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co">## 2 after --mpheno will use bmi as phenotype</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="va">$GCTA</span> <span class="at">--grm</span> <span class="va">$OUTPUT</span>/rat6k_autosome <span class="at">--reml</span> <span class="at">--pheno</span> <span class="va">$INPUT</span>/data/phenotype/pheno.fam <span class="at">--mpheno</span> 2 <span class="at">--out</span> <span class="va">$OUTPUT</span>/bmi_h2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="read-h2" class="level2">
<h2 class="anchored" data-anchor-id="read-h2">read h2</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>tempo <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/bodylen_h2.hsq"</span>)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Source<span class="sc">==</span><span class="st">"V(G)/Vp"</span>) </span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>bodylen_h2 <span class="ot">=</span> tempo <span class="sc">%&gt;%</span> <span class="fu">pull</span>(Variance)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>bodylen_se <span class="ot">=</span> tempo <span class="sc">%&gt;%</span> <span class="fu">pull</span>(SE)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>tempo <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/bmi_h2.hsq"</span>)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Source<span class="sc">==</span><span class="st">"V(G)/Vp"</span>) </span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>bmi_h2 <span class="ot">=</span> tempo <span class="sc">%&gt;%</span> <span class="fu">pull</span>(Variance)</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>bmi_se <span class="ot">=</span> tempo <span class="sc">%&gt;%</span> <span class="fu">pull</span>(SE)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="read-grm-matrix" class="level2">
<h2 class="anchored" data-anchor-id="read-grm-matrix">read grm matrix</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>grm_mat <span class="ot">&lt;-</span> <span class="fu">read_GRMBin</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/rat6k_autosome.grm"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="read-predicted-expression" class="level2">
<h2 class="anchored" data-anchor-id="read-predicted-expression">read predicted expression</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>read_pred_expr <span class="ot">=</span> <span class="cf">function</span>(filename)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="do">##usage: Br_pred_expr = read_pred_expr(glue("{OUTPUT}/Br-hki-rat6k__predict.txt"))</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  pred_expr <span class="ot">&lt;-</span> vroom<span class="sc">::</span><span class="fu">vroom</span>(filename) <span class="sc">%&gt;%</span> </span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>FID) <span class="sc">%&gt;%</span>  <span class="co"># Remove the FID column</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">IID =</span> <span class="fu">str_split</span>(IID, <span class="st">"_"</span>, <span class="at">simplify =</span> <span class="cn">TRUE</span>)[, <span class="dv">1</span>])  <span class="co"># Keep the first part of IID</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  pred_expr</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="show-calibration-of-type-i-error-remove-from-tutorial" class="level2">
<h2 class="anchored" data-anchor-id="show-calibration-of-type-i-error-remove-from-tutorial">show calibration of type I error REMOVE FROM TUTORIAL</h2>
<section id="simulate-genetic-effect-sizes-for-nsnp" class="level3">
<h3 class="anchored" data-anchor-id="simulate-genetic-effect-sizes-for-nsnp">simulate genetic effect sizes for nsnp</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="do">## on the terminal generate</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="do">## $PLINK --bfile $INPUT/data/genotype/rat6k_autosome --freq --out rat6k_autosome </span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="do">## simulate unrelated Y</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>nsim <span class="ot">=</span> <span class="dv">100</span> </span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="do">## to get nsip, read bim file</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="co">#tempo = read_tsv(glue("{INPUT}/data/genotype/rat6k_autosome.bim"),col_names = FALSE)</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>tempo <span class="ot">=</span> <span class="fu">read_table</span>(<span class="fu">glue</span>(<span class="st">"{INPUT}/data/genotype/rat6k_autosome.frq"</span>),<span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>tempo <span class="ot">=</span> tempo <span class="sc">%&gt;%</span> <span class="fu">select</span>(SNP,A1,A2,MAF)</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>nsnp <span class="ot">=</span> <span class="fu">nrow</span>(df_freq)</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a><span class="do">##set.seed(29444)</span></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>semilla <span class="ot">=</span> <span class="dv">32240</span></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a><span class="do">##semilla = round(runif(1)*1e5)</span></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(semilla)</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>scoremat <span class="ot">=</span> <span class="fu">matrix</span>(<span class="fu">rnorm</span>(nsnp<span class="sc">*</span>nsim),nsnp,nsim)</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a><span class="do">## divide by maf</span></span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>scoremat <span class="ot">=</span> <span class="fu">sweep</span>(scoremat,<span class="dv">1</span>, <span class="fu">sqrt</span>(<span class="dv">2</span><span class="sc">*</span>tempo<span class="sc">$</span>MAF<span class="sc">*</span>(<span class="dv">1</span><span class="sc">-</span>tempo<span class="sc">$</span>MAF)), <span class="st">"/"</span> )</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>tempo <span class="ot">=</span> <span class="fu">cbind</span>(tempo, scoremat)</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(tempo, <span class="at">file =</span> <span class="fu">glue</span>(<span class="st">"{OUTPUT}/sim/sim_weights-{semilla}.txt"</span>), <span class="at">col_names =</span> <span class="cn">FALSE</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="use-plink-to-calculate-the-sim-phenotype-y_-_k-x_k" class="level3">
<h3 class="anchored" data-anchor-id="use-plink-to-calculate-the-sim-phenotype-y_-_k-x_k">use plink to calculate the sim phenotype $y_ = _k X_k $</h3>
<blockquote class="blockquote">
<p>An alternative way to simulate Y with related individuals would be to simulate unrelated normal rv. for each rat, then pre-multiply by the correlation matrix <span class="math inline">\(\Sigma^{0.5}\)</span>. This approach would make it more obvious that premultiplying by <span class="math inline">\(\Sigma^{-0.5}\)</span> would yield a an uncorrelated trait across rats.</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate PRS using plink</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="co"># https://www.cog-genomics.org/plink/1.9/score</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="co"># --score my.scores 3 2 1</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="co"># reads variant IDs from column 3, allele codes from column 2, and scores from column 1.</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="co"># &gt; head(tempo)</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="co"># A tibble: 6 × 6</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="co">#     X1 X2               X3      X4 X5    X6   </span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="co">#  &lt;dbl&gt; &lt;chr&gt;         &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="co">#1     1 1_1643610_C_T     0 1643610 T     C    </span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="co">#2     1 1_1646409_T_G     0 1646409 T     G    </span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a><span class="co">## BEFORE 1/30/2024 --score $OUTPUT/sim/$WEIGHTS 2 6 $((6+SIMID)) \</span></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a><span class="co">## new column order</span></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; headleft(tempo,6) tempo has df with snp info and weights N(0,1)/sqrt(2 p (1-p))</span></span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a><span class="co">#            SNP A1 A2    MAF           1          2</span></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a><span class="co">#1 1_1643610_C_T  T  C 0.4521  0.01446167 -0.1048587</span></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a><span class="co">#2 1_1646409_T_G  T  G 0.1253  0.88769963 -0.7607204</span></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a><span class="co">#3 1_1658435_A_G  G  A 0.4511  2.66557665 -0.7946988</span></span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the range of SIMID values</span></span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a><span class="va">NSIM</span><span class="op">=</span>100  <span class="co"># Replace with your actual value</span></span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a><span class="va">SEMILLA</span><span class="op">=</span>32240</span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a><span class="va">WEIGHTS</span><span class="op">=</span><span class="st">"sim_weights-</span><span class="va">$SEMILLA</span><span class="st">.txt"</span></span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a><span class="bu">time</span> <span class="er">(</span></span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> <span class="kw">((</span><span class="va">SIMID</span><span class="op">=</span><span class="dv">1</span><span class="kw">;</span> <span class="va">SIMID</span><span class="op">&lt;=</span><span class="va">$NSIM</span><span class="kw">;</span> <span class="va">SIMID</span><span class="op">++</span><span class="kw">));</span> <span class="cf">do</span></span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>    <span class="va">$PLINK</span> <span class="at">--bfile</span> <span class="va">$INPUT</span>/data/genotype/rat6k_autosome <span class="dt">\</span></span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a>          <span class="at">--score</span> <span class="va">$OUTPUT</span>/sim/<span class="va">$WEIGHTS</span> 1 3 <span class="va">$((</span><span class="dv">4</span><span class="op">+</span><span class="va">SIMID))</span> <span class="dt">\</span></span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a>          <span class="at">--out</span> <span class="va">$OUTPUT</span>/sim/tempo/PRS_output_<span class="va">$SIMID</span>-<span class="va">$SEMILLA</span></span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Print SIMID every 10 iterations</span></span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="kw">((</span><span class="va">SIMID</span> <span class="op">%</span> <span class="dv">10</span> <span class="op">==</span> <span class="dv">0</span><span class="kw">));</span> <span class="cf">then</span></span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a>      <span class="bu">echo</span> <span class="st">"Processed </span><span class="va">$SIMID</span><span class="st"> simulations"</span></span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a>    <span class="cf">fi</span></span>
<span id="cb16-35"><a href="#cb16-35" aria-hidden="true" tabindex="-1"></a>  <span class="cf">done</span></span>
<span id="cb16-36"><a href="#cb16-36" aria-hidden="true" tabindex="-1"></a><span class="kw">)</span></span>
<span id="cb16-37"><a href="#cb16-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-38"><a href="#cb16-38" aria-hidden="true" tabindex="-1"></a><span class="co">#Processed 100 simulations</span></span>
<span id="cb16-39"><a href="#cb16-39" aria-hidden="true" tabindex="-1"></a><span class="co">#real    2m1.887s</span></span>
<span id="cb16-40"><a href="#cb16-40" aria-hidden="true" tabindex="-1"></a><span class="co">#user    1m45.025s</span></span>
<span id="cb16-41"><a href="#cb16-41" aria-hidden="true" tabindex="-1"></a><span class="co">#sys     0m13.337s</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="format-simulated-scores-into-matrix" class="level3">
<h3 class="anchored" data-anchor-id="format-simulated-scores-into-matrix">format simulated scores into matrix</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># semilla = 20</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="co"># set.seed(semilla) </span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>h2 <span class="ot">=</span> bodylen_h2</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>tailo<span class="ot">=</span><span class="st">"-32240"</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(F)</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  simy <span class="ot">=</span> <span class="fu">suppressMessages</span>(<span class="fu">read_table</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/sim/tempo/PRS_output_1{tailo}.profile"</span>)))</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  nsim <span class="ot">=</span> <span class="dv">100</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  phenomat <span class="ot">=</span> <span class="fu">matrix</span>(<span class="cn">NA</span>,<span class="fu">nrow</span>(simy),nsim)</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(phenomat) <span class="ot">=</span> simy<span class="sc">$</span>IID</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>  phenomat[,<span class="dv">1</span>] <span class="ot">=</span> simy<span class="sc">$</span>SCORE</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(simid <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span>nsim)</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>    simy <span class="ot">=</span> <span class="fu">suppressMessages</span>(<span class="fu">read_table</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/sim/tempo/PRS_output_{simid}{tailo}.profile"</span>)))</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>    phenomat[,simid] <span class="ot">=</span> simy<span class="sc">$</span>SCORE</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>  <span class="do">## scale phenomat</span></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>  phenomat <span class="ot">=</span> <span class="fu">scale</span>(phenomat)</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>  <span class="do">## add noise</span></span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>  <span class="co">#nx = nrow(phenomat)</span></span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a>  <span class="co">#phenomat = sqrt(h2) * phenomat + sqrt(1 - h2) * scale(matrix(rnorm(nx*nsim),nx, nsim))</span></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>  simpheno <span class="ot">=</span> <span class="fu">cbind</span>(simy[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], <span class="fu">as.data.frame</span>(phenomat))</span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>  <span class="do">## CHECK WHAT NAME TO USE</span></span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>  <span class="co">#saveRDS(simpheno,file=glue("{OUTPUT}/sim/tempo/simpheno_{h2}-{tailo}.RDS"))</span></span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(simpheno,<span class="at">file=</span><span class="fu">glue</span>(<span class="st">"{OUTPUT}/sim/tempo/simpheno.RDS"</span>)) <span class="do">## no error component</span></span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> </span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a>  simpheno <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="at">file=</span><span class="fu">glue</span>(<span class="st">"{OUTPUT}/sim/tempo/simpheno.RDS"</span>)) <span class="do">## no error component</span></span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a>  <span class="co">#simpheno=readRDS(file=glue("{OUTPUT}/sim/tempo/simpheno_{h2}-{tailo}.RDS"))</span></span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a>  nsim <span class="ot">=</span> <span class="fu">ncol</span>(simpheno) <span class="sc">-</span> <span class="dv">2</span> <span class="do">## subtract FID, and IID columns</span></span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="visualize-raw-and-corrected-pvalues" class="level3">
<h3 class="anchored" data-anchor-id="visualize-raw-and-corrected-pvalues">visualize raw and corrected pvalues</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>myplot <span class="ot">&lt;-</span> <span class="cf">function</span>(tempres, <span class="at">post_titulo=</span><span class="st">""</span>,<span class="at">semilla=</span><span class="st">""</span>) {</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a data frame with specific columns</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">p0.01_yes =</span> <span class="fu">apply</span>(tempres<span class="sc">$</span>pmat_correct, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">mean</span>(x <span class="sc">&lt;</span> <span class="fl">0.01</span>)),</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">p0.01_no =</span> <span class="fu">apply</span>(tempres<span class="sc">$</span>pmat_raw, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">mean</span>(x <span class="sc">&lt;</span> <span class="fl">0.01</span>)),</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">p0.05_yes =</span> <span class="fu">apply</span>(tempres<span class="sc">$</span>pmat_correct, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">mean</span>(x <span class="sc">&lt;</span> <span class="fl">0.05</span>)),</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">p0.05_no =</span> <span class="fu">apply</span>(tempres<span class="sc">$</span>pmat_raw, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">mean</span>(x <span class="sc">&lt;</span> <span class="fl">0.05</span>)),</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">p0.10_yes =</span> <span class="fu">apply</span>(tempres<span class="sc">$</span>pmat_correct, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">mean</span>(x <span class="sc">&lt;</span> <span class="fl">0.10</span>)), </span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">p0.10_no =</span> <span class="fu">apply</span>(tempres<span class="sc">$</span>pmat_raw, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">mean</span>(x <span class="sc">&lt;</span> <span class="fl">0.10</span>))</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ... [rest of your code for creating df] ...</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Pivot the data frame to long format, specifying the columns to keep</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>df_long <span class="ot">&lt;-</span> <span class="fu">pivot_longer</span>(df, <span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"p"</span>))</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>df_long <span class="ot">&lt;-</span> df_long <span class="sc">%&gt;%</span> <span class="fu">separate</span>(name,<span class="at">into =</span> <span class="fu">c</span>(<span class="st">"threshold"</span>,<span class="st">"corrected"</span>),<span class="at">sep=</span><span class="st">"_"</span>) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">proportion=</span>value)</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename the name column to replace p0.xx with p&lt;0.xx</span></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>df_long <span class="ot">&lt;-</span> df_long <span class="sc">%&gt;%</span></span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">threshold =</span> <span class="fu">gsub</span>(<span class="st">"p0</span><span class="sc">\\</span><span class="st">."</span>, <span class="st">"p&lt;0."</span>, threshold))</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create boxplots with mean</span></span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>  pp <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df_long, <span class="fu">aes</span>(<span class="at">x =</span> threshold, <span class="at">y =</span> proportion, <span class="at">fill =</span> corrected)) <span class="sc">+</span></span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_summary</span>(<span class="at">fun =</span> mean, <span class="at">geom =</span> <span class="st">"point"</span>, <span class="at">shape =</span> <span class="dv">3</span>, <span class="at">size =</span> <span class="dv">2</span>, <span class="at">stroke =</span> <span class="dv">2</span>, <span class="at">color =</span> <span class="st">"blue"</span>,                  <span class="co">#position = position_dodge(width = 0.8)) +</span></span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>                 <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="sc">-</span><span class="fl">0.1</span>)) <span class="sc">+</span></span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>    <span class="co">#stat_summary(fun = mean, geom = "crossbar",  size = .5, color = "blue") +</span></span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a>    <span class="co">#stat_summary(fun = mean, geom = "crossbar",  size = .5, color = "darkgray") +</span></span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">c</span>(<span class="fl">0.01</span>, <span class="fl">0.05</span>, <span class="fl">0.10</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ggtitle(glue("Type I Error Calibration {semilla} {post_titulo}")) +</span></span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"significance"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"false positive rate"</span>)</span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a>  pp</span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb18-36"><a href="#cb18-36" aria-hidden="true" tabindex="-1"></a><span class="co">#myplot(tempres_sim,post_titulo = glue("nsam:", {nsam},"\n"),semilla)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="define-function-lmm-with-grm" class="level2">
<h2 class="anchored" data-anchor-id="define-function-lmm-with-grm">define function lmm with GRM</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="do">## HERE WE USE THE FULL GRM MATRIX AND CALCULATE THE INVERSE OF THE SIGMA MATRIX</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="do">## define lmm association function </span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>lmmGRM <span class="ot">=</span> <span class="cf">function</span>(pheno, grm_mat, h2, pred_expr,<span class="at">pheno_id_col=</span><span class="dv">1</span>,<span class="at">pheno_value_cols=</span><span class="dv">6</span><span class="sc">:</span><span class="dv">6</span>,<span class="at">out=</span><span class="cn">NULL</span>)</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  <span class="do">## input pheno is a data frame with id column pheno_id_col=1 by default</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="do">## phenotype values are in pheno_value_cols, 6:6 by default (SCORE column location in plink output), it can have more than one phenotype</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  <span class="do">## but h2 has to be the same, this is useful when running simulations with different h2</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>  <span class="do">## call lmmXcan(pheno %&gt;% select(IID,SCORE))</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  <span class="do">## format pheno to matrix form</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>  phenomat <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(pheno[,pheno_value_cols])</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(phenomat) <span class="ot">&lt;-</span> pheno[[pheno_id_col]]</span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>  <span class="do">## turn pred_expr into matrix with rownames =IID, keep only IIDs in ymat</span></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>  exp_mat <span class="ot">=</span> <span class="fu">as.matrix</span>(pred_expr <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>IID))</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(exp_mat) <span class="ot">=</span> pred_expr<span class="sc">$</span>IID</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>  <span class="do">## align pheno and expr matrices</span></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a>  idlist <span class="ot">=</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(phenomat), <span class="fu">rownames</span>(exp_mat))</span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a>  nsam <span class="ot">=</span> <span class="fu">length</span>(idlist)</span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a>  <span class="do">## CALCULATE SIGMA</span></span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a>  ID_mat <span class="ot">=</span> <span class="fu">diag</span>(<span class="fu">rep</span>(<span class="dv">1</span>,nsam))</span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true" tabindex="-1"></a>  <span class="co">#testing_scale_grm = TRUE</span></span>
<span id="cb19-27"><a href="#cb19-27" aria-hidden="true" tabindex="-1"></a>  <span class="co">#if(testing_scale_grm) grm_mat = sweep( sweep(grm_mat,2, 1/sqrt(diag(grm_mat)),"*"), 1, 1/sqrt(diag(grm_mat)),"*")    </span></span>
<span id="cb19-28"><a href="#cb19-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-29"><a href="#cb19-29" aria-hidden="true" tabindex="-1"></a>  Sigma <span class="ot">=</span> grm_mat[idlist,idlist] <span class="sc">*</span> h2 <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">-</span> h2) <span class="sc">*</span> ID_mat</span>
<span id="cb19-30"><a href="#cb19-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-31"><a href="#cb19-31" aria-hidden="true" tabindex="-1"></a>  Sig_eigen <span class="ot">=</span> <span class="fu">eigen</span>(Sigma)</span>
<span id="cb19-32"><a href="#cb19-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(Sig_eigen<span class="sc">$</span>vectors) <span class="ot">=</span> <span class="fu">rownames</span>(Sigma)</span>
<span id="cb19-33"><a href="#cb19-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-34"><a href="#cb19-34" aria-hidden="true" tabindex="-1"></a>  isighalf <span class="ot">=</span> Sig_eigen<span class="sc">$</span>vectors <span class="sc">%*%</span> <span class="fu">diag</span>( <span class="dv">1</span> <span class="sc">/</span> <span class="fu">sqrt</span>(  Sig_eigen<span class="sc">$</span>values  ) ) <span class="sc">%*%</span> <span class="fu">t</span>(Sig_eigen<span class="sc">$</span>vectors)</span>
<span id="cb19-35"><a href="#cb19-35" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-36"><a href="#cb19-36" aria-hidden="true" tabindex="-1"></a>  <span class="do">## perform raw association</span></span>
<span id="cb19-37"><a href="#cb19-37" aria-hidden="true" tabindex="-1"></a>  cormat_raw <span class="ot">=</span> <span class="fu">matrix_lm</span>(phenomat[idlist,, <span class="at">drop =</span> <span class="cn">FALSE</span>], exp_mat[idlist,])</span>
<span id="cb19-38"><a href="#cb19-38" aria-hidden="true" tabindex="-1"></a>  pmat_raw <span class="ot">=</span> <span class="fu">cor2pval</span>(cormat_raw,nsam)</span>
<span id="cb19-39"><a href="#cb19-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(pmat_raw) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"cor_"</span>, <span class="st">"pval_"</span>, <span class="fu">colnames</span>(pmat_raw))</span>
<span id="cb19-40"><a href="#cb19-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-41"><a href="#cb19-41" aria-hidden="true" tabindex="-1"></a>  <span class="do">## perform corrected association</span></span>
<span id="cb19-42"><a href="#cb19-42" aria-hidden="true" tabindex="-1"></a>  cormat_correct <span class="ot">=</span> <span class="fu">matrix_lm</span>(isighalf<span class="sc">%*%</span> phenomat[idlist,, <span class="at">drop =</span> <span class="cn">FALSE</span>], isighalf <span class="sc">%*%</span> exp_mat[idlist,])</span>
<span id="cb19-43"><a href="#cb19-43" aria-hidden="true" tabindex="-1"></a>  pmat_correct <span class="ot">=</span> <span class="fu">cor2pval</span>(cormat_correct,nsam)</span>
<span id="cb19-44"><a href="#cb19-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(pmat_correct) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"cor_"</span>, <span class="st">"pval_"</span>, <span class="fu">colnames</span>(pmat_correct))</span>
<span id="cb19-45"><a href="#cb19-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-46"><a href="#cb19-46" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(out))</span>
<span id="cb19-47"><a href="#cb19-47" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb19-48"><a href="#cb19-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">saveRDS</span>(cormat_correct,<span class="at">file =</span> <span class="fu">glue</span>(<span class="st">"{out}_cormat_correct.RDS"</span>))</span>
<span id="cb19-49"><a href="#cb19-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">saveRDS</span>(pmat_correct,  <span class="at">file =</span> <span class="fu">glue</span>(<span class="st">"{out}_pmat_correct.RDS"</span>))</span>
<span id="cb19-50"><a href="#cb19-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">saveRDS</span>(cormat_raw,    <span class="at">file =</span> <span class="fu">glue</span>(<span class="st">"{out}_cormat_raw.RDS"</span>))</span>
<span id="cb19-51"><a href="#cb19-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">saveRDS</span>(pmat_raw,      <span class="at">file =</span> <span class="fu">glue</span>(<span class="st">"{out}_pmat_raw.RDS"</span>))</span>
<span id="cb19-52"><a href="#cb19-52" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb19-53"><a href="#cb19-53" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb19-54"><a href="#cb19-54" aria-hidden="true" tabindex="-1"></a>    <span class="at">cormat_correct=</span>cormat_correct, </span>
<span id="cb19-55"><a href="#cb19-55" aria-hidden="true" tabindex="-1"></a>    <span class="at">pmat_correct=</span>pmat_correct, </span>
<span id="cb19-56"><a href="#cb19-56" aria-hidden="true" tabindex="-1"></a>    <span class="at">cormat_raw=</span>cormat_raw, </span>
<span id="cb19-57"><a href="#cb19-57" aria-hidden="true" tabindex="-1"></a>    <span class="at">pmat_raw=</span>pmat_raw)</span>
<span id="cb19-58"><a href="#cb19-58" aria-hidden="true" tabindex="-1"></a>  res</span>
<span id="cb19-59"><a href="#cb19-59" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-60"><a href="#cb19-60" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="run-ratxcan-null-regression" class="level3">
<h3 class="anchored" data-anchor-id="run-ratxcan-null-regression">run ratxcan null regression</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="do">##pred_expr = read_pred_expr(glue("{OUTPUT}/Br-hki-rat6k__predict.txt"))</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>add_noise <span class="ot">=</span> <span class="cf">function</span>(simpheno,h2)</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>  phenomat <span class="ot">=</span> <span class="fu">as.matrix</span>(simpheno <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>IID,<span class="sc">-</span>FID))</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>  phenomat <span class="ot">=</span> <span class="fu">scale</span>(phenomat)</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>  <span class="do">## add noise</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>  nx <span class="ot">=</span> <span class="fu">nrow</span>(phenomat)</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>  nsim<span class="ot">=</span><span class="fu">ncol</span>(phenomat)</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>  phenomat <span class="ot">=</span> <span class="fu">sqrt</span>(h2) <span class="sc">*</span> phenomat <span class="sc">+</span> <span class="fu">sqrt</span>(<span class="dv">1</span> <span class="sc">-</span> h2) <span class="sc">*</span> <span class="fu">scale</span>(<span class="fu">matrix</span>(<span class="fu">rnorm</span>(nx<span class="sc">*</span>nsim),nx, nsim))</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cbind</span>(simpheno[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], <span class="fu">as.data.frame</span>(phenomat))</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>recalculate<span class="ot">=</span><span class="cn">FALSE</span></span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(h2 <span class="cf">in</span> <span class="fu">c</span>(<span class="fl">0.1</span>,<span class="fl">0.2</span>,<span class="fl">0.4</span>,<span class="fl">0.6</span>,<span class="fl">0.8</span>))</span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"---"</span>,h2,<span class="st">"---</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(recalculate)</span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>  { </span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>    tic<span class="ot">=</span><span class="fu">Sys.time</span>()</span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>    tempres <span class="ot">&lt;-</span> <span class="fu">lmmGRM</span>(<span class="fu">add_noise</span>(simpheno,h2),</span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>                      grm_mat, h2,pred_expr,<span class="at">pheno_id_col=</span><span class="dv">1</span>, </span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a>                      <span class="at">pheno_value_cols=</span><span class="dv">2</span><span class="sc">+</span>(<span class="dv">1</span><span class="sc">:</span>nsim))</span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a>    toc<span class="ot">=</span><span class="fu">Sys.time</span>()</span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(toc <span class="sc">-</span> tic)</span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">saveRDS</span>(tempres,<span class="at">file=</span><span class="fu">glue</span>(<span class="st">"{OUTPUT}/tempres-null-h2-{h2}.RDS"</span>))</span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> tempres <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/tempres-null-h2-{h2}.RDS"</span>))</span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">myplot</span>(tempres_sim)</span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggsave</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/calib-figure-h2-{h2}.png"</span>),<span class="at">width=</span><span class="dv">6</span>,<span class="at">height=</span><span class="dv">5</span>)</span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/hist-p-corrected-{h2}.png"</span>))</span>
<span id="cb20-31"><a href="#cb20-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hist</span>(tempres<span class="sc">$</span>pmat_correct,<span class="at">main=</span><span class="fu">glue</span>(<span class="st">"corrected p-values - {h2}"</span>))</span>
<span id="cb20-32"><a href="#cb20-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb20-33"><a href="#cb20-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/hist-p-raw-{h2}.png"</span>))</span>
<span id="cb20-34"><a href="#cb20-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hist</span>(tempres<span class="sc">$</span>pmat_raw,<span class="at">main=</span><span class="fu">glue</span>(<span class="st">"raw p-values - {h2}"</span>))</span>
<span id="cb20-35"><a href="#cb20-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb20-36"><a href="#cb20-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/qqunif-compare-raw-corrected-{h2}.png"</span>))</span>
<span id="cb20-37"><a href="#cb20-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">qqunif.compare</span>(tempres<span class="sc">$</span>pmat_raw,tempres<span class="sc">$</span>pmat_correct,<span class="at">BF=</span><span class="cn">FALSE</span>,<span class="at">BF2=</span><span class="cn">FALSE</span>,<span class="at">main=</span><span class="st">"qqunif {h2}"</span>)</span>
<span id="cb20-38"><a href="#cb20-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb20-39"><a href="#cb20-39" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb20-40"><a href="#cb20-40" aria-hidden="true" tabindex="-1"></a><span class="do">##%HERE</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="simulate-y-sigma12epsilon-and-run-assoc-with-expr_mat" class="level3">
<h3 class="anchored" data-anchor-id="simulate-y-sigma12epsilon-and-run-assoc-with-expr_mat">simulate <span class="math inline">\(Y = Sigma^{1/2}\epsilon\)</span> and run assoc with expr_mat</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>nsam<span class="ot">=</span><span class="fu">nrow</span>(grm_mat)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>ind<span class="ot">=</span><span class="dv">1</span><span class="sc">:</span>nsam</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="co">#ind=1:1000</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>test_mat <span class="ot">=</span> grm_mat[ind,ind]</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>nsam<span class="ot">=</span><span class="fu">nrow</span>(test_mat)</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>Sigma <span class="ot">=</span> test_mat <span class="sc">*</span> h2 <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">-</span> h2) <span class="sc">*</span> <span class="fu">diag</span>(<span class="fu">rep</span>(<span class="dv">1</span>,nsam))</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>Sig_eigen <span class="ot">=</span> <span class="fu">eigen</span>(Sigma)</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(Sig_eigen<span class="sc">$</span>vectors) <span class="ot">=</span> <span class="fu">rownames</span>(Sigma)</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a><span class="do">##sighalf = Sig_eigen$vectors %*% diag( sqrt(  Sig_eigen$values  ) ) %*% t(Sig_eigen$vectors)</span></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a><span class="do">## make this multiplication more efficient using sweep</span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>sighalf <span class="ot">=</span> Sig_eigen<span class="sc">$</span>vectors <span class="sc">%*%</span> <span class="fu">sweep</span>(<span class="fu">t</span>(Sig_eigen<span class="sc">$</span>vectors),<span class="dv">1</span>,<span class="fu">sqrt</span>(  Sig_eigen<span class="sc">$</span>values ),<span class="st">"*"</span>)</span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(ii <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>sim_sigma_pheno <span class="ot">=</span> sighalf <span class="sc">%*%</span> <span class="fu">matrix</span>(<span class="fu">rnorm</span>(nsam <span class="sc">*</span> nsim), nsam, nsim) </span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>sim_sigma_pheno<span class="ot">=</span><span class="fu">cbind</span>(<span class="at">FID=</span><span class="fu">rownames</span>(sim_sigma_pheno),<span class="at">IID=</span><span class="fu">rownames</span>(sim_sigma_pheno),<span class="fu">as.data.frame</span>(sim_sigma_pheno))</span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>tic<span class="ot">=</span><span class="fu">Sys.time</span>()</span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>tempres_sigma_pheno <span class="ot">&lt;-</span> <span class="fu">lmmGRM</span>(sim_sigma_pheno,grm_mat, h2,pred_expr,<span class="at">pheno_id_col=</span><span class="dv">1</span>, <span class="at">pheno_value_cols=</span><span class="dv">2</span><span class="sc">+</span>(<span class="dv">1</span><span class="sc">:</span>nsim))</span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>toc<span class="ot">=</span><span class="fu">Sys.time</span>()</span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>toc <span class="sc">-</span> tic</span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>pp<span class="ot">&lt;-</span><span class="fu">myplot</span>(tempres_sigma_pheno,<span class="at">post_titulo =</span> <span class="fu">glue</span>(<span class="st">"sigma_pheno n= {nsam} - ii={ii}"</span>))</span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(ii,<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(pp)</span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/calib-sim-sigma-n{nsam}-ii{ii}.png"</span>))</span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="simulate-y-grm12-h2-1---h2-epsilon-and-run-assoc-with-expr_mat" class="level3">
<h3 class="anchored" data-anchor-id="simulate-y-grm12-h2-1---h2-epsilon-and-run-assoc-with-expr_mat">simulate <span class="math inline">\(Y = GRM^{1/2} h2 + (1 - h2) \epsilon\)</span> and run assoc with expr_mat</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>nsam<span class="ot">=</span><span class="fu">nrow</span>(grm_mat)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>ind<span class="ot">=</span><span class="dv">1</span><span class="sc">:</span>nsam</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="co">#ind=1:1000</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>test_mat <span class="ot">=</span> grm_mat[ind,ind]</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>nsam<span class="ot">=</span><span class="fu">nrow</span>(test_mat)</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>test_eigen <span class="ot">=</span> <span class="fu">eigen</span>(test_mat)</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(test_eigen<span class="sc">$</span>vectors) <span class="ot">=</span> <span class="fu">rownames</span>(test_mat)</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a><span class="do">## show smallest eigenvalues of GRM</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sort</span>(test_eigen<span class="sc">$</span>values) <span class="sc">%&gt;%</span> head, <span class="st">"these numbers should be non negative </span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a><span class="do">## force eigenvalues to be nonnegative</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>test_eigen<span class="sc">$</span>values <span class="ot">=</span> <span class="fu">pmax</span>(test_eigen<span class="sc">$</span>values,<span class="dv">0</span>)</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>grmhalf <span class="ot">=</span> test_eigen<span class="sc">$</span>vectors <span class="sc">%*%</span> <span class="fu">sweep</span>(<span class="fu">t</span>(test_eigen<span class="sc">$</span>vectors), <span class="dv">1</span>, <span class="fu">sqrt</span>(test_eigen<span class="sc">$</span>values),<span class="st">"*"</span>)</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(ii <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>  <span class="do">## simulate phenomat as rv with h2*grm as cov + indep noise term (1-h2)</span></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>sim_grm_pheno <span class="ot">=</span> <span class="fu">sqrt</span>(h2) <span class="sc">*</span> grmhalf <span class="sc">%*%</span> <span class="fu">matrix</span>(<span class="fu">rnorm</span>(nsam <span class="sc">*</span> nsim), nsam, nsim) <span class="sc">+</span> <span class="fu">sqrt</span>(<span class="dv">1</span><span class="sc">-</span>h2) <span class="sc">*</span> <span class="fu">matrix</span>(<span class="fu">rnorm</span>(nsam <span class="sc">*</span> nsim), nsam, nsim)</span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>sim_grm_pheno<span class="ot">=</span><span class="fu">cbind</span>(<span class="at">FID=</span><span class="fu">rownames</span>(sim_grm_pheno),<span class="at">IID=</span><span class="fu">rownames</span>(sim_grm_pheno),<span class="fu">as.data.frame</span>(sim_grm_pheno))</span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>tic<span class="ot">=</span><span class="fu">Sys.time</span>()</span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>tempres_grm_pheno <span class="ot">&lt;-</span> <span class="fu">lmmGRM</span>(sim_grm_pheno,grm_mat, h2,pred_expr,<span class="at">pheno_id_col=</span><span class="dv">1</span>, <span class="at">pheno_value_cols=</span><span class="dv">2</span><span class="sc">+</span>(<span class="dv">1</span><span class="sc">:</span>nsim))</span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a>toc<span class="ot">=</span><span class="fu">Sys.time</span>()</span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a>toc <span class="sc">-</span> tic</span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a>pp<span class="ot">&lt;-</span><span class="fu">myplot</span>(tempres_grm_pheno,<span class="at">post_titulo =</span> <span class="fu">glue</span>(<span class="st">"sigma_pheno n= {nsam} - ii={ii}"</span>))</span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(ii,<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(pp)</span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/calib-sim-grm-n{nsam}-ii{ii}.png"</span>))</span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="run-bodylen-regression-with-br-expr-lmmgrm" class="level3">
<h3 class="anchored" data-anchor-id="run-bodylen-regression-with-br-expr-lmmgrm">run bodylen regression with BR expr lmmGRM</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>pred_expr <span class="ot">=</span> <span class="fu">read_pred_expr</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/Br-hki-rat6k__predict.txt"</span>))</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>recalculate <span class="ot">=</span> <span class="cn">FALSE</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>trait <span class="ot">=</span> <span class="st">"bodylen"</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>h2 <span class="ot">=</span> bodylen_h2</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>h2se <span class="ot">=</span> bodylen_h2<span class="sc">+</span>bodylen_se</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a><span class="do">##pheno, grm_mat, h2, pred_expr,pheno_id_col=1,pheno_value_cols=6:6,out=NULL</span></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(recalculate)</span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>  tic<span class="ot">=</span><span class="fu">Sys.time</span>()</span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>  tempres_h2 <span class="ot">&lt;-</span> <span class="fu">lmmGRM</span>(pheno,grm_mat, h2, pred_expr,<span class="at">pheno_id_col=</span><span class="dv">1</span>, <span class="at">pheno_value_cols=</span><span class="fu">which</span>(<span class="fu">colnames</span>(pheno)<span class="sc">==</span>trait))</span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>  toc<span class="ot">=</span><span class="fu">Sys.time</span>()</span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a>  toc <span class="sc">-</span> tic</span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(tempres_h2,<span class="fu">glue</span>(<span class="st">"{OUTPUT}/{trait}-BR-tempres_h2.RDS"</span>))</span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span></span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a>tempres_h2 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/{trait}-BR-tempres_h2.RDS"</span>))</span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/{trait}-BR-lmmGRM.png"</span>))</span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif.compare</span>(tempres_h2<span class="sc">$</span>pmat_raw,tempres_h2<span class="sc">$</span>pmat_correct,<span class="at">main=</span><span class="fu">glue</span>(<span class="st">"ratxcan {trait}"</span>) )</span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb23-22"><a href="#cb23-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-23"><a href="#cb23-23" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(recalculate)</span>
<span id="cb23-24"><a href="#cb23-24" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb23-25"><a href="#cb23-25" aria-hidden="true" tabindex="-1"></a>tic<span class="ot">=</span><span class="fu">Sys.time</span>()</span>
<span id="cb23-26"><a href="#cb23-26" aria-hidden="true" tabindex="-1"></a>tempres_h2se <span class="ot">&lt;-</span> <span class="fu">lmmGRM</span>(pheno,grm_mat, h2se,pred_expr,<span class="at">pheno_id_col=</span><span class="dv">1</span>, <span class="at">pheno_value_cols=</span><span class="fu">which</span>(<span class="fu">colnames</span>(pheno)<span class="sc">==</span>trait))</span>
<span id="cb23-27"><a href="#cb23-27" aria-hidden="true" tabindex="-1"></a>toc<span class="ot">=</span><span class="fu">Sys.time</span>()</span>
<span id="cb23-28"><a href="#cb23-28" aria-hidden="true" tabindex="-1"></a>toc <span class="sc">-</span> tic</span>
<span id="cb23-29"><a href="#cb23-29" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(tempres_h2se,<span class="fu">glue</span>(<span class="st">"{OUTPUT}/{trait}-BR-tempres_h2se.RDS"</span>))</span>
<span id="cb23-30"><a href="#cb23-30" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span></span>
<span id="cb23-31"><a href="#cb23-31" aria-hidden="true" tabindex="-1"></a>tempres_h2se <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/{trait}-BR-tempres_h2se.RDS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="run-bmi-regression-with-lmmgrm" class="level3">
<h3 class="anchored" data-anchor-id="run-bmi-regression-with-lmmgrm">run bmi regression with lmmGRM</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>recalculate <span class="ot">=</span> <span class="cn">FALSE</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>h2 <span class="ot">&lt;-</span> bmi_h2</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>h2se <span class="ot">=</span> bmi_h2 <span class="sc">+</span> bmi_se</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>trait <span class="ot">=</span> <span class="st">"bmi"</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>pred_expr <span class="ot">=</span> <span class="fu">read_pred_expr</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/Br-hki-rat6k__predict.txt"</span>))</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(recalculate)</span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>tic<span class="ot">=</span><span class="fu">Sys.time</span>()</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>h2 <span class="ot">=</span> bmi_h2 <span class="sc">+</span> bmi_se</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>tempres_h2 <span class="ot">&lt;-</span> <span class="fu">lmmGRM</span>(pheno,grm_mat, h2, pred_expr, <span class="at">pheno_id_col=</span><span class="dv">1</span>, <span class="at">pheno_value_cols=</span><span class="fu">which</span>(<span class="fu">colnames</span>(pheno)<span class="sc">==</span>trait))</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>toc<span class="ot">=</span><span class="fu">Sys.time</span>()</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>toc <span class="sc">-</span> tic</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(tempres_h2,<span class="fu">glue</span>(<span class="st">"{OUTPUT}/{trait}-BR-tempres_h2.RDS"</span>))</span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> tempres_h2 <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/{trait}-BR-tempres_h2.RDS"</span>))</span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/{trait}-BR-lmmGRM.png"</span>))</span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif.compare</span>(tempres_h2<span class="sc">$</span>pmat_correct,tempres_h2<span class="sc">$</span>pmat_raw,<span class="at">main=</span><span class="fu">glue</span>(<span class="st">"ratxcan {trait}"</span>) )</span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(recalculate)</span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a>tic<span class="ot">=</span><span class="fu">Sys.time</span>()</span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>tempres_h2se <span class="ot">&lt;-</span> <span class="fu">lmmGRM</span>(pheno,grm_mat, h2se,pred_expr,<span class="at">pheno_id_col=</span><span class="dv">1</span>, <span class="at">pheno_value_cols=</span><span class="fu">which</span>(<span class="fu">colnames</span>(pheno)<span class="sc">==</span>trait))</span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a>toc<span class="ot">=</span><span class="fu">Sys.time</span>()</span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>toc <span class="sc">-</span> tic</span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(tempres_h2se,<span class="fu">glue</span>(<span class="st">"{OUTPUT}/{trait}-BR-tempres_h2se.RDS"</span>))</span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> tempres_h2se <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/{trait}-BR-tempres_h2se.RDS"</span>))</span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/{trait}-BR-lmmGRM_h2_plus_se.png"</span>))</span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif.compare</span>(tempres_h2se<span class="sc">$</span>pmat_correct,tempres_h2se<span class="sc">$</span>pmat_raw,<span class="at">main=</span><span class="fu">glue</span>(<span class="st">"ratxcan {trait}"</span>) )</span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="bodylen-and-bmi-vs-5-brain-region-expr-association" class="level3">
<h3 class="anchored" data-anchor-id="bodylen-and-bmi-vs-5-brain-region-expr-association">bodylen and bmi vs 5 brain region expr association</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>recalculate <span class="ot">=</span> <span class="cn">FALSE</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="co">#tissuelist = c("AC", "IL", "LH", "PL", "VO")</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(trait <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"bodylen"</span>,<span class="st">"bmi"</span>))</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(trait <span class="sc">==</span> <span class="st">"bmi"</span>) h2 <span class="ot">=</span> bmi_h2<span class="sc">+</span>bmi_se <span class="cf">else</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(trait <span class="sc">==</span> <span class="st">"bodylen"</span>) h2 <span class="ot">=</span> bodylen_h2 <span class="sc">+</span> bodylen_se <span class="cf">else</span> <span class="fu">stop</span>(<span class="st">"unknown trait"</span>)</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(trait)</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(tis <span class="cf">in</span> tissuelist)</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(tis)</span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(recalculate)</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>    {</span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>      pred_expr <span class="ot">=</span> <span class="fu">read_pred_expr</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/{tis}-filtered__predict.txt"</span>))</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>      <span class="do">## run lmmGRM</span></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>      tic<span class="ot">=</span><span class="fu">Sys.time</span>()</span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>      tempres_h2se <span class="ot">&lt;-</span> </span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>        <span class="fu">lmmGRM</span>(pheno,grm_mat,</span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>               h2,</span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>               pred_expr,</span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a>               <span class="at">pheno_id_col=</span><span class="dv">1</span>,</span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>               <span class="at">pheno_value_cols=</span><span class="fu">which</span>(<span class="fu">colnames</span>(pheno)<span class="sc">==</span>trait) )</span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a>      toc<span class="ot">=</span><span class="fu">Sys.time</span>()</span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a>      <span class="fu">print</span>(toc <span class="sc">-</span> tic)</span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a>      <span class="do">## save results</span></span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a>      <span class="fu">saveRDS</span>(tempres_h2se,<span class="fu">glue</span>(<span class="st">"{OUTPUT}/{trait}-{tis}-tempres_h2se.RDS"</span>))</span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span></span>
<span id="cb25-32"><a href="#cb25-32" aria-hidden="true" tabindex="-1"></a>      tempres_h2se <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/{trait}-{tis}-tempres_h2se.RDS"</span>))</span>
<span id="cb25-33"><a href="#cb25-33" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb25-34"><a href="#cb25-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/{trait}-{tis}-lmmGRM_h2_plus_se.png"</span>))</span>
<span id="cb25-35"><a href="#cb25-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">qqunif.compare</span>(tempres_h2se<span class="sc">$</span>pmat_raw,tempres_h2se<span class="sc">$</span>pmat_correct,<span class="at">main=</span><span class="fu">glue</span>(<span class="st">"ratxcan {trait} {tis}"</span>) )</span>
<span id="cb25-36"><a href="#cb25-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dev.off</span>()</span>
<span id="cb25-37"><a href="#cb25-37" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb25-38"><a href="#cb25-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-39"><a href="#cb25-39" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="download-rat-gene-annotation-and-human-phenomexcan" class="level3">
<h3 class="anchored" data-anchor-id="download-rat-gene-annotation-and-human-phenomexcan">download rat gene annotation and human phenomexcan</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ensembl = biomaRt::useMart("ensembl", dataset = "rnorvegicus_gene_ensembl")</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="co">#annotation = biomaRt::getBM(attributes = c('ensembl_gene_id','external_gene_name', 'start_position', 'end_position', 'chromosome_name'),mart = ensembl)</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>new_ensembl_query<span class="ot">=</span><span class="cn">FALSE</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(new_ensembl_query)</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  human <span class="ot">=</span> biomaRt<span class="sc">::</span><span class="fu">useEnsembl</span>(<span class="at">biomart=</span><span class="st">'ensembl'</span>, <span class="at">dataset=</span><span class="st">"hsapiens_gene_ensembl"</span>, <span class="at">mirror =</span> <span class="st">"useast"</span>)</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a><span class="co">#attributes &lt;- listAttributes(human)</span></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>attributes <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span>, <span class="st">"external_gene_name"</span>, <span class="st">"rnorvegicus_homolog_ensembl_gene"</span>, <span class="st">"rnorvegicus_homolog_associated_gene_name"</span>)</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>orth.rats <span class="ot">=</span> biomaRt<span class="sc">::</span><span class="fu">getBM</span>(attributes, <span class="at">filters=</span><span class="st">"with_rnorvegicus_homolog"</span>,<span class="at">values=</span><span class="cn">TRUE</span>, <span class="at">mart =</span> human, <span class="at">uniqueRows=</span><span class="cn">TRUE</span>)</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(orth.rats,<span class="at">file=</span><span class="fu">glue</span>(<span class="st">"{INPUT}/data/expression/orth.rats.RDS"</span>))</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span></span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>  orth.rats <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="at">file=</span><span class="fu">glue</span>(<span class="st">"{INPUT}/data/expression/orth.rats.RDS"</span>))</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>phenomexcan_results <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="fu">glue</span>(<span class="st">"{INPUT}/data/phenomexcan/phenomexcan_results_signs.RDS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="create-ratxcan-association-table-to-save-as-csv-files" class="level2">
<h2 class="anchored" data-anchor-id="create-ratxcan-association-table-to-save-as-csv-files">create ratxcan association table to save as csv files</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>generate_results_table <span class="ot">=</span> <span class="cf">function</span>(tempres,<span class="at">out=</span><span class="cn">NULL</span>,phenoname)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="do">## phenoname: "Body mass index (BMI) (21001_raw)" or "Standing height"</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>  pmat <span class="ot">=</span> tempres<span class="sc">$</span>pmat_correct</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">gene=</span><span class="fu">rownames</span>(pmat), <span class="at">p_correc=</span>pmat, <span class="at">cor_correct=</span>tempres<span class="sc">$</span>cormat_correct, <span class="at">p_raw=</span>tempres<span class="sc">$</span>pmat_raw)</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(df) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"gene"</span>,<span class="st">"p_correct"</span>,<span class="st">"cor_correct"</span>,<span class="st">"p_raw"</span>)</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(df) <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>  <span class="do">## annotate</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> df <span class="sc">%&gt;%</span> </span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(gene_annotation <span class="sc">%&gt;%</span> </span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>                <span class="fu">select</span>(<span class="at">genename=</span>gene_name,</span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>                       gene_id,</span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>                       chr,</span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>                       start), <span class="at">by=</span><span class="fu">c</span>(<span class="st">"gene"</span><span class="ot">=</span><span class="st">"gene_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(p_correct)</span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>  <span class="do">## write csv file</span></span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(out)) <span class="fu">write_csv</span>(df,<span class="at">file =</span> out)</span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>  df</span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a>df_all <span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a>tis<span class="ot">=</span><span class="st">"BR"</span></span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a>trait<span class="ot">=</span><span class="st">"bodylen"</span></span>
<span id="cb27-28"><a href="#cb27-28" aria-hidden="true" tabindex="-1"></a><span class="co">#tissuelist = c("AC", "IL", "LH", "PL", "VO")</span></span>
<span id="cb27-29"><a href="#cb27-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-30"><a href="#cb27-30" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(tis <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"BR"</span>,tissuelist))</span>
<span id="cb27-31"><a href="#cb27-31" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb27-32"><a href="#cb27-32" aria-hidden="true" tabindex="-1"></a>  RDSpath <span class="ot">=</span> <span class="fu">glue</span>(<span class="st">"{OUTPUT}/{trait}-{tis}-tempres_h2se.RDS"</span>)</span>
<span id="cb27-33"><a href="#cb27-33" aria-hidden="true" tabindex="-1"></a>  tempres <span class="ot">=</span> <span class="fu">readRDS</span>(RDSpath)</span>
<span id="cb27-34"><a href="#cb27-34" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> <span class="fu">generate_results_table</span>(tempres, <span class="co">#out=glue("{OUTPUT}/{trait}-results.csv"),</span></span>
<span id="cb27-35"><a href="#cb27-35" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">phenoname=</span><span class="st">"Standing height"</span>)</span>
<span id="cb27-36"><a href="#cb27-36" aria-hidden="true" tabindex="-1"></a>  df<span class="sc">$</span>trait<span class="ot">=</span>trait</span>
<span id="cb27-37"><a href="#cb27-37" aria-hidden="true" tabindex="-1"></a>  df<span class="sc">$</span>tissue <span class="ot">=</span> tis</span>
<span id="cb27-38"><a href="#cb27-38" aria-hidden="true" tabindex="-1"></a>  df_all <span class="ot">=</span> <span class="fu">rbind</span>(df_all,df)</span>
<span id="cb27-39"><a href="#cb27-39" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-40"><a href="#cb27-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-41"><a href="#cb27-41" aria-hidden="true" tabindex="-1"></a>trait<span class="ot">=</span><span class="st">"bmi"</span></span>
<span id="cb27-42"><a href="#cb27-42" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(tis <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"BR"</span>,tissuelist))</span>
<span id="cb27-43"><a href="#cb27-43" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb27-44"><a href="#cb27-44" aria-hidden="true" tabindex="-1"></a>  RDSpath <span class="ot">=</span> <span class="fu">glue</span>(<span class="st">"{OUTPUT}/{trait}-{tis}-tempres_h2se.RDS"</span>)</span>
<span id="cb27-45"><a href="#cb27-45" aria-hidden="true" tabindex="-1"></a>  tempres <span class="ot">=</span> <span class="fu">readRDS</span>(RDSpath)</span>
<span id="cb27-46"><a href="#cb27-46" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">=</span> <span class="fu">generate_results_table</span>(tempres, </span>
<span id="cb27-47"><a href="#cb27-47" aria-hidden="true" tabindex="-1"></a>                                <span class="do">##out=glue("{OUTPUT}/{trait}-results.csv"),</span></span>
<span id="cb27-48"><a href="#cb27-48" aria-hidden="true" tabindex="-1"></a>                                <span class="at">phenoname=</span><span class="st">"Body mass index (BMI) (21001_raw)"</span>)</span>
<span id="cb27-49"><a href="#cb27-49" aria-hidden="true" tabindex="-1"></a>  df<span class="sc">$</span>trait<span class="ot">=</span>trait</span>
<span id="cb27-50"><a href="#cb27-50" aria-hidden="true" tabindex="-1"></a>  df<span class="sc">$</span>tissue <span class="ot">=</span> tis</span>
<span id="cb27-51"><a href="#cb27-51" aria-hidden="true" tabindex="-1"></a>  df_all <span class="ot">=</span> <span class="fu">rbind</span>(df_all,df)</span>
<span id="cb27-52"><a href="#cb27-52" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-53"><a href="#cb27-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-54"><a href="#cb27-54" aria-hidden="true" tabindex="-1"></a><span class="co">#saveRDS(df_all,file = glue("{OUTPUT}/all-assoc-results.RDS"))</span></span>
<span id="cb27-55"><a href="#cb27-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-56"><a href="#cb27-56" aria-hidden="true" tabindex="-1"></a><span class="do">## pivot wider</span></span>
<span id="cb27-57"><a href="#cb27-57" aria-hidden="true" tabindex="-1"></a>df_wider <span class="ot">&lt;-</span> df_all <span class="sc">%&gt;%</span> <span class="fu">select</span>(gene,trait,tissue,p_correct)  <span class="sc">%&gt;%</span> <span class="fu">pivot_wider</span>(</span>
<span id="cb27-58"><a href="#cb27-58" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_from =</span> tissue, </span>
<span id="cb27-59"><a href="#cb27-59" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_from =</span> p_correct,</span>
<span id="cb27-60"><a href="#cb27-60" aria-hidden="true" tabindex="-1"></a>    <span class="at">id_cols =</span> <span class="fu">c</span>(gene, trait)</span>
<span id="cb27-61"><a href="#cb27-61" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb27-62"><a href="#cb27-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-63"><a href="#cb27-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-64"><a href="#cb27-64" aria-hidden="true" tabindex="-1"></a>acat <span class="ot">=</span> <span class="cf">function</span>(pvec) </span>
<span id="cb27-65"><a href="#cb27-65" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb27-66"><a href="#cb27-66" aria-hidden="true" tabindex="-1"></a>  pvec <span class="ot">=</span> pvec[<span class="sc">!</span><span class="fu">is.na</span>(pvec)]</span>
<span id="cb27-67"><a href="#cb27-67" aria-hidden="true" tabindex="-1"></a>  TT <span class="ot">=</span> <span class="fu">sum</span>( <span class="fu">tan</span>( (<span class="fl">0.5</span> <span class="sc">-</span> pvec) <span class="sc">*</span>pi ) )</span>
<span id="cb27-68"><a href="#cb27-68" aria-hidden="true" tabindex="-1"></a>  .<span class="dv">5</span> <span class="sc">-</span> <span class="fu">atan</span>(TT <span class="sc">/</span> <span class="fu">length</span>(pvec)) <span class="sc">/</span> pi</span>
<span id="cb27-69"><a href="#cb27-69" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-70"><a href="#cb27-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-71"><a href="#cb27-71" aria-hidden="true" tabindex="-1"></a>df_wider<span class="sc">$</span>p_acat_5 <span class="ot">=</span> <span class="fu">apply</span>(df_wider <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">all_of</span>(tissuelist)) ,<span class="dv">1</span>,<span class="cf">function</span>(x) <span class="fu">acat</span>(x) ) </span>
<span id="cb27-72"><a href="#cb27-72" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb27-73"><a href="#cb27-73" aria-hidden="true" tabindex="-1"></a>df_wider<span class="sc">$</span>p_acat_6 <span class="ot">=</span> <span class="fu">apply</span>(df_wider <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">"BR"</span>,tissuelist)),) ,<span class="dv">1</span>,<span class="cf">function</span>(x) <span class="fu">acat</span>(x) ) </span>
<span id="cb27-74"><a href="#cb27-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-75"><a href="#cb27-75" aria-hidden="true" tabindex="-1"></a>df_wider <span class="ot">=</span> df_wider <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(gene_annotation <span class="sc">%&gt;%</span> <span class="fu">select</span>(gene_id,gene_name,chr,start),<span class="at">by=</span><span class="fu">c</span>(<span class="st">"gene"</span><span class="ot">=</span><span class="st">"gene_id"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="collect-signs" class="level2">
<h2 class="anchored" data-anchor-id="collect-signs">collect signs</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="do">## pivot wider</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>df_wider_cor <span class="ot">&lt;-</span> df_all <span class="sc">%&gt;%</span> <span class="fu">select</span>(gene,trait,tissue,cor_correct)  <span class="sc">%&gt;%</span> <span class="fu">pivot_wider</span>(</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_from =</span> tissue, </span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_from =</span> cor_correct,</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">id_cols =</span> <span class="fu">c</span>(gene, trait)</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>df_wider_cor<span class="sc">$</span>max_value_original_sign <span class="ot">=</span> <span class="fu">apply</span>(df_wider_cor[, <span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>)], <span class="dv">1</span>, <span class="cf">function</span>(x) {</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>    idx_max_magnitude <span class="ot">=</span> <span class="fu">which.max</span>(<span class="fu">abs</span>(x))</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(x[idx_max_magnitude])</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>df_wider_cor<span class="sc">$</span>best_sign <span class="ot">=</span> <span class="fu">sign</span>(df_wider_cor<span class="sc">$</span>max_value_original_sign)</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(ctrait <span class="cf">in</span> traitlist)</span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>{tempo <span class="ot">=</span> df_wider_cor <span class="sc">%&gt;%</span> </span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trait <span class="sc">==</span> ctrait) <span class="sc">%&gt;%</span> </span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(df_wider <span class="sc">%&gt;%</span> </span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>               <span class="fu">select</span>(gene,trait,p_acat_6), <span class="at">by=</span><span class="fu">c</span>(<span class="st">"gene"</span><span class="ot">=</span><span class="st">"gene"</span>,<span class="st">"trait"</span><span class="ot">=</span><span class="st">"trait"</span>))  <span class="sc">%&gt;%</span></span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(phenomexcan_results <span class="sc">%&gt;%</span></span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a>              <span class="fu">filter</span>(phenotype<span class="sc">==</span>phenolist[ctrait]) <span class="sc">%&gt;%</span></span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a>              <span class="fu">select</span>(<span class="at">p_human=</span>pvalue,</span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a>                     <span class="at">hugo_gene=</span>external_gene_name,</span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a>                     gene_id,rnorvegicus_homolog_ensembl_gene,</span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a>                     dir_effect_most_signif),</span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a>            <span class="at">by=</span><span class="fu">c</span>(<span class="st">"gene"</span><span class="ot">=</span><span class="st">"rnorvegicus_homolog_ensembl_gene"</span>),<span class="at">relationship =</span> <span class="st">"many-to-many"</span>)</span>
<span id="cb28-27"><a href="#cb28-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-28"><a href="#cb28-28" aria-hidden="true" tabindex="-1"></a>ind<span class="ot">=</span>tempo<span class="sc">$</span>p_acat_6 <span class="sc">&lt;</span> <span class="fl">1e-4</span>;</span>
<span id="cb28-29"><a href="#cb28-29" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"---"</span>,ctrait,<span class="st">"---</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb28-30"><a href="#cb28-30" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">fisher.test</span>(tempo<span class="sc">$</span>best_sign[ind], tempo<span class="sc">$</span>dir_effect_most_signif[ind]))</span>
<span id="cb28-31"><a href="#cb28-31" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><mark> no concordance of signs </mark></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>df_wider<span class="sc">$</span>qval <span class="ot">=</span> <span class="cn">NA</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="do">## calculate qvalues</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(ctrait <span class="cf">in</span> traitlist)</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>  ind <span class="ot">=</span> df_wider<span class="sc">$</span>trait <span class="sc">==</span> ctrait</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>  qq <span class="ot">=</span> <span class="fu">qvalue</span>(df_wider<span class="sc">$</span>p_acat_6[ind])</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>  df_wider<span class="sc">$</span>qval[ind] <span class="ot">=</span> qq<span class="sc">$</span>qvalues</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="enrichment-of-human-gene-analysis" class="level2">
<h2 class="anchored" data-anchor-id="enrichment-of-human-gene-analysis">enrichment of human gene analysis</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="do">## annotate with human genes</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="do">## add human pvalues</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="co">#ctrait = "bmi"</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(ctrait <span class="cf">in</span> traitlist)</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">=</span> df_wider <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trait <span class="sc">==</span> ctrait) <span class="sc">%&gt;%</span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(phenomexcan_results <span class="sc">%&gt;%</span> </span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>              <span class="fu">filter</span>(phenotype<span class="sc">==</span>phenolist[ctrait]) <span class="sc">%&gt;%</span></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>              <span class="fu">select</span>(<span class="at">p_human=</span>pvalue,<span class="at">hugo_gene=</span>external_gene_name,</span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>                     gene_id,rnorvegicus_homolog_ensembl_gene, dir_effect_most_signif), </span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>            <span class="at">by=</span><span class="fu">c</span>(<span class="st">"gene"</span><span class="ot">=</span><span class="st">"rnorvegicus_homolog_ensembl_gene"</span>),</span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>            <span class="at">relationship =</span> <span class="st">"many-to-many"</span>)</span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>  mhuman <span class="ot">=</span> <span class="dv">17573</span></span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>  mrat <span class="ot">=</span> <span class="fu">nrow</span>(tempo)</span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"fisher test"</span>,ctrait,<span class="st">"p_acat_6 </span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(thres <span class="cf">in</span> <span class="fu">c</span>(<span class="fl">0.01</span>,<span class="fl">0.05</span>,<span class="fl">0.1</span>))</span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">'---'</span>,thres,<span class="st">'---</span><span class="sc">\n</span><span class="st">'</span>)</span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">with</span>(tempo, <span class="fu">print</span>(<span class="fu">fisher.test</span>(p_acat_6<span class="sc">&lt;</span>thres<span class="sc">/</span>mrat,p_human<span class="sc">&lt;</span>thres<span class="sc">/</span>mhuman)) )</span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">with</span>(tempo, <span class="fu">print</span>(<span class="fu">table</span>(p_acat_6<span class="sc">&lt;</span>thres<span class="sc">/</span>mrat,p_human<span class="sc">&lt;</span>thres<span class="sc">/</span>mhuman)) )</span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"fisher test"</span>,ctrait,<span class="st">"p_acat_5 </span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb30-27"><a href="#cb30-27" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(thres <span class="cf">in</span> <span class="fu">c</span>(<span class="fl">0.01</span>,<span class="fl">0.05</span>,<span class="fl">0.1</span>))</span>
<span id="cb30-28"><a href="#cb30-28" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb30-29"><a href="#cb30-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">'---'</span>,thres,<span class="st">'---</span><span class="sc">\n</span><span class="st">'</span>)</span>
<span id="cb30-30"><a href="#cb30-30" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(thres<span class="sc">==</span><span class="fl">0.05</span>) <span class="fu">cat</span>(<span class="st">'---</span><span class="sc">\n</span><span class="st">--- UPDATE PAPER WITH THIS 0.05 ---</span><span class="sc">\n</span><span class="st">---</span><span class="sc">\n</span><span class="st">'</span>)</span>
<span id="cb30-31"><a href="#cb30-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">with</span>(tempo, <span class="fu">print</span>(<span class="fu">fisher.test</span>(p_acat_5<span class="sc">&lt;</span>thres<span class="sc">/</span>mrat,p_human<span class="sc">&lt;</span>thres<span class="sc">/</span>mhuman)) )</span>
<span id="cb30-32"><a href="#cb30-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">with</span>(tempo, <span class="fu">print</span>(<span class="fu">table</span>(p_acat_5<span class="sc">&lt;</span>thres<span class="sc">/</span>mrat,p_human<span class="sc">&lt;</span>thres<span class="sc">/</span>mhuman)) )</span>
<span id="cb30-33"><a href="#cb30-33" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb30-34"><a href="#cb30-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-35"><a href="#cb30-35" aria-hidden="true" tabindex="-1"></a>  ind <span class="ot">=</span> tempo<span class="sc">$</span>p_acat_6 <span class="sc">&lt;</span> <span class="fl">0.05</span><span class="sc">/</span>mrat </span>
<span id="cb30-36"><a href="#cb30-36" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ind = tempo$p_acat_6 &lt; p_fdr05</span></span>
<span id="cb30-37"><a href="#cb30-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(ctrait)</span>
<span id="cb30-38"><a href="#cb30-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">qqunif.compare</span>(tempo<span class="sc">$</span>p_human,tempo<span class="sc">$</span>p_human[ind],<span class="at">BH=</span><span class="cn">FALSE</span>,<span class="at">BF2=</span><span class="cn">FALSE</span>,<span class="at">main=</span><span class="fu">glue</span>(<span class="st">"human gene assoc for rat {ctrait} genes"</span>),<span class="at">cex2=</span><span class="fl">1.5</span>)</span>
<span id="cb30-39"><a href="#cb30-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/qqunif-compare-{ctrait}.png"</span>),<span class="at">width =</span> <span class="dv">640</span>,<span class="at">height =</span> <span class="dv">480</span>)</span>
<span id="cb30-40"><a href="#cb30-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">qqunif.compare</span>(tempo<span class="sc">$</span>p_human,tempo<span class="sc">$</span>p_human[ind],<span class="at">BH=</span><span class="cn">FALSE</span>,<span class="at">BF2=</span><span class="cn">FALSE</span>,<span class="at">col=</span><span class="st">'maroon1'</span>,<span class="at">col2=</span><span class="st">'dodgerblue'</span>,<span class="at">cex2=</span><span class="fl">1.5</span>)</span>
<span id="cb30-41"><a href="#cb30-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb30-42"><a href="#cb30-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"wilcoxon test"</span>)</span>
<span id="cb30-43"><a href="#cb30-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">wilcox.test</span>(tempo<span class="sc">$</span>p_human[<span class="sc">!</span>ind],tempo<span class="sc">$</span>p_human[ind]))</span>
<span id="cb30-44"><a href="#cb30-44" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-45"><a href="#cb30-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-46"><a href="#cb30-46" aria-hidden="true" tabindex="-1"></a>  ind <span class="ot">=</span> tempo<span class="sc">$</span>p_acat_5 <span class="sc">&lt;</span> <span class="fl">0.05</span><span class="sc">/</span>(tempo <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(p_acat_5)) <span class="sc">%&gt;%</span> <span class="fu">nrow</span>()) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(tempo<span class="sc">$</span>p_acat_5)</span>
<span id="cb30-47"><a href="#cb30-47" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ind = tempo$p_acat_6 &lt; p_fdr05</span></span>
<span id="cb30-48"><a href="#cb30-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(ctrait)</span>
<span id="cb30-49"><a href="#cb30-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">qqunif.compare</span>(tempo<span class="sc">$</span>p_human,tempo<span class="sc">$</span>p_human[ind],<span class="at">BH=</span><span class="cn">FALSE</span>,<span class="at">BF2=</span><span class="cn">FALSE</span>,<span class="at">main=</span><span class="fu">glue</span>(<span class="st">"human gene assoc for rat {ctrait} genes"</span>),<span class="at">cex2=</span><span class="fl">1.5</span>)</span>
<span id="cb30-50"><a href="#cb30-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/qqunif-compare-{ctrait}-p_acat_05.png"</span>),<span class="at">width =</span> <span class="dv">640</span>,<span class="at">height =</span> <span class="dv">480</span>)</span>
<span id="cb30-51"><a href="#cb30-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">qqunif.compare</span>(tempo<span class="sc">$</span>p_human,tempo<span class="sc">$</span>p_human[ind],<span class="at">BH=</span><span class="cn">FALSE</span>,<span class="at">BF2=</span><span class="cn">FALSE</span>,<span class="at">col=</span><span class="st">'maroon1'</span>,<span class="at">col2=</span><span class="st">'dodgerblue'</span>,<span class="at">cex2=</span><span class="fl">1.5</span>)</span>
<span id="cb30-52"><a href="#cb30-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb30-53"><a href="#cb30-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"wilcoxon test"</span>)</span>
<span id="cb30-54"><a href="#cb30-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">wilcox.test</span>(tempo<span class="sc">$</span>p_human[<span class="sc">!</span>ind],tempo<span class="sc">$</span>p_human[ind]))</span>
<span id="cb30-55"><a href="#cb30-55" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-56"><a href="#cb30-56" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-57"><a href="#cb30-57" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-58"><a href="#cb30-58" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(F) <span class="fu">write_csv</span>(tempo <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(p_acat_6) <span class="sc">%&gt;%</span></span>
<span id="cb30-59"><a href="#cb30-59" aria-hidden="true" tabindex="-1"></a>              <span class="fu">select</span>(gene_name,p_acat_6,chr,start,p_human,hugo_gene,</span>
<span id="cb30-60"><a href="#cb30-60" aria-hidden="true" tabindex="-1"></a>                     trait,gene,gene_id,BR,AC,IL,LH,PL,VO,p_acat_5,qval),</span>
<span id="cb30-61"><a href="#cb30-61" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> <span class="fu">glue</span>(<span class="st">"{OUTPUT}/ratxcan-{ctrait}-results-qval.csv"</span>))</span>
<span id="cb30-62"><a href="#cb30-62" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-63"><a href="#cb30-63" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="enrichment-of-crouses-consensus-genes" class="level2">
<h2 class="anchored" data-anchor-id="enrichment-of-crouses-consensus-genes">enrichment of Crouse’s consensus genes</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>consensus_df <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_excel</span>(<span class="fu">glue</span>(<span class="st">"{INPUT}/data/expression/Table S10 Fat_consensus_alpha_0.01 PG.xlsx"</span>),<span class="at">skip=</span><span class="dv">1</span>)</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(consensus_df)</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a><span class="do">## enrichment of consensus genes among bmi associated genes in rats</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>tempo <span class="ot">=</span> df_wider <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trait<span class="sc">==</span><span class="st">"bmi"</span>) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(consensus_df <span class="sc">%&gt;%</span> <span class="fu">select</span>(ensembl_gene_id, description...<span class="dv">3</span>),<span class="at">by=</span><span class="fu">c</span>(<span class="st">"gene"</span><span class="ot">=</span><span class="st">"ensembl_gene_id"</span>)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(gene,p_acat_6,description...<span class="dv">3</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">consensus =</span> <span class="sc">!</span><span class="fu">is.na</span>(description...<span class="dv">3</span>))</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif.compare</span>(tempo<span class="sc">$</span>p_acat_6,tempo<span class="sc">$</span>p_acat_6[tempo<span class="sc">$</span>consensus], <span class="at">main=</span><span class="st">"no enrich. of consensus genes in rat bmi genes"</span>)</span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a><span class="do">## enrichment of consensus genes among bmi associated genes in humans</span></span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a>tempo <span class="ot">=</span> phenomexcan_results <span class="sc">%&gt;%</span> <span class="fu">filter</span>(phenotype<span class="sc">==</span><span class="st">"Body mass index (BMI) (21001_raw)"</span>) <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(consensus_df <span class="sc">%&gt;%</span> <span class="fu">select</span>(ensembl_gene_id, description...<span class="dv">3</span>),<span class="at">by=</span><span class="fu">c</span>(<span class="st">"rnorvegicus_homolog_ensembl_gene"</span><span class="ot">=</span><span class="st">"ensembl_gene_id"</span>)) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">consensus =</span> <span class="sc">!</span><span class="fu">is.na</span>(description...<span class="dv">3</span>)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="at">gene=</span>rnorvegicus_homolog_ensembl_gene,<span class="at">p_human=</span>pvalue,consensus) </span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif.compare</span>(tempo<span class="sc">$</span>p_human,tempo<span class="sc">$</span>p_human[tempo<span class="sc">$</span>consensus], <span class="at">main=</span><span class="st">"no enrich. of consensus genes in human bmi genes"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<blockquote class="blockquote">
<p>Crouse et al’s consensus genes are not enriched among human nor rat BMI genes</p>
</blockquote>
</section>
<section id="summarize-results-for-the-paper" class="level2">
<h2 class="anchored" data-anchor-id="summarize-results-for-the-paper">summarize results for the paper</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>BF_thres <span class="ot">=</span> <span class="fl">0.05</span><span class="sc">/</span> (df_wider <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trait<span class="sc">==</span>ctrait) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(p_acat_6)) <span class="sc">%&gt;%</span> <span class="fu">nrow</span>())</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(ctrait <span class="cf">in</span> traitlist)</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"BF threshold "</span>, <span class="fu">signif</span>(BF_thres,<span class="dv">3</span>),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"---"</span>,ctrait,<span class="st">"---</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>res <span class="ot">=</span> df_wider <span class="sc">%&gt;%</span> </span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trait<span class="sc">==</span>ctrait) <span class="sc">%&gt;%</span> </span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_acat_6 <span class="sc">&lt;</span> BF_thres) <span class="sc">%&gt;%</span></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(trait,p_acat_6,gene_name,chr,start,gene) <span class="sc">%&gt;%</span></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>   <span class="fu">left_join</span>(phenomexcan_results <span class="sc">%&gt;%</span> <span class="fu">filter</span>(phenotype<span class="sc">==</span>phenolist[ctrait]) <span class="sc">%&gt;%</span>              <span class="fu">select</span>(rnorvegicus_homolog_ensembl_gene,<span class="at">p_human=</span>pvalue,<span class="at">hugo=</span>gene_name) , <span class="at">by=</span><span class="fu">c</span>(<span class="st">"gene"</span><span class="ot">=</span><span class="st">"rnorvegicus_homolog_ensembl_gene"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(p_acat_6)</span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(res)</span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>res <span class="ot">=</span> res <span class="sc">%&gt;%</span> <span class="fu">select</span>(gene_name) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">nrow</span>()</span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"---</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(res,<span class="st">" BF significant genes</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a>ntests <span class="ot">=</span> (df_wider <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trait<span class="sc">==</span>ctrait) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(p_acat_5)) <span class="sc">%&gt;%</span> <span class="fu">nrow</span>()) </span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true" tabindex="-1"></a>BF_thres <span class="ot">=</span> <span class="fl">0.05</span><span class="sc">/</span>ntests</span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(ctrait <span class="cf">in</span> traitlist)</span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"BF threshold "</span>, <span class="fu">signif</span>(BF_thres,<span class="dv">3</span>),<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb32-23"><a href="#cb32-23" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"---"</span>,ctrait,<span class="st">"--- p_acat_5 - UPDATE PAPER WITH THIS</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb32-24"><a href="#cb32-24" aria-hidden="true" tabindex="-1"></a>res <span class="ot">=</span> df_wider <span class="sc">%&gt;%</span> </span>
<span id="cb32-25"><a href="#cb32-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(trait<span class="sc">==</span>ctrait) <span class="sc">%&gt;%</span> </span>
<span id="cb32-26"><a href="#cb32-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_acat_5 <span class="sc">&lt;</span> BF_thres) <span class="sc">%&gt;%</span></span>
<span id="cb32-27"><a href="#cb32-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(trait,p_acat_5,gene_name,chr,start,gene) <span class="sc">%&gt;%</span></span>
<span id="cb32-28"><a href="#cb32-28" aria-hidden="true" tabindex="-1"></a>   <span class="fu">left_join</span>(phenomexcan_results <span class="sc">%&gt;%</span> <span class="fu">filter</span>(phenotype<span class="sc">==</span>phenolist[ctrait]) <span class="sc">%&gt;%</span>              <span class="fu">select</span>(rnorvegicus_homolog_ensembl_gene,<span class="at">p_human=</span>pvalue,<span class="at">hugo=</span>gene_name) , <span class="at">by=</span><span class="fu">c</span>(<span class="st">"gene"</span><span class="ot">=</span><span class="st">"rnorvegicus_homolog_ensembl_gene"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-29"><a href="#cb32-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(p_acat_5)</span>
<span id="cb32-30"><a href="#cb32-30" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(res)</span>
<span id="cb32-31"><a href="#cb32-31" aria-hidden="true" tabindex="-1"></a>res <span class="ot">=</span> res <span class="sc">%&gt;%</span> <span class="fu">select</span>(gene_name) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> <span class="fu">nrow</span>()</span>
<span id="cb32-32"><a href="#cb32-32" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"---</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb32-33"><a href="#cb32-33" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(res,<span class="st">" BF significant genes  p_acat_5 - UPDATE PAPER WITH THIS</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb32-34"><a href="#cb32-34" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-35"><a href="#cb32-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-36"><a href="#cb32-36" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(BF_thres,<span class="st">" BF threshold for  p_acat_5 - UPDATE PAPER WITH THIS</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb32-37"><a href="#cb32-37" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(ntests,<span class="st">" number of tests  p_acat_5 - UPDATE PAPER WITH THIS</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="gene-set-enrichment" class="level2">
<h2 class="anchored" data-anchor-id="gene-set-enrichment">gene set enrichment</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>ctrait <span class="ot">=</span> <span class="st">"bmi"</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>thres <span class="ot">=</span> <span class="fl">1e-2</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>top_genes_thres <span class="ot">=</span> df_wider <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trait <span class="sc">==</span> ctrait) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(p_acat_6 <span class="sc">&lt;</span> thres) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene)</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(top_genes_thres,<span class="at">file=</span><span class="st">"~/Downloads/top_genes_thres.txt"</span>)</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>top_genes_n <span class="ot">=</span> df_wider <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trait <span class="sc">==</span> ctrait) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(p_acat_6) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(gene) <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">500</span>)</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(top_genes_n,<span class="at">file=</span><span class="st">"~/Downloads/top_genes_n.txt"</span>)</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a><span class="co">#top_genes_hugo_thres = df_wider %&gt;% filter(trait == ctrait) %&gt;% filter(p_acat_6 &lt; thres) %&gt;% inner_join(orth.rats,by=c("gene"="rnorvegicus_homolog_ensembl_gene")) %&gt;% pull(external_gene_name)</span></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a><span class="co">#write(top_genes_hugo,file="~/Downloads/top_genes_hugo_thres.txt")</span></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(gprofiler2))</span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>gostres <span class="ot">&lt;-</span> <span class="fu">gost</span>(<span class="at">query =</span> top_genes_thres,<span class="at">organism=</span><span class="st">"rnorvegicus"</span>)</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a><span class="co">#gostres &lt;- gost(query = top_genes_thres,organism="hsapiens")</span></span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a><span class="fu">gostplot</span>(gostres)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="define-ggplot-based-manhattan-plotting-function" class="level2">
<h2 class="anchored" data-anchor-id="define-ggplot-based-manhattan-plotting-function">define ggplot based manhattan plotting function</h2>
<p><a href="https://github.com/hakyimlab/Rat_Genomics_Paper_Pipeline/blob/main/analysis/03.PrediXcan.Rmd">from Natasha</a></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="do">## here</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="do">## qq_manhattan(tempo %&gt;% rename(pvalue=p_acat_6))</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>gg_manhattan <span class="ot">&lt;-</span> <span class="cf">function</span>(df, <span class="at">titulo=</span><span class="st">""</span>,<span class="at">significance_threshold =</span> <span class="fl">0.05</span>) {</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>  <span class="do">## USAGE: gg_manhattan(df,0.05)</span></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>  <span class="do">## df has columns: pvalue, chr (numeric), and start (position)</span></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>  <span class="do">## significance threshold gets divided by the number of tests</span></span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>  <span class="do">## </span></span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate cumulative base pair positions</span></span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>  data_cum <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(chr) <span class="sc">%&gt;%</span></span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">max_bp =</span> <span class="fu">as.numeric</span>(<span class="fu">max</span>(start)), <span class="at">.groups =</span> <span class="st">'drop'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">bp_add =</span> <span class="fu">lag</span>(<span class="fu">cumsum</span>(max_bp), <span class="at">default =</span> <span class="dv">0</span>))</span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a>  gwas_data <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(data_cum, <span class="at">by =</span> <span class="st">"chr"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">bp_cum =</span> start <span class="sc">+</span> bp_add)</span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate axis labels</span></span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a>  axis_set <span class="ot">&lt;-</span> gwas_data <span class="sc">%&gt;%</span></span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(chr) <span class="sc">%&gt;%</span></span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarize</span>(<span class="at">center =</span> <span class="fu">mean</span>(bp_cum), <span class="at">.groups =</span> <span class="st">'drop'</span>)</span>
<span id="cb34-23"><a href="#cb34-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-24"><a href="#cb34-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Determine the ylim based on the most significant p-value</span></span>
<span id="cb34-25"><a href="#cb34-25" aria-hidden="true" tabindex="-1"></a>  ylim <span class="ot">&lt;-</span> gwas_data <span class="sc">%&gt;%</span></span>
<span id="cb34-26"><a href="#cb34-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(pvalue <span class="sc">==</span> <span class="fu">min</span>(pvalue)) <span class="sc">%&gt;%</span></span>
<span id="cb34-27"><a href="#cb34-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">ylim =</span> <span class="fu">abs</span>(<span class="fu">floor</span>(<span class="fu">log10</span>(pvalue))) <span class="sc">+</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-28"><a href="#cb34-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(ylim)</span>
<span id="cb34-29"><a href="#cb34-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-30"><a href="#cb34-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate the genome-wide significance level</span></span>
<span id="cb34-31"><a href="#cb34-31" aria-hidden="true" tabindex="-1"></a>  sig <span class="ot">&lt;-</span> significance_threshold <span class="sc">/</span> <span class="fu">nrow</span>(df)</span>
<span id="cb34-32"><a href="#cb34-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-33"><a href="#cb34-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Construct the Manhattan plot</span></span>
<span id="cb34-34"><a href="#cb34-34" aria-hidden="true" tabindex="-1"></a>  manhattan_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(gwas_data, <span class="fu">aes</span>(<span class="at">x =</span> bp_cum, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pvalue), <span class="at">color =</span> <span class="fu">as.factor</span>(chr), <span class="at">size =</span> <span class="sc">-</span><span class="fu">log10</span>(pvalue))) <span class="sc">+</span></span>
<span id="cb34-35"><a href="#cb34-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(sig), <span class="at">color =</span> <span class="st">"grey40"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb34-36"><a href="#cb34-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.0001</span>), <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb34-37"><a href="#cb34-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.75</span>, <span class="at">shape =</span> <span class="dv">19</span>) <span class="sc">+</span> <span class="co"># Simplified shape decision for clarity</span></span>
<span id="cb34-38"><a href="#cb34-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_label_repel</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">ifelse</span>(pvalue <span class="sc">&lt;=</span> sig, gene_name, <span class="st">""</span>)), <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb34-39"><a href="#cb34-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>, ylim)) <span class="sc">+</span></span>
<span id="cb34-40"><a href="#cb34-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">labels =</span> axis_set<span class="sc">$</span>chr, <span class="at">breaks =</span> axis_set<span class="sc">$</span>center) <span class="sc">+</span></span>
<span id="cb34-41"><a href="#cb34-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">"dodgerblue4"</span>, <span class="st">"midnightblue"</span>), <span class="fu">length</span>(<span class="fu">unique</span>(axis_set<span class="sc">$</span>chr)))) <span class="sc">+</span></span>
<span id="cb34-42"><a href="#cb34-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_size_continuous</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="fl">0.5</span>, <span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb34-43"><a href="#cb34-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, <span class="at">y =</span> <span class="fu">expression</span>(<span class="sc">-</span>log[<span class="dv">10</span>](<span class="fu">italic</span>(p)))) <span class="sc">+</span></span>
<span id="cb34-44"><a href="#cb34-44" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb34-45"><a href="#cb34-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb34-46"><a href="#cb34-46" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb34-47"><a href="#cb34-47" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb34-48"><a href="#cb34-48" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid.minor.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb34-49"><a href="#cb34-49" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb34-50"><a href="#cb34-50" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">vjust =</span> <span class="dv">0</span>),</span>
<span id="cb34-51"><a href="#cb34-51" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>))</span>
<span id="cb34-52"><a href="#cb34-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-53"><a href="#cb34-53" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(titulo <span class="sc">!=</span><span class="st">""</span>) manhattan_plot <span class="ot">=</span> manhattan_plot <span class="sc">+</span> <span class="fu">ggtitle</span>(titulo)</span>
<span id="cb34-54"><a href="#cb34-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(manhattan_plot)</span>
<span id="cb34-55"><a href="#cb34-55" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-manhattan-plots-of-bmi-and-body-length" class="level2">
<h2 class="anchored" data-anchor-id="plot-manhattan-plots-of-bmi-and-body-length">plot manhattan plots of bmi and body length</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(ctrait <span class="cf">in</span> traitlist)</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">=</span> df_wider <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trait<span class="sc">==</span>ctrait) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">chr=</span><span class="fu">as.numeric</span>(chr))</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>  gg <span class="ot">=</span> <span class="fu">gg_manhattan</span>(tempo <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">pvalue=</span>p_acat_6), <span class="at">titulo=</span>ctrait)</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(gg)</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggsave</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/{ctrait}-manhattan-p_acat_6.png"</span>))</span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(ctrait <span class="cf">in</span> traitlist)</span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">=</span> df_wider <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trait<span class="sc">==</span>ctrait) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">chr=</span><span class="fu">as.numeric</span>(chr))</span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>  gg <span class="ot">=</span> <span class="fu">gg_manhattan</span>(tempo <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">pvalue=</span>p_acat_5) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(pvalue)), <span class="at">titulo=</span>ctrait)</span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(gg)</span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggsave</span>(<span class="fu">glue</span>(<span class="st">"{OUTPUT}/{ctrait}-manhattan-p_acat_5.png"</span>))</span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="double-checking-that-the-saved-qvalues-are-correct" class="level2">
<h2 class="anchored" data-anchor-id="double-checking-that-the-saved-qvalues-are-correct">double checking that the saved qvalues are correct</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>kk5 <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/haekyungim/Downloads/ratxcan-bmi-results - ratxcan-bmi-results(1).tsv"</span>)</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>kk6 <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">"/Users/haekyungim/Downloads/ratxcan-bodylen-results - ratxcan-bodylen-results(1).tsv"</span>)</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="fu">all.equal</span>(<span class="fu">qvalue</span>(kk5<span class="sc">$</span>p_acat_6)<span class="sc">$</span>qvalues,kk5<span class="sc">$</span>qval)</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a><span class="fu">all.equal</span>(<span class="fu">qvalue</span>(kk6<span class="sc">$</span>p_acat_6)<span class="sc">$</span>qvalues,kk6<span class="sc">$</span>qval)</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">qvalue</span>(kk5<span class="sc">$</span>p_acat_6)<span class="sc">$</span>qvalues,kk5<span class="sc">$</span>qval); <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">qvalue</span>(kk6<span class="sc">$</span>p_acat_6)<span class="sc">$</span>qvalues,kk6<span class="sc">$</span>qval); <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a><span class="co"># kk1 = read_tsv("/Users/haekyungim/Downloads/ratxcan-bmi-results - ratxcan-bmi-results.tsv")</span></span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a><span class="co"># kk2 = read_tsv("/Users/haekyungim/Downloads/with-qval-ratxcan-bmi-results - ratxcan-bmi-results.tsv")</span></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a><span class="co"># kk3 = read_tsv("/Users/haekyungim/Downloads/ratxcan-bodylen-results - ratxcan-bodylen-results.tsv")</span></span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a><span class="co"># kk4 = read_tsv("/Users/haekyungim/Downloads/with-qval-ratxcan-bodylen-results - ratxcan-bodylen-results.tsv")</span></span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a><span class="co"># all.equal(kk1$p_acat_6,kk2$p_acat_6)</span></span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a><span class="co"># all.equal(kk3$p_acat_6,kk4$p_acat_6)</span></span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a><span class="co"># all.equal(qvalue(kk1$p_acat_6)$qvalues,kk2$qval)</span></span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a><span class="co"># all.equal(qvalue(kk3$p_acat_6)$qvalues,kk4$qval)</span></span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a><span class="do">## small difference between qvalues were calculated before joining with human data</span></span>
<span id="cb36-20"><a href="#cb36-20" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">qvalue</span>(kk1<span class="sc">$</span>p_acat_6)<span class="sc">$</span>qvalues,kk2<span class="sc">$</span>qval); <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb36-21"><a href="#cb36-21" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">qvalue</span>(kk3<span class="sc">$</span>p_acat_6)<span class="sc">$</span>qvalues,kk4<span class="sc">$</span>qval); <span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>



</section>

<div class="quarto-listing quarto-listing-container-default" id="listing-listing">
<div class="list quarto-listing-default">

</div>
<div class="listing-no-matching d-none">
No matching items
</div>
</div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>