<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.361">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Natasha Santhanam">
<meta name="dcterms.date" content="2021-10-12">

<title>Rat Genomics Pipeline Rendered - PhenomeXcan Query</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-listing/list.min.js"></script>
<script src="../site_libs/quarto-listing/quarto-listing.js"></script>
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script>

  window.document.addEventListener("DOMContentLoaded", function (_event) {
    const listingTargetEl = window.document.querySelector('#listing-listing .list');
    if (!listingTargetEl) {
      // No listing discovered, do not attach.
      return; 
    }

    const options = {
      valueNames: [{ data: ['index'] },{ data: ['categories'] },{ data: ['listing-date-sort'] },{ data: ['listing-file-modified-sort'] }],
      
      searchColumns: [],
    };

    window['quarto-listings'] = window['quarto-listings'] || {};
    window['quarto-listings']['listing-listing'] = new List('listing-listing', options);

    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  });

  window.addEventListener('hashchange',() => {
    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  })
  </script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Rat Genomics Pipeline Rendered</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html" rel="" target="">
 <span class="menu-text">paper_code</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../other_code.html" rel="" target="">
 <span class="menu-text">more_code</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#query-phenomexcan-association-with-top-phenotypes-for-list-of-genes" id="toc-query-phenomexcan-association-with-top-phenotypes-for-list-of-genes" class="nav-link active" data-scroll-target="#query-phenomexcan-association-with-top-phenotypes-for-list-of-genes">query PhenomeXcan association with top phenotypes for list of genes</a></li>
  <li><a href="#enrichment-analysis" id="toc-enrichment-analysis" class="nav-link" data-scroll-target="#enrichment-analysis">Enrichment Analysis</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">PhenomeXcan Query</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Natasha Santhanam </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">10/12/2021</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">source</span>(data.dir <span class="sc">%&amp;%</span> <span class="st">"helpers.R"</span>, <span class="at">chdir =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>phenomexcan_con <span class="ot">&lt;-</span> <span class="fu">get_db</span>()</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dbListTables</span>(phenomexcan_con)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="query-phenomexcan-association-with-top-phenotypes-for-list-of-genes" class="level2">
<h2 class="anchored" data-anchor-id="query-phenomexcan-association-with-top-phenotypes-for-list-of-genes">query PhenomeXcan association with top phenotypes for list of genes</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>input <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>input<span class="sc">$</span>pheno <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Obesity"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>input<span class="sc">$</span>limit <span class="ot">=</span> <span class="dv">30000</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>obesity_genes <span class="ot">&lt;-</span>  <span class="fu">suppressMessages</span>(<span class="fu">get_results_from_data_db</span>(input))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>input <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>input<span class="sc">$</span>pheno <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Body fat percentage"</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>input<span class="sc">$</span>limit <span class="ot">=</span> <span class="dv">30000</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>body_fat_genes <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">get_results_from_data_db</span>(input))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>input <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>input<span class="sc">$</span>pheno <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Body mass index (BMI) (21001_raw)"</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>input<span class="sc">$</span>limit <span class="ot">=</span> <span class="dv">30000</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>BMI_genes <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">get_results_from_data_db</span>(input))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>input <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>input<span class="sc">$</span>pheno <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Fasting Glucose"</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>input<span class="sc">$</span>limit <span class="ot">=</span> <span class="dv">30000</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>glucose_genes <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">get_results_from_data_db</span>(input))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>input <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>input<span class="sc">$</span>pheno <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Height"</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>input<span class="sc">$</span>limit <span class="ot">=</span> <span class="dv">30000</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>height_genes <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">get_results_from_data_db</span>(input))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Generate a table of all human MultiXcan results</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">#matrix - humans (rows are genes and columns are traits (fat, BMI, Obesity))</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>listphenos <span class="ot">&lt;-</span> <span class="fu">list</span>(BMI_genes, body_fat_genes, obesity_genes, glucose_genes, height_genes)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>pheno_Multi_humans <span class="ot">&lt;-</span> <span class="fu">data_frame</span>(<span class="at">gene_name =</span> <span class="fu">as.character</span>())</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(l <span class="cf">in</span> listphenos) {</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  trait <span class="ot">&lt;-</span> l<span class="sc">$</span>phenotype[<span class="dv">1</span>]</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> l <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(gene_name, pvalue))</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tempo)[<span class="dv">2</span>] <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">"pvalue"</span>, trait, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  pheno_Multi_humans <span class="ot">&lt;-</span> <span class="fu">full_join</span>(pheno_Multi_humans, tempo, <span class="at">by =</span> <span class="st">"gene_name"</span>)</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>human_genes <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pheno_Multi_humans<span class="sc">$</span>gene_name)</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="co">#pheno_humans &lt;- as.matrix(pheno_humans %&gt;% dplyr::select(-c(gene_name)))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>human <span class="ot">=</span> <span class="fu">useEnsembl</span>(<span class="at">biomart=</span><span class="st">'ensembl'</span>, <span class="at">dataset=</span><span class="st">"hsapiens_gene_ensembl"</span>, <span class="at">mirror =</span> <span class="st">"uswest"</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co">#human = useMart("ensembl", dataset = "hsapiens_gene_ensembl", mirror = "uswest")</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>attributes <span class="ot">&lt;-</span> <span class="fu">listAttributes</span>(human)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>attributes <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span>, <span class="st">"external_gene_name"</span>, <span class="st">"rnorvegicus_homolog_ensembl_gene"</span>, <span class="st">"rnorvegicus_homolog_associated_gene_name"</span>)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>orth.rats <span class="ot">=</span> <span class="fu">getBM</span>(attributes, <span class="at">filters=</span><span class="st">"with_rnorvegicus_homolog"</span>,<span class="at">values=</span><span class="cn">TRUE</span>, <span class="at">mart =</span> human, <span class="at">uniqueRows=</span><span class="cn">TRUE</span>)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>human_genes <span class="ot">&lt;-</span> human_genes <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">external_gene_name =</span> <span class="st">`</span><span class="at">pheno_Multi_humans$gene_name</span><span class="st">`</span>)</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>human_genes <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(human_genes, orth.rats, <span class="at">by =</span> <span class="st">"external_gene_name"</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(external_gene_name, rnorvegicus_homolog_associated_gene_name))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Generate a table of all rat MultiXcan results</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/MultiXcan/results"</span>, <span class="at">pattern =</span> <span class="st">"assoc.txt"</span>,<span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>pheno_Multi_rat<span class="ot">&lt;-</span> <span class="fu">data_frame</span>(<span class="at">gene =</span> <span class="fu">as.character</span>())</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  trait <span class="ot">&lt;-</span> <span class="fu">substr</span>(fila, <span class="dv">66</span>, <span class="fu">str_length</span>(fila)<span class="sc">-</span><span class="dv">18</span>)</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">fread</span>(fila) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, pvalue))</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tempo)[<span class="dv">2</span>] <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">"pvalue"</span>, trait, <span class="at">sep=</span><span class="st">"."</span>)</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>  pheno_Multi_rat<span class="ot">&lt;-</span> <span class="fu">full_join</span>(pheno_Multi_rat, tempo, <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="co">#pheno_Multi_rat &lt;- read_tsv("/Users/natashasanthanam/Downloads/rat_metabolic_MultiXcan_pval_assoc.txt", col_names = TRUE)</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>pheno_Multi_rat <span class="ot">&lt;-</span> pheno_Multi_rat <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">gene_name =</span> orth.rats[<span class="fu">match</span>(pheno_Multi_rat<span class="sc">$</span>gene, orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene),<span class="dv">2</span>]<span class="sc">$</span>external_gene_name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="enrichment-analysis" class="level1">
<h1>Enrichment Analysis</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">source_gist</span>(<span class="st">"38431b74c6c0bf90c12f"</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">source_gist</span>(<span class="st">"1e9053c8f35c30396429350a08f33ea7"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> full_df <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">gene_name =</span> orth.rats[<span class="fu">match</span>(full_df<span class="sc">$</span>gene, orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene), <span class="dv">2</span>]<span class="sc">$</span>external_gene_name)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif</span>(BMI_genes<span class="sc">$</span>pvalue, <span class="at">col=</span> <span class="st">"maroon1"</span> )</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(pheno_Multi_humans)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"pvalue_BMI"</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>BMI_genes_rats <span class="ot">&lt;-</span> full_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(metabolic_trait <span class="sc">==</span> <span class="st">"Body Mass Index (BMI) with tail"</span>)  <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pvalue <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> .[[<span class="st">"gene_name"</span>]]</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="fu">qqpoints</span>(BMI_genes <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene_name <span class="sc">%in%</span> BMI_genes_rats) <span class="sc">%&gt;%</span> .[[<span class="st">"pvalue"</span>]],<span class="at">pch=</span><span class="st">'+'</span>, <span class="at">col =</span> <span class="st">"dodgerblue"</span>) </span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif</span>(height_genes<span class="sc">$</span>pvalue, <span class="at">col =</span> <span class="st">"maroon1"</span>)</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>height_genes_rats <span class="ot">&lt;-</span> full_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(metabolic_trait <span class="sc">==</span> <span class="st">"Body length including tail"</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pvalue <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> .[[<span class="st">"gene_name"</span>]]</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a><span class="fu">qqpoints</span>(height_genes <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene_name <span class="sc">%in%</span> height_genes_rats) <span class="sc">%&gt;%</span> .[[<span class="st">"pvalue"</span>]],<span class="at">pch=</span><span class="st">'+'</span>, <span class="at">col=</span><span class="st">"dodgerblue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif</span>(BMI_genes<span class="sc">$</span>pvalue, <span class="at">col=</span> <span class="st">"maroon1"</span>) </span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>BMI_genes_rats <span class="ot">&lt;-</span> pheno_Multi_rat  <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pvalue.bmi_bodylength_wo_tail <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> .[[<span class="st">"gene_name"</span>]] <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">qqpoints</span>(BMI_genes <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene_name <span class="sc">%in%</span> BMI_genes_rats) <span class="sc">%&gt;%</span> .[[<span class="st">"pvalue"</span>]],<span class="at">pch=</span><span class="st">'+'</span>, <span class="at">col=</span><span class="st">"dodgerblue"</span>) </span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif</span>(height_genes<span class="sc">$</span>pvalue, <span class="at">col=</span> <span class="st">"maroon1"</span>)</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>height_genes_rats <span class="ot">&lt;-</span> pheno_Multi_rat <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pvalue.bodylength_w_tail <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> .[[<span class="st">"gene_name"</span>]] <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="fu">qqpoints</span>(height_genes <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene_name <span class="sc">%in%</span> height_genes_rats) <span class="sc">%&gt;%</span> .[[<span class="st">"pvalue"</span>]],<span class="at">pch=</span><span class="st">'+'</span>, <span class="at">col=</span><span class="st">"dodgerblue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>all_bmi_genes <span class="ot">&lt;-</span> BMI_genes <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">in_rats =</span> <span class="fu">ifelse</span>(BMI_genes<span class="sc">$</span>gene_name <span class="sc">%in%</span> BMI_genes_rats, <span class="dv">1</span>, <span class="dv">0</span>))</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>all_height_genes <span class="ot">&lt;-</span> height_genes <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">in_rats =</span> <span class="fu">ifelse</span>(height_genes<span class="sc">$</span>gene_name <span class="sc">%in%</span> height_genes_rats, <span class="dv">1</span>, <span class="dv">0</span>))</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(pvalue <span class="sc">~</span> in_rats, <span class="at">data=</span>all_bmi_genes) </span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(pvalue <span class="sc">~</span> in_rats, <span class="at">data=</span>all_height_genes) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>



<!-- -->

</section>

<div class="quarto-listing quarto-listing-container-default" id="listing-listing">
<div class="list quarto-listing-default">

</div>
<div class="listing-no-matching d-none">
No matching items
</div>
</div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb14" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "05.PhenomeXcan_MultiXcan.Rmd"</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Natasha Santhanam"</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2/14/2022"</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="an">output:</span><span class="co"> html_document</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>---</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>title: "PhenomeXcan Query"</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>author: "Natasha Santhanam"</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>date: "10/12/2021"</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>output: html_document</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>---</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup, include=FALSE}</span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RSQLite)</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a><span class="st">"%&amp;%"</span> <span class="ot">=</span> <span class="cf">function</span>(a,b) <span class="fu">paste</span>(a,b,<span class="at">sep=</span><span class="st">""</span>)</span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>data.dir <span class="ot">&lt;-</span> <span class="st">"/Users/natashasanthanam/Github/rat-genomic-analysis/"</span></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">source</span>(data.dir <span class="sc">%&amp;%</span> <span class="st">"helpers.R"</span>, <span class="at">chdir =</span> <span class="cn">TRUE</span>))</span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>phenomexcan_con <span class="ot">&lt;-</span> <span class="fu">get_db</span>()</span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a><span class="fu">dbListTables</span>(phenomexcan_con)</span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a><span class="fu">## query PhenomeXcan association with top phenotypes for list of genes</span></span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a><span class="in">```{r query PhenomeXcan for top genes associated with Obesity }</span></span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>input <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>input<span class="sc">$</span>pheno <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Obesity"</span>)</span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>input<span class="sc">$</span>limit <span class="ot">=</span> <span class="dv">30000</span></span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>obesity_genes <span class="ot">&lt;-</span>  <span class="fu">suppressMessages</span>(<span class="fu">get_results_from_data_db</span>(input))</span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a><span class="in">```{r query PhenomeXcan for top genes associated with Body fat percentage }</span></span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a>input <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a>input<span class="sc">$</span>pheno <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Body fat percentage"</span>)</span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a>input<span class="sc">$</span>limit <span class="ot">=</span> <span class="dv">30000</span></span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a>body_fat_genes <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">get_results_from_data_db</span>(input))</span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-48"><a href="#cb14-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-49"><a href="#cb14-49" aria-hidden="true" tabindex="-1"></a><span class="in">```{r query PhenomeXcan for top genes associated with BMI }</span></span>
<span id="cb14-50"><a href="#cb14-50" aria-hidden="true" tabindex="-1"></a>input <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb14-51"><a href="#cb14-51" aria-hidden="true" tabindex="-1"></a>input<span class="sc">$</span>pheno <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Body mass index (BMI) (21001_raw)"</span>)</span>
<span id="cb14-52"><a href="#cb14-52" aria-hidden="true" tabindex="-1"></a>input<span class="sc">$</span>limit <span class="ot">=</span> <span class="dv">30000</span></span>
<span id="cb14-53"><a href="#cb14-53" aria-hidden="true" tabindex="-1"></a>BMI_genes <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">get_results_from_data_db</span>(input))</span>
<span id="cb14-54"><a href="#cb14-54" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-55"><a href="#cb14-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-56"><a href="#cb14-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-57"><a href="#cb14-57" aria-hidden="true" tabindex="-1"></a><span class="in">```{r query PhenomeXcan for top genes associated with Fasting Glucose }</span></span>
<span id="cb14-58"><a href="#cb14-58" aria-hidden="true" tabindex="-1"></a>input <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb14-59"><a href="#cb14-59" aria-hidden="true" tabindex="-1"></a>input<span class="sc">$</span>pheno <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Fasting Glucose"</span>)</span>
<span id="cb14-60"><a href="#cb14-60" aria-hidden="true" tabindex="-1"></a>input<span class="sc">$</span>limit <span class="ot">=</span> <span class="dv">30000</span></span>
<span id="cb14-61"><a href="#cb14-61" aria-hidden="true" tabindex="-1"></a>glucose_genes <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">get_results_from_data_db</span>(input))</span>
<span id="cb14-62"><a href="#cb14-62" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-63"><a href="#cb14-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-64"><a href="#cb14-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-65"><a href="#cb14-65" aria-hidden="true" tabindex="-1"></a><span class="in">```{r query PhenomeXcan for top genes associated with Height }</span></span>
<span id="cb14-66"><a href="#cb14-66" aria-hidden="true" tabindex="-1"></a>input <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb14-67"><a href="#cb14-67" aria-hidden="true" tabindex="-1"></a>input<span class="sc">$</span>pheno <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Height"</span>)</span>
<span id="cb14-68"><a href="#cb14-68" aria-hidden="true" tabindex="-1"></a>input<span class="sc">$</span>limit <span class="ot">=</span> <span class="dv">30000</span></span>
<span id="cb14-69"><a href="#cb14-69" aria-hidden="true" tabindex="-1"></a>height_genes <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">get_results_from_data_db</span>(input))</span>
<span id="cb14-70"><a href="#cb14-70" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-71"><a href="#cb14-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-72"><a href="#cb14-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-73"><a href="#cb14-73" aria-hidden="true" tabindex="-1"></a>Generate a table of all human MultiXcan results</span>
<span id="cb14-74"><a href="#cb14-74" aria-hidden="true" tabindex="-1"></a><span class="in">```{r create phenotype matrix in Humans}</span></span>
<span id="cb14-75"><a href="#cb14-75" aria-hidden="true" tabindex="-1"></a><span class="co">#matrix - humans (rows are genes and columns are traits (fat, BMI, Obesity))</span></span>
<span id="cb14-76"><a href="#cb14-76" aria-hidden="true" tabindex="-1"></a>listphenos <span class="ot">&lt;-</span> <span class="fu">list</span>(BMI_genes, body_fat_genes, obesity_genes, glucose_genes, height_genes)</span>
<span id="cb14-77"><a href="#cb14-77" aria-hidden="true" tabindex="-1"></a>pheno_Multi_humans <span class="ot">&lt;-</span> <span class="fu">data_frame</span>(<span class="at">gene_name =</span> <span class="fu">as.character</span>())</span>
<span id="cb14-78"><a href="#cb14-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-79"><a href="#cb14-79" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(l <span class="cf">in</span> listphenos) {</span>
<span id="cb14-80"><a href="#cb14-80" aria-hidden="true" tabindex="-1"></a>  trait <span class="ot">&lt;-</span> l<span class="sc">$</span>phenotype[<span class="dv">1</span>]</span>
<span id="cb14-81"><a href="#cb14-81" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> l <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(gene_name, pvalue))</span>
<span id="cb14-82"><a href="#cb14-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tempo)[<span class="dv">2</span>] <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">"pvalue"</span>, trait, <span class="at">sep=</span><span class="st">"_"</span>)</span>
<span id="cb14-83"><a href="#cb14-83" aria-hidden="true" tabindex="-1"></a>  pheno_Multi_humans <span class="ot">&lt;-</span> <span class="fu">full_join</span>(pheno_Multi_humans, tempo, <span class="at">by =</span> <span class="st">"gene_name"</span>)</span>
<span id="cb14-84"><a href="#cb14-84" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-85"><a href="#cb14-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-86"><a href="#cb14-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-87"><a href="#cb14-87" aria-hidden="true" tabindex="-1"></a>human_genes <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pheno_Multi_humans<span class="sc">$</span>gene_name)</span>
<span id="cb14-88"><a href="#cb14-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-89"><a href="#cb14-89" aria-hidden="true" tabindex="-1"></a><span class="co">#pheno_humans &lt;- as.matrix(pheno_humans %&gt;% dplyr::select(-c(gene_name)))</span></span>
<span id="cb14-90"><a href="#cb14-90" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-91"><a href="#cb14-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-92"><a href="#cb14-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-93"><a href="#cb14-93" aria-hidden="true" tabindex="-1"></a><span class="in">```{r find orthologs in rats}</span></span>
<span id="cb14-94"><a href="#cb14-94" aria-hidden="true" tabindex="-1"></a>human <span class="ot">=</span> <span class="fu">useEnsembl</span>(<span class="at">biomart=</span><span class="st">'ensembl'</span>, <span class="at">dataset=</span><span class="st">"hsapiens_gene_ensembl"</span>, <span class="at">mirror =</span> <span class="st">"uswest"</span>)</span>
<span id="cb14-95"><a href="#cb14-95" aria-hidden="true" tabindex="-1"></a><span class="co">#human = useMart("ensembl", dataset = "hsapiens_gene_ensembl", mirror = "uswest")</span></span>
<span id="cb14-96"><a href="#cb14-96" aria-hidden="true" tabindex="-1"></a>attributes <span class="ot">&lt;-</span> <span class="fu">listAttributes</span>(human)</span>
<span id="cb14-97"><a href="#cb14-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-98"><a href="#cb14-98" aria-hidden="true" tabindex="-1"></a>attributes <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span>, <span class="st">"external_gene_name"</span>, <span class="st">"rnorvegicus_homolog_ensembl_gene"</span>, <span class="st">"rnorvegicus_homolog_associated_gene_name"</span>)</span>
<span id="cb14-99"><a href="#cb14-99" aria-hidden="true" tabindex="-1"></a>orth.rats <span class="ot">=</span> <span class="fu">getBM</span>(attributes, <span class="at">filters=</span><span class="st">"with_rnorvegicus_homolog"</span>,<span class="at">values=</span><span class="cn">TRUE</span>, <span class="at">mart =</span> human, <span class="at">uniqueRows=</span><span class="cn">TRUE</span>)</span>
<span id="cb14-100"><a href="#cb14-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-101"><a href="#cb14-101" aria-hidden="true" tabindex="-1"></a>human_genes <span class="ot">&lt;-</span> human_genes <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">external_gene_name =</span> <span class="st">`</span><span class="at">pheno_Multi_humans$gene_name</span><span class="st">`</span>)</span>
<span id="cb14-102"><a href="#cb14-102" aria-hidden="true" tabindex="-1"></a>human_genes <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(human_genes, orth.rats, <span class="at">by =</span> <span class="st">"external_gene_name"</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(external_gene_name, rnorvegicus_homolog_associated_gene_name))</span>
<span id="cb14-103"><a href="#cb14-103" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-104"><a href="#cb14-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-105"><a href="#cb14-105" aria-hidden="true" tabindex="-1"></a>Generate a table of all rat MultiXcan results</span>
<span id="cb14-106"><a href="#cb14-106" aria-hidden="true" tabindex="-1"></a><span class="in">```{r make rat multiXcan matrix, eval=FALSE}</span></span>
<span id="cb14-107"><a href="#cb14-107" aria-hidden="true" tabindex="-1"></a>filelist <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"/gpfs/data/im-lab/nas40t2/natasha/rat_genomics/MultiXcan/results"</span>, <span class="at">pattern =</span> <span class="st">"assoc.txt"</span>,<span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-108"><a href="#cb14-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-109"><a href="#cb14-109" aria-hidden="true" tabindex="-1"></a>pheno_Multi_rat<span class="ot">&lt;-</span> <span class="fu">data_frame</span>(<span class="at">gene =</span> <span class="fu">as.character</span>())</span>
<span id="cb14-110"><a href="#cb14-110" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fila <span class="cf">in</span> filelist) {</span>
<span id="cb14-111"><a href="#cb14-111" aria-hidden="true" tabindex="-1"></a>  trait <span class="ot">&lt;-</span> <span class="fu">substr</span>(fila, <span class="dv">66</span>, <span class="fu">str_length</span>(fila)<span class="sc">-</span><span class="dv">18</span>)</span>
<span id="cb14-112"><a href="#cb14-112" aria-hidden="true" tabindex="-1"></a>  tempo <span class="ot">&lt;-</span> <span class="fu">fread</span>(fila) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(gene, pvalue))</span>
<span id="cb14-113"><a href="#cb14-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(tempo)[<span class="dv">2</span>] <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">"pvalue"</span>, trait, <span class="at">sep=</span><span class="st">"."</span>)</span>
<span id="cb14-114"><a href="#cb14-114" aria-hidden="true" tabindex="-1"></a>  pheno_Multi_rat<span class="ot">&lt;-</span> <span class="fu">full_join</span>(pheno_Multi_rat, tempo, <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb14-115"><a href="#cb14-115" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-116"><a href="#cb14-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-117"><a href="#cb14-117" aria-hidden="true" tabindex="-1"></a><span class="co">#pheno_Multi_rat &lt;- read_tsv("/Users/natashasanthanam/Downloads/rat_metabolic_MultiXcan_pval_assoc.txt", col_names = TRUE)</span></span>
<span id="cb14-118"><a href="#cb14-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-119"><a href="#cb14-119" aria-hidden="true" tabindex="-1"></a>pheno_Multi_rat <span class="ot">&lt;-</span> pheno_Multi_rat <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">gene_name =</span> orth.rats[<span class="fu">match</span>(pheno_Multi_rat<span class="sc">$</span>gene, orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene),<span class="dv">2</span>]<span class="sc">$</span>external_gene_name)</span>
<span id="cb14-120"><a href="#cb14-120" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-121"><a href="#cb14-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-122"><a href="#cb14-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-123"><a href="#cb14-123" aria-hidden="true" tabindex="-1"></a><span class="fu"># Enrichment Analysis</span></span>
<span id="cb14-124"><a href="#cb14-124" aria-hidden="true" tabindex="-1"></a><span class="in">```{r enrichment for fasting glucose}</span></span>
<span id="cb14-125"><a href="#cb14-125" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">source_gist</span>(<span class="st">"38431b74c6c0bf90c12f"</span>)</span>
<span id="cb14-126"><a href="#cb14-126" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">source_gist</span>(<span class="st">"1e9053c8f35c30396429350a08f33ea7"</span>)</span>
<span id="cb14-127"><a href="#cb14-127" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-128"><a href="#cb14-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-129"><a href="#cb14-129" aria-hidden="true" tabindex="-1"></a><span class="in">```{r enrichment for BMI and Height using prediXcan}</span></span>
<span id="cb14-130"><a href="#cb14-130" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> full_df <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">gene_name =</span> orth.rats[<span class="fu">match</span>(full_df<span class="sc">$</span>gene, orth.rats<span class="sc">$</span>rnorvegicus_homolog_ensembl_gene), <span class="dv">2</span>]<span class="sc">$</span>external_gene_name)</span>
<span id="cb14-131"><a href="#cb14-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-132"><a href="#cb14-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-133"><a href="#cb14-133" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif</span>(BMI_genes<span class="sc">$</span>pvalue, <span class="at">col=</span> <span class="st">"maroon1"</span> )</span>
<span id="cb14-134"><a href="#cb14-134" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(pheno_Multi_humans)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"pvalue_BMI"</span></span>
<span id="cb14-135"><a href="#cb14-135" aria-hidden="true" tabindex="-1"></a>BMI_genes_rats <span class="ot">&lt;-</span> full_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(metabolic_trait <span class="sc">==</span> <span class="st">"Body Mass Index (BMI) with tail"</span>)  <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pvalue <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> .[[<span class="st">"gene_name"</span>]]</span>
<span id="cb14-136"><a href="#cb14-136" aria-hidden="true" tabindex="-1"></a><span class="fu">qqpoints</span>(BMI_genes <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene_name <span class="sc">%in%</span> BMI_genes_rats) <span class="sc">%&gt;%</span> .[[<span class="st">"pvalue"</span>]],<span class="at">pch=</span><span class="st">'+'</span>, <span class="at">col =</span> <span class="st">"dodgerblue"</span>) </span>
<span id="cb14-137"><a href="#cb14-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-138"><a href="#cb14-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-139"><a href="#cb14-139" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif</span>(height_genes<span class="sc">$</span>pvalue, <span class="at">col =</span> <span class="st">"maroon1"</span>)</span>
<span id="cb14-140"><a href="#cb14-140" aria-hidden="true" tabindex="-1"></a>height_genes_rats <span class="ot">&lt;-</span> full_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(metabolic_trait <span class="sc">==</span> <span class="st">"Body length including tail"</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pvalue <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> .[[<span class="st">"gene_name"</span>]]</span>
<span id="cb14-141"><a href="#cb14-141" aria-hidden="true" tabindex="-1"></a><span class="fu">qqpoints</span>(height_genes <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene_name <span class="sc">%in%</span> height_genes_rats) <span class="sc">%&gt;%</span> .[[<span class="st">"pvalue"</span>]],<span class="at">pch=</span><span class="st">'+'</span>, <span class="at">col=</span><span class="st">"dodgerblue"</span>)</span>
<span id="cb14-142"><a href="#cb14-142" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-143"><a href="#cb14-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-144"><a href="#cb14-144" aria-hidden="true" tabindex="-1"></a><span class="in">```{r enrichment with MultiXcan}</span></span>
<span id="cb14-145"><a href="#cb14-145" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif</span>(BMI_genes<span class="sc">$</span>pvalue, <span class="at">col=</span> <span class="st">"maroon1"</span>) </span>
<span id="cb14-146"><a href="#cb14-146" aria-hidden="true" tabindex="-1"></a>BMI_genes_rats <span class="ot">&lt;-</span> pheno_Multi_rat  <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pvalue.bmi_bodylength_wo_tail <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> .[[<span class="st">"gene_name"</span>]] <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb14-147"><a href="#cb14-147" aria-hidden="true" tabindex="-1"></a><span class="fu">qqpoints</span>(BMI_genes <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene_name <span class="sc">%in%</span> BMI_genes_rats) <span class="sc">%&gt;%</span> .[[<span class="st">"pvalue"</span>]],<span class="at">pch=</span><span class="st">'+'</span>, <span class="at">col=</span><span class="st">"dodgerblue"</span>) </span>
<span id="cb14-148"><a href="#cb14-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-149"><a href="#cb14-149" aria-hidden="true" tabindex="-1"></a><span class="fu">qqunif</span>(height_genes<span class="sc">$</span>pvalue, <span class="at">col=</span> <span class="st">"maroon1"</span>)</span>
<span id="cb14-150"><a href="#cb14-150" aria-hidden="true" tabindex="-1"></a>height_genes_rats <span class="ot">&lt;-</span> pheno_Multi_rat <span class="sc">%&gt;%</span> <span class="fu">filter</span>(pvalue.bodylength_w_tail <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> .[[<span class="st">"gene_name"</span>]] <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb14-151"><a href="#cb14-151" aria-hidden="true" tabindex="-1"></a><span class="fu">qqpoints</span>(height_genes <span class="sc">%&gt;%</span> <span class="fu">filter</span>(gene_name <span class="sc">%in%</span> height_genes_rats) <span class="sc">%&gt;%</span> .[[<span class="st">"pvalue"</span>]],<span class="at">pch=</span><span class="st">'+'</span>, <span class="at">col=</span><span class="st">"dodgerblue"</span>)</span>
<span id="cb14-152"><a href="#cb14-152" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb14-153"><a href="#cb14-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-154"><a href="#cb14-154" aria-hidden="true" tabindex="-1"></a><span class="in">```{r calculate Enrichment value with MW test}</span></span>
<span id="cb14-155"><a href="#cb14-155" aria-hidden="true" tabindex="-1"></a>all_bmi_genes <span class="ot">&lt;-</span> BMI_genes <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">in_rats =</span> <span class="fu">ifelse</span>(BMI_genes<span class="sc">$</span>gene_name <span class="sc">%in%</span> BMI_genes_rats, <span class="dv">1</span>, <span class="dv">0</span>))</span>
<span id="cb14-156"><a href="#cb14-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-157"><a href="#cb14-157" aria-hidden="true" tabindex="-1"></a>all_height_genes <span class="ot">&lt;-</span> height_genes <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">in_rats =</span> <span class="fu">ifelse</span>(height_genes<span class="sc">$</span>gene_name <span class="sc">%in%</span> height_genes_rats, <span class="dv">1</span>, <span class="dv">0</span>))</span>
<span id="cb14-158"><a href="#cb14-158" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-159"><a href="#cb14-159" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(pvalue <span class="sc">~</span> in_rats, <span class="at">data=</span>all_bmi_genes) </span>
<span id="cb14-160"><a href="#cb14-160" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(pvalue <span class="sc">~</span> in_rats, <span class="at">data=</span>all_height_genes) </span>
<span id="cb14-161"><a href="#cb14-161" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>