<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Rat Genomics Pipeline Rendered - About</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-listing/list.min.js"></script>
<script src="site_libs/quarto-listing/quarto-listing.js"></script>
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<script>

  window.document.addEventListener("DOMContentLoaded", function (_event) {
    const listingTargetEl = window.document.querySelector('#listing-listing .list');
    if (!listingTargetEl) {
      // No listing discovered, do not attach.
      return; 
    }

    const options = {
      valueNames: ['listing-date','listing-title','listing-author','listing-description',{ data: ['index'] },{ data: ['categories'] },{ data: ['listing-date-sort'] },{ data: ['listing-file-modified-sort'] }],
      
      searchColumns: ["listing-date","listing-title","listing-author","listing-description"],
    };

    window['quarto-listings'] = window['quarto-listings'] || {};
    window['quarto-listings']['listing-listing'] = new List('listing-listing', options);

    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  });

  window.addEventListener('hashchange',() => {
    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  })
  </script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Rat Genomics Pipeline Rendered</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html">
 <span class="menu-text">Home</span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<div class="quarto-about-jolla">
  <img src="profile.jpg" class="about-image
  round " style="height: 15em; width: 15em;">
 <header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">About</h1>
</div>
<div class="quarto-title-meta">
  </div>
</header><main class="content" id="quarto-document-content">
<p>About this blog</p>
<p>:::{#quarto-listing-pipeline .hidden} <span class="hidden" data-render-id="quarto-enable-math-inline"><span class="math inline">\(e = mC^2\)</span></span></p>
<p>:::{.hidden render-id=“pipeline-listing-listing”}</p>
:::{.list .quarto-listing-default}
<div class="quarto-post image-right" data-index="0" data-listing-date-sort="1644213600000" data-listing-file-modified-sort="1700328487980" data-listing-reading-time-sort="15.315">
<div class="body">
<a href="./posts/01.Heritability_Sparsity.html">
<h3 class="no-anchor listing-title">
01.Heritability_Sparsity
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
First…
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/01.Heritability_Sparsity.html">
<div class="listing-date">
Feb 7, 2022
</div>
<div class="listing-author">
natasha.santhanam
</div>
</a>
</div>
</div>
<div class="quarto-post image-right" data-index="1" data-listing-date-sort="1644213600000" data-listing-file-modified-sort="1700329273541" data-listing-reading-time-sort="11.8">
<div class="body">
<a href="./posts/01.Heritability_Sparsity_SM.html">
<h3 class="no-anchor listing-title">
01.Heritability_Sparsity Sabrina
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
For each gene, we calculate heritability from its local GRM. Start with creating list of genes for each of the gene expression file:
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/01.Heritability_Sparsity_SM.html">
<div class="listing-date">
Feb 7, 2022
</div>
<div class="listing-author">
natasha.santhanam, Sabrina
</div>
</a>
</div>
</div>
<div class="quarto-post image-right" data-index="2" data-listing-date-sort="1644213600000" data-listing-file-modified-sort="1700329033511" data-listing-reading-time-sort="8.085">
<div class="body">
<a href="./posts/02.Prediction_Model_Pipeline.html">
<h3 class="no-anchor listing-title">
<ol start="2" type="1">
<li>Prediction_Model_Pipeline
</li></ol></h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
Data from here - genoGex.Rdata has everything we need in it There are 5 ‘gex’ RDS…
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/02.Prediction_Model_Pipeline.html">
<div class="listing-date">
Feb 7, 2022
</div>
<div class="listing-author">
Tyson Miller
</div>
</a>
</div>
</div>
<div class="quarto-post image-right" data-index="3" data-listing-date-sort="1644213600000" data-listing-file-modified-sort="1700329032645" data-listing-reading-time-sort="9.98">
<div class="body">
<a href="./posts/03.PrediXcan.html">
<h3 class="no-anchor listing-title">
03.PrediXcan_MultiXcan
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
Sabrina’s note: I have no idea what is going on here, I don’t know where bimbam file came from. I tried running this script from genotype files generated from previous steps…
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/03.PrediXcan.html">
<div class="listing-date">
Feb 7, 2022
</div>
<div class="listing-author">
Natasha Santhanam
</div>
</a>
</div>
</div>
<div class="quarto-post image-right" data-index="4" data-listing-date-sort="1650517200000" data-listing-file-modified-sort="1700328025676" data-listing-reading-time-sort="1.68">
<div class="body">
<a href="./posts/04.GEMMA_LMM_analysis.html">
<h3 class="no-anchor listing-title">
04.GEMMA_LMM_analysis
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
Gemma uses a LMM with…
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/04.GEMMA_LMM_analysis.html">
<div class="listing-date">
Apr 21, 2022
</div>
<div class="listing-author">
Natasha Santhanam
</div>
</a>
</div>
</div>
<div class="quarto-post image-right" data-index="5" data-listing-date-sort="1644213600000" data-listing-file-modified-sort="1700328025688" data-listing-reading-time-sort="5.25">
<div class="body">
<a href="./posts/06.PTRS_Creation.html">
<h3 class="no-anchor listing-title">
06.PTRS_Creation
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
The <code>orth.rats</code> file contains gives a dictionary between human genes and the corresponding gene in rats.
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/06.PTRS_Creation.html">
<div class="listing-date">
Feb 7, 2022
</div>
<div class="listing-author">
Natasha Santhanam
</div>
</a>
</div>
</div>
<div class="quarto-post image-right" data-index="6" data-listing-date-sort="1644213600000" data-listing-file-modified-sort="1700329030676" data-listing-reading-time-sort="3.43">
<div class="body">
<a href="./posts/07.SPTRS_Creation.html">
<h3 class="no-anchor listing-title">
07 SPTRS_Creation
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
Set up the conda environment
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/07.SPTRS_Creation.html">
<div class="listing-date">
Feb 7, 2022
</div>
<div class="listing-author">
Natasha Santhanam
</div>
</a>
</div>
</div>
<div class="quarto-post image-right" data-index="7" data-listing-date-sort="1644213600000" data-listing-file-modified-sort="1700329028979" data-listing-reading-time-sort="2.725">
<div class="body">
<a href="./posts/08.Compare_all_PTRS.html">
<h3 class="no-anchor listing-title">
08.Compare_all_PTRS
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
Match genes in weights file in humans to Rat expression
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/08.Compare_all_PTRS.html">
<div class="listing-date">
Feb 7, 2022
</div>
<div class="listing-author">
Natasha Santhanam
</div>
</a>
</div>
</div>
<div class="quarto-post image-right" data-index="8" data-listing-date-sort="1651640400000" data-listing-file-modified-sort="1700328025704" data-listing-reading-time-sort="3.765">
<div class="body">
<a href="./posts/09.PRS_Rats.html">
<h3 class="no-anchor listing-title">
09.PRS_Rats
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
first extract out 500 random rats that we’ll use as baseline
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/09.PRS_Rats.html">
<div class="listing-date">
May 4, 2022
</div>
<div class="listing-author">
Natasha Santhanam
</div>
</a>
</div>
</div>
<div class="quarto-post image-right" data-index="9" data-listing-date-sort="1656219600000" data-listing-file-modified-sort="1700328025709" data-listing-reading-time-sort="2.525">
<div class="body">
<a href="./posts/10.PTRS_simulation_weights.html">
<h3 class="no-anchor listing-title">
10.PTRS_simulation_weights
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
Read in files
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/10.PTRS_simulation_weights.html">
<div class="listing-date">
Jun 26, 2022
</div>
<div class="listing-author">
Natasha Santhanam
</div>
</a>
</div>
</div>
<div class="quarto-post image-right" data-index="10" data-listing-file-modified-sort="1700329024538" data-listing-reading-time-sort="0.015">
<div class="body">
<a href="./posts/about.html">
<h3 class="no-anchor listing-title">
About
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
Describe your project.
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/about.html"> </a>
</div>
</div>
<div class="quarto-post image-right" data-index="11" data-listing-date-sort="1644818400000" data-listing-file-modified-sort="1700328025714" data-listing-reading-time-sort="4.225">
<div class="body">
<a href="./posts/Coregualtion_across_Tissues_Species.html">
<h3 class="no-anchor listing-title">
Coregulation_across_Tissues_Species
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
Generate Correlation Matrices for 100 rats - compare predicted Expression across tissues
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/Coregualtion_across_Tissues_Species.html">
<div class="listing-date">
Feb 14, 2022
</div>
<div class="listing-author">
Natasha Santhanam
</div>
</a>
</div>
</div>
<div class="quarto-post image-right" data-index="12" data-listing-date-sort="1644818400000" data-listing-file-modified-sort="1700328025719" data-listing-reading-time-sort="6.345">
<div class="body">
<a href="./posts/GTEx_n_genes_rat_comparison.html">
<h3 class="no-anchor listing-title">
GTEx_comparison_rat_n_genes
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
Gather Genotype, Gene epression data, snp annotation file and gtf (made with ensembl notation)
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/GTEx_n_genes_rat_comparison.html">
<div class="listing-date">
Feb 14, 2022
</div>
<div class="listing-author">
Natasha Santhanam
</div>
</a>
</div>
</div>
<div class="quarto-post image-right" data-index="13" data-listing-date-sort="1646805600000" data-listing-file-modified-sort="1700328025762" data-listing-reading-time-sort="1.77">
<div class="body">
<a href="./posts/generate_sparsity_figs.html">
<h3 class="no-anchor listing-title">
Generate Sparsity Figures
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
we will use this for comparison later
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/generate_sparsity_figs.html">
<div class="listing-date">
Mar 9, 2022
</div>
<div class="listing-author">
Natasha Santhanam
</div>
</a>
</div>
</div>
<div class="quarto-post image-right" data-index="14" data-listing-date-sort="1644818400000" data-listing-file-modified-sort="1700328025757" data-listing-reading-time-sort="3.875">
<div class="body">
<a href="./posts/generate_plink_format_genotypes.html">
<h3 class="no-anchor listing-title">
Generate_plink_format_genotypes
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
Data from here - genoGex.Rdata has everything we need in it There are 5 ‘gex’ RDS…
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/generate_plink_format_genotypes.html">
<div class="listing-date">
Feb 14, 2022
</div>
<div class="listing-author">
Natasha Santhanam
</div>
</a>
</div>
</div>
<div class="quarto-post image-right" data-index="15" data-listing-date-sort="1645682400000" data-listing-file-modified-sort="1700328025724" data-listing-reading-time-sort="3.275">
<div class="body">
<a href="./posts/Human_PTRS_Comparison.html">
<h3 class="no-anchor listing-title">
Human_PTRS_Performance_Comparison
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
Yanyu’s PTRS weights estimate the effect of genes on a given trait, in this case we pick height and BMI.
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/Human_PTRS_Comparison.html">
<div class="listing-date">
Feb 24, 2022
</div>
<div class="listing-author">
Natasha Santhanam
</div>
</a>
</div>
</div>
<div class="quarto-post image-right" data-index="16" data-listing-date-sort="1644818400000" data-listing-file-modified-sort="1700329031662" data-listing-reading-time-sort="2.315">
<div class="body">
<a href="./posts/05.PhenomeXcan_MutliXcan.html">
<h3 class="no-anchor listing-title">
PhenomeXcan Query
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
Generate a table of all human MultiXcan results
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/05.PhenomeXcan_MutliXcan.html">
<div class="listing-date">
Feb 14, 2022
</div>
<div class="listing-author">
Natasha Santhanam
</div>
</a>
</div>
</div>
<div class="quarto-post image-right" data-index="17" data-listing-date-sort="1644818400000" data-listing-file-modified-sort="1700328025735" data-listing-reading-time-sort="5.36">
<div class="body">
<a href="./posts/Rat_QC_Merging.html">
<h3 class="no-anchor listing-title">
Rat QC and Comparison Check
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
Apurva sent bimbam files that I converted to map/ped and then bim/bed/fam files using my own pipeline. Pipeline is in rat_compare_genotypes_metabolic.Rmd file.
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/Rat_QC_Merging.html">
<div class="listing-date">
Feb 14, 2022
</div>
<div class="listing-author">
Natasha Santhanam
</div>
</a>
</div>
</div>
<div class="quarto-post image-right" data-index="18" data-listing-date-sort="1644818400000" data-listing-file-modified-sort="1700328262150" data-listing-reading-time-sort="6.39">
<div class="body">
<a href="./posts/Rat_Predictability_comp_Humans.html">
<h3 class="no-anchor listing-title">
Rat_Predictability_comp_Humans
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
#Figure comparing pred R2 vs obs R2 in Ac tissue in Rats
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/Rat_Predictability_comp_Humans.html">
<div class="listing-date">
Feb 14, 2022
</div>
<div class="listing-author">
Natasha Santhanam
</div>
</a>
</div>
</div>
<div class="quarto-post image-right" data-index="19" data-listing-date-sort="1644213600000" data-listing-file-modified-sort="1700328025741" data-listing-reading-time-sort="1.845">
<div class="body">
<a href="./posts/Sparsity_Badger_Template.html">
<h3 class="no-anchor listing-title">
Sparsity_Badger_Template
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">

</div>
</a>
</div>
<div class="metadata">
<a href="./posts/Sparsity_Badger_Template.html">
<div class="listing-date">
Feb 7, 2022
</div>
<div class="listing-author">
natasha.santhanam
</div>
</a>
</div>
</div>
<div class="quarto-post image-right" data-index="20" data-listing-date-sort="1645423200000" data-listing-file-modified-sort="1700328025751" data-listing-reading-time-sort="1.195">
<div class="body">
<a href="./posts/convert_genotypes_old.html">
<h3 class="no-anchor listing-title">
convert_genotypes_old
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
<strong>Note:</strong> This is old script to convert Palmer lab genotype data in an .Rdata object to plink format. For our analysis, we used a different process to reformat, documented here
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/convert_genotypes_old.html">
<div class="listing-date">
Feb 21, 2022
</div>
</a>
</div>
</div>
<div class="quarto-post image-right" data-index="21" data-listing-date-sort="1644818400000" data-listing-file-modified-sort="1700328025772" data-listing-reading-time-sort="3.2">
<div class="body">
<a href="./posts/rat_compare_genotypes_metabolic.html">
<h3 class="no-anchor listing-title">
rat_compare_genotypes
</h3>
<div class="listing-subtitle">
</div>
<div class="listing-description">
Min. 1st Qu. Median Mean 3rd Qu. Max. 0.6856 0.7511 0.7851 0.7833 0.8211 0.8801
</div>
</a>
</div>
<div class="metadata">
<a href="./posts/rat_compare_genotypes_metabolic.html">
<div class="listing-date">
Feb 14, 2022
</div>
<div class="listing-author">
Natasha Santhanam
</div>
</a>
</div>
</div>
:::
<div class="listing-no-matching d-none">
No matching items
</div>
<p>:::</p>
<p>:::</p>


<div class="quarto-listing quarto-listing-container-default" id="listing-listing"></div></main> 
  <hr class="about-sep">
   <div class="about-links">
  <a href="https://twitter.com" class="about-link">
    <i class="bi bi-twitter"></i>
     <span class="about-link-text">Twitter</span>
  </a>
  <a href="https://linkedin.com" class="about-link">
    <i class="bi bi-linkedin"></i>
     <span class="about-link-text">LinkedIn</span>
  </a>
  <a href="https://github.com" class="about-link">
    <i class="bi bi-github"></i>
     <span class="about-link-text">Github</span>
  </a>
</div>
</div>
 <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>